---
format:
  pdf:
    keep-tex: true
    pdfengine: xelatex
    geometry: 
      - "top=1in"
      - "bottom=1in"
      - "left=1in"
      - "right=1in"
    code-annotations: hover
---

::: {.flushright data-latex="1" linestretch="1"}
The University of Texas at Austin\
Department of Integrative Biology\
2415 Speedway\
Austin, TX 78712
:::

 

::: {.flushright data-latex="1" linestretch="1"}
October 16th, 2024
:::
 

Dear Editors,

Thank you for considering this work for publication in *eLife*.

Our work is a key methodological development in understanding genetic variation in genotype-by-environment interactions, or GxE. GxE interactions must vary across the genome; that is, different genes and genetic variants can exhibit different responses to the environment. Yet current statistical models to test for GxE don't have robust frameworks to specify and compete different models of GxE effects, instead focusing on finding a single, best fitting form of GxE covariance. We use multivariate adaptive shrinkage (mash [@urbut2018]) to determine the regions of the switchgrass genome with different types of GxE, specifically regions of the genome with effects on phenology that covary with several weather-based 'cues'. We use a greedy algorithm to compete and select types of GxE and GxWeather that improve our models, then use mash to identify which SNP effects are best modeled with each type of GxE or GxWeather.

Second, to our knowledge, this work is the first to use a statistically unbiased test to detect the relative frequencies of loci with and without sign-changing GxE. Here, rather than using a standard statistical test such as the false discovery rate to ask "Are these two effects different?" - as we reasonably expect two effects to be, even if this difference cannot be measured - we use the local false sign rate to ask a more meaningful question: Can we be confident in the sign of these two effects? In contrast with previous empirical work which finds few genetic effects that differ in sign, we use the local false sign rate to find thousands of genetic effects with antagonistic pleiotropy for the start of vegetative growth for switchgrass.

Finally, with current advances in high-throughput measurement, and the increased resources devoted to experiments in natural and semi-natural environments, we must consider how to share information across these experiments and how to associate natural variation in genomes with natural variation in complex field environments. Advances in methods to find the environmental cues that matter should allow researchers to determine the adaptations important to responses to the natural environment, and how the importance of these genetic mechanisms changes as the environment itself changes. Our manuscript should be of broad interest to *eLife* readership for the insights it provides into these important issues.

 

 

::: {.center data-latex="1"}
Best,
:::

::: {.flushright data-latex="1"}
Alice MacQueen and Tom Juenger
:::
