---
title: Quarto Template for PNAS Submissions
format:
  pnas-pdf:
    keep-tex: true  
author:
  - name: Author One
    affiliations:
      - name: Harvard University
        id: a
        department: Department of Government
        city: Cambridge
        state: MA
        postal-code: 02138
      - name: Harvard University
        id: c
        department: Department of Statistics
        city: Cambridge
        state: MA
        postal-code: 02138
    attributes:
      equal-contributor: true
  - name: Author Two
    affiliations:
      - name: Yale University
        id: b
        department: Department of Political Science
        city: New Haven
        state: CT
        postal-code: 06511
    email: alicem@uchicago.edu
    attributes:
      equal-contributor: true
      corresponding: true
  - name: Author Three
    affiliations:
      - ref: a
abstract: |
  The timing of vegetative growth and flowering in plants ("phenological timings") multiple depend on environmental cues; thus, phenological traits can have important genotype-by-environment (GxE) interactions. Here, we map genotype-by-weather effects on phenological timings in multiple populations of two highly divergent switchgrass (Panicum virgatum) populations spanning the central United States. We distinguish GxWeather patterns covarying with interpretable weather-based cues and agnostic, empirically-inferred GxWeather patterns. Most GxWeather effects belong to the latter category and do not covary with weather-based cues. However, >26% of Gulf population effects on flowering time covary with photoperiod cues, while >__% of Midwest upland population SNPs affecting flowering had effects that covaried with __ cues. An independent pseudo-F2 cross of Gulf and Midwest individuals mapped 23 additive QTLs for flowering at the same common gardens, all with significant mash associations and ten with enrichment of highly significant mash associations. We demonstrate that we can identify QTL with GxE and assign them to specific weather-based cues and/or data-driven patterns. Breeding for particular alleles at these loci could change flowering responsiveness to photoperiod cues in switchgrass. More broadly, this approach could be used to identify genetic marker-environment interactions in any species with related populations phenotyped in multiple environments.
runningauthor: "One, Two, and Three"
significance: |
  The timing of plant seasonal development (phenology) has major impacts on fitness because of the steep price of plant-environment mismatches. Here we map the genetic basis of two phenological events, the start of above-ground growth and flowering, in two genetically and phenologically distinct populations of switchgrass. We do this at eight field locations spanning the latitudinal range of both switchgrass populations. Our approach allows us to identify regions of the genome that covary with specific weather-related environmental features at every location. For flowering, these features differed by population: the Midwest population had genetic variation that primarily covaried with cumulative growing degree days, a temperature-related measure, while the Gulf population had genetic variation that primarily covaried with photoperiod, a day-length-related measure.
author_contributions:  "T.E.J. designed research. D.B.L. contributed plant material and resources. J.B., D.B.L., and T.E.J. designed and executed field experiments. A.R.B., P.A.F., F.B.F., D.B.L., R.B.M., F.M.R., Y.W., and T.E.J. hosted field experiments. A.H.M. and L.Z. conducted statistical and computational analyses. The manuscript was written by A.H.M., L.Z., and T.E.J. with contributions from all authors."
conflict_of_interest: "Please declare any competing interests here."
equal-contributor: "A.O.(Author One) contributed equally to this work with A.T. (Author Two) (remove if not applicable)."
keywords:  [allele-by-environment effect variation, antagonistic pleiotropy, photoperiod, cumulative growing degree days, genetic variation]
acknowledgements: | 
  We thank the Brackenridge Field laboratory, the Ladybird Johnson Wildflower Center, and the Juenger laboratory for support with plant care and propagation. This material is based upon work supported in part by the Great Lakes Bioenergy Research Center, U.S. Department of Energy, Office of Science, Office of Biological and Environmental Research under Award Numbers DE-SC0018409 and DE-FC02-07ER64494, the US Department of Energy Awards DESC0014156 to T.E.J., DE-SC0017883 to D.B.L, National Science Foundation PGRP Awards IOS0922457 and IOS1444533 to T.E.J, and the Long-term Ecological Research Program (DEB 1832042) at the Kellogg Biological Station.
bibliography: bibliography.bib  
fontsize: "9pt"
classoption:
 - twocolumn
 - twoside
---

\firstpage[20]{2}

The timing of plant vegetative and reproductive development are major components of plant fitness affected by multiple external environmental cues (e.g. degree of winter chilling, day length, temperature, and water availability) that signal existing or upcoming growing conditions (1--3). Genetic responses to environmental cues determine the speed, timing, and energy apportioned to vegetative and reproductive growth and shape both the individual's lifespan and its lifetime production of viable seed. Day length (or photoperiod) is one of the most predictable environmental cues, and genetic sensitivity to photoperiod protects plants from potentially fatal consequences of phenological responses to temperature cues at the "wrong" time of year. However, the usefulness of specific environmental cues depends on both features of the environment and the species' adaptive strategies (4). Species with wide natural distributions can have multiple distinct environmentally-cued phenological responses: for example, populations of sunflower (Helianthus annuus) exhibit day-neutral, facultative short day, and facultative long-day flowering responses, which vary with their environments (5, 6). Distinct genetic responses in different environments are known as genotype by environment interactions, or GxE.

Flowering time, in particular, is a common subject of GxE research (5--11), a key output of selection driving adaptation to local environments (3, 12, 13), and a selection target for crop improvement to adapt crops to local or future environments (14). Changing flowering responsiveness to photoperiod cues has allowed geographic range expansion and increased yields in a number of cereal species (15--19) and other crops (20, 21). Recent statistical advances in studying phenological GxE have involved determining critical environmental indices before the phenological event occurs, such as photothermal time within a critical growth window (10). However, most studies of flowering GxE focus on finding a single, best fitting form of genotype-environment covariance, despite the key expectation that different genetic subpopulations, and even different genomic regions, have likely evolved distinct patterns of GxE. Additionally, despite theoretical predictions that local adaptation should involve antagonistic pleiotropy, or sign-changing GxE, at the level of individual loci (22--25), previous work has found limited evidence of antagonistic pleiotropy (12, 26), but has been limited by a known statistical bias that reduced detection of SNP effects that differ in sign (26--28). Thus, despite substantial interest in the frequencies of various forms of GxE, the prevalence of antagonistic pleiotropy relative to other forms of GxE remains unknown.

Switchgrass (Panicum virgatum) is considered a short-day plant with reproductive development strongly linked to day of the year (29). However, as part of its wide environmental adaptation across the eastern half of North America, its photoperiodicity has been predicted to differ by plant latitude of origin (30, 31). We previously found divergent Midwest and Gulf genetic subpopulations of switchgrass which segregate for distinct sets of environmental adaptations, based on the analysis of two measures of fitness (i.e.,biomass and survival) among 732 genotypes (32). The Midwest genetic subpopulation is primarily composed of individuals from the well-studied upland switchgrass ecotype (33, 34), while the Gulf subpopulation has individuals from the lowland ecotype and the phenotypically intermediate coastal ecotype (32).

Here, we test how the Midwest and Gulf subpopulations differ in their phenological adaptations across environments, and hence their phenological GxE. We phenotype a diversity panel of hundreds of switchgrass genotypes from the Midwest and Gulf subpopulations for the timing of vegetative development ("green-up") and reproductive development (flowering) at eight common garden locations spanning 17 degrees of latitude. These gardens cover the majority of the latitudinal and climatic range of switchgrass and capture the most comprehensive picture to date of the environmental variation this species encounters. After single-site estimation of SNP effects for these phenotypes, we use multivariate adaptive shrinkage (mash) to jointly estimate SNP effects in each subpopulation and phenotype at all eight sites (35). Mash allows us to identify and specify multiple ways genetic effects linked to SNPs may covary with the environment, and does not have a statistical bias in detecting SNP effects with the same or opposite signs (39). To confirm our genetic mapping of GxE, we compare to mapping results from an outbred pseudo-F2 cross grown at the same sites. Taken together, our results allow us to describe the environmental cues and genetic variation affecting phenology in two divergent natural populations of switchgrass.

# Results

In our diversity panel of tetraploid switchgrass (32), genotypes from the Gulf and Midwest genetic subpopulations had distinct phenological timings and distinct patterns of phenological correlations across our eight common garden sites (Fig. 1A,B). At the three Texas common gardens (hereafter 'Texas' gardens), located within the natural range of the Gulf subpopulation, Gulf green-up occurred before Midwestern green-up, and Gulf flowering occurred after Midwestern flowering (Fig. 1A). At the four northernmost common gardens (hereafter 'North' gardens), located within the natural range of the Midwest subpopulation, both Gulf green-up and flowering occurred after Midwest green-up and flowering. At the Oklahoma common garden, located near the natural range limits of both the Gulf and the Midwest subpopulations, Gulf and Midwest green-up occurred over the same time period. These patterns led to strong negative phenotypic correlations for green-up between the North and Texas gardens, particularly in the Gulf and Both subpopulations, and contributed to positive phenotypic correlations for flowering time of larger magnitude at more northern gardens (Fig. 1B).

Narrow-sense heritabilities (h2) indicated that rank-changing GxE for these phenotypes was present across the common gardens (Fig 1C). h2 were typically high at individual gardens: 59% on average for green-up date, and 87% for flowering date. However, h2 were variable across gardens, and green-up dates were uncorrelated (r2 \< 0.2) or negatively correlated between pairs of gardens (Fig. 1B). These negative and small correlations undoubtedly contributed to the low h2 values for green-up and flowering date when estimated jointly at all eight gardens: h2 was 0.8% for green-up and 23.2% for flowering date.

### Mapping major patterns of genotype-by-environment effects on green-up and flowering

We next explored how genetic variation in phenology covaried with environmental cues by using mash to jointly discover trait-associated SNPs across all common gardens for the Midwest, Gulf, and 'Both' subpopulations (Fig 1D, SI Appendix, Datasets 1-6). We first estimated SNP effects using GWAS on site-specific BLUPs for each phenotype. We then used mash on two subsets of these SNP effects, first to determine the covariance structures that significantly improved the mash model log likelihood when included, then to re-estimate the set of SNP effects with the largest effect estimates in univariate GWAS (hereafter "strong" effects) using the set of significant covariance matrices.

Three types of covariance matrices were tested for inclusion: "canonical" covariance matrices, with simple patterns of effects; "data-driven" matrices derived from common patterns of SNP effects in the data, and "hypothesis-based" covariance matrices that captured the covariance of weather-based phenological cues for cloned genotypes grown at multiple common gardens (Table 1, SI Appendix, Section S2). Mash first assigns mixture proportions for each SNP onto each provided covariance matrix using maximum likelihood. Then, in a second step, each SNP's effect estimate undergoes shrinkage such that it is more aligned with the covariance structure inferred in the first step. The hypothesis-based covariance matrices are an important advantage mash offers for studying patterns of GxE: they allow hypothesis testing of specific environmental drivers for each SNP effect, and loadings on these matrices provide information on genome-wide patterns of SNP-environment interaction. Say that a SNP in FLC controls flowering in a temperature-dependent manner. In that case, that SNP effect could have a high mixture proportion, or mass, on a covariance matrix created using a temperature-based environmental cue, such as average temperature on the day prior to flowering. In our data, we would infer that the effect of that SNP on flowering was caused by a response to that or a correlated environmental cue.

The phenotypic correlations for green up date had moderate negative correlations between the Texas and North gardens, particularly in the Gulf and in Both subpopulations (Fig 1B). If these phenotypic correlations have a genetic component, then they might be controlled by SNP effects that differed in sign across these regions, SNP effects that are large in one garden or region and non-significant in others, or a combination of these patterns. Five hypothesis-based covariance matrices were selected by the greedy mash algorithm, one to two per subpopulation. Of these five, three had mass on them in mash models of the strong effects (Fig 2A, 2B). Two of these three matrices had negative covariances between sites between Texas and North gardens (Fig 2A), while one had all positive or near-zero covariances. SNP-associated phenotypic effects covaried with different weather-based cues in the Gulf & in Both subpopulations (Fig 2B). In total, 65% of the posterior weight of strong SNP effects in the mash model of Gulf green-up fell on a covariance matrix of daylength 14 days prior to the date of green-up. The covariance matrix for this weather cue was very similar to the pattern of phenotypic correlation for green-up in the Gulf subpopulation (Fig 1B; Fig 2A). Mash models of Midwest and Both subpopulation green-up did not include this covariance matrix; the Midwest had no weight on any hypothesis-based matrix, while Both subpopulations had non-zero weights on two additional hypothesis-based matrices, average temperature one day prior to green-up, and the day length change in seconds in the day prior to green-up (Fig 2B). The average temperature covariance matrix had negative covariances between Texas and North gardens, though not as strong as the negative phenotypic correlations seen in Both subpopulations (Fig 1B; Fig 2A). Only the Gulf subpopulation and Both subpopulations had mass on any hypothesis-based covariance matrices (Fig 2C).

For flowering date, distinct weather-based cues captured SNP-associated effect patterns in the Gulf and Midwest subpopulations. 33% of SNP effects on flowering in the Gulf subpopulation covaried with cumulative rainfall in the seven days prior to flowering (Fig 2D). 22.6% of SNP effects on flowering in the Midwest subpopulation contrived with day length change in the two days prior to flowering (Fig 2D), which had negative covariances between Texas and North gardens. Neither covariance matrix was selected for in Both subpopulations, and no hypothesis-based covariance matrices had mass in Both subpopulations (Fig 2E). In five of the six mash models of strong effects, the hypothesis-based covariance matrices captured a minority of the posterior weights of the strong effects (Figure 2C,E); the majority of this mass was on various canonical covariance matrices. These matrices included simple heterozygosity, with intermediate, positive covariances between all gardens, and garden-specific effects present only at one garden.

We next characterized the pairwise patterns of effects where we were confident in the sign of the effect at both gardens. We used the local false sign rate (lfsr), an analogue of the lfdr that establishes confidence in the effect sign, not the effect's difference from zero, to determine significance. We required significance (p \< 0.05) in both gardens to include effects. This means that our tests for antagonistic pleiotropy, or a sign change between conditions, carry an equal statistical burden to those for effects with the same sign. We also summarized the pairwise patterns of effect sign and magnitude within and between the Texas and North regions (Table 2).

For greenup date for the Gulf subpopulation, hundreds to thousands of pairwise effects exhibited antagonistic pleiotropy, or a difference in effect sign, between pairs of Texas and North gardens (Fig. 3A). 78.7% of pairwise comparisons between North and Texas gardens had a difference in sign, while only 28.6% and 0.2% of North-North or Texas-Texas comparisons had a difference in sign, respectively (Table 2). The majority of pairwise effects for greenup for the Midwest (\>55%) and Both (\>85%) subpopulations were the same sign, and effects frequently differed in magnitude between the MO and OK garden and other gardens (Table 2; Fig 3A).

For flowering date for the Gulf subpopulation, less than 2% of pairwise effects exhibited antagonistic pleiotropy, or a difference in effect sign, within or between regions (Table 2). More effects differed in magnitude between the Texas and North regions than within these regions (42.7% vs \<20%; Figure 3B). The Midwest population had relatively few significant effects for flowering, but a large proportion of these differed in sign between Texas and North regions (42.7%) or within the North region (65.4%). Finally, in Both subpopulations, less than 20% of pairwise effects differed in sign (Table 2). Most differences in sign were between TX1, the southernmost garden, and all other gardens (Fig 3B). Similarly, more effects that differed in magnitude included gardens in the Texas region (52.3-55.9%), and most effect pairs in the North region were not distinguishable (91.5%).

### Confirmation of genotype-by-environment effects using an independent mapping population

We sought additional experimental support for our mash intervals using an independent pseudo-F2 mapping population created from Gulf & Midwest individuals and grown at the same sites (Fig. 4A,B). We conducted quantitative trait loci (QTL) mapping of flowering as functions of four environmental cues that we also used as covariance matrices in mash, and identified eight QTL for flowering date, six QTL for flowering GDD, ten QTL for flowering day length, and eight QTL flowering day length change, all of which showed QTL by environment interactions (SI Appendix, Fig. S3). All QTL for flowering overlapped one or more homologs from rice or A. thaliana with functionally validated roles in flowering (SI Appendix, Dataset 7). All flowering QTL intervals contained at least one SNP significant in at least one mash run at a log10-transformed Bayes Factor \> 2, or in the 1% tail of significance, whichever was stricter (SI Appendix, Dataset 8). We also looked for enrichments of mash SNPs in the 1% tail of significance (the 'mash 1% tail') within each QTL interval. At the 5% level, five QTL had enrichments of SNPs in the mash 1% tail. Overall, there were eleven significant enrichments (p \< 0.05, hypergeometric test) of SNPs in the mash 1% tail in the QTL intervals. Our QTL intervals had more enrichments of SNPs in the mash 1% tail than were found for all but eleven of these sets of random genomic intervals (Fig. 4C, p = 0.011). Thus, we were able to experimentally support our mash intervals with a QTL mapping experiment using a separate mapping population.

# Discussion

As the climate and the natural environment change, it is increasingly critical to understand how patterns of gene-environment and plant-environment interactions will change in response. To do this, we must understand the current patterns of trait covariation across environments, the genetic underpinnings of these patterns, and the cases where this covariation can be altered. Here, we demonstrate that we can associate multiple patterns of GxE with specific genomic regions using a switchgrass diversity panel grown at eight common gardens, and also that we can assign specific SNP-associated patterns of GxE to both weather-based cues and to other, data-driven patterns. We use this approach to study GxE in both green-up and flowering phenological data in the deeply genetically diverged Gulf and Midwest populations of switchgrass.

Our analysis of green-up in the Gulf and in Both subpopulations revealed substantial antagonistic pleiotropy in effects between the Texas and North gardens (Figure 3A). This result supports theoretical models that local adaptation should involve antagonistic pleiotropy at the level of individual loci (22--25), and is the first experimental work using QTL mapping and GWAS across common gardens to find antagonistic pleiotropy to be common in small genomic regions (12, 36, 37).

Our analysis of flowering showed that the Gulf and Midwest subpopulations have two distinct photoperiod-related flowering responses: the Midwest subpopulation is day neutral, and flowering is cued primarily by a cumulative GDD threshold; in contrast, the Gulf subpopulation is photoperiod sensitive, and flowering is cued by the transition to shortening days. This result was supported by observations that expressing flowering date as a function of the day length at flowering increased its heritability in the Gulf subpopulation, while expressing flowering date as a function of cumulative GDD between green-up and flowering increased the heritability of flowering in the Midwest subpopulation (Fig. 1D). The genomic regions affecting flowering found in mash were also supported by QTL from an independent mapping population (Fig. 4C).

Identifying the environmental cues that are predictive of, or even correlated with, plant phenotypic responses remains a major challenge to studies interrogating gene action across many natural environments. Clearly, the photoperiod and cumulative GDD cues we identify here are functions of the genotypes measured, are not predictive, and capture only a minority of SNP effects on flowering. We know still less about the overwintering parameters that affect green-up, which is reflected in our lower ability to assign SNP effects on green-up to weather-based cues. More generally, it is difficult to predict the time scales over which individuals may integrate environmental cues, particularly in perennial species which may integrate these cues over longer time scales. Mash offers an opportunity to specify multiple environmental cues and compete them to explain patterns of SNP effects, allowing us to detect how important these cues are genome-wide, and how strongly each cue influences each SNP. This is a key development to further improve our understanding of genetic variation in GxE.

# Materials and Methods

Whenever possible, plant material will be shared upon request. Source data and code to replicate these analyses are available at: https://github.com/Alice-MacQueen/pvdiv-phenology-gxe.git. SNP data to replicate these analyses are available from the UT dataverse at https://doi.org/link.

### Genotype-by-environment effects on green-up and flowering as functions of weather-based cues

In 2019, we scored two phenological events every two days in two mapping populations of switchgrass, a diversity panel and a pseudo-F2 cross, planted at eight common garden locations (32, 34, 37). We scored green-up date as the day of the year when 50% of the tiller area of the crown of the plant cut the previous year had green growth. Flowering date was the day of the year when 50% of the plant tillers had panicles undergoing anthesis. We scored green-up and flowering as day of the year, then linked these dates to multiple weather-based environmental factors measured daily at each common garden (SI Appendix, Section S1, Table S1).

The formation and resequencing of the diversity panel has been described previously (32). The diversity panel contained 134 sequenced, clonally propagated individuals from the Midwest genetic subpopulation, and 229 from the Gulf genetic subpopulation. To allow for the possibility that different subpopulations had different strengths of connection between our phenotypes and genotypes (38), we conducted three sets of genetic analyses: on Gulf and Midwest genotypes separately, and on both subpopulations together ('Both'). Analyses to determine narrow-sense heritability (h2) for green-up and flowering were done using linear mixed models and followed (32). Details on these models can be found in (SI Appendix, Section S3,S4).

### Mapping major patterns of genotype-by-environment effects on green-up and flowering

To evaluate the prevalence and kinds of covariance patterns of SNP effects across our common gardens, we used multivariate adaptive shrinkage (mash) on SNP effect estimates from the diversity panel (35). Mash is a statistical method that allows estimation and comparison of many effects jointly across many different conditions; it improves on previous methods by allowing multiple, arbitrary correlations in effect sizes among conditions (SI Appendix, Section S5). To obtain SNP effect estimates, we first conducted univariate genome-wide association at each common garden for green-up and flowering date. We then analyzed SNP effects for the top 19K relatively unlinked (r2 \< 0.2) SNPs per condition using mash, as in (32). Details on these models can be found in (SI Appendix, Section S6).

We generated hypothesis-based covariance matrices derived from correlations in environmental cues in the green-up or flowering date windows for the three subpopulations (SI Appendix, Table S1, Section S1). These covariance matrices represent correlations between identical genotypes drawn from a specific population at pairs of common gardens; covariances near one mean that the population has a strong, positive linear relationship in individual responses at that pair of gardens, while covariances near zero mean that there is no relationship within the population for individual responses at that pair of gardens. Mash SNP effects will undergo strong shrinkage towards one another in the first case, and little shrinkage in the second case. Mash also generates data-driven covariance matrices corresponding to major patterns of SNP effects present in the data. We generated six data-driven matrices per mash run, five (denoted DD_PCA_1 through DD_PCA_5) produced by singular value decomposition (SVD) of an overall matrix, denoted 'DD_tPCA'. We used SVD to present vectors of garden-specific effects for each numbered DD_PCA matrix, because the first eigenvector of a SVD explains 100% of the variation in these matrices, and each value in this eigenvector represents one garden-specific effect.

Last, we characterized the overall patterns of antagonistic pleiotropy in the set of SNPs where there was pairwise significance of effects at pairs of gardens. To do this, we used the 'get_GxE' function of the switchgrassGWAS R package. First, this determines the set of SNPs with evidence of significant effects in both conditions for all pairs of conditions using local false sign rates (lfsr) as the significance criteria. Second, to determine antagonistic pleiotropy, this function determines if effects significant in both conditions are of opposite sign.

Using the lfsr rather than the local false discovery rate (lfdr) is a critical change in our ability to detect antagonistic pleiotropy. The lfdr, like other measures of FDR, focuses on if we have enough evidence to reject the null hypothesis that an effect j is 0 -- that there is a significant effect. Previous studies of antagonistic pleiotropy (e.g. (37)) have used the lfdr or equivalent statistical tests to detect antagonistic pleiotropy. These tests were conservative, in that they required two non-zero effects of different signs, while tests for differential sensitivity required only one non-zero effect. This previous work recognized that this testing bias could lead to undercounting occurrences of antagonistic pleiotropy (26, 27), and sought to reduce it by permutation (28). However, using the lfsr to test for antagonistic pleiotropy does not undercount occurrences of antagonistic pleiotropy, as this statistic answers a fundamentally different question. For each effect j, the lfsr_j is defined as the probability that we make an error in the sign of effect j if we were forced to declare the effect positive or negative (Stephens paper). Thus, rather than asking "Are these two effects different?" - as we reasonably expect two effects to be, even if this difference cannot be measured - the local false sign rate answers a more meaningful question: Can we be confident in the sign of this effect?

In addition, the get_GxE function also sets an arbitrary threshold to count an effect as changing in magnitude between environments, commonly known as differential sensitivity. For differential sensitivity, this function determines if effects significant in both conditions are of the same sign and of a magnitude (not tested for significance) that differs by a factor of 0.4 or more. The remaining effects that are significant in both conditions have the same effect sign and similar effect magnitudes and we denote these effects as having no GxE. The distinction between effects with different magnitudes is arbitrary but useful to fully characterize how effects vary across environments. and our use of the lfsr to determine significance and our specification that SNP effects must be significant in both conditions to be included means that our tests for antagonistic pleiotropy carry an equal statistical burden to those measuring differential sensitivity and effects without GxE.

### Confirmation of genotype-by-environment effects using an independent mapping population

To confirm candidate genomic regions and patterns of allelic effects found in the diversity panel, we analyzed flowering in an outbred pseudo-F2 cross between four individuals, two Midwest and two Gulf individuals. The formation of this mapping population has been described previously (34); additional details on QTL mapping can be found in SI Appendix, Section S6. To be directly comparable to the diversity panel data, only 2019 phenology data from the pseudo-F2 cross from the same eight common garden sites were used. To compare our QTL enrichments of significant mash associations to the null expectation, we used permutation to choose 1000 sets of 23 genomic regions of the same size randomly distributed throughout the genome, then calculated enrichments of the mash 1% tail in these random intervals.

# Figures and Tables

![Figure 1. Characterization of green-up and flowering dates from the switchgrass diversity panel. (a) Map and trait histograms of green-up and flowering dates across two genetically distinct switchgrass subpopulations and eight common gardens. Purple represents individuals from the Midwest genetic subpopulation, and pink individuals from the Gulf subpopulation. Vertical dashed lines indicate the summer solstice. Common gardens are arranged in latitudinal order. (b) Phenotypic correlations between clonal replicates planted at eight common gardens, within and between two genetic subpopulations.  (c) Narrow sense heritability of green-up and flowering within single common gardens (purple) and across all eight common gardens (green), within and between two genetic subpopulations. (d) Flow diagram of the methods applied to the green-up and flowering dates to jointly estimate SNP effects across all sites. Mash was fit to SNP effect data and used to find covariance matrices that improved the mash model likelihood using a large set of randomly selected, relatively unlinked SNP effects; this model was applied to a \"strong\" set of SNP effects with large effect sizes in the univariate GWAS.](images/Fig1_200dpi.png)

![Figure 2.](images/Figure_2_Covariance_matrices_and_posterior_weights.png)

## Author Affiliations {.unnumbered}

Include department, institution, and complete address, with the ZIP/postal code, for each author. Use lower case letters to match authors with institutions, as shown in the example. PNAS strongly encourages authors to supply an [ORCID identifier](https://orcid.org/) for each author. Individual authors must link their ORCID account to their PNAS account at [www.pnascentral.org](http://www.pnascentral.org/). For proper authentication, authors must provide their ORCID at submission and are not permitted to add ORCIDs on proofs.

## Manuscript Length {.unnumbered}

A standard 6-page article is approximately 4,000 words, 50 references, and 4 medium-size graphical elements (i.e., figures and tables). The preferred length of articles remains at 6 pages, but PNAS will allow articles up to a maximum of 12 pages.

## References {.unnumbered}

References should be cited in numerical order as they appear in text; this will be done automatically via bibtex, e.g. \cite{belkin2002using} and \cite{berard1994embedding,coifman2005geometric,phdthesis,masterthesis}. All references cited in the main text should be included in the main manuscript file.

## Data Archival {.unnumbered}

PNAS must be able to archive the data essential to a published article. Where such archiving is not possible, deposition of data in public databases, such as GenBank, ArrayExpress, Protein Data Bank, Unidata, and others outlined in the [Information for Authors](https://www.pnas.org/author-center/editorial-and-journal-policies#materials-and-data-availability), is acceptable.

## Figures and Tables {.unnumbered}

| **Weather variable**                                                      | **Time frame**                                               | **\# Covariance Matrices tested in greedy mash algorithm** |
|---------------------------|-----------------------|----------------------|
| cumulative GDD at 12C in the time frame                                   | 1, 2, 3, 4, 5, 6, 7, 14, 21, 28 days prior to green up date  | 60, 30 per phenotype and 10 per phenotype & genetic group  |
| cumulative rainfall in the time frame                                     | 1, 2, 3, 4, 5, 6, 7, 14, 21, 28 days prior to green up date  | 60, 30 per phenotype and 10 per phenotype &  genetic group |
| day length (hours) on a specific day indicated by the time frame          | 1, 2, 3, 4, 5, 6, 7, and 14 days prior to green up date      | 48, 24 per phenotype and 8 per phenotype &  genetic group  |
| day length change (seconds) on a specific day indicated by the time frame | 1, 2, 3, 4, 5, 6, 7, and 14 days prior to green up date      | 48, 24 per phenotype and 8 per phenotype & genetic group   |
| average temperature in the time frame                                     | 1, 2, 3, 4, 5, 6, 7, 14, 21, 28 days prior to green up date  | 60, 30 per phenotype and 10 per phenotype & genetic group  |

: Weather variables and time frames around the green up and flowering date used to construct the hypothesis-based covariance matrices. The correlations between values of these weather variables for genetically identical plants grown in different gardens were used to fill off-diagonal cells of the covariance matrices. Narrow-sense heritabilities for these values at each garden were used for the diagonal cells.

Figures can be included using Quarto's syntax, code chunks, or with LaTeX.

For simple figures, Quarto syntax is typically easiest. Full details are available in [the Quarto documentation](https://quarto.org/docs/authoring/figures.html).

![Placeholder image of a frog with a long example legend to show justification setting.](figs/frog.pdf){#fig-frog}

Tables can similarly be included using Quarto's syntax, code chunks, or with LaTeX. I tend to use code chunks, but LaTeX works very well for classic tables, like regression tables or summary statistics.

```{=tex}
\begin{table}[t!]
\centering
\caption{Comparison of the fitted potential energy surfaces and ab initio benchmark electronic energy calculations}
\begin{tabular}{lrrr}
Species & CBS & CV & G3 \\
\midrule
1. Acetaldehyde & 0.0 & 0.0 & 0.0 \\
2. Vinyl alcohol & 9.1 & 9.6 & 13.5 \\
3. Hydroxyethylidene & 50.8 & 51.2 & 54.0\\
\bottomrule
\end{tabular}

\addtabletext{nomenclature for the TSs refers to the numbered species in the table.}
\end{table}
```
### Digital Figures {.unnumbered}

EPS, high-resolution PDF, and PowerPoint are preferred formats for figures that will be used in the main manuscript. Authors may submit PRC or U3D files for 3D images; these must be accompanied by 2D representations in TIFF, EPS, or high-resolution PDF format. Color images must be in RGB (red, green, blue) mode. Include the font files for any text.

Images must be provided at final size, preferably 1 column width (8.7cm). Figures wider than 1 column should be sized to 11.4cm or 17.8cm wide. Numbers, letters, and symbols should be no smaller than 6 points (2mm) and no larger than 12 points (6mm) after reduction and must be consistent.

Figures and tables should be labelled and referenced in the standard way using the `\label{}` and `\ref{}` commands.

Figure @fig-frog shows an example of how to insert a column-wide figure. To insert a figure wider than one column, please use the `figure*` environment. Figures wider than one column should be sized to 11.4 cm or 17.8 cm wide. Use `SCfigure*` environment for a wide figure with side legends, such as @fig-side.

In Quarto, we can do these by setting the `fig-env` command to `figure*` or `SCfigure*`

```{r}
#| label: fig-side
#| fig-cap: "This legend would be placed at the side of the figure, rather than below it."
#| fig-env: "SCfigure*"
#| echo: false
# tell it the options as comments with a | and a space, as above.
# set echo: false to avoid printing this text
knitr::include_graphics('figs/frog.pdf')
```

## Tables {.unnumbered}

Tables should be included in the main manuscript file and should not be uploaded separately.

## Single column equations {.unnumbered}

Authors may use 1- or 2-column equations in their article, according to their preference.

To allow an equation to span both columns, use the `\begin{figure*}...\end{figure*}` environment mentioned above for figures.

Note that the use of the `widetext` environment for equations is not recommended, and should not be used.

```{=tex}
\begin{figure*}[bt!]
\begin{align*}
(x+y)^3&=(x+y)(x+y)^2\\
       &=(x+y)(x^2+2xy+y^2) \numberthis \label{eqn:example} \\
       &=x^3+3x^2y+3xy^3+x^3.
\end{align*}
\end{figure*}
```
# Material and Methods {.unnumbered}

## Supporting Information Appendix (SI) {.unnumbered}

Authors should submit SI as a single separate SI Appendix PDF file, combining all text, figures, tables, movie legends, and SI references. SI will be published as provided by the authors; it will not be edited or composed. Additional details can be found in the [PNAS Author Center](https://www.pnas.org/authors/submitting-your-manuscript#manuscript-formatting-guidelines). The PNAS Overleaf SI template can be found [here](https://www.overleaf.com/latex/templates/pnas-template-for-supplementary-information/wqfsfqwyjtsd). Refer to the SI Appendix in the manuscript at an appropriate point in the text. Number supporting figures and tables starting with S1, S2, etc.

Authors who place detailed materials and methods in an SI Appendix must provide sufficient detail in the main text methods to enable a reader to follow the logic of the procedures and results and also must reference the SI methods. If a paper is fundamentally a study of a new method or technique, then the methods must be described completely in the main text.

### SI Datasets {.unnumbered}

Supply .xlsx, .csv, .txt, .rtf, or .pdf files. This file type will be published in raw format and will not be edited or composed.

### SI Movies {.unnumbered}

Supply Audio Video Interleave (avi), Quicktime (mov), Windows Media (wmv), animated GIF (gif), or MPEG files. Movie legends should be included in the SI Appendix file. All movies should be submitted at the desired reproduction size and length. Movies should be no more than 10MB in size.

# References

\bibsplit[2]<!-- Use \bibsplit to split the references from the body of the text. Value "[2]" represents the number of reference in the left column (Note: Please avoid single column figures & tables on this page.) -->

::: {#refs}
:::
