---
title: Diverse Genotype-by-Weather Interactions in Switchgrass
pdf-engine: pdflatex
format:
  pnas-pdf:
    keep-tex: true  
  docx:
    highlight-style: github
    fig-width: 7
    fig-asp: 1.62
    execute:
      echo: false
      cache: false
author:
  - name: Alice H. MacQueen
    affiliations:
      - name: University of Texas at Austin
        id: a
        department: Department of Integrative Biology
        city: Austin
        state: TX
        postal-code: 78712
    email: alicem@uchicago.edu
    attributes:
      equal-contributor: true
      corresponding: true
  - name: Li Zhang
    affiliations:
      - ref: a
    attributes:
      equal-contributor: true
  - name: Samuel A. Smith
    affiliations:
        - ref: a
    attributes:
      equal-contributor: true
  - name: Jason Bonnette
    affiliations:
          - ref: a
  - name: Arvid R. Boe
    affiliations:
      - name: South Dakota State University
        id: b
        department: Department of Agronomy
        city: Brookings
        state: SD
        postal-code: 57006
  - name: Phillip A. Fay
    affiliations:
      - name: USDA-ARS
        id: c
        department: Grassland, Soil and Water Research Laboratory
        city: Temple
        state: TX
        postal-code: 76502
  - name: Felix B. Fritschi
    affiliations:
      - name: University of Missouri
        id: d
        department: Division of Plant Sciences
        city: Columbia
        state: MO
        postal-code: 65211
  - name: David B. Lowry
    affiliations:
      - name: Michigan State University
        id: e
        department: Department of Plant Biology
        city: East Lansing
        state: MI
        postal-code: 48824
  - name: Robert B. Mitchell
    affiliations:
      - name: USDA-ARS
        id: f
        department: Wheat, Sorghum, and Forage Research Unit
        city: Lincoln
        state: NE
        postal-code: 68583
  - name: Francis M. Rouquette Jr
    affiliations:
      - name: Texas A&M University
        id: g
        department: Texas A&M AgriLife Research and Extension Center
        city: Overton
        state: TX
        postal-code: 75684
  - name: Yanqi Wu
    affiliations:
      - name: Oklahoma State University
        id: h
        department: Department of Plant and Soil Sciences
        city: Stillwater
        state: OK
        postal-code: 74078
  - name: Arbel Harpak
    affiliations:
      - ref: a
  - name: Thomas E. Juenger
    affiliations:
      - ref: a
    attributes:
      corresponding: true
    email: tjuenger@utexas.edu
abstract: |
  The timing of vegetative and reproductive growth in plants ("phenological timings") depends on genetic effects, environmental (e.g., weather) cues, and possibly their interaction. Quantifying these GxWeather interactions can aid prediction and manipulation of phenological timings. Here, we map GxWeather effects on phenological timings in two highly divergent switchgrass (*Panicum virgatum*) populations using repeated plantings of cloned individuals from these populations at eight sites across the central United States. Some genetic effects covary with  interpretable weather-based cues. For example, in the Gulf population, 65% of genetic effects on the timing of vegetative growth covary with daylength 14 days prior to green-up date, and 33% of genetic effects on the timing of flowering covary with cumulative rainfall in the seven days prior to flowering. However, most variation in genetic effects across sites is site-specific and can not be attributed to variation in weather variables alone. We demonstrate that we can identify genetic variation with GxWeather and assign these loci to specific weather-based cues or other patterns. Breeding for particular alleles at the GxWeather loci we identified could change flowering responsiveness in a photoperiod and rainfall-specific way. More broadly, our approach proposes a refined, powerful characterization of genotype-by-environment interactions in any species phenotyped in multiple environments.
runningauthor: "MacQueen"
significance: |
  The timing of plant seasonal development (phenology) has major impacts on fitness because of the steep fitness cost of plant-environment mismatches. We infer the mixture of ways that genetic effects on phenological traits depend on plant environment, focusing on how effects covary with weather prior to the phenological event (GxWeather). Most effects do not have GxWeather, but a minority do. GxWeather is population-specific. The majority of effects on the timing of vegetative growth in the Gulf population have GxWeather: these effects differ in sign across environments and covary with a photoperiod cue two weeks prior. A minority of effects on flowering date in the Gulf and Midwest populations had GxWeather, covarying with a cumulative rainfall and a photoperiod cue, respectively.
author_contributions:  "T.E.J. designed research. D.B.L. contributed plant material and resources. J.B., D.B.L., and T.E.J. designed and executed field experiments. A.R.B., P.A.F., F.B.F., D.B.L., R.B.M., F.M.R., Y.W., and T.E.J. hosted field experiments. A.H.M., L.Z., and S.A.S. conducted statistical and computational analyses. The manuscript was written by A.H.M. with contributions from all authors."
conflict_of_interest: "The authors declare no conflicts of interest."
equal-contributor: "A.H.M. contributed equally to this work with L.Z. and S.A.S."
keywords:  [allele-by-environment effect variation, antagonistic pleiotropy, photoperiod, cumulative rainfall, genetic variation]
acknowledgements: | 
  We thank the Brackenridge Field laboratory, the Ladybird Johnson Wildflower Center, and the Juenger laboratory for support with plant care and propagation. This material is based upon work supported in part by the Great Lakes Bioenergy Research Center, U.S. Department of Energy, Office of Science, Office of Biological and Environmental Research under Award Numbers DE-SC0018409 and DE-FC02-07ER64494, the US Department of Energy Awards DESC0014156 to T.E.J., DE-SC0017883 to D.B.L, National Science Foundation PGRP Awards IOS0922457 and IOS1444533 to T.E.J, and the Long-term Ecological Research Program (DEB 1832042) at the Kellogg Biological Station.
bibliography: bibliography.bib 
csl: pnas.csl
fontsize: "9pt"
classoption:
 - twocolumn
 - twoside
---

\firstpage[17]{2}

Plant phenological timings are major components of plant fitness affected by multiple external environmental cues (e.g. degree of winter chilling, day length, temperature, and water availability) that signal existing or upcoming growing conditions [@bauerle_photoperiodic_2012; @andres2012genetic; @korner2010phenology]. Genetic responses to environmental cues determine the speed, timing, and energy apportioned to vegetative and reproductive growth and shape both the individual's lifespan and its lifetime production of viable seed. Day length (or photoperiod) is one of the most predictable environmental cues, and genetic sensitivity to photoperiod protects plants from potentially fatal consequences of phenological responses to temperature cues at the "wrong" time of year. However, the usefulness of specific environmental cues depends on both features of the environment, such as cue predictability and relevance, and the species' adaptive strategies @botero2015evolutionary. Species with wide natural distributions can have multiple distinct environmentally cued phenological responses: for example, populations of sunflower (*Helianthus annuus*) exhibit day-neutral, facultative short day, and facultative long-day flowering responses, which vary with their environments [@blackman2013interacting; @henry2014transitions]. Distinct genetic responses in different environments are known as genotype-by-environment interactions, or GxE.

Flowering time, or the transition from vegetative to reproductive growth, is a common subject of GxE research [@blackman2013interacting; @henry2014transitions; @aagren2017adaptive; @brachi2010linkage; @dittmar2014flowering; @li2018genomic; @romero2017study], a key output of selection driving adaptation to local environments [@andres2012genetic; @wadgymar2017identifying; @blumel2015flowering], and a selection target for crop improvement to adapt crops to local or future environments @jung2009flowering. Changing flowering responsiveness to photoperiod cues has allowed geographic range expansion and increased yields in several cereal species [@turner2005pseudo; @faure2012mutation; @hung2012zmcct; @zakhrabekova2012induced; @yang2013oself3] and other crops [@pin2012multifaceted; @weller2019parallel]. Recent statistical advances in studying phenological GxE have involved determining critical environmental indices before the phenological event occurs, such as photothermal time within a critical growth window @li2018genomic. However, most studies of flowering GxE focus on finding a single, best fitting form of genotype-environment covariance, despite the key expectation that different genetic subpopulations, and even different genomic regions, have likely evolved distinct patterns of GxE [@Weine2023.06.21.545998]. Additionally, despite theoretical predictions that local adaptation should involve antagonistic pleiotropy, or sign-changing GxE, at the level of individual loci [@levene1953genetic; @felsenstein1976theoretical; @kawecki2004conceptual; @hedrick1986genetic], previous empirical work has found limited evidence of antagonistic pleiotropy [@des2013genotype; @wadgymar2017identifying]. However, this work has been limited by a known statistical bias that reduced detection of genetic effects that differ in sign [@des2013genotype; @anderson2013genetic; @anderson2011evolutionary]. Thus, despite substantial interest in the frequencies of various forms of GxE, the prevalence of antagonistic pleiotropy relative to other forms of GxE remains unknown.

Previous research suggests that switchgrass phenological timings should have GxWeather and that these timings could differ by genetic subpopulation. Switchgrass is considered a short-day plant with reproductive development strongly linked to day of the year @mitchell1997predicting. However, as part of its wide environmental adaptation across the eastern half of North America, its photoperiodicity has been predicted to differ by plant latitude of origin [@parrish2005biology; @casler2004latitudinal]. We previously found divergent Midwest and Gulf genetic subpopulations of switchgrass with distinct sets of environmental adaptations, in that both populations had distinct genetic variation associated with each of two fitness proxies, biomass and overwinter survival @lovell2021genomic. The Midwest genetic subpopulation is primarily composed of individuals from the well-studied upland switchgrass ecotype [@porter1966analysis; @milano2016genetic], while the Gulf subpopulation has individuals from the well-studied lowland ecotype and the phenotypically intermediate coastal ecotype @lovell2021genomic.

Here, we estimate GxWeather for two phenological timings in switchgrass by assigning patterns of genetic effects on phenology across gardens to many patterns of weather covariance at these gardens. To do this, we phenotype a diversity panel of hundreds of switchgrass genotypes from the Midwest and Gulf subpopulations for the timing of vegetative and reproductive development at eight common garden locations spanning 17 degrees of latitude. These gardens cover the majority of the latitudinal and climatic range of switchgrass and capture the most comprehensive picture to date of the environmental variation this species encounters. We define multiple ways phenological timings might covary with weather (Table 1) and additional ways phenological timings might vary by site (SI Appendix, Table S1, Section SX), then jointly re-estimate genetic effects on these timings at all eight sites using the set of these covariance matrices that significantly improved the modeled log-likelihood when included (SI Appendix, Section SX) @urbut2019flexible. We use the Bayesian framework *mash* (multivariate adaptive shrinkage) developed by @urbut2019flexible, to refine effect size estimates from genome-wide association (GWAS) conducted on individuals at each of the eight sites. *mash* allows us to identify and specify multiple covariance structures among genetic effect estimates across sites, including structures that represent covariance in weather variables of interest. Importantly, this method circumvents statistical biases in detecting genetic effects with the same or opposite signs @10.1093/biostatistics/kxw041. To confirm our genetic mapping of GxWeather, we compare our posterior genetic effect estimates from *mash* to mapping results from an outbred pseudo-F2 cross grown at the same sites. Our analyses allow us to describe the weather cues and genetic variation affecting phenology in two divergent natural populations of switchgrass.

# Results

Genotypes from the Gulf and Midwest subpopulations had distinct phenological timings and distinct patterns of phenological correlations across our eight common garden sites (@fig-map). At the three Texas common gardens (hereafter 'Texas' gardens), located within the natural range of the Gulf subpopulation, Gulf vegetative growth (hereafter 'green-up') occurred before Midwestern green-up, and Gulf reproductive growth (hereafter 'flowering') occurred after Midwestern flowering (@fig-map A). At the four northernmost common gardens (hereafter 'North' gardens), located within the natural range of the Midwest subpopulation, both Gulf green-up and flowering occurred after Midwest green-up and flowering. At the Oklahoma common garden, located near the natural range limits of both the Gulf and the Midwest subpopulations, Gulf and Midwest green-up occurred over the same period, and Gulf flowering occurred after Midwestern flowering (@fig-map A). These patterns led to strong negative phenotypic correlations for the timing of the start of vegetative growth between the North and Texas gardens, particularly in the Gulf and across all individuals from the Gulf and Midwest subpopulations (hereafter, 'Both' subpopulations), and contributed to positive phenotypic correlations for the timing of flowering that had larger magnitudes at more northern gardens (@fig-map B).

Narrow-sense heritabilities (h^2^) suggested that rank-changing GxE for these phenotypes was present across the common gardens (@fig-map C). h^2^ were typically high at individual gardens: 59% on average for green-up date, and 87% for flowering date. However, h^2^ were variable across gardens, and green-up dates were uncorrelated (r^2^ \< 0.2) or negatively correlated between pairs of gardens (@fig-map B). These negative and small correlations undoubtedly contributed to the low h^2^ values for green-up and flowering date when estimated jointly at all eight gardens: h^2^ was 0.8% for green-up and 23.2% for flowering date.

### Inference of GxWeather effects for vegetative and reproductive timing using *mash*

We next looked for evidence of GxWeather by jointly re-estimating genetic effects across all eight common gardens using the *mash* model, which can flexibly capture many modes of covariance across gardens (@fig-map D; SI Appendix, Datasets 1-6). We specified three qualitative categories of covariance structures across gardens: "canonical" covariance, with simple patterns of effect size covariance introduced in the initial *mash* manuscript; "data-driven" covariances derived from common patterns of SNP effects observed in the data, and "GxWeather" covariance, estimated from the covariance of empirical weather patterns at each garden at specific times before the phenological event (Table 1; SI Appendix, Section S2). For example, the more similar that the amount of rainfall prior to flowering is for genotypes A and B between gardens in MI and MO, the nearer these covariances will be to one. If rainfall amounts are uncorrelated, covariance will be near zero; if there is rank changing (e.g. high rainfall before flowering of A at MO but not MI, and high rainfall before flowering for B at MI but not MO), this covariance will be near negative one. We specified multiple covariance structures for the canonical, data-driven, and GxWeather categories, each of which represented a specific pattern of covariance in genetic effect size estimates that could be included in the model. The phenotypic correlations for the timing of vegetative growth had moderate negative correlations between the Texas and North gardens, particularly in the Gulf subpopulation and Both subpopulations (@fig-map B). If these phenotypic correlations have a genetic component, they could be partially or completely captured by the covariance structures specified in the *mash* model.

The GxWeather covariance structures allow hypothesis testing of specific weather variables as cues for the start of vegetative and reproductive growth. Say that a SNP in FLC controls flowering in a photoperiod-dependent manner. In that case, the joint estimate of effects for that SNP could have a high mixture proportion, or mass, on a covariance matrix created using a photoperiod-based environmental cue, such as day length at some interval prior to flowering. In our data, we would infer that the effect of that SNP on flowering was caused by a response to the environmental cue used to construct the GxWeather covariance structure with the largest mass. Overall, loadings of genetic effects on these GxWeather matrices in the model, along with the posterior effect size estimates, provide information on genome-wide patterns of SNP-environment interaction.

We used a greedy algorithm to select covariance matrices from each model's set that significantly improved the model likelihood. Five GxWeather covariance structures were selected, zero to two per subpopulation: two for green-up date, and three for flowering date. Of these five, three had mass on them in *mash* models of the strong effects (@fig-covar A-B). Two of these three matrices had negative covariances between sites between Texas and North gardens (@fig-covar A), while one had all positive or near-zero covariances. SNP-associated phenotypic effects covaried with different weather-based cues in the Gulf & in Both subpopulations (@fig-covar B). In total, 65% of the posterior weight of strong SNP effects in the *mash* model of Gulf green-up fell on a covariance matrix constructed using the covariance of day length 14 days prior to the date of green-up. The covariance matrix for this weather cue was visually similar to the observed pattern of phenotypic correlation for the timing of vegetative growth in the Gulf subpopulation (@fig-map B; @fig-covar A). *Mash* models of the timing of vegetative growth in the Midwest subpopulation and Both subpopulations did not include this GxWeather covariance; the Midwest had no weight on any GxWeather covariance, while Both subpopulations had non-zero weights on two additional GxWeather covariance types, average temperature one day prior to green-up, and the day length change in seconds in the day prior to green-up (@fig-covar B). The average temperature covariance matrix had negative covariances between Texas and North gardens, though not as strong as the negative phenotypic correlations seen in Both subpopulations (@fig-map B; @fig-covar A). Only the Gulf subpopulation and Both subpopulations had mass on any GxWeather covariance matrices (@fig-covar C) for vegetative growth.

For flowering date, distinct GxWeather covariance structures captured covariance in effect sizes for SNPs in the Gulf and Midwest subpopulations. 33% of SNP effects on flowering in the Gulf subpopulation covaried with cumulative rainfall in the seven days prior to flowering (@fig-covar D). 22.6% of SNP effects on flowering in the Midwest subpopulation covaried with day length change in the two days prior to flowering (@fig-covar D), which had negative covariances between Texas and North gardens. Neither of these covariance matrices significantly improved the log-likelihood of the *mash* model of Both subpopulations, and no GxWeather covariance matrices had non-zero mass in models of Both subpopulations (@fig-covar E). In five of the six *mash* models of strong effects, the GxWeather covariance matrices captured a minority of the posterior weights of the strong effects (@fig-covar C,E); in these five models, the majority of this mass was on various canonical covariance matrices. These matrices included simple heterozygosity, with intermediate, positive covariances between all gardens, and garden-specific effects present only at one garden.

We next characterized the pairwise patterns of effects where we were confident in the sign of the effect at both gardens. If the overall pattern of phenotypic expression differs between a pair of gardens, this difference may be comprised of SNP effects that differ in sign between these gardens (effects with antagonistic pleiotropy), SNP effects that differ in magnitude (effects that are large in one garden or region and smaller in others), and SNP effects that are indistinguishable in the two gardens (@fig-effects A). We used the local false sign rate (lfsr), an analogue of the lfdr that establishes confidence in the effect sign, not the effect's difference from zero, to determine significance. We required lfsr significance (p \< 0.05) in both gardens to include effects. This means that our tests for antagonistic pleiotropy, or a sign change between conditions, carry an equal statistical burden to those for effects with the same sign.

For green-up date for the Gulf subpopulation, hundreds to thousands of pairwise effects exhibited antagonistic pleiotropy, or a difference in effect sign, between pairs of Texas and North gardens (@fig-effects B; SI Appendix, Fig SX). 78.7% of pairwise comparisons between North and Texas gardens had a difference in sign, while only 28.6% and 0.2% of North-North or Texas-Texas comparisons had a difference in sign, respectively (@fig-effects B). The majority of pairwise effects for greenup for the Midwest (\>55%) and Both (\>85%) subpopulations were the same sign, and effects most often differed in magnitude between gardens within and between the regions (@fig-effects B).

For flowering date for the Gulf subpopulation, less than 2% of pairwise effects exhibited antagonistic pleiotropy, or a difference in effect sign, within or between regions (@fig-effects B). More effects differed in magnitude between the Texas and North regions than within these regions (42.7% vs \<20%; @fig-effects B), while the majority of effects had no GxE. The Midwest population had relatively few significant effects for flowering, but a large proportion of these differed in sign between Texas and North regions (42.7%) or within the North region (65.4%). Finally, in Both subpopulations, less than 20% of pairwise effects differed in sign (@fig-effects B). Most differences in sign were between TX1, the southernmost garden, and all other gardens (SI Appendix, Fig SX). Similarly, more effects that differed in magnitude included gardens in the Texas region (52.3-55.9%), and most effect pairs in the North region were not distinguishable (91.5%).

### Confirmation of effects on phenology using an independent mapping population

We sought additional experimental support for our joint re-estimates of SNP effects using an independent pseudo-F2 mapping population created from Gulf & Midwest individuals and grown at the same sites (@fig-qtl A,B). We conducted quantitative trait loci (QTL) mapping of flowering as functions of five environmental cues that we also used as covariance matrices in *mash*, and identified eight QTL for flowering date, eight QTL for flowering as a function of day length change two days prior, one QTL for the start of vegetative growth, and two QTL for vegetative growth as a function of daylength change one day prior, all of which showed QTL by environment interactions (SI Appendix, Fig. S3). All QTL for flowering overlapped one or more homologs from rice or *A. thaliana* with functionally validated roles in flowering (SI Appendix, Dataset 7). All flowering and green-up QTL intervals contained at least one SNP significant in at least one *mash* run at a log10-transformed Bayes Factor \> 2, or in the 1% tail of significance, whichever was stricter (SI Appendix, Dataset 8). We also looked for enrichments of *mash* SNPs in the 1% tail of significance (the '*mash* 1% tail') within each QTL interval. At the 5% level, three QTL had enrichments of SNPs in the *mash* 1% tail. Overall, there were five significant enrichments (p \< 0.05, hypergeometric test) of SNPs in the *mash* 1% tail in the QTL intervals. Thus, we were able to experimentally support some of our re-estimates of SNP effects with a QTL mapping experiment using a separate mapping population.

# Discussion

As the climate and the natural environment change, it is increasingly critical to understand how patterns of plant-environment interactions will change in response. To do this, we must understand the current patterns of trait covariation across environments, the genetic underpinnings of these patterns, and the cases where this covariation can be altered. Here, we demonstrate that we can associate multiple patterns of GxWeather with specific genomic regions using a switchgrass diversity panel grown at eight common gardens. We can assign genetic effects to both GxWeather patterns with interpretable weather-based cues, and to agnostic, site-based patterns. We use this approach to study GxWeather for the timings of vegetative and reproductive development in the deeply genetically diverged Gulf and Midwest subpopulations of switchgrass.

Our analysis of the timing of vegetative growth in the Gulf and in Both subpopulations revealed substantial antagonistic pleiotropy in effects between the Texas and North gardens (@fig-effects B). This result supports theoretical models that local adaptation should involve antagonistic pleiotropy at the level of individual loci [@kawecki2004conceptual; @felsenstein1976theoretical; @levene1953genetic; @hedrick1986genetic], and is the first experimental work using GWAS across common gardens to find antagonistic pleiotropy to be common in small genomic regions [@wadgymar2017identifying; @savolainen2013ecological; @lowry2019qtl].

Our analysis of the timing of flowering showed that the Gulf and Midwest subpopulations have distinct GxWeather: flowering timing in the Midwest subpopulation has photoperiod-related variation, in that flowering timing covaries with a day length change signal two days before flowering occurs. In contrast, the Gulf subpopulation does not have variation in flowering based on a photoperiod cue. Instead, the Gulf subpopulation has variation in flowering that covaries with the rainfall that occurs in the week prior to flowering. Three genomic regions affecting flowering that we re-estimated across all eight sites were also supported by QTL from an independent mapping population at these sites (@fig-qtl B).

Identifying the environmental cues that are predictive of, or even correlated with, plant phenotypic responses remains a major challenge to studies interrogating gene action across many natural environments. The GxWeather photoperiod and cumulative rainfall cues we identify here are functions of the genotypes measured and capture only a minority of SNP effects on flowering. We could only assign SNP effects to a GxWeather covariance structures in four of the six phenotype & genetic subpopulations we modeled. It's likely we did not include some GxWeather important to these phenological cues - for example, overwintering parameters that might cause variation in the start of vegetative growth. More generally, it is difficult to predict the time scales over which individuals may integrate environmental cues, particularly in perennial species which may integrate these cues over longer time scales. If this integration time itself varies between individuals, the covariance structures we modeled cannot reflect this, though these structures would likely be highly correlated with GxWeather structures we did include. Our approach offers an opportunity to specify multiple environmental cues and compete them to explain patterns of genetic effects, allowing us to detect how important these cues are genome-wide, and how strongly each cue influences each SNP. This is a key development to further improve our understanding of genetic variation in GxE.

# Figures and Tables

```{=tex}
\clearpage
\onecolumn
\begin{table}[t!]
\centering
\caption{Weather variables and time frames prior to the start of vegetative and reproductive growth used to construct the hypothesis-based covariance matrices. Covariance matrices were constructed for, and tested on, three subpopulations and two phenological traits. The correlations between values of these weather variables for genetically identical plants grown in different gardens were used to fill off-diagonal cells of the covariance matrices. Narrow-sense heritabilities for these values at each garden were used for the diagonal cells.}
\begin{tabular}{lrr}
Weather variable & Time Frame & Covariance Matrices \\
\midrule
1. cumulative growing degree days at 12C in the time frame & (1-7), 14, 21, 28 days prior to the phenological date & 60 \\
2. cumulative rainfall in the time frame & (1-7), 14, 21, 28 days prior to the phenological date & 60  \\
3. day length (hours) on a specific day indicated by the time frame & (1-7) and 14 days prior to the phenological date & 48 \\
4. day length change (seconds) on a specific day indicated by the time frame & (1-7) and 14 days prior to the phenological date & 48 \\
5. average temperature in the time frame & (1-7), 14, 21, 28 days prior to the phenological date & 60 \\
\bottomrule
\end{tabular}

\end{table}
\clearpage
\twocolumn
```
![Characterization of the timing of vegetative (green-up) and reproductive (flowering) growth in the switchgrass diversity panel. (a) Map and trait histograms of green-up and flowering dates across two genetically distinct switchgrass subpopulations and eight common gardens. Purple represents individuals from the Midwest genetic subpopulation, and pink individuals from the Gulf subpopulation; map positions represent the original collection locations for the genotypes, and shapes represent the ecotype of the genotype. Vertical dashed lines indicate the summer solstice. Common gardens are arranged in latitudinal order. (b) Phenotypic correlations between clonal replicates planted at eight common gardens, within and between two genetic subpopulations. (c) Narrow sense heritability of green-up and flowering within single common gardens (purple) and across all eight common gardens (green), within and between two genetic subpopulations. (d) Flow diagram of the methods applied to the green-up and flowering dates to jointly estimate SNP effects across all sites. *Mash* was fit to SNP effect data and used to find covariance matrices that improved the *mash* model likelihood using a large set of randomly selected, relatively unlinked SNP effects; this model was applied to a "strong" set of SNP effects with large effect sizes in the univariate GWAS.](images/Fig1_200dpi.png){#fig-map}

![Example hypothesis-driven covariance matrices specified in *mash* and the posterior weights placed on all covariance matrices. (a) Left column: Five example hypothesized covariance matrices specified for the green-up date or flowering date phenotype; these matrices were created from environment-specific correlations across eight common gardens, and are described in Table 1. Common gardens are arranged in latitudinal order within the matrices. Right column: Five example canonical covariance matrices. Canonical matrices (purple) have simple interpretations, such as equal effects across all common gardens, or effects specific to a single common garden. (b,d) Total posterior weight placed on each covariance matrix type specified for (b) green-up date and (d) flowering date *mash* models, within and between two genetic subpopulations. Hypothesized covariance matrices (green). Covariance matrices included in *mash* that had zero posterior weight in all three *mash* runs on the genetic subpopulations, such as the identity matrix, are not shown. (c,e) Total posterior weight placed on covariance matrices that were hypothesized or canonical, for the (c) green-up date phenotype and (e) flowering date phenotype.](images/Figure_2_recolor_covariance_matrices_and_posterior_weights.png){#fig-covar}

![Types of GxE present between of pairs of jointly re-estimated effects in eight common gardens, for effects with lfsr \< 0.05. a) Examples of effect patterns at three pairs of sites with three types of GxE. Sign: Effects that differ in sign at these pairs of gardens (p \< 0.05, lfsr). Magnitude: Effects identical in sign (p \< 0.05, lfsr) that differ in magnitude by a factor of \>0.4. None: Effects not distinguishable by magnitude nor sign of the effect. b) The fraction of effects with each GxE type for the start of vegetative growth (Greenup Date) and reproductive growth (Flowering Date), within and between two genetic subpopulations. Common gardens are grouped by the larger region they came from: North gardens are within the natural range of the Midwest subpopulation, while Texas gardens are within the natural range of the Gulf subpopulation.](images/Figure_3_GxE.png){#fig-effects}

![Overlaps of QTL from an outbred pseudo-F2 cross and with jointly re-estimated SNP effects in the 1% tail of significance from a diversity panel. Dotted lines indicate permutation-based significance thresholds for each weather-related function. Stars indicate QTL with significant enrichment for SNPs in the 1% *mash* tail; G, M, and B indicate which subpopulation had enrichment: G - Gulf subpopulation, M -Midwest subpopulation, B - both subpopulations. Rug plots show genomic locations of SNPs in the 1% *mash* tail for flowering date for each subpopulation. a) QTL mapping for the start of vegetative growth (Greenup Date), and three weather-related functions of greenup date. b) QTL mapping for the start of reproductive growth (Flowering Date), and two weather-related functions of flowering date.](images/Figure_4_QTL_Overlaps.png){#fig-qtl}

# Materials and Methods

Whenever possible, plant material will be shared upon request. Source data and code to replicate these analyses are available at: https://github.com/Alice-MacQueen/pvdiv-phenology-gxe.git. SNP data to replicate these analyses are available from the UT dataverse at https://doi.org/link.

### Genotype-by-environment effects on green-up and flowering as functions of weather-based cues

In 2019, we scored two phenological events every two days in two mapping populations of switchgrass, a diversity panel and a pseudo-F2 cross, planted at eight common garden locations [@lovell2021; @milano2016genetic; @lowry2019qtl]. We scored the start of vegetative growth, or green-up date, as the day of the year when 50% of the tiller area of the crown of the plant cut the previous year had green growth. The start of reproductive growth, or flowering date, was the day of the year when 50% of the plant tillers had panicles undergoing anthesis. We scored green-up and flowering as day of the year, then linked these dates to multiple weather-based environmental factors measured daily at each common garden (SI Appendix, Section S1, Table S1).

The formation and resequencing of the diversity panel has been described previously @lovell2021genomic. The diversity panel contained 134 sequenced, clonally propagated individuals from the Midwest genetic subpopulation, and 229 from the Gulf genetic subpopulation. To allow for the possibility that different subpopulations had different strengths of connection between our phenotypes and genotypes [@kortefarlow2013], we conducted three sets of genetic analyses: on Gulf and Midwest genotypes separately, and on both subpopulations together ('Both' subpopulations). Analyses to determine narrow-sense heritability (h^2^) for green-up and flowering were done using linear mixed models and followed @lovell2021genomic. Details on these models can be found in (SI Appendix, Section S2).

### Mapping major patterns of genotype-by-environment effects on green-up and flowering

To evaluate the prevalence and kinds of covariance patterns of SNP effects across our common gardens, we used multivariate adaptive shrinkage (*mash*) on SNP effect estimates from the diversity panel [@urbut2019flexible]. To do this, we first conducted univarate genome-wide association (GWAS) on site-specific best linear unbiased predictors (BLUPs) for the timing of vegetative growth and flowering, using the switchgrassGWAS package and the methods in [@lovell2021]. We used the effect estimates for single nucleotide polymorphisms (SNPs) from these GWAS as preliminary effect estimates for the two traits at each garden. We then jointly modeled genetic effect estimates, and the mixture of ways these effects might covary, across all eight common gardens, using a two-part procedure. We first modeled genetic effects with different combinations of covariance structures, using the effect size estimates from a subset of [19K]{.underline} relatively unlinked (r^2^ \< 0.2), randomly selected SNPs used in the GWAS. In these models (hereafter referred to as "random"), we then specified a variety of potential covariance structures among genetic effect size estimates from each garden (as expanded on below) and used a greedy algorithm to iteratively select and add covariance structures that significantly improved the log likelihood of the *mash* model. Details on these models can be found in (SI Appendix, Section S3). Given the high correlation between some of the covariance patterns we tested, this approach allowed us to select only the subset of patterns of covariance that significantly improved the fit of the jointly inferred structure of genetic effect estimates present in the data. We then selected 19K relatively unlinked (r^2^ \< 0.2) SNPs from each univariate GWAS that had the largest effect estimates in each GWAS (hereafter "strong" effects), which gave sets of [**48K - 80K**]{.underline} SNPs, and jointly re-estimated these effect size estimates at all eight sites using the set of significant covariance matrices that most improved the model fit on the random effects. Notably, the majority of these "strong" effects were not significant in univariate GWAS.

We generated hypothesis-based covariance matrices derived from correlations in environmental cues in the green-up or flowering date windows for the three populations (Table 1; SI Appendix, Table S1, Section S1). These covariance matrices represent correlations between identical genotypes drawn from a specific population at pairs of common gardens; covariances near one mean that the population has a strong, positive linear relationship in individual responses at that pair of gardens, while covariances near zero mean that there is no relationship within the population for individual responses at that pair of gardens. *Mash* SNP effects will undergo strong shrinkage towards one another in the first case, and little shrinkage in the second case. *Mash* also generates data-driven covariance matrices corresponding to major patterns of SNP effects present in the data. We generated six data-driven matrices per *mash* run, five produced by singular value decomposition (SVD) of an overall matrix.

Last, we characterized the overall patterns of antagonistic pleiotropy in the set of SNPs where there was pairwise significance of effects at pairs of gardens. To do this, we used the 'get_GxE' function of the switchgrassGWAS R package. First, this determines the set of SNPs with evidence of significant effects in both conditions for all pairs of conditions using local false sign rates (lfsr) as the significance criteria. Second, to determine antagonistic pleiotropy, this function determines if effects significant in both conditions are of opposite sign.

Using the lfsr rather than the local false discovery rate (lfdr) is a critical change in our ability to detect antagonistic pleiotropy. The lfdr, like other measures of FDR, focuses on if we have enough evidence to reject the null hypothesis that an effect j is 0, or that there is a significant effect. Previous studies of antagonistic pleiotropy (e.g. [@lowry2019qtl]) have used the lfdr or equivalent statistical tests to detect antagonistic pleiotropy. These tests were conservative, in that they required two non-zero effects of different signs, while tests for differential sensitivity required only one non-zero effect. This previous work recognized that this testing bias could lead to undercounting occurrences of antagonistic pleiotropy [@anderson2011evolutionary; @des2013genotype], and sought to reduce it by permutation [@anderson2013genetic]. However, using the lfsr to test for antagonistic pleiotropy does not undercount occurrences of antagonistic pleiotropy, as this statistic answers a fundamentally different question. For each effect j, the lfsr~j~ is defined as the probability that we make an error in the sign of effect j if we were forced to declare the effect positive or negative @10.1093/biostatistics/kxw041. Thus, rather than asking "Are these two effects different?" - as we reasonably expect two effects to be, even if this difference cannot be measured - the local false sign rate answers a more meaningful question: Can we be confident in the sign of this effect?

In addition, the get_GxE function also sets an arbitrary threshold to count an effect as changing in magnitude between environments, commonly known as differential sensitivity or a change in amplitude of the effect. For differential sensitivity, this function determines if effects significant in both conditions are of the same sign and of a magnitude (not tested for significance) that differs by a factor of 0.4 or more. The remaining effects that are significant in both conditions have the same effect sign and similar effect magnitudes and we denote these effects as having no GxE. The distinction between effects with different magnitudes is arbitrary but useful to fully characterize how effects vary across environments to ultimately influence phenotypes. Our use of the lfsr to determine significance and our specification that SNP effects must be significant in both conditions to be included means that our tests for antagonistic pleiotropy carry an equal statistical burden to those measuring differential sensitivity and effects without GxE.

### Confirmation of genotype-by-environment effects using an independent mapping population

To confirm candidate genomic regions and patterns of allelic effects found in the diversity panel, we analyzed flowering in an outbred pseudo-F2 cross between four individuals, two Midwest and two Gulf individuals. The formation of this mapping population has been described previously [@milano2016genetic]; additional details on QTL mapping can be found in SI Appendix, Section S4. To be directly comparable to the diversity panel data, only 2019 phenology data from the pseudo-F2 cross from the same eight common garden sites were used.

# References

\bibsplit[2]<!-- Use \bibsplit to split the references from the body of the text. Value "[2]" represents the number of reference in the left column (Note: Please avoid single column figures & tables on this page.) -->

::: {#refs}
:::
