---
title: Diverse Genotype-by-Weather Interactions in Switchgrass
pdf-engine: pdflatex
format:
  pnas-si-pdf:
    keep-tex: false  
author:
  - name: Alice H. MacQueen
    affiliations:
      - name: University of Texas at Austin
        id: a
        department: Department of Integrative Biology
        city: Austin
        state: TX
        postal-code: 78712
    email: alicem@uchicago.edu
    attributes:
      equal-contributor: true
      corresponding: true
  - name: Li Zhang
    affiliations:
      - ref: a
    attributes:
      equal-contributor: true
  - name: Samuel A. Smith
    affiliations:
        - ref: a
    attributes:
      equal-contributor: true
  - name: Jason Bonnette
    affiliations:
          - ref: a
  - name: Arvid R. Boe
    affiliations:
      - name: South Dakota State University
        id: b
        department: Department of Agronomy
        city: Brookings
        state: SD
        postal-code: 57006
  - name: Phillip A. Fay
    affiliations:
      - name: USDA-ARS
        id: c
        department: Grassland, Soil and Water Research Laboratory
        city: Temple
        state: TX
        postal-code: 76502
  - name: Felix B. Fritschi
    affiliations:
      - name: University of Missouri
        id: d
        department: Division of Plant Sciences
        city: Columbia
        state: MO
        postal-code: 65211
  - name: David B. Lowry
    affiliations:
      - name: Michigan State University
        id: e
        department: Department of Plant Biology
        city: East Lansing
        state: MI
        postal-code: 48824
  - name: Robert B. Mitchell
    affiliations:
      - name: USDA-ARS
        id: f
        department: Wheat, Sorghum, and Forage Research Unit
        city: Lincoln
        state: NE
        postal-code: 68583
  - name: Francis M. Rouquette Jr
    affiliations:
      - name: Texas A&M University
        id: g
        department: Texas A&M AgriLife Research and Extension Center
        city: Overton
        state: TX
        postal-code: 75684
  - name: Yanqi Wu
    affiliations:
      - name: Oklahoma State University
        id: h
        department: Department of Plant and Soil Sciences
        city: Stillwater
        state: OK
        postal-code: 74078
  - name: Arbel Harpak
    affiliations:
      - ref: a
  - name: Thomas E. Juenger
    affiliations:
      - ref: a
    attributes:
      corresponding: true
    email: tjuenger@utexas.edu
bibliography: bibliography.bib  
fontsize: "9pt"
classoption:
 - twoside
---

\noindent This PNAS template for Supporting Information (SI) may be used to organize your supporting material. The template is intended to provide a clearly organized PDF file that will ensure readers can easily navigate to sections or specific figures and tables. Movie files or large datasets can be presented as separate files. Further information is available in the [PNAS Author Center](https://www.pnas.org/authors/submitting-your-manuscript).

## Using the template {.unnumbered}

Specify the title, author list, and corresponding authors with the same YAML specification as in the [corresponding PNAS Quarto template](https://github.com/christopherkenny/pnas). The cover page will be automatically generated with the relevant description of the SI.

Figures should be placed on separate pages with legends set immediately below each figure. Table titles should be set immediately above each table. Note that tables extending beyond the width of the page can be included in the PDF or provided as separate dataset files. Oversized/nonstandard page sizes are accepted as part of your SI Appendix file.

References cited in the SI text should be included in a separate reference list at the end of this SI file: @varga2016multilingual and @olsen1992optimal.

Supporting information for Brief Reports is limited to extended methods, essential supporting datasets, and videos (no additional tables or figures). Supporting figures and tables are not allowed for Brief Reports.

## Submitting SI {.unnumbered}

Save your completed SI file as a PDF for submission. Further submission instructions are available [here](https://www.pnas.org/page/authors/submission#preparation).

## Subhead {.unnumbered}

Type or paste text here. This should be additional explanatory text such as an extended technical description of results, full details of mathematical models, etc.

# Supplementary Methods

## Section S1. Covariance matrices generated to jointly re-estimate SNP effects across eight sites {.unnumbered}

A trait measured in two environments can be considered two traits that are genetically correlated [@falconer1952]. By extension, a phenological trait measured at eight common gardens can be modeled as eight genetically correlated traits represented by a covariance matrix, where diagonal entries represent narrow-sense heritability at a common garden, or the additive genetic variance present in the population at that garden, and non-diagonal entries represent the genetic correlation between the trait in a pair of environments.

We were interested in the types of genetic correlation that could be expressed by different alleles across the genome and reasoned that different loci could have different patterns of genetic correlation. For example, one locus's effects could covary with a weather-based cues, if it had an allele that was responsive to this cue. Alternatively, a locus might have an effect at only one garden, for example if a herbivore or pathogen was only present at one garden.

We specified three qualitative categories of covariance structures across gardens: "canonical" covariance, with simple patterns of effect size covariance introduced in the initial *mash* manuscript; "data-driven" covariances derived from common patterns of SNP effects observed in the data and introduced in the initial *mash* manuscript, and "GxWeather" covariance, estimated from the covariance of empirical weather patterns at each garden over specific time frames before the phenological event.

Canonical covariance matrices were created by *mash* and fell into four groups: an identity matrix, an equal effects matrix, singleton matrices, and simple heterozygousity matrices. The identity matrix has values of zero for covariance between different gardens and correlations of one within gardens, and represents the situation in which the effects in different conditions are independent. The equal effects matrix has values of one for all matrix elements, and represents identical effects across all conditions. Each singleton matrix has a value of one for a single garden and zeros for all other elements, representing a garden-specific effect. Three simple heterozygousity matrices are generated, which have covariances of 0.25, 0.5, or 0.75 for all off-diagonal matrix elements, and correlations of one for diagonal entries, representing moderately positively correlated effects across all sites.

Data-driven covariance matrices can also be created by *mash* and fall into two groups: an overall pattern of covariance between all conditions, denoted as $tPCA$, and the first five principle components of a singular value decomposition of the $tPCA$, denoted as `PCA_1`$PCA_1$ through $PCA_5$[@Urbut2019].

To create the GxWeather covariance matrices we transformed each genotype's green-up and flowering date at each garden to create date-related values that were summary functions for a weather variable across a specified date range. For example, the cumulative sum of the rainfall that occurred seven days prior to the flowering date was used to create one possible GxWeather cue variable for flowering date. The code for these calculations can be found at git\@github.com:Alice-MacQueen/pvdiv-phenology-gxe/R/Analysis_v1.2_mash_using_greedy_covar.qmd, starting at line 208.

We defined both phenological dates as functions of five weather variables and eight to ten time frames (Table 1): Cumulative growing degree days (GDD) variables for each time frame were calculated as $GDD = \sum^{DATE}_{DATE - i} max(T_{mean} - T_{base}, 0)$, where $T_{mean}$ is the daily average temperature, defined as $(T_{max} + T_{min}) / 2$, $T_{base}$ is the base temperature of 12Â°C for switchgrass, $T_{max}$ is the maximum daily temperature, $T_{min}$ is the minimum daily temperature, $DATE$ is the phenological date, and $i$ is the time frame, or the number of days prior to the phenological date that GDD is summed across [@kiniry2005; @behrman2013]. Second, cumulative rainfall values in each time frame were calculated as $C_{rain} = \sum^{DATE}_{DATE - i} PRCP$, where PRCP is the daily rainfall measured in millimeters, $DATE$ is the phenological date, and $i$ is the time frame, or the number of days prior to the phenological date that rainfall is summed across. Third, day length in hours was determined for each time frame as a specific single day prior to the phenological date, as indicated by the time frame $i$, e.g. $DATE - i$. Day length was calculated as a function of latitude and day of the year as in [@forsythe1995]. Fourth, day length change in seconds was determined for a specific single day prior to the phenological date as indicated by the time frame, as $DYLN_{change (s)} = (DYLN_{DATE - i} - DYLN_{DATE-i-1})*60*60$, where $DYLN$ in the day length in hours, $DATE$ is the phenological date, and $i$ is the number of days prior to the phenological date. Fifth, average temperature in each time frame was defined as $T_{ave} = \sum^{DATE}_{DATE - i} (T_{max} + T_{min}) / 2$, where $T_{max}$ is the maximum daily temperature, $T_{min}$ is the minimum daily temperature, $DATE$ is the phenological date, and $i$ is the time frame, or the number of days prior to the phenological date that average temperature is summed across.

We then generated GxWeather covariance matrices derived from correlations in these GxWeather cue variables for the three subpopulations. These covariance matrices represent the correlations between genotypes for these weather-related variables across our eight common gardens. To create these matrices, we determined the correlations between these weather-related cue variables for identical genotypes grown at different common gardens and used these correlations to fill the off-diagonals of a eight-by-eight common garden matrix. For the diagonal of these matrices, we used the narrow-sense heritability for each weather-related cue variable at each garden, calculated using the same methods as Section S3. Matrices need to be positive semi-definite to be used in *mash*, so when the resultant matrices were not positive semi-definite, the `Matrix::nearPD` function was used to compute the nearest positive definite matrix to the correlation matrix [@Matrix]. Finally, all covariance matrices are rescaled so that the maximum diagonal element of the rescaled matrix is 1, as in [@Urbut2019]. Code used to generate these matrices can be found at git\@github.com:Alice-MacQueen/pvdiv-phenology-gxe.git/R/Analysis_v1.2_mash_using_greedy_covar.qmd, starting at line 536.

We defined distinct sets of GxWeather covariance matrices for the two phenological dates and for the three genetic subgroups. We used the same five weather variables to transform both green-up and flowering date, but defined the weather-based covariance matrices relative to each of the two phenological dates. GxWeather covariance matrices were also defined separately for individual genotypes from the Gulf, Midwest, and Both subpopulations.

## Section S2. Greedy mash algorithm used to select covariance matrices that significantly improved the model log-likelihood {.unnumbered}

We generated 48 - 60 GxWeather covariance matrices for each of our five weather variables over different time frames. These GxWeather covariance matrices were in many cases highly correlated.

selection of the GxWeather

> describing that the weather-based cues are themselves correlated, and picking and choosing among weather cues is a model building problem â to address the issue, we used a heuristic greedy algorithm to pick the weather-based covariance structures that improved the likelihood of the mash model

> A lot of the results interpretation is based on interpreting the SNP loadings on the specified covariance matrices. This is a secondary use of mash (the primary being the refinement of effect sizes), and while they do interpret these somewhat in the mash paper, these loadings need to be interpreted with caution. If these covariance matrices are similar, mash will somewhat arbitrarily assign weight to any of the matrices because all lead to the same fit to the effect-size data. The mash algorithm doesnât give the full posterior distribution on the loadings so you canât check for posterior correlations there. It seems likely to me that the hypothesis-driven and data-driven covariance matrices are somewhat correlated here, and the correlations may differ between green-up and flowering because of the better correlation between the environmental metrics and flowering than green-up.

Type or paste text here. Break this section up into subheads as needed

## Section S3. Narrow-sense heritability estimation {.unnumbered}

In the diversity panel, we determined narrow-sense heritabilities (h^2^) for green-up and flowering dates at single gardens using genomic relationship matrices calculated using the van Raden method [@vanraden2008]. Genomic relationship matrices were calculated within each subpopulation (Midwest and Gulf) and for both genetic subpopulations (Both). We used rrBLUP [@rrBLUP] to specify mixed models of the form:

$$
\mathbf{y} = 1 + Zu + e
$$

$$
Var(u) = G\sigma_u^2
$$

$$
Var(e) = I\sigma_e^2
$$

in which the vector **y** represents the flowering date or green-up date values for that garden, Z the design matrix for random effects, u the whole genome additive genetic effect, and e the residual. Matrix G is the whole genomic relationship matrix based on all SNPs retained for subpopulation-specific analyses. I is the rank-y identity matrix. Phenotypic variance $\sigma_p^2$ is $\sigma_u^2 + \sigma_e^2$. Narrow-sense heritability is then $h^2 = (\sigma_u^2 / \sigma_p^2)$.

These models were run for each of the eight gardens, and across all gardens by adding an additional environmental effect of site without an interaction term. This resulted in 54 models: 3 sets of populations (the Gulf, Midwest, and Both subpopulations) for 9 garden sets (all eight gardens separately, and all eight gardens together) and two phenotypes (green-up date and flowering date).

## Section S4. Outbred pseudo-F2 mapping population and Quantitative Trait Locus mapping {.unnumbered}

To confirm candidate genomic regions and patterns of allelic effects found in the diversity panel, we analyzed flowering in an outbred pseudo-F2 cross between four grandparents, two Midwest and two Gulf individuals. The formation of this mapping population has been described previously [@Milano2016]. The parents of this cross were DAC, an early flowering Midwest individual, VS16, a late flowering Midwest individual, AP13, an early flowering Gulf individual, and WBC, a late flowering Gulf individual. We made F1 crosses of the two early flowering genotypes, AP13xDAC, and the two late flowering genotypes, WBCxVS16. We then clonally propagated and planted the four parents, the two F1 genotypes (AP13xDAC, and VS16xWBC), and 801 F2 genotypes at eight field sites in May-July of 2015. To be directly comparable to the diversity panel data, only 2019 phenology data from the pseudo-F2 cross from the same eight common garden sites were used here.

Details on the genetic map construction, map polishing and fine-scale reordering can be accessed on [DataDryad](https://datadryad.org/stash/dataset/doi:10.5061/dryad.ghx3ffbjv). QTL mapping was conducted with R/qtl2 [@broman2019]. We performed a genome scan with a linear mixed model that accounts for the relationships among individuals and for environmental covariates (i.e., field sites). The full model can be expressed as:

$$
phenotype = \mu + QTL + E + QTL*E + kinship + e
$$

where $\mu$ is the population mean, QTL is the marker genetic effect, E is the environmental effects (here, common garden), QTL\*E is the interaction between marker genetic and environmental effects, kinship corresponds to the background polygenic variation, and e is the error term. The genome scan was accomplished with the âscan1â function. The statistical significance of the genome scan was established by performing a stratified (i.e., stratifying on common garden) permutation test (n=1000) using âscan1permâ function. The estimated QTL effect was obtained using âscan1coefâ function in R/qtl2.

Figures should be cross-referenced like @fig-frog. Each figure should be on its own page. You can control this by placing a pagebreak shortcode, with `{{< pagebreak >}}`.

This can also be done with LaTeX's `\newpage` command. You may find `\FloatBarrier` useful if your figures or tables are running off to the wrong pages.

{{< pagebreak >}}

![Figure](figs/frog.eps){#fig-frog}

\FloatBarrier <!-- always leave a blank line between this and figures --->

{{< pagebreak >}}

\FloatBarrier

<!-- use these FloatBarrier calls to avoid true-to-the-theme jumping ahead of the frog --->

| Species           |  CBS |   CV |   G3 |
|:------------------|-----:|-----:|-----:|
| Acetaldehyde      |  0.0 |  0.0 |  0.0 |
| Vinyl alcohol     |  9.1 |  9.6 | 13.5 |
| Hydroxyethylidene | 50.8 | 51.2 | 54.0 |

: Simple table example {#tbl-simple}

{{< pagebreak >}}

## Datasets {.unnumbered}

If your document relies on movies or datasets, please list them here with their captions. Use the `movie{your caption}` and `\dataset{file_name.ext}{your caption}` commands to do so.

Datasets 1 through 6 are csv files where the first two columns have the following definitions:

-   Marker: The SNP marker in the format Chromosome_Position

-   log10BF: log10(Bayes Factor) of the significance of the marker effect in the mash model

The remaining column names follow the pattern Effect\_\[Mean/StandardError/lfsr\]\_\[Subpopulation\]\_\[Phenotype\]\_\[Garden\], where Mean and Standard Error are estimates of the effect mean and standard error, lfsr is the local false sign rate statistic for the effect, and \[Subpopulation\], \[Phenotype\], and \[Garden\] follow the conventions of Figure 1.

\dataset{Dataset_1_Gulf_vegetative_growth_date.csv}{SNP-associated effects for the start of vegetative growth jointly re-estimated in the Gulf genetic subpopulation.}

\dataset{Dataset_2_Midwest_vegetative_growth_date.csv}{SNP-associated effects and standard errors for the start of vegetative growth jointly re-estimated in the Midwest genetic subpopulation.}

\dataset{Dataset_3_Both_subpopulations_vegetative_growth_date.csv}{SNP-associated effects and standard errors for the start of vegetative growth jointly re-estimated in both the Midwest and Gulf genetic subpopulations.}

\dataset{Dataset_4_Gulf_reproductive_growth_date.csv}{SNP-associated effects and standard errors for the start of reproductive growth jointly re-estimated in the Gulf genetic subpopulation.}

\dataset{Dataset_5_Midwest_reproductive_growth_date.csv}{SNP-associated effects and standard errors for the start of reproductive growth jointly re-estimated in the Midwest genetic subpopulation.}

\dataset{Dataset_6_Both_subpopulations_reproductive_growth_date.csv}{SNP-associated effects and standard errors for the start of reproductive growth jointly re-estimated in both the Midwest and Gulf genetic subpopulations.}

\dataset{Dataset_7_QTL_intervals_with_functionally_validated_homologs.csv}{Genes in the quantitative trait loci (QTL) regions shown in Figure 4 that have functionally validated homologs in rice for vegetative or reproductive development. The first 9 columns of this table provide QTL information, the remainder provide annotation information for the switchgrass genes in the QTL interval with homologs in A. thaliana or rice that have functional validation in flowering or vegetative-growth related traits. Columns 20-30 provide the titles of papers providing functional validation of the rice homolog of the switchgrass gene.}

\dataset{Dataset_8_blah.csv}{Overlap between quantitative trait loci (QTL) for the start of vegetative or reproductive growth with significant mash effect estimates (with a log10(Bayes Factor) > 2) found using diversity panels of the Midwest, Gulf, and Both genetic subpopulations. The first five columns of this table provide QTL information, and columns 6-10 provide mash Marker information for markers within the QTL regions.}

# References

<!-- references for the SI go here. The below div controls its printing --->

::: {#refs}
:::
