---
title: "Weather covariance matrices as U_hyp for GR50"
author: "Alice MacQueen"
date: 2021-01-05
output: html_document
---

# TRIAL script
# -------------------------
NB: Trial script on mash-env-cov branch to use weather covariance matrices as 
U_hyp in a mash run of GR50 phenotypes from 8 gardens.


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(bigsnpr)
library(switchgrassGWAS)
library(mashr)
library(rlist)
```

# Load project data

```{r}
source("~/Github/Functions_ggplot-theme-adjustments_2018-01-03.R")
workingdir <- file.path("~", "Github", "pvdiv-phenology-gxe")
outputdir <- file.path(workingdir, "analysis", "gwas", "BLUPs", "single_sites")
mashdir <- file.path(workingdir, "analysis", "mash")
metadata <- read_rds(file.path(workingdir, "data", "metadata.rds"))
# pvdiv PLANT_ID metadata
sites <- read_rds(file.path(workingdir, "data", "sites.rds"))
# pvdiv common garden metadata
subpops <- read_rds(file.path(workingdir, "data", "subpop_color_coding.rds"))
phe_gr_df <- read_rds(file.path(workingdir, "data",
                                "Weather_related_phe_for_asreml_h2_GR50.rds"))
# all greenup related phenotypes, including greenup as functions of weather variables.
phe_fl_df <- read_rds(file.path(workingdir, "data",
                                "Weather_related_phe_for_asreml_h2_FL50.rds"))
# all flowering related phenotypes, including functions of weather variables.
k_full <- read_rds(file.path(workingdir, "data",
                    "Kinship_van_Raden_630_individuals_SNPs_r2_20percent.rds"))
# kinship matrix for 630 tetraploid individuals.

## vectors with subsets of gardens, phenotypes, and genetic subpopulations.
site_v <- list(TX1 = "TX1", TX2 = "TX2", TX3 = "TX3", OK = "OK", MO = "MO",
               NE = "NE", MI = "MI", SD = "SD")#, eight_sites = c("TX1", "TX2", "TX3", "OK", "MO", "NE", "MI", "SD"), tx_sites = c("TX1", "TX2", "TX3"), north_sites = c("MO", "NE", "MI", "SD"))
subpop_v <- list(Gulf_and_Midwest = "363g_12.1M", Midwest = "134g_8.8M", 
                 Gulf = "229g_10.3M")
subpop_v2 <- list(Gulf_and_Midwest = c("Gulf", "Midwest"), Midwest = "Midwest", 
                 Gulf = "Gulf")
phe_single <- c("FL50", "dyln_fl50", "cgdd_12c_gr2fl", "crain_gr2fl",
               "crain_1d", "GR50", "cgdd_12c_10d")
phe_mult <- list(flowering_5 = c("FL50", "dyln_fl50", "cgdd_12c_gr2fl",
                                 "crain_gr2fl", "crain_1d"), 
                 greenup_2 = c("GR50", "cgdd_12c_10d"))
inputfiles <- read_delim(file.path(workingdir, "analysis", "gwas",
                                   "inputkinship.txt"),  delim = " ", 
                                   col_names = "SNPfiles") 
# paths to large SNP files
```

# 1. Weather U_hyp

Make weather corr/covar matrices 

1. Make dataframe of each env variable across all 8 sites
2. Compute correlations with pairwise complete observations
3. Change within-site covariance to coefficient of variation for each site. NB: mash does set the maximum of the diagonal to 1... hmm. I wonder why.

Do this for three pop subsets for 'cgdd_12c_10d' and maybe for 'cgdd_12c_18d' and 'cgdd_12c_5d'.

Function to plot covariance matrices to look at their differences
```{r}
pvdiv_plot_cov <- function(covdf){
  U1 <- cor_phe
  if(isSymmetric(U1)){
        for(i in 1:nrow(U1)){
          for(j in 1:ncol(U1)){
            if(i < j){
              U1[i, j] <- NA
            }
          }
        }
  }
  U1 <- as_tibble(U1, rownames = "rowU", .name_repair = "unique") %>%
          pivot_longer(cols = -.data$rowU, names_to = "colU",
                       values_to = "covar") %>%
          filter(!is.na(.data$covar))
  U1$colU <- factor(U1$colU, levels = colnames(cor_phe))
  U1$rowU <- factor(U1$rowU, levels = colnames(cor_phe))
  
  U1_covar <- U1 %>%
    ggplot(aes(x = .data$rowU, y = .data$colU)) +
    switchgrassGWAS::theme_oeco +
    geom_tile(aes(fill = .data$covar), na.rm = TRUE) +
    scale_fill_gradientn(colors = c("#440154FF", "#3B528BFF", "#2C728EFF",
                                    "white", "#27AD81FF", "#5DC863FF",
                                    "#FDE725FF"), limits = c(-1,1)) +
    geom_text(aes(label = round(.data$covar, 1)), color = "darkgrey") +
    theme(legend.position = c(0.2, 0.9),
          axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) +
    xlab("") + ylab("") 
  return(U1_covar)
}

```

## 1a. Make U_hyp

Specify in the phe_hyp vector which functions of weather I want to make U_hyp for mash.
```{r}
U_hyp <- list()
ggcov <- list()
phe_hyp <- c("GR50", "cgdd_12c_10d", "cgdd_12c_18d", "cgdd_12c_5d", "tave_10d")

for(i in seq_along(phe_hyp)){
  for(k in seq_along(subpop_v2)){
    
  cov_df <- phe_gr_df %>%
    ungroup() %>%
    left_join(select(sites, SITE, manu_site), by = "SITE") %>%
    filter(SUBPOP %in% subpop_v2[[k]]) %>%
    select(PLANT_ID, SUBPOP, manu_site, phe_hyp[i]) %>%
    group_by(PLANT_ID, SUBPOP, manu_site) %>%
    mutate(IND = row_number()) %>%
    pivot_wider(names_from = manu_site, values_from = phe_hyp[i]) %>%
    mutate(PLANT_ID = as.factor(PLANT_ID),
           IND = as.factor(IND)) %>%
    select(-IL) %>%
    select(PLANT_ID, SUBPOP, IND, MI, MO, NE, OK, SD, TX1, TX2, TX3)

  cor_phe <- cor(cov_df[,-(1:3)], use = "pairwise")
  diag(cor_phe) <- cov_df %>%
    ungroup() %>%
    summarise(MI = sqrt(var(MI, na.rm = TRUE))/mean(MI, na.rm = TRUE),
              MO = sqrt(var(MO, na.rm = TRUE))/mean(MO, na.rm = TRUE),
              NE = sqrt(var(NE, na.rm = TRUE))/mean(NE, na.rm = TRUE),
              OK = sqrt(var(OK, na.rm = TRUE))/mean(OK, na.rm = TRUE),
              SD = sqrt(var(SD, na.rm = TRUE))/mean(SD, na.rm = TRUE),
              TX1 = sqrt(var(TX1, na.rm = TRUE))/mean(TX1, na.rm = TRUE),
              TX2 = sqrt(var(TX2, na.rm = TRUE))/mean(TX2, na.rm = TRUE),
              TX3 = sqrt(var(TX3, na.rm = TRUE))/mean(TX3, na.rm = TRUE)) %>%
    as_vector()

  ggcov1 <- pvdiv_plot_cov(cor_phe) + ggtitle(label = paste0(phe_hyp[i], "_", names(subpop_v2)[k]))
  ggcov <- list.prepend(ggcov, ggcov1)
  U_hyp <- list.prepend(U_hyp, cor_phe)
  names(U_hyp)[1] <- paste0(phe_hyp[i], "_", names(subpop_v2)[k])
  }
}

ggcov
```

# 2. Make mashr inputs for 8 GR50 phenotypes & run mash.

Some univariate GWAS for the Midwest had population structure issues. Mostly for greenup - even GR50 at MO was quite bad. Drop MO for GR50. Otherwise they look ok for FL50 and GR50.

## 2a. Gulf & Gulf&Midwest.

### mash inputs 
NB: Do this on server, where the large GWAS_datatable objects were created.
```{r}
library
k = 3 # subpop: G&M, M, Gulf
for(k in c(1, 3)){
phe_suffix <- paste("GR50", subpop_v[k], 
                    sep = "*.*")
snp <- snp_attach(inputfiles$SNPfiles[k])
gwas_rds <- pvdiv_results_in_folder(path = file.path(outputdir), 
                                      pattern = paste0("GWAS_datatable*.*",
                                                       phe_suffix))
phenotypes_1 <- str_sub(gwas_rds, start = 16, end = -42) %>%
    str_replace(., "_$", "")
phenotypes <- paste0(names(subpop_v)[k], "_", phenotypes_1)
numSNPs <- round(ceiling(1200000/length(phenotypes)^2)/1000)*1000
out1phe <- pvdiv_bigsnp2mashr(path = file.path(outputdir), 
                           snp = snp, gwas_rds = gwas_rds, 
                           phenotypes = phenotypes, numSNPs = numSNPs, 
                           clump = TRUE, scaled = TRUE, saveoutput = TRUE,
                           suffix = paste0("GR50_", names(subpop_v)[k]))
}
```

Use scp to move these output files (which all have the same suffix) to genegenie.

e.g. 
`scp -i Github/jls_alice alice@popgen.biosci.utexas.edu:~/Github/pvdiv-phenology-gxe/analysis/gwas/BLUPs/single_sites/*19000topSNPs_GR50_Midwest*  /home/alice/Github/pvdiv-phenology-gxe/analysis/gwas/BLUPs/single_sites/Midwest/`

`scp -i ~/Github/jls_alice alice@popgen.biosci.utexas.edu:~/Github/pvdiv-phenology-gxe/analysis/gwas/BLUPs/single_sites/Part-One-Output*  /home/alice/Github/pvdiv-phenology-gxe/analysis/gwas/BLUPs/single_sites/
`

### run mash 
```{r}
for(k in c(1,3,2)){

  numSNPs <- 19000
  m_out <- mash_standard_run(path = file.path(outputdir), #, names(subpop_v)[k]), # use to run on genegenie within mash dirs
                             numSNPs = numSNPs, 
                             U_hyp = U_hyp,
                             suffix = paste0("GR50_", names(subpop_v)[k]),
                             saveoutput = TRUE)
}

inputfiles
```


## 2b. Midwest, dropping some gardens.


# -----------------------------

# 3. Analyse mash output
## -- analysis outline --
•	How many SNP effects load onto each covariance matrix?
•	Mashhattans colored by covariance matrices.
•	What patterns do the data-driven matrices capture that our weather matrices don’t capture?
•	We characterized patterns of differential sensitivity and antagonistic pleiotropy between all SNPs and all pairs of gardens.
•	We looked for enrichments of genes that could potentially affect flowering in the 20kb genomic windows surrounding the SNPs with significant effects on flowering.
•	Rerun greenup mash and rewrite greenup mash section.

## 3a. Covar Loadings
How many SNP effects load onto each covariance matrix?
```{r}
gulfm <- readRDS("~/Github/pvdiv-phenology-gxe/analysis/mash/Gulf/Strong_Effects19000SNPs_GR50_Gulf.rds")

gandm <- readRDS("~/Github/pvdiv-phenology-gxe/analysis/mash/Gulf_and_Midwest/Strong_Effects19000SNPs_GR50_Gulf_and_Midwest.rds")
```
44765 SNPs tested for G&M, 45931 SNPs tested for Gulf.

Midwest I haven't run successfully yet as of 2021-01-12 - I will need to remove some locations with all-zero effects, probably.

### ii. Gulf
```{r}
length(get_significant_results(gulfm))  # 9142/45931  # 19.9%
get_significant_results(gulfm)[1:10]


covargulf <- mash_plot_covar(gulfm, suffix = "19000SNPs_GR50_Gulf", saveoutput = TRUE)
covargulf
get_U_by_mass(gulfm)
mash_plot_Ulist(gulfm, range = get_U_by_mass(gulfm), limits = FALSE)
mash_plot_manhattan_by_condition(gulfm, saveoutput = FALSE)
mash_plot_sig_by_condition(gulfm, saveoutput = FALSE)
```

### ii. G&M
```{r}
length(get_significant_results(gandm))  # 15104/44765  # 33.7%
get_significant_results(gandm)[1:10]


covargam <- mash_plot_covar(gandm, suffix = "19000SNPs_GR50_Gulf_and_Midwest", saveoutput = TRUE)
covargam
get_U_by_mass(gandm)
mash_plot_Ulist(gandm, range = get_U_by_mass(gandm), limits = FALSE)
mash_plot_manhattan_by_condition(gandm, saveoutput = FALSE)
mash_plot_sig_by_condition(gandm, saveoutput = FALSE)
```

```{r}
covargam$covar_df %>% 
  mutate(Type = case_when(grepl("ED_", `Covariance Matrix`) ~ "data-driven",
                          grepl("single_effect", `Covariance Matrix`) ~ "canonical",
                          grepl("simple_het", `Covariance Matrix`) ~ "canonical",
                          grepl("no_effects", `Covariance Matrix`) ~ "canonical",
                          grepl("identity", `Covariance Matrix`) ~ "canonical",
                          grepl("equal_effects", `Covariance Matrix`) ~ "canonical",
                          grepl("_and_Midwest$", `Covariance Matrix`) ~ "GAM_hyp",
                          grepl("Midwest$", `Covariance Matrix`) ~ "Midw_hyp",
                          grepl("Gulf$", `Covariance Matrix`) ~ "Gulf_hyp",
                          TRUE ~ "other"
                          )) %>%
  group_by(Type) %>%
  summarise(Mass = sum(Mass))
```
Gulf: 20% canonical, 68.5% data-driven, 9.5% Gulf hyp, 1.6% Midwest hyp
Gulf & Midwest: 28% canonical, 62% data-driven, 5.5% Gulf, 1.2% Midwest, 2.8% G&M

Not stunning, especially compared to how the flowering hyp do.

## 3b.	Mashhattans 
Mashhattans colored by covariance matrices.

### ii. Plot setup

Get posterior weight dataframe for the covariance matrix grid. Per SNP/row, entries in this grid sum to 1.

Get log10BayesFactor for mash SNPs also and combine these into a dataframe for ggplot.
```{r}
m <- gandm
log10bf_df <- switchgrassGWAS:::get_log10bf(m = m) %>%
    as.data.frame() %>%
    rownames_to_column(var = "value") %>%
    mutate(value = as.integer(.data$value)) %>%
    as_tibble() %>%
    left_join(switchgrassGWAS:::get_marker_df(m = m), by = "value") %>%
    dplyr::rename(log10BayesFactor = .data$V1) %>%
    dplyr::select(-.data$value)

#df <- get_topcov(m)

pw_df <- as_tibble(m$posterior_weights)
pw_df <- switchgrassGWAS:::get_marker_df(m = m) %>%
  bind_cols(pw_df)

ggman_df <- pw_df %>%
  #left_join(m_meta) %>%
  separate(.data$Marker, into = c("Chr", "Pos"), remove = FALSE, sep = "_",
           extra = "merge", convert = TRUE) %>%
  left_join(log10bf_df, by = "Marker") %>%
  arrange(.data$Chr, .data$Pos) 

log10BF <- expression(paste("log"[10], plain("(Bayes Factor)")))

mash_plot_covar(m)
```

Make a list containing all of the ggplots for all entries in the covariance grid represented by `pw_df`.
Then view all the ggplots in gglist.
```{r}
gglist <- list()
for(i in seq_along(names(pw_df)[-(1:2)])){
var <- names(pw_df)[i+2]   
gglist[[i]] <- ggman_df %>%
  arrange(!! sym(var)) %>%
  ggplot(aes(x = .data$Pos, y = .data$log10BayesFactor)) +
  switchgrassGWAS::theme_oeco +
  geom_point(aes(color = !! sym(var), fill = !! sym(var),
                 shape = as.factor(.data$Chr))) +
  facet_wrap(~ .data$Chr, nrow = 1, scales = "free_x", strip.position = "bottom") +
  scale_color_viridis_c(option = "D", end = 0.95) + 
  scale_fill_viridis_c(option = "D", end = 0.95) +
  theme(strip.text = element_text(size = 8),
        axis.text.y = element_text(size = 8),
        axis.text.x = element_blank(),
        legend.text = element_text(size = 8),
        axis.ticks.x = element_blank(),
        panel.background = element_rect(fill=NA)) +
  labs(x = "Chromosome", y = log10BF) +
  scale_x_continuous(expand = c(0.32, 0.32)) +
  scale_shape_manual(values = rep(c(21,22),9), guide = FALSE) +
  geom_hline(yintercept = 2, linetype = 2) +
  ggtitle(var)
}
gglist
```

Which covariance grids have SNPs with log10BF > 2 that load strongly onto them?
For Gulf mash:
GR50_Gulf.6: many, almost all < 2
GR50_Midwest.6: many, almost all < 2
equal_effects.7: many, almost all < 2
GR50_Gulf.6.7: many, almost all < 2
equal_effects.8: some, all <2
ED_PCA_1.9 many, many > 2
ED_PCA_2.9 many, few > 2
ED_PCA_4.9 some, few > 2
ED_tPCA.9: many, many > 2
ED_PCA_3.10: many, few > 2
ED_PCA_5.10: some, few > 2
OK.10 many, some > 2, many >0.8 loading & > 2
SD.10 many, some > 2
cgdd_12c_G&M.10: few, few > 2, some with 80%+ loading here


ED_tPCA.12: many, many > 2, many >0.8 loading esp at high log10BF
SD.12: many, many > 2, many > 0.8 loading esp at high log10BF

TX1.15: few bg, few >2

```{r}
var <- names(ggman_df)[ncol(ggman_df)-1]  
ggman_df %>%
  pivot_longer(cols = null:var, names_to = "covar_grid", values_to = "loading") %>%
  filter(loading > 0.5) %>%
  group_by(log10BayesFactor > 2, covar_grid, loading > 0.8) %>%
  tally()
```
## 3c. Data-driven mat

What patterns do the data-driven matrices capture that our weather matrices don’t capture?

SVD and plots of effects for the ED_PCA matrices.

## 3d. DS & AP

We characterized patterns of differential sensitivity and antagonistic pleiotropy between all SNPs and all pairs of gardens.


## 3e. Enrichments
We looked for enrichments of genes that could potentially affect flowering in the 20kb genomic windows surrounding the SNPs with significant effects on flowering.

## 3f. Loglikelihood

log likelihood of models with and without U_hyp. 



