---
title: "QTL_interval_plotting"
author: "Alice MacQueen"
date: "1/27/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(qtl)
library(qtl2)
library(qtl2convert)
library(tidyverse)
```

#

```{r prep QTL_Eff df}
###QTL Effect plot
###load the original cross and reduce the markers from 4700 to 1185
load('../phenology-fourway/5-QTLmappingForAlice/sexAveraged.cross.rda', verbose = T)
map  =  lapply(pull.map(cross), function(x) x[1,])
# reduced_markers  =  unlist(
#   lapply(
#     reduce_markers(
#       lapply(pull.map(cross), function(x)
#         x[1,]), min_distance = 1), names))
#
# mars2drop  =  markernames(cross)[!markernames(cross) %in% reduced_markers]
# cross  =  drop.markers(cross, mars2drop)
cross = calc.genoprob(cross, step=1, error.prob = 1e-3,map.function = "kosambi",stepwidth = "max")
######### Converting probalibity from qtl1 to qtl2
probs  =  probs_qtl_to_qtl2(cross)$probs
##calculate the kinship matrix for single site
kinship_loco_Single = calc_kinship(probs,use_allele_probs = F,type = "loco")

###extract the genotype ID from the original cross file, so we can align the phenology file with genotype ID
ID <- data.frame(ID=cross$pheno$id)
ID <- ID %>% mutate_at(vars('ID'), list(as.character)) %>% mutate_at(vars('ID'), list(as.numeric))

######get the phenotype at each site
phenology = read.csv('../phenology-fourway/5-QTLmappingForAlice/FL50_phenotypes_Alice.csv', header = TRUE)
colnames(phenology)[1:2] = c('ID','Covariate')
str(phenology) ###make sure your phenology traits are numeric, if not, change them into numeric
phenos <- phenology %>% mutate_at(vars('ID'), list(as.character)) %>%
  mutate_at(vars('ID'), list(as.numeric))%>% group_by(Covariate,ID ) %>%
  summarize_all(mean, na.rm=T) %>% ungroup()%>% as.data.frame()
```

```{r setup QTL effects for all traits}
###QTL effect for each trait
EffectMatrix = vector('list', length=0)
t=0
for (envi in unique(phenos$Covariate)){
    t=t+1
    pheno = phenos %>% filter(Covariate==envi) %>% dplyr::select(-Covariate) %>%
      select(ID, FL50) %>% right_join(ID) %>%  column_to_rownames('ID') %>% as.matrix()

    ##ran effect scan for each chromosome
    eff_out = vector('list',length = 0)
    se_out = vector('list',length = 0)
    add_contrasts=cbind(mu=c(1,1,1,1), add1=c(-1, 1, -1, 1),add2=c(-1, -1, 1, 1), dom=c(-1, 1, 1, -1))

    z = 0
    for (chr in chrnames(cross)){
      print(c(chr, envi,t))
      z= z+1
      eff = scan1coef(probs[,chr],pheno,kinship = kinship_loco_Single[chr], se=T, contrasts = add_contrasts)
      se = attr(eff, 'SE')
      eff_out[[z]] =  eff %>% as.data.frame() %>% rownames_to_column('MARKER')
      se_out[[z]] =  se %>% as.data.frame() %>% rownames_to_column('MARKER')
    }
    eff_out = bind_rows(eff_out)
    se_out = bind_rows(se_out)

    se_out2 = se_out %>% dplyr::select(-mu) %>% gather('CROSS','SE',-MARKER)
    eff_out2 = eff_out %>% dplyr::select(-mu) %>% gather('CROSS','EFF',-MARKER) %>% mutate(SE=se_out2$SE, Covariate=envi)

    EffectMatrix[[t]] = eff_out2
  }

EffectMatrix2 = bind_rows(EffectMatrix)

###unlist nmap and merge with Effect file
nmap = data.frame(pos = unlist(map))
nnmap = nmap %>% rownames_to_column('MARKER') %>% rowwise() %>%
  mutate(chr = str_sub(MARKER, 1,2), MARKER=str_sub(MARKER, 4,nchar(MARKER)))%>%
  ungroup() %>% mutate(pos=round(pos,4))

flank_marker_full = read.csv('../phenology-fourway/5-QTLmappingForAlice/QTLsWithFlankMarkers_FL50_phenotypes_Alice_7_SITES_Full_model.csv')
flank_marker_full = flank_marker_full %>% mutate_at(vars(chr),list(as.character))

for (i in 1:nrow(flank_marker_full)){
  tmp = nnmap %>% filter(chr==flank_marker_full$chr[i] & pos==round(flank_marker_full$pos[i],4))
  flank_marker_full$MARKER[i] = tmp$MARKER
}

QTL_full_model = flank_marker_full %>% dplyr::rename(trait=lodcolumn)



```

```{r ggplots of QTL effects}
QTLdf <- QTL_full_model %>% left_join(EffectMatrix2)%>%
  mutate(CROSS = str_replace(CROSS,'add2','C x D')) %>%
  mutate(CROSS = str_replace(CROSS,'add1','A x B')) %>%
  mutate(Garden = case_when(Covariate == "BRKG" ~ "SD",
                            Covariate == "CLMB" ~ "MO",
                            Covariate == "KBSM" ~ "MI",
                            Covariate == "KING" ~ "TX1",
                            Covariate == "LINC" ~ "NE",
                            Covariate == "PKLE" ~ "TX2",
                            Covariate == "STIL" ~ "OK",
                            TRUE ~ "Other"),
         Cross = case_when(CROSS == "A x B" ~ "AP13 x DAC",
                           CROSS == "C x D" ~ "VS16 x WBC",
                           CROSS == "dom" ~ "Dominance",
                           TRUE ~ "Other")) %>%
  separate(MARKER, into = c("Chr", "Pos"), sep = "_", convert = TRUE, 
           remove = FALSE) %>%
  mutate(Pos_plot = paste0(Chr, "\n", round(Pos, 1), " Mb"))

QTLdf$Covariate = factor(QTLdf$Covariate, levels=c('KING','PKLE','STIL','CLMB',
                                                       'LINC','KBSM','BRKG'))
QTLdf$Garden = factor(QTLdf$Garden, levels = c("TX1", "TX2", "OK", "MO", 
                                                   "NE", "MI", "SD"))

QTLdf %>% 
  filter(Cross != "Dominance") %>% # trait == 'FL50' &
  ggplot(aes(x = Garden, y = EFF, group = Cross)) +
  geom_point(size = 2) + facet_grid(Cross + trait ~ Pos_plot) +
  geom_hline(yintercept = 0, linetype=2) + 
  geom_errorbar(aes(ymin=EFF-SE, ymax=EFF+SE), size=0.6, width=0.5) +
  xlab("") +
  ylab('QTL Effect') + 
  coord_flip() + 
  theme_bw()+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        text=element_text(size=12, face="bold"),
        axis.title=element_text(size=12,face="bold"),
        axis.text.x = element_text(size = 10, angle = 45, hjust = 1, vjust = 1))
# shared QTL for more than one trait differ in LOD score, but not meaningfully in effect patterns across gardens. Useful to know.

QTLplot <- QTLdf %>%
  arrange(desc(lod)) %>%
  group_by(Chr, Pos, Garden, Cross) %>%
  slice(1) %>% 
  filter(Cross != "Dominance") # trait == 'FL50' &

QTLplot$Pos_plot <- factor(QTLplot$Pos_plot, levels = c("Chr02K\n8.6 Mb",
                                                        "Chr02K\n64.4 Mb",
                                                        "Chr02N\n19.5 Mb",
                                                        "Chr02N\n58 Mb",
                                                        "Chr02N\n60.3 Mb",
                                                        "Chr02N\n60.9 Mb",
                                                        "Chr02N\n69.4 Mb",
                                                        "Chr03K\n18.6 Mb",
                                                        "Chr04K\n4.7 Mb",
                                                        "Chr04K\n11.8 Mb",
                                                        "Chr05N\n2.3 Mb",
                                                        "Chr05N\n3.2 Mb",
                                                        "Chr05N\n3.4 Mb",
                                                        "Chr05N\n45.9 Mb",
                                                        "Chr05N\n59.9 Mb",
                                                        "Chr05N\n64.9 Mb",
                                                        "Chr05N\n65.3 Mb",
                                                        "Chr07K\n37 Mb",
                                                        "Chr07N\n45.2 Mb",
                                                        "Chr08N\n6.8 Mb",
                                                        "Chr08N\n16.5 Mb",
                                                        "Chr09K\n16.3 Mb",
                                                        "Chr09N\n15.7 Mb"))
#levels(as.factor(QTLplot$Pos_plot))  

QTLplot %>%
  ggplot(aes(x = Garden, y = EFF, group = Cross)) +
  geom_col(size = 1, aes(fill = trait)) + 
  facet_grid(Cross ~ Pos_plot) +
  geom_hline(yintercept = 0, linetype=2) + 
  geom_errorbar(aes(ymin=EFF-SE, ymax=EFF+SE, color = trait), size=0.6, width=0.5) +
  xlab("") + scale_y_continuous(limits = c(-4.25, 6.5), breaks = c(-3, 0, 3)) +
  ylab('QTL Effect') + 
  coord_flip() + 
  switchgrassGWAS::theme_oeco + 
  scale_color_manual(values = c("#CC0066", "#FFCC00", "#006600", "#6600FF")) +
  scale_fill_manual(values = c("#CC0066", "#FFCC00", "#006600", "#6600FF")) +
  theme(legend.position = "top",
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        text = element_text(size = 10, face="bold"),
        strip.text = element_text(size = 10, face="bold"),
        axis.title = element_text(size = 10,face="bold"),
        axis.text.x = element_text(size = 10, angle = 45, hjust = 1, vjust = 1),
        )# +
    #scale_y_continuous(expand = c(0.2, 0.2)) 
save_plot(filename = "../manuscript/plots/Figure_4_QTL_effects_by_trait.png", base_height = 4.5, base_width = 13.5, plot = last_plot())

QTLdf %>%
  filter(trait == 'CGDD_12C') 
QTLdf %>% 
  filter(trait == 'dyln_fl50')
QTLdf %>% 
  filter(trait == 'dyln_change_sec') 


```

I could combine this with some plots pulled from the mash object using `mash_plot_effects` to do a side-by-side comparison for the paper. That would go well maybe with the Figure 4 I started today, although of course I can't think of a good way to validate this statistically any more than I did in the other script. 
```{r}
library(switchgrassGWAS)
m_gulf_fl_hyp <- readRDS("~/Github/pvdiv-phenology-gxe/analysis/mash/Gulf/Strong_Effects_19000_SNPs_FL50_Gulf_Uhyp.rds")
m_midw_fl_hyp <- readRDS("~/Github/pvdiv-phenology-gxe/analysis/mash/Midwest/Strong_Effects_33000_SNPs_FL50_Midwest_Uhyp.rds")
m_gam_fl_hyp <- readRDS("~/Github/pvdiv-phenology-gxe/analysis/mash/Gulf_and_Midwest/Strong_Effects_19000_SNPs_FL50_Gulf_and_Midwest_Uhyp.rds")

get_post_weights <- function(m){
  log10bf_df <- switchgrassGWAS:::get_log10bf(m = m) %>%
      as.data.frame() %>%
      rownames_to_column(var = "value") %>%
      dplyr::mutate(value = as.integer(.data$value)) %>%
      as_tibble() %>%
      left_join(switchgrassGWAS:::get_marker_df(m = m), by = "value") %>%
      dplyr::rename(log10BayesFactor = .data$V1) %>%
      dplyr::select(-.data$value)
  
  pw_df <- as_tibble(m$posterior_weights)
  pw_df <- switchgrassGWAS:::get_marker_df(m = m) %>%
    bind_cols(pw_df)
  
  ggman_df <- pw_df %>%
    #left_join(m_meta) %>%
    tidyr::separate(.data$Marker, into = c("Chr", "Pos"), remove = FALSE, sep = "_",
             extra = "merge", convert = TRUE) %>%
    dplyr::left_join(log10bf_df, by = "Marker") %>%
    dplyr::arrange(.data$Chr, .data$Pos) 
  return(ggman_df)
}

pw_gf <- get_post_weights(m = m_gulf_fl_hyp) %>%
  mutate(subpop = "Gulf",
         phenotype = "flowering")
pw_mf <- get_post_weights(m_midw_fl_hyp) %>%
  mutate(subpop = "Midwest",
         phenotype = "flowering")
pw_gamf <- get_post_weights(m_gam_fl_hyp) %>%
  mutate(subpop = "Both",
         phenotype = "flowering")
```

```{r}
qtldf <- read_csv("~/Github/pvdiv-phenology-gxe/phenology-fourway/5-QTLmappingForAlice/QTLsWithFlankMarkers_FL50_phenotypes_Alice_7_SITES_Full_model.csv")

qtldf <- qtldf %>%
  separate(flank_lo, into = c("CHR", "Pos_Mb_lo"), sep = "_", convert = TRUE) %>%
  separate(flank_hi, into = c("flank", "Pos_Mb_hi"), sep = "_", convert = TRUE) %>%
  mutate(Pos_Mb_lo = round(Pos_Mb_lo, digits = 2),
         Pos_Mb_hi = round(Pos_Mb_hi, digits = 2)) %>%
  dplyr::select(CHR, Pos_Mb_lo, Pos_Mb_hi, lodcolumn, lod, everything()) %>%
  arrange(CHR, Pos_Mb_lo)

qtlplot <- qtldf %>%
  mutate(phe = case_when(lodcolumn == "FL50" ~ "50% flowering",
                              lodcolumn == "dyln_fl50" ~ "Daylength",
                              lodcolumn == "dyln_change_sec" ~ "daylength change (s)",
                              lodcolumn == "CGDD_12C" ~ "GDD, GR to FL",
                              lodcolumn == "crain_gr2fl" ~ "Rainfall, GR to FL")) %>%
  #filter(phe %in% c("Daylength", "GDD, GR to FL")) %>%
  mutate(POS_lo = Pos_Mb_lo * 1000000,
         POS_hi = Pos_Mb_hi * 1000000)

qtlplot$phe <- factor(qtlplot$phe, 
                      levels = c("50% flowering", "Rainfall, GR to FL",
                                 "GDD, GR to FL", "Daylength",
                                          "daylength change (s)",
                                          "Rainfall, 1 day",
                                                  #"Rainfall, 2d sum",
                                                  "Rainfall, 3d sum",
                                                  "Rainfall, 5d sum"#,
                                                  #"Rainfall, 7d sum"
                                                  ))
```
# Functionally validated flowering time genes in QTL intervals (Table S2)
```{r}
qtl_anno_df <- qtlplot %>%
  dplyr::rename(start = POS_lo, end = POS_hi)
txdb <- loadDb("~/Github/pvdiv-genome/Pvirgatum_516_v5.1.gene.txdb.sqlite")
source("~/Github/Functions_ggplot-theme-adjustments_2018-01-03.R")
workingdir <- file.path("~", "Github", "pvdiv-phenology-gxe")
ftanno <- read_csv(file.path(workingdir, "data/Pvirgatum_v5.1.annotation_FTGeneHomologs.csv"))
ftkey <- read_delim(file.path(workingdir, "data", "OsGene_keyword_table.txt"),
                    delim = "\t")
ftanno %>%
  filter(!is.na(OsGene)) %>%
  left_join(ftkey, by = c("OsGene" = "Symbol")) %>%
  filter(Keyword %in% c("flowering time", "heading date")) %>% unique()

osgene_keyword_df <- ftkey %>%
  dplyr::select(-RAPdb, -MSU) %>%
  group_by(Symbol, Keyword) %>%
  dplyr::slice(1) %>%
  pivot_wider(names_from = Keyword, values_from = Title)

osgene_key <- osgene_keyword_df %>%
  dplyr::select(Symbol, `flowering time`, `heading date`, anther, flower,
                `floral meristem`, `floral meristem determinacy`, flowering,
                `flower development`, `carpel differentiation`, stamen, 
                `stamen number`) %>%
  pivot_longer(cols = -Symbol, names_to = "Keyword", values_to = "Title") %>%
  filter(!is.na(Title)) %>%
  pivot_wider(names_from = Keyword, values_from = Title)

os_annos <- ftanno %>%
  filter(OsGene %in% osgene_key$Symbol | !is.na(AtGene)) %>%
  mutate(Pos_Mb = round(start/10000)/100,
         #Pos_Mb_e = floor(start/10000)/100,
         POS_bin = round(Pos_Mb*100/2)) %>%# ,
         #POS_bin_hi = round(Pos_Mb_e*100/2))
  dplyr::select(OsGene, AtGene, locusName, everything()) %>%
  dplyr::rename(`Gene ID` = locusName)


qtl_annos <- switchgrassGWAS::pvdiv_table_topsnps(df = qtl_anno_df, type = "table", txdb = txdb) %>%
    inner_join(os_annos, by = c("Gene ID")) %>%
    left_join(osgene_key, by = c("OsGene" = "Symbol")) %>%
    dplyr::select(where(not_all_na)) %>%
    group_by(CHR, pos, `Gene ID`) %>%
    dplyr::slice(1) 

names(qtl_annos)
all_na_ft <- rowSums(is.na(qtl_annos[,58:65]))
add_column(qtl_annos, FT_Osgene_NA = all_na_ft, .before = TRUE) %>%
  group_by(CHR, pos, FT_Osgene_NA < 8) %>%
  tally()
flowering_qtl_annos <- add_column(qtl_annos, FT_Osgene_NA = all_na_ft, .before = TRUE) %>%
  filter(FT_Osgene_NA < 8) %>%
  dplyr::select(CHR, pos, phe, lod, Pos_Mb_lo, Pos_Mb_hi, ci_lo, ci_hi, `Gene ID`,  gene_start:gene_width, `Arabidopsis thaliana homolog`:`Rice gene annotation`, OsGene, anther:`floral meristem determinacy`)
  
flowering_qtl_annos %>%
  write_csv("~/Github/pvdiv-phenology-gxe/manuscript/tables/Final/Table_S2_QTL_intervals_with_functionally_validated_flowering_genes.csv")
```


```{r}
pw_3f <- pw_gf %>%
  full_join(pw_mf) %>%
  full_join(pw_gamf) %>%
  arrange(Chr, Pos) %>%
  select(Chr:Pos, log10BayesFactor, subpop, value, phenotype, everything()) %>%
  filter(log10BayesFactor > 2)
pw_list <- list(Gulf = pw_gf, Midwest = pw_mf, Both = pw_gamf)
m_list <- list(Gulf = m_gulf_fl_hyp, Midwest = m_midw_fl_hyp, Both = m_gam_fl_hyp)
names(m_list)
qtlplot <- qtlplot %>%
  arrange(desc(lod)) %>%
  mutate(pos = as.integer(pos)) %>%
  group_by(CHR, pos) %>%
  slice(1) %>%
  arrange(CHR, pos)
mash_plot_manhattan_by_condition(m_gulf_fl_hyp)
mash_plot_effects(m_midw_fl_hyp, n = 5)
```

Make 4 plots for Figure 4: 
i = 4, j = 1 Both BF 7
10, 3 BF 4.37 Gulf

15, 5: 5.1 BF Midwest
16, 1 7.6 BF Both
17, 1 7.6 BF Both (same SNP as 16)
```{r}
i <- c(4, 10, 15, 16)
j <- c(1, 3, 5, 1)
f4df <- tibble(i = i, j = j)

ggdf_list <- list()

k=3
for(k in 1:nrow(f4df)){
  pw_int <- pw_3f %>%
    filter(.data$Chr %in% qtlplot$CHR[f4df$i[k]] & between(.data$Pos, 
                                                   qtlplot$POS_lo[f4df$i[k]], 
                                                   qtlplot$POS_hi[f4df$i[k]])) %>%
    arrange(desc(log10BayesFactor))
  
  plot_df <- 
    mash_plot_effects(m = m_list[[which(names(m_list) == pw_int$subpop[f4df$j[k]])]], 
                    i = pw_int$value[f4df$j[k]])
  ggb1 <- plot_df$effect_df %>%
    mutate(Pos_plot = plot_df$marker,
           subpop = pw_int$subpop[f4df$j[k]]) %>%
    mutate(Conditions = str_remove(Conditions, "Gulf_and_Midwest_")) %>%
  mutate(Conditions = str_remove(Conditions, "Gulf_")) %>%
  mutate(Conditions = str_remove(Conditions, "Midwest_")) %>%
  separate(Conditions, into = c("Phenotype", "Garden"), sep = -3) %>%
  mutate(Phenotype = str_remove(Phenotype, "_$"),
         Garden = str_remove(Garden, "^_")) %>%
  mutate(Garden = str_replace(Garden, "0_M", "MI"),
         Phenotype = str_replace(Phenotype, "FL5$", "FL50"))
  ggb1$Garden <- factor(ggb1$Garden, levels = c("TX1", "TX2", "TX3", "OK", "MO", "NE", "MI", "SD"))
  ggdf_list[[k]] <- ggb1
}

ggdf <- ggdf_list[[1]] %>%
  full_join(ggdf_list[[2]]) %>%
  full_join(ggdf_list[[3]]) %>%
  full_join(ggdf_list[[4]]) 

ggdf %>%
  ggplot() + 
  geom_point(mapping = aes(x = Garden, y = .data$mn)) +
  switchgrassGWAS::theme_oeco +
  geom_errorbar(mapping = aes(ymin = .data$mn - .data$se,
                              ymax = .data$mn + .data$se,
                              x = Garden), width = 0.3) +
  geom_hline(yintercept = 0, lty = 2) +
  facet_wrap(~Pos_plot+subpop, nrow = 1) +
  labs(x = "", y = "Effect Size (% total variation)") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
  coord_flip()

save_plot(filename = "../manuscript/plots/Figure_4_mash_SNP_effects.png", plot = last_plot(), base_height = 2.4, base_width = 5.5)
```

```{r}
	Chr06N_41610882
plot_df <- mash_plot_effects(m_midw_fl_hyp, i = 41793)

ggb1 <- plot_df$effect_df %>%
    mutate(Pos_plot = plot_df$marker,
           subpop = "Midwest") %>%
    mutate(Conditions = str_remove(Conditions, "Gulf_and_Midwest_")) %>%
  mutate(Conditions = str_remove(Conditions, "Gulf_")) %>%
  mutate(Conditions = str_remove(Conditions, "Midwest_")) %>%
  separate(Conditions, into = c("Phenotype", "Garden"), sep = -3) %>%
  mutate(Phenotype = str_remove(Phenotype, "_$"),
         Garden = str_remove(Garden, "^_")) %>%
  mutate(Garden = str_replace(Garden, "0_M", "MI"),
         Phenotype = str_replace(Phenotype, "FL5$", "FL50"))
ggb1$Garden <- factor(ggb1$Garden, levels = c("TX1", "TX2", "TX3", "OK", "MO", "NE", "MI", "SD"))
  
  ggb1 %>%
  ggplot() + 
  geom_point(mapping = aes(x = Garden, y = .data$mn)) +
  switchgrassGWAS::theme_oeco +
  geom_errorbar(mapping = aes(ymin = .data$mn - .data$se,
                              ymax = .data$mn + .data$se,
                              x = Garden), width = 0.3) +
  geom_hline(yintercept = 0, lty = 2) +
  facet_wrap(~Pos_plot+subpop, nrow = 1) +
  labs(x = "", y = "Effect Size (%)") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
  coord_flip()
  
  save_plot(filename = "../manuscript/plots/Figure_DOE_mash_SNP_effects.png", plot = last_plot(), base_height = 2.4, base_width = 2)
```

