---
title: "Analysis_manuscript_mash_GxE"
author: "Alice MacQueen"
date: "10/8/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(cowplot)
library(bigsnpr)
library(rlist)
#library(mashr)
#install_github("alice-macqueen/switchgrassGWAS")
library(switchgrassGWAS)
source("~/Github/Functions_ggplot-theme-adjustments_2018-01-03.R")
```

THEN, for the next section, do GWAS on the BLUPs from the eight single site models. For these traits:
all single sites...
FL50 (actually, maybe not)
dyln_fl50
cgdd_12c_gr2fl
crain_gr2fl
crain_1d
--
GR50 (or maybe not)
cgdd_12c_10d

Then do mash runs. First for single traits across all eight sites, to make U_hyp for the next step - take the U_ed with large percentage variance explained. Make U_hyp with those covariances but no covariance with anything else... 
Then run all the flowering-related traits together. Talk about how then SNPs can have effects that are strongest in response to one type of environmental cue. And how cues can be present or absent at different sites. And how this kind of analysis can help reveal all the patterns at play, because it doesn't make assumptions about how effects are structured.

Maybe bring back the QTL maps after this and talk about how patterns of effects differ across sites in these models, too. Though the Genstat model is less sensitive...

# Genotype-by-environment effects of flowering as functions of environmental cues

While the presence of different associations at the Texas and North gardens demonstrated GxE in flowering at a continental scale, our analysis of G and GxE within each of our eight common gardens also suggested the presence of GxE for rainfall, GDD, and photoperiod cues for flowering, for which variation was more visible outside of each subpopulations’ native range. We evaluated GxE in flowering as a function of five environmental cues: as a function of daylength at flowering (‘flowering daylength’), as a function of daylength change from the previous day on the day of flowering (‘flowering daylength change’) as a function of cumulative GDD between greenup and flowering (‘flowering GDD’), as a function of rainfall on the day of flowering (‘flowering rainfall’), and as a function of rainfall between greenup and flowering (‘cumulative rainfall’). We conducted univariate GWAS at each common garden for these flowering functions, then analyzed the allelic effects of unlinked SNPs across common garden sites for the top XK SNPs using mash. When the same SNP set is used in multiple univariate GWAS, a subsequent mash analysis shares information on patterns of effect size and direction for SNPs across these GWAS, improving the power to detect significant, shared results.

## Make phenotype df for GWAS
```{r}
workingdir <- file.path("~", "Github", "pvdiv-phenology-gxe")
outputdir <- file.path(workingdir, "analysis", "heritability", "kinship_subpop")
site_v <- list(TX1 = "TX1", TX2 = "TX2", TX3 = "TX3", OK = "OK", MO = "MO",
               NE = "NE", MI = "MI", SD = "SD", eight_sites = c("TX1", "TX2", "TX3", "OK", "MO", "NE", "MI", "SD"), tx_sites = c("TX1", "TX2", "TX3"), north_sites = c("MO", "NE", "MI", "SD"))
subpop_v <- list(Gulf = "Gulf", Midwest = "Midwest", Gulf_and_Midwest = c("Gulf", "Midwest"))
phe_eight <- c("FL50", "dyln_fl50", "cgdd_12c_gr2fl")
phe_txano <- c("FL50", "dyln_fl50", "cgdd_12c_gr2fl", "crain_gr2fl",
               "crain_1d", "GR50", "cgdd_12c_10d")
phe_single <- c("FL50", "dyln_fl50", "cgdd_12c_gr2fl", "crain_gr2fl",
               "crain_1d", "GR50", "cgdd_12c_10d")
phe_lists <- list(TX1 = phe_single, TX2 = phe_single, TX3 = phe_single, OK = phe_single, MO = phe_single,
               NE = phe_single, MI = phe_single, SD = phe_single, eight_sites = phe_eight, tx_sites = phe_txano, north_sites = phe_txano)

i = 9
j = 1
k = 1
phe_gwas <- tibble()

for(j in seq_along(subpop_v)){
    for(i in seq_along(site_v)){
      phe_v <- phe_lists[[i]]
      for(k in seq_along(phe_v)){
        phe_raw <- read_csv(file.path(outputdir, names(site_v)[i],
                                      paste0("blups_", names(site_v)[i], "_",
                                             names(subpop_v)[j], "_", phe_v[k],
                                             ".csv")))
        names(phe_raw)[2] <- paste0(phe_v[k], "_", names(site_v)[i])
        phe_proc <- phe_raw %>%
          separate(Effect, into = c("factor", "PLANT_ID"), sep = "\\)_") %>%
          filter(!is.na(PLANT_ID) & !grepl("^SITE", factor)) %>%
          left_join(metadata, by = "PLANT_ID") %>%
          filter(SUBPOP %in% subpop_v[[j]]) %>%
          select(PLANT_ID, 3)
        if(i == 1 & k == 1){
          phe_gwas <- phe_proc
        } else {
          phe_gwas <- phe_gwas %>%
            left_join(phe_proc, by = "PLANT_ID")
        }
      }
    }
  phe_gwas <- phe_gwas %>%
  select(PLANT_ID, ends_with("eight_sites"), ends_with("tx_sites"), ends_with("north_sites"), everything())
  write_csv(phe_gwas, path = file.path(workingdir, "analysis", "gwas", "BLUPs",
                                       paste0("BLUP_phenotypes_",
                                              "for_gwas_GR_FL_",
                                              names(subpop_v)[j],
                                              ".csv")))
}
```

## Run GWAS 
In Plots_manuscript_v0.3.Rmd

## Run mash
Then do mash runs. First for single traits across all eight sites, to make U_hyp for the next step - take the U_ed with large percentage variance explained. Make U_hyp with those covariances but no covariance with anything else... 
Then run all the flowering-related traits together. Talk about how then SNPs can have effects that are strongest in response to one type of environmental cue. And how cues can be present or absent at different sites. And how this kind of analysis can help reveal all the patterns at play, because it doesn't make assumptions about how effects are structured.
```{r}
workingdir <- file.path("~", "Github", "pvdiv-phenology-gxe")
outputdir <- file.path(workingdir, "analysis", "gwas", "BLUPs", "single_sites")
site_v <- list(TX1 = "TX1", TX2 = "TX2", TX3 = "TX3", OK = "OK", MO = "MO",
               NE = "NE", MI = "MI", SD = "SD")#, eight_sites = c("TX1", "TX2", "TX3", "OK", "MO", "NE", "MI", "SD"), tx_sites = c("TX1", "TX2", "TX3"), north_sites = c("MO", "NE", "MI", "SD"))
subpop_v <- list(Gulf_and_Midwest = "363g_12.1M", Midwest = "134g_8.8M", Gulf = "229g_10.3M")
phe_single <- c("FL50", "dyln_fl50", "cgdd_12c_gr2fl", "crain_gr2fl",
               "crain_1d", "GR50", "cgdd_12c_10d")
inputfiles <- read_delim(file.path(workingdir, "analysis", "gwas",
                                   "inputkinship.txt"),  delim = " ", 
                                   col_names = "SNPfiles")
```

single traits across all eight sites...
```{r}
i=1 # site
j=1 # phe
k=3 # subpop: G&M, M, Gulf
for(j in seq_along(phe_single)){
  for(k in c(1,3)){
phe_suffix <- paste(phe_single[j], subpop_v[k], 
                    sep = "*.*")
snp <- snp_attach(inputfiles$SNPfiles[k])
gwas_rds <- pvdiv_results_in_folder(path = file.path(outputdir), 
                                      pattern = paste0("GWAS_datatable*.*",
                                                       phe_suffix))
phenotypes_1 <- str_sub(gwas_rds, start = 16, end = -42) %>%
    str_replace(., "_$", "")
phenotypes <- paste0(names(subpop_v)[k], "_", phenotypes_1)
out1phe <- pvdiv_bigsnp2mashr(path = file.path(outputdir), 
                           snp = snp, gwas_rds = gwas_rds, 
                           phenotypes = phenotypes, numSNPs = 1000, 
                           clump = TRUE, scaled = TRUE, saveoutput = TRUE,
                           suffix = paste0("single_phe_", names(subpop_v)[k], 
                                           "_", phe_single[j]))

m_out <- mash_standard_run(path = outputdir, numSNPs = 1000, 
                           suffix = paste0("single_phe_", names(subpop_v)[k], 
                                           "_", phe_single[j]),
                           saveoutput = TRUE)
  }
}
```
Some of the Midwest traits don't work well. Haven't run that subpop yet. 

Looking at the results for the other subsets first before I go back and fix the Midwest.

# View single trait mash outputs
```{r}
i=1 # site
j=7 # phe
k=1 # subpop: G&M, M, Gulf
suffix <- paste0("single_phe_", names(subpop_v)[k], "_", phe_single[j])

m_out <- readRDS(file.path(outputdir, paste0("Strong_Effects1000SNPs_", 
                                                     suffix, ".rds")))
switchgrassGWAS::get_U_by_mass(m_out)
switchgrassGWAS::mash_plot_covar(m_out)
switchgrassGWAS::mash_plot_Ulist(m_out, range = get_U_by_mass(m_out))
mash_plot_sig_by_condition(m_out)
mash_plot_manhattan_by_condition(m_out)
mash_plot_pairwise_sharing(m_out, reorder = FALSE)

```

Keep covariance matrices with point mass > 0.05 and add them to U_hyp.
Add them as single effects (the remaining phe are 0)
And add them as combinations if possible.


1. Make mash inputs for five & two phenotypes for eight sites. Standardize phenotypes and clump SNPs.
2. Find trait order.  The trait order is the same order as the 'phenotypes' vector.
3. Make U_hyp in trait order.
4. Run mash with U_hyp.
# ------- load for m_single to full mash part --
```{r}
workingdir <- file.path("~", "Github", "pvdiv-phenology-gxe")
outputdir <- file.path(workingdir, "analysis", "gwas", "BLUPs", "single_sites")
site_v <- list(TX1 = "TX1", TX2 = "TX2", TX3 = "TX3", OK = "OK", MO = "MO",
               NE = "NE", MI = "MI", SD = "SD")#, eight_sites = c("TX1", "TX2", "TX3", "OK", "MO", "NE", "MI", "SD"), tx_sites = c("TX1", "TX2", "TX3"), north_sites = c("MO", "NE", "MI", "SD"))
subpop_v <- list(Gulf_and_Midwest = "363g_12.1M", Midwest = "134g_8.8M", Gulf = "229g_10.3M")
phe_mult <- list(flowering_5 = c("FL50", "dyln_fl50", "cgdd_12c_gr2fl", "crain_gr2fl",
               "crain_1d"), greenup_2 = c("GR50", "cgdd_12c_10d"))
inputfiles <- read_delim(file.path(workingdir, "analysis", "gwas",
                                   "inputkinship.txt"),  delim = " ", 
                                   col_names = "SNPfiles")
```
# make mashr inputs for 2 greenup and 5 flowering phenotypes via bigsnp2mashr
```{r}
j=2
k=3
for(j in seq_along(phe_mult)){
  for(k in c(1,3)){
phe_suffix <- paste(phe_mult[[j]], subpop_v[k], 
                    sep = "*.*")
snp <- snp_attach(inputfiles$SNPfiles[k])
gwas_rds <- c()
for(i in seq_along(phe_suffix)){
gwas_rds1 <- pvdiv_results_in_folder(path = file.path(outputdir), 
                                      pattern = paste0("GWAS_datatable*.*",
                                                       phe_suffix[i]))
gwas_rds <- c(gwas_rds, gwas_rds1)
}
phenotypes_1 <- str_sub(gwas_rds, start = 16, end = -42) %>%
    str_replace(., "_$", "")

phenotypes <- paste0(names(subpop_v)[k], "_", phenotypes_1)
numSNPs <- round(ceiling(1200000/length(phenotypes)^2)/1000)*1000
out1phe <- pvdiv_bigsnp2mashr(path = file.path(outputdir), 
                           snp = snp, gwas_rds = gwas_rds, 
                           phenotypes = phenotypes, numSNPs = numSNPs, 
                           clump = TRUE, scaled = TRUE, saveoutput = TRUE,
                           suffix = paste0(names(phe_mult)[j], "_phe_",
                                           names(subpop_v)[k], "_subpop"))
  }
}
```

# Now make U_hyp in trait order. Try programmatically. Tests here
```{r}
v <- matrix(data = 0, nrow=16, ncol=16)

i=2 # site
j=1 # phe
k=1 # subpop: G&M, M, Gulf
# test generate U_ed with point mass > 0.05 from one m_single run.
suffix <- paste0("single_phe_", names(subpop_v)[k], "_", phe_mult[[j]][i])

m_single <- readRDS(file.path(outputdir, paste0("Strong_Effects1000SNPs_", 
                                                     suffix, ".rds")))

U_single <- m_single$fitted_g$Ulist[get_U_by_mass(m_single)]
names(m_single$fitted_g$Ulist)
U_s <- U_single[!names(U_single) %in% c("equal_effects", "identity", "simple_het_1", "simple_het_2", "simple_het_3")]

q=1
U_hyp <- list()

# test generate U_hyp for multiple phenotypes from one m_single run.
for(q in seq_along(names(U_s))){
    s_length <- dim(U_s[[1]])[1]
    hyp1 <- v
    hyp1[c((i*s_length-s_length+1):(i*s_length)),
         c((i*s_length-s_length+1):(i*s_length))] <- U_s[[q]]
    U_hyp <- list.append(U_hyp, hyp1)
}
```

# -------------------------
# Create U_hyp programmatically
```{r}
library(tidyverse)
library(cowplot)
library(bigsnpr)
library(rlist)
library(switchgrassGWAS)
workingdir <- file.path("~", "Github", "pvdiv-phenology-gxe")
outputdir <- file.path(workingdir, "analysis", "gwas", "BLUPs", "single_sites")
site_v <- list(TX1 = "TX1", TX2 = "TX2", TX3 = "TX3", OK = "OK", MO = "MO",
               NE = "NE", MI = "MI", SD = "SD")#, eight_sites = c("TX1", "TX2", "TX3", "OK", "MO", "NE", "MI", "SD"), tx_sites = c("TX1", "TX2", "TX3"), north_sites = c("MO", "NE", "MI", "SD"))
subpop_v <- list(Gulf_and_Midwest = "363g_12.1M", Midwest = "134g_8.8M", Gulf = "229g_10.3M")
phe_mult <- list(flowering_5 = c("FL50", "dyln_fl50", "cgdd_12c_gr2fl", "crain_gr2fl",
               "crain_1d"), greenup_2 = c("GR50", "cgdd_12c_10d"))
```


```{r}
j=2 # phe_mult: 1 is fl
i=1
k=3 # subpop 1 = Gulf_and_Midwest   3 = Gulf



U_single_list <- list()
# Make a list (from each phenotype) of lists of U_ed with point mass > 0.05.
for(i in seq_along(phe_mult[[j]])){
  suffix <- paste0("single_phe_", names(subpop_v)[k], "_", phe_mult[[j]][i])
  m_single <- readRDS(file.path(outputdir, paste0("Strong_Effects1000SNPs_", 
                                                  suffix, ".rds")))
  U_single1 <- m_single$fitted_g$Ulist[get_U_by_mass(m_single)]
  U_s1 <- U_single1[!names(U_single1) %in% c("equal_effects", "identity", "simple_het_1", "simple_het_2", "simple_het_3")]
  U_single_list <- list.append(U_single_list, U_s1)
}

# Using U_single_list, generate U_hyp for multiple phenotypes.

## part 1: determine size of matrix that U_ed will need to be
s_total <- 0

for(i in seq_along(U_single_list)){
    s_length <- dim(as.data.frame(U_single_list[[i]][1]))[1]
    s_total <- s_total+s_length 
  }

## part 2: Define null U_ed matrix
v <- matrix(data = 0, nrow=s_total, ncol=s_total) 
U_hyp <- list()
for(i in seq_along(U_single_list)){
  for(q in seq_along(names(U_single_list[[i]]))){
    s_length <- dim(as.data.frame(U_single_list[[i]][q]))[1]
    U_s1 <- matrix(unlist(U_single_list[[i]][q]), nrow = s_length, ncol = s_length)
    hyp1 <- v
    hyp1[c((i*s_length-s_length+1):(i*s_length)),c((i*s_length-s_length+1):(i*s_length))] <- U_s1
    U_hyp <- list.append(U_hyp, hyp1)
  }
}

# part 3: determine U_ed with the same names
all_names <- c()
for(i in seq_along(U_single_list)){
  all_names <- c(all_names, names(U_single_list[[i]]))
}
U_hyp_comb_df <- enframe(all_names) %>% group_by(value) %>% tally() %>% filter(n > 1)  # these U_ed names have two or more phenotypes with point mass > 0.05.

n=1
U_hyp_comb <- list()
for(n in seq_along(U_hyp_comb_df$value)){
  name_U <- U_hyp_comb_df$value[n]
  for(i in seq_along(U_single_list)){
    for(q in seq_along(names(U_single_list[[i]]))){
      if(i == 1 & q == 1){
        s_length <- dim(as.data.frame(U_single_list[[i]][q]))[1]
        hyp1 <- v
      }
      if(name_U == names(U_single_list[[i]])[q]){
        U_s1 <- matrix(unlist(U_single_list[[i]][q]), nrow = s_length, 
                       ncol = s_length)
        hyp1[c((i*s_length-s_length+1):(i*s_length)),c((i*s_length-s_length+1):(i*s_length))] <- U_s1
      }
    }
  }
  
  U_hyp_comb <- list.append(U_hyp_comb, hyp1)
}
# part 4: Combine U_single into a combined U_hyp if the names of the U_ed match. 
# Probably there are better ways to do this but this is the best I can think of, apart from doing every last possible combination. That will also suck because I don't have any way to generate covariances between phenotypes. So, eh. Most likely none of these will get used.
 
#View(U_hyp_comb[[2]])
U_hyp <- c(U_hyp, U_hyp_comb)
```
So between the 'single phenotype' U_hyp and the combined U_hyp, that will give me 16 U_hyp for the five flowering phenotypes, and 8 U_hyp for the greenup phenotypes.

Got it! Now need to either save these U_hyp, or just use them in mash after I generate the relevant list.

# Run mash on multiple phenotypes using these U_hyp

j = 2; k = 1: greenup, Gulf


```{r}
j=2 # phe_mult 1 = flowering 2 = greenup
k=3 # subpop 3 = Gulf   1 = G&M

phe_suffix <- paste(phe_mult[[j]], subpop_v[k], 
                    sep = "*.*")
gwas_rds <- c()
for(i in seq_along(phe_suffix)){
gwas_rds1 <- pvdiv_results_in_folder(path = file.path(outputdir), 
                                      pattern = paste0("GWAS_datatable*.*",
                                                       phe_suffix[i]))
gwas_rds <- c(gwas_rds, gwas_rds1)
}
phenotypes_1 <- str_sub(gwas_rds, start = 16, end = -42) %>%
    str_replace(., "_$", "")

phenotypes <- paste0(names(subpop_v)[k], "_", phenotypes_1)
numSNPs <- round(ceiling(1200000/length(phenotypes)^2)/1000)*1000

U_ed <- readRDS(paste0("analysis/gwas/BLUPs/single_sites/Mash_data_driven_covariances_", numSNPs, "_", names(phe_mult)[j], "_phe_", names(subpop_v)[k],
                       "_subpop", ".rds"))
U_hyp <- readRDS(paste0("analysis/gwas/BLUPs/single_sites/Mash_data_driven_covariances_", numSNPs, "_", names(phe_mult)[j], "_phe_", names(subpop_v)[3],
                       "_subpop", ".rds"))

m_out2 <- mash_standard_run(path = outputdir, numSNPs = numSNPs, 
                           U_hyp = U_hyp, U_ed = U_ed,
                           suffix = paste0(names(phe_mult)[j], "_phe_",
                                           names(subpop_v)[k], "_subpop"),
                           saveoutput = TRUE)

```

Another option, I suppose, is just running mash on multiple phenotypes for a very small subset of strong SNPs... just a thought. Would give covar between phenotypes, too, when now I just have covar within phenotypes. Really this is just using a much smaller subset of strong SNPs to define U_ed than I have been doing.

For whatever reason (unknown as of 2020-11-02), finding the data-driven covariance matrices has taken more than a whole weekend for flowering phenotypes (5) in the Gulf subpop. I fear errors/the worst but will let it run a while longer to see. In case it's the number of conditions, I tried to run a separate mash on greenup phenotypes (2) in the Gulf subpop. We'll see if that completes faster. 

```{r}
(18+17+16+15+14+13+12+11+10+9+8+7+6+5+4+3+2+1)*3
10/513
```


```{r}
subpop_v  # 1 is G&M  3 is Gulf
numSNPs <- 5000
m1 <- readRDS(paste0("../analysis/gwas/BLUPs/single_sites/Strong_Effects", numSNPs, "SNPs_", names(phe_mult)[j], "_phe_", names(subpop_v)[1],
                       "_subpop", ".rds"))
m3 <- readRDS(paste0("../analysis/gwas/BLUPs/single_sites/Strong_Effects", numSNPs, "SNPs_", names(phe_mult)[j], "_phe_", names(subpop_v)[3],
                       "_subpop", ".rds"))

mash_plot_Ulist(m1,1)
mash_plot_Ulist(m3,1)
mash_plot_covar(m1)
mash_plot_covar(m3)
m1$loglik
m3$loglik
length(get_significant_results(m3))
```

```{r}
m <- readRDS(paste0("../analysis/gwas/BLUPs/single_sites/Strong_Effects5000SNPs_greenup_2_phe_Gulf_and_Midwest_subpop_Gulf_Uhyp.rds"))

mash_plot_covar(m)
mash_plot_sig_by_condition(m)
mash_plot_manhattan_by_condition(m)
mash_plot_pairwise_sharing(m)
mash_plot_effects(m, n = 6)

switchgrassGWAS::get_U_by_mass(m)
```

```{r}
library(mashr)
mash_plot_covar <- function(m, saveoutput = FALSE, suffix = ""){
  df <- get_estimated_pi(m)
  df <- enframe(df, name = "Covariance Matrix", value = "Mass") %>%
    mutate(`Covariance Matrix` = str_replace(.data$`Covariance Matrix`,
                                             "(Stand_)??Bhat_?",
                                             "single_effect_"),
           `Covariance Matrix` = str_replace(.data$`Covariance Matrix`,
                                             "^null$", "no_effects")) %>%
    arrange(desc(.data$`Covariance Matrix`))
  df$`Covariance Matrix` <- factor(df$`Covariance Matrix`,
                                   levels = (df$`Covariance Matrix`))
  ggobject <- ggplot(df) +
    geom_bar(aes(x = .data$Mass, y = .data$`Covariance Matrix`),
             stat = "identity") +
    switchgrassGWAS::theme_oeco

  if(saveoutput == TRUE){
    if(!(str_sub(suffix, end = 1) %in% c("", "_"))){
      suffix <- paste0("_", suffix)
    }
    if(str_sub(suffix, end = 1) %in% c("")){
      suffix <- paste0("_", get_date_filename())
    }
    save_plot(paste0("Covariance_matrix_mass_plot", suffix,
                     ".png"), plot = ggobject, base_aspect_ratio = 1,
              base_height = 4)
  }
  return(list(covar_df = df, ggobject = ggobject))
}
```

# ---------------------

# Greenup mash results

#' @param posterior_weights Vector containing the posterior
#'     probability of each mixture component in Ulist for the data.
```{r}
length(get_significant_results(m))
get_significant_results(m)[1:10]
get_significant_results(m)[1:10]

str(m$posterior_weights)

n=7500
i = get_significant_results(m)[n]

# largest probability that this SNP loads on this covariance matrix
max(m$posterior_weights[i,])
which(m$posterior_weights[i,] == max(m$posterior_weights[i,]))

# second largest probability that this SNP loads on this covariance matrix
max( m$posterior_weights[i,][m$posterior_weights[i,]!=max(m$posterior_weights[i,])])
which(m$posterior_weights[i,] == max(m$posterior_weights[i,][m$posterior_weights[i,]!=max(m$posterior_weights[i,])]))

sum(m$posterior_weights[,1])

mash_plot_covar(m, saveoutput = TRUE)

colnames(m$posterior_weights)
Ulist <- switchgrassGWAS:::get_Ulist(m)
names(Ulist)

df <- mashr:::get_estimated_pi(m, dimension = "all")
```

Gulf
```{r}
inputdir <- file.path("../analysis/gwas/BLUPs/single_sites", "Gulf")
mash_out <- pvdiv_results_in_folder(inputdir, "Strong")
mash_out
m <- read_rds(file.path(inputdir, mash_out[11]))
mash_plot_covar(m, saveoutput = TRUE)
get_U_by_mass(m)
mash_plot_Ulist(m, range = get_U_by_mass(m))
mash_plot_effects(m, n = 1, saveoutput = TRUE)
mash_plot_sig_by_condition(m, saveoutput = TRUE)
mash_plot_manhattan_by_condition(m, saveoutput = TRUE)
mash_plot_pairwise_sharing(effectRDS = file.path(inputdir, mash_out[1])) # sign
mash_plot_pairwise_sharing(effectRDS = file.path(inputdir, mash_out[2])) # sign & size
Gulf_greenup_gxe <- get_GxE(m)

mash_plot_pairwise_sharing(corrmatrix = Gulf_greenup_gxe$S_AP/Gulf_greenup_gxe$S_all_pairwise, saveoutput = TRUE)
mash_plot_pairwise_sharing(corrmatrix = Gulf_greenup_gxe$S_DS/Gulf_greenup_gxe$S_all_pairwise, reorder = TRUE, saveoutput = TRUE)
mash_plot_pairwise_sharing(corrmatrix = Gulf_greenup_gxe$S_2_no, reorder = FALSE)
mash_plot_pairwise_sharing(corrmatrix = Gulf_greenup_gxe$S_all_pairwise, reorder = FALSE)

```

```{r}
821+2309+789+1401+1090+1277+658+1075
9420/10581
```

# SVD of a covariance matrix

Work through an two-condition example here to help me understand how to explain this.

Sample points from a bivariate Gaussian distribution with a standard deviation of 3 in roughly the lower left-upper right direction and of 1 in the orthogonal direction. Because the x and y components co-vary, the variances of x and y do not fully describe the distribution. A 2 × 2 covariance matrix is needed; the directions of the arrows correspond to the eigenvectors of this covariance matrix and their lengths to the square roots of the eigenvalues.
```{r}
cov1 <- matrix(c(1, -0.4, 0.3, -0.4, 1, 0.7, 0.3, 0.7, 1), ncol = 3 )

b1 <- svd(cov1)

b1$d[1] / sum(b1$d) # Proportion of variance explained by first eigenvector 80%
b1$d[2] / sum(b1$d) # 20%
b1$d[3] / sum(b1$d) # 20%
# b1$u symmetric with v if first matrix is symmetric
b1$v
b1$d

#sum(b1$d[1:2]) / sum(b1$d) # First 2 eigenvectors explain 100% of total variance.
for(g in 1:3){
barplot(b1$v[,g]/cov1[which.max(abs(cov1[,g])),g])
}

L = chol(cov1)
nvars = dim(L)[1]
nobs=200
# Random variables that follow an M correlation matrix
r = t(L) %*% matrix(rnorm(nvars*nobs), nrow=nvars, ncol=nobs)
r = t(r)

rdata = as.data.frame(r)
names(rdata) = c('cond1', 'cond2', 'cond3')
library(lattice)
splom(rdata)
```
The closer the covariance between x and y is to zero, the more equal the eigenvalues will be, and the more equal the proportion of variance explained by each eigenvector will be.
If the covariance is negative the first eigenvector will show a sign change.

# ------------------
# Gulf & Midwest
```{r}
inputdir <- file.path("../analysis/gwas/BLUPs/single_sites", "Gulf_and_Midwest")
mash_out <- pvdiv_results_in_folder(inputdir, "Strong")
mash_out
m <- read_rds(file.path(inputdir, mash_out[13]))

mash_plot_covar(m)
get_U_by_mass(m)
mash_plot_Ulist(m, range = get_U_by_mass(m))
mash_plot_effects(m, n = 1)
mash_plot_sig_by_condition(m)
mash_plot_manhattan_by_condition(m)
mash_plot_pairwise_sharing(effectRDS = file.path(inputdir, mash_out[1])) # sign
mash_plot_pairwise_sharing(effectRDS = file.path(inputdir, mash_out[2])) # sign & size
Gulf_greenup_gxe <- get_GxE(m)
```

```{r}
get_U_by_mass(m)
i=2
names(Ulist)[i]
Ulist <- switchgrassGWAS:::get_Ulist(m)
b1 <- svd(Ulist[[i]])
b1$d[1] / sum(b1$d) # Proportion of variance explained by first eigenvector 11.5
b1$d[2] / sum(b1$d) # get to 60%
b1$d[3] / sum(b1$d) # 
b1$d[4] / sum(b1$d) # 
b1$d[5] / sum(b1$d) # 


sum(b1$d[1:2]) / sum(b1$d) # First 5 eigenvectors explain 1% or more variance, and 27.7% of total variance.
for(g in 1:5){
barplot(b1$v[,g]/Ulist[[1]][which.max(abs(Ulist[[1]][,g])),g])
}

ggb1 <- tibble(`First eigenvector (PVE = 100%)` = b1$v[,1]/Ulist[[1]][which.max(abs(Ulist[[1]][,1])),1],
       Phenotype = switchgrassGWAS:::get_colnames(m)) %>%
  mutate(Phenotype = str_remove(Phenotype, "Stand_BhatGulf_")) %>%
  separate(Phenotype, into = c("Function", "Garden"), sep = -3) %>%
  mutate(Function = str_remove(Function, "_$"),
         Garden = str_remove(Garden, "^_"))
ggb1$Garden <- factor(ggb1$Garden, levels = c("TX1", "TX2", "TX3", "OK", "MO", "NE", "MI", "SD"))
ggb1_plot <- ggb1 %>%
  ggplot(aes(x = Garden, y = `First eigenvector (PVE = 100%)`)) +
  geom_bar(aes(group = Function, color = Function, fill = Function), stat = "identity", position = "dodge") +
  theme(legend.position = "top", panel.spacing.x = unit(0.1, 'cm'), axis.text.x = element_text(hjust = 1, angle = 45)) +
  geom_hline(yintercept = 0, linetype = 2) +
  scale_color_viridis_d(option = "B", end = 0.7, 
                      direction = -1) +
  scale_fill_viridis_d(option = "B", begin = 0.1, end = 0.7, direction = -1) 
save_plot(paste0("../manuscript/plots/SVD_of_greenup_Gulf_", names(Ulist)[i], ".png"), base_height = 4, plot = ggb1_plot)


ggb2 <- tibble(`First eigenvector (PVE = 21.7%)` = b1$v[,2]/Ulist[[1]][which.max(abs(Ulist[[1]][,2])),2],
       Phenotype = switchgrassGWAS:::get_colnames(m)) %>%
  mutate(Phenotype = str_remove(Phenotype, "Stand_BhatGulf_")) %>%
  separate(Phenotype, into = c("Function", "Garden"), sep = -3) %>%
  mutate(Function = str_remove(Function, "_$"),
         Garden = str_remove(Garden, "^_"))
ggb2$Garden <- factor(ggb1$Garden, levels = c("TX1", "TX2", "TX3", "OK", "MO", "NE", "MI", "SD"))
ggb2_plot <- ggb2 %>%
  ggplot(aes(x = Garden, y = `First eigenvector (PVE = 21.7%)`)) +
  geom_bar(aes(group = Function, color = Function, fill = Function), stat = "identity", position = "dodge") +
  theme(legend.position = "top", panel.spacing.x = unit(0.1, 'cm'), axis.text.x = element_text(hjust = 1, angle = 45)) +
  geom_hline(yintercept = 0, linetype = 2) +
  scale_color_viridis_d(option = "B", end = 0.7, 
                      direction = -1) +
  scale_fill_viridis_d(option = "B", begin = 0.1, end = 0.7, direction = -1) 
save_plot(paste0("../manuscript/plots/SVD_of_greenup_Gulf_", names(Ulist)[i], "Eigenvector_2.png"), base_height = 4, plot = ggb2_plot)

```
GO enrichment amongst things that load onto different covariance matrices?
