---
title: "Analyses v0.2"
author: "Alice MacQueen"
date: "7/17/2020"
output: html_document
---

Add any libraries/scripts that need to be loaded for these analyses here.
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(cowplot)
library(viridis)
library(lubridate)
library(switchgrassGWAS)
source("~/Github/Functions_ggplot-theme-adjustments_2018-01-03.R")
```

# Load data
Add any data that needs to be loaded for these analyses here.
```{r}
wea_df <- readRDS("~/Github/pvdiv-phenology-gxe/data/Weather_with_PTT_2019.rds")
metadata <- readRDS("~/Github/pvdiv-phenology-gxe/data/metadata.rds")
sites <- readRDS("~/Github/pvdiv-phenology-gxe/data/sites.rds")
phenotypes <- readRDS("~/Github/pvdiv-phenology-gxe/data/Phenology_CGDD_PTT_and_rainfall_phenotypes.rds")
jday_phe <- readRDS("~/Github/pvdiv-phenology-gxe/data/Julian_date_GDD_and_rainfall_phenotypes_4wcr_and_gwas.rds")
```
```{r}
jday_phe <- wea_df %>%
  group_by(SITE) %>%
  mutate(dyln_change_sec = (DYLN - lag(DYLN))*60*60,
         FL50 = yday(DOY)) %>%
  select(DOY, DYLN, dyln_change_sec, everything(), -PTT_12C) %>%
  right_join(jday_phe, by = c("SITE", "FL50")) %>%
  select(-(TMAX:CPTT_11C))
jday_phe <- metadata %>%
  rename(Lat_origin = LATITUDE, Long_origin = LONGITUDE) %>%
  select(PLANT_ID, SUBPOP, Lat_origin, Long_origin) %>%
  right_join(jday_phe) %>%
  left_join(sites)

jday_phe$SUBPOP <- factor(jday_phe$SUBPOP, levels = c("4X", "Atlantic", "Gulf", "Midwest", "8X"))
jday_phe$SITE <- factor(jday_phe$SITE, levels = c("KING", "PKLE", "TMPL", "OVTN", "STIL", "CLMB", "LINC", "FRMI", "KBSM", "BRKG"))
jday_phe$manu_site <- factor(jday_phe$manu_site, levels = rev(c("TX1", "TX2", "TX3", "TX4", "OK", "MO", "NE", "IL", "MI", "SD")))
```


# Overview
Here I'm putting the code for any new analyses that I'm running to do new writeup for the manuscript v0.2.

# Greenup h2
Greenup as a function of Julian date had extremely low heritability within subpopulations (0.005 +/- 0.003 Gulf; 0.021 +/0 0.011 for Midwest) across all eight common gardens, but high heritability across both subpopulations (h2 = 0.987 +/- 0.006).

<did any environmental cue beat GR50 to estimate greenup?>.

## define environmental cues for greenup
Probably all CGDD related.
```{r}

asreml_fl <- jday_phe %>%
  select(PLANT_ID:Long_origin, PLOT_GL, SITE, FL50, GR50, CGDD_12C, CRAIN, CRAIN_5d, DYLN, dyln_change_sec) %>%
  rename(dyln_fl50 = DYLN, cgdd_12c_gr2fl = CGDD_12C, crain_gr2fl = CRAIN) %>%
  left_join(wea_df) %>%
  group_by(SITE, PLOT_GL) %>%
  filter(between(DOY, as_date(FL50, origin = "2019-01-01") - 6, as_date(FL50, origin = "2019-01-01"))) %>%
  mutate(crain_2d = RAIN_SM + lag(RAIN_SM),
         crain_3d = RAIN_SM + lag(RAIN_SM) + lag(RAIN_SM, n = 2),
         crain_7d = sum(RAIN_SM)) %>%
  filter(DOY == as_date(FL50, origin = "2019-01-01")) %>%
  filter(SUBPOP %in% c("Midwest", "Gulf", "Atlantic") & !(SITE %in% c("OVTN"))) %>%
  select(-(TMAX:TMIN_SM), -(GDD_13C_SM:CPTT_13C)) %>%
  rename(crain_1d = RAIN_SM, crain_5d = CRAIN_5d)
saveRDS(asreml_fl, file = "../data/Weather_related_phe_for_asreml_h2_FL50.rds")
asreml_gr <- jday_phe %>%
  select(PLANT_ID:Long_origin, PLOT_GL, SITE, GR50) %>%
  left_join(wea_df) %>%
  group_by(SITE, PLOT_GL) %>%
  filter(DOY <= as_date(GR50, origin = "2019-01-01")) %>%
  mutate(cgdd_12c_jan2gr = sum(GDD_12C_SM),
         cgdd_8c_jan2gr = sum(GDD_8C_SM)
         ) %>%
  select(-(GDD_13C_SM:CPTT_13C), -(TMAX:Longitude), GDD_12C_SM) %>%
  filter(between(DOY, as_date(GR50, origin = "2019-01-01") - 17, as_date(GR50, origin = "2019-01-01"))) %>%
  mutate(tmax_18d = mean(TMAX_SM),
         tmin_18d = mean(TMIN_SM),
         tave_18d = (tmax_18d + tmin_18d)/2,
         cgdd_12c_18d = mean(GDD_12C_SM)) %>%
  filter(between(DOY, as_date(GR50, origin = "2019-01-01") - 9, as_date(GR50, origin = "2019-01-01"))) %>%
  mutate(tmax_10d = mean(TMAX_SM),
         tmin_10d = mean(TMIN_SM),
         tave_10d = (tmax_10d + tmin_10d)/2,
         cgdd_12c_10d = mean(GDD_12C_SM)) %>%
  filter(between(DOY, as_date(GR50, origin = "2019-01-01") - 4, as_date(GR50, origin = "2019-01-01"))) %>%
  mutate(tmax_5d = mean(TMAX_SM),
         tmin_5d = mean(TMIN_SM),
         tave_5d = (tmax_5d + tmin_5d)/2,
         cgdd_12c_5d = mean(GDD_12C_SM))%>%
  filter(DOY == as_date(GR50, origin = "2019-01-01")) %>%
  filter(SUBPOP %in% c("Midwest", "Gulf", "Atlantic") & !(SITE %in% c("OVTN"))) %>%
  select(-(TMAX_SM:RAIN_SM), -Day_of_year)
saveRDS(asreml_gr, file = "../data/Weather_related_phe_for_asreml_h2_GR50.rds")
```

## ASReml h2 greenup analysis
```{r}
library(tidyverse)
library(asreml)
library(switchgrassGWAS)
library(cowplot)
```

```{r}
source("~/Github/Functions_ggplot-theme-adjustments_2018-01-03.R")
workingdir <- file.path("~", "Github", "pvdiv-phenology-gxe")
outputdir <- file.path(workingdir, "analysis", "heritability")
metadata <- read_rds(file.path(workingdir, "data", "metadata.rds"))
sites <- read_rds(file.path(workingdir, "data", "sites.rds"))
subpops <- read_rds(file.path(workingdir, "data", "subpop_color_coding.rds"))
phe_gr_df <- read_rds(file.path(workingdir, "data",
                                "Weather_related_phe_for_asreml_h2_GR50.rds"))
phe_fl_df <- read_rds(file.path(workingdir, "data",
                                "Weather_related_phe_for_asreml_h2_FL50.rds"))
k_full <- read_rds("~/Github/pvdiv-genome/tensite_twoyear/Kinship_van_Raden_630_individuals_SNPs_r2_20percent.rds")

```



### loop to make heritability table using ASReml mixed models and vpredict.
```{r}
site_v <- list(TX1 = "TX1", TX2 = "TX2", TX3 = "TX3", OK = "OK", MO = "MO",
               NE = "NE", IL = "IL", MI = "MI", SD = "SD", eight_sites = c("TX1", "TX2", "TX3", "OK", "MO", "NE", "MI", "SD"), nine_sites = c("TX1", "TX2", "TX3", "OK", "MO", "NE", "IL", "MI", "SD"))
subpop_v <- list(Gulf = "Gulf", Midwest = "Midwest", Gulf_and_Midwest = c("Gulf", "Midwest"), Atlantic = "Atlantic", three_subpops = c("Atlantic", "Gulf", "Midwest"))
phe_fl <- c("FL50", "dyln_fl50", "dyln_change_sec", "cgdd_12c_gr2fl",
           "crain_gr2fl", "crain_1d", "crain_2d", "crain_3d", "crain_5d",
           "crain_7d")
phe_gr <- c("GR50", "DYLN", "cgdd_12c_jan2gr", "cgdd_8c_jan2gr", "tmax_18d", 
            "tmin_18d", "tave_18d", "cgdd_12c_18d", "tmax_10d", "tmin_10d",
            "tave_10d", "cgdd_12c_10d", "tmax_5d", "tmin_5d", "tave_5d",
            "cgdd_12c_5d")
phe_dfs <- list(phe_gr_df, phe_fl_df)
phe_lists <- list(phe_gr, phe_fl)
h2_table <- tibble()

for(p in 2){#seq_along(phe_dfs)){
  phe_v <- phe_lists[[p]]
  phe <- phe_dfs[[p]]
  
  for(i in seq_along(site_v)){
    for(j in seq_along(subpop_v)){
      for(k in seq_along(phe_v)){
        phe_single <- phe %>%
    left_join(sites, by = "SITE") %>%
    filter(manu_site %in% site_v[[i]] & SUBPOP %in% subpop_v[[j]]) %>%
    rename(phenotype = eval(phe_v[k])) %>%
    mutate(PLANT_ID = as_factor(PLANT_ID),
           SITE = as_factor(SITE),
           PLOT_GL = as_factor(PLOT_GL)) %>%
    as.data.frame()
  
  if(length(site_v[[i]]) > 1){  # need to add site as a random factor if 
    # there is more than one site included in the dataset.
    tryCatch({
     asr_out <- asreml(phenotype ~ 1,
                       random = ~vm(PLANT_ID, k_full) +
                         ~idv(SITE):vm(PLANT_ID, k_full),
                       residual = ~idv(units),
                       data = phe_single, 
                       workspace = "3gb")
  
    h2_est <- vpredict(asr_out, h2 ~ V2/(V1+V2+V3)) %>% as_tibble() %>%
      mutate(site = names(site_v)[i],
             subpop = names(subpop_v)[j],
             phe = phe_v[k],
             bic = summary(asr_out)$bic[1],
             loglik = summary(asr_out)$loglik)
    }, error=function(e){cat("ERROR :",conditionMessage(e), "\n")})
  } else {
    tryCatch({
    asr_out <- asreml(phenotype ~ 1,
                       random = ~vm(PLANT_ID, k_full),
                       residual = ~idv(units),
                       data = phe_single, 
                       workspace = "3gb")
    
    h2_est <- vpredict(asr_out, h2 ~ V1/(V1+V2)) %>% as_tibble() %>%
      mutate(site = names(site_v)[i],
             subpop = names(subpop_v)[j],
             phe = phe_v[k],
             bic = summary(asr_out)$bic[1],
             loglik = summary(asr_out)$loglik)
  }, error=function(e){cat("ERROR :",conditionMessage(e), "\n")})
  }
  
  if(i == 1 & j == 1 & k == 1){
    h2_table <- h2_est
  } else {
    h2_table <- add_row(h2_table, h2_est)
  }
  tryCatch({
      save_plot(filename=file.path(outputdir, paste0("ASReml_performance_",
                                                 names(site_v)[i], "_",
                                                 names(subpop_v)[j],
                                                 "_", phe_v[k], ".png")), 
            plot = plot(asr_out))
    }, error=function(e){cat("ERROR :",conditionMessage(e), "\n")})
  
      }
    }
  }
  write_csv(h2_table, file.path(outputdir, paste0("h2_df_", phe_v[1],
                                                  "_related_", 
                                                  "functions_of_weather_v3.csv")))
}

```

## BLUPs for G 

We next evaluated genetic effects of greenup as functions of GDD and minimum temperature 10 days prior to greenup, and genetic effects of flowering as functions of daylength at flowering and GDD between greenup and flowering. To do this, we calculated BLUPs for each individual using mixed models with common garden and a kinship matrix as random effects. We then conducted genome-wide association on these BLUPs both within and between genetic subpopulations.
```{r}
site_v <- list(eight_sites = c("TX1", "TX2", "TX3", "OK", "MO", "NE", "MI", "SD"))
subpop_v <- list(Gulf = "Gulf", Midwest = "Midwest", Gulf_and_Midwest = c("Gulf", "Midwest"))
phe_fl <- c("FL50", "dyln_fl50", "dyln_change_sec", "cgdd_12c_gr2fl")
phe_gr <- c("GR50", "tmin_10d", "cgdd_12c_10d")
phe_dfs <- list(phe_gr_df, phe_fl_df)
phe_lists <- list(phe_gr, phe_fl)
h2_table <- tibble()

i = 1
j = 1
k = 1
p = 2

for(p in seq_along(phe_dfs)){
  phe_v <- phe_lists[[p]]
  phe <- phe_dfs[[p]]
  
  for(i in seq_along(site_v)){
    for(j in seq_along(subpop_v)){
      for(k in seq_along(phe_v)){
        phe_single <- phe %>%
    left_join(sites, by = "SITE") %>%
    filter(manu_site %in% site_v[[i]] & SUBPOP %in% subpop_v[[j]]) %>%
    rename(phenotype = eval(phe_v[k])) %>%
    mutate(PLANT_ID = as_factor(PLANT_ID),
           SITE = as_factor(SITE),
           PLOT_GL = as_factor(PLOT_GL)) %>%
    as.data.frame()
  
  if(length(site_v[[i]]) > 1){  # need to add site as a random factor if 
    # there is more than one site included in the dataset.
    tryCatch({
     asr_out <- asreml(phenotype ~ 1,
                       random = ~vm(PLANT_ID, k_full) +
                         ~idv(manu_site),
                       residual = ~idv(units),
                       data = phe_single, 
                       workspace = "3gb")
  
    h2_est <- vpredict(asr_out, h2 ~ V2/(V1+V2+V3)) %>% as_tibble() %>%
      mutate(site = names(site_v)[i],
             subpop = names(subpop_v)[j],
             phe = phe_v[k],
             bic = summary(asr_out)$bic[1],
             loglik = summary(asr_out)$loglik)
    blup <- summary(asr_out, coef=TRUE)$coef.random %>% as_tibble(rownames = "Effect")
    write_csv(blup, file.path(outputdir, paste0("blups_", names(site_v)[i], "_",
                                                 names(subpop_v)[j],
                                                 "_", phe_v[k], ".csv")))
    }, error=function(e){cat("ERROR :",conditionMessage(e), "\n")})
  } else {
    tryCatch({
    asr_out <- asreml(phenotype ~ 1,
                       random = ~vm(PLANT_ID, k_full),
                       residual = ~idv(units),
                       data = phe_single, 
                       workspace = "3gb")
    
    h2_est <- vpredict(asr_out, h2 ~ V1/(V1+V2)) %>% as_tibble() %>%
      mutate(site = names(site_v)[i],
             subpop = names(subpop_v)[j],
             phe = phe_v[k],
             bic = summary(asr_out)$bic[1],
             loglik = summary(asr_out)$loglik)
  }, error=function(e){cat("ERROR :",conditionMessage(e), "\n")})
  }
  
  if(i == 1 & j == 1 & k == 1){
    h2_table <- h2_est
  } else {
    h2_table <- add_row(h2_table, h2_est)
  }
  tryCatch({
      save_plot(filename=file.path(outputdir, paste0("ASReml_performance_",
                                                 names(site_v)[i], "_",
                                                 names(subpop_v)[j],
                                                 "_", phe_v[k], ".png")), 
            plot = plot(asr_out))
    }, error=function(e){cat("ERROR :",conditionMessage(e), "\n")})
  
      }
    }
  }
  
}

```

After this I'll need to select from each csv just the individuals in the subpop that I included phenotypes for, to do GWAS on. Then do GWAS with bigsnpr, and go ahead and look at those results. 

# gwas

set up phenotype df for gwas
```{r}
site_v <- list(eight_sites = c("TX1", "TX2", "TX3", "OK", "MO", "NE", "MI", "SD"))
subpop_v <- list(Gulf = "Gulf", Midwest = "Midwest", Gulf_and_Midwest = c("Gulf", "Midwest"))
phe_fl <- c("FL50", "dyln_fl50", "dyln_change_sec", "cgdd_12c_gr2fl")
phe_gr <- c("GR50", "tmin_10d", "cgdd_12c_10d")
phe_dfs <- list(phe_gr_df, phe_fl_df)
phe_lists <- list(phe_gr, phe_fl)

i = 1
j = 1
k = 1
p = 2
phe_gwas <- tibble()

for(j in seq_along(subpop_v)){
  for(p in seq_along(phe_dfs)){
    phe_v <- phe_lists[[p]]
    phe <- phe_dfs[[p]]
    for(i in seq_along(site_v)){
      for(k in seq_along(phe_v)){
        phe_raw <- read_csv(file.path(outputdir, paste0("blups_",
                                                        names(site_v)[i], "_",
                                                        names(subpop_v)[j],
                                                        "_", phe_v[k],
                                                        ".csv")))
        names(phe_raw)[2] <- paste0(phe_v[k], "_G_BLUP")
        phe_proc <- phe_raw %>%
          separate(Effect, into = c("factor", "PLANT_ID"), sep = "\\)_") %>%
          filter(!is.na(PLANT_ID)) %>%
          left_join(metadata, by = "PLANT_ID") %>%
          filter(SUBPOP %in% subpop_v[[j]]) %>%
          select(PLANT_ID, 3)
        if(i == 1 & k == 1 & p == 1){
          phe_gwas <- phe_proc
        } else {
          phe_gwas <- phe_gwas %>%
            left_join(phe_proc, by = "PLANT_ID")
        }
      }
    }
  }
write_csv(phe_gwas, path = file.path(outputdir, paste0("G_BLUP_phenotypes_",
                                                       "for_gwas_GR_FL_",
                                                       names(subpop_v)[j],
                                                       ".csv")))
}
```

```{r}
library(bigsnpr)
library(tidyverse)
library(switchgrassGWAS)
library(AnnotationDbi)
txdb <- loadDb(file = file.path("~", "Github", "pvdiv-genome",
                                "Pvirgatum_516_v5.1.gene.txdb.sqlite"))
workingdir <- file.path("~", "Github", "pvdiv-phenology-gxe")
inputdir <- file.path(workingdir, "analysis", "heritability")
outputdir <- file.path("~", "Github", "pvdiv-phenology-gxe", "analysis", "gwas", "G_BLUP")
subpop_v <- list(Gulf_and_Midwest = c("Gulf", "Midwest"), Midwest = "Midwest", Gulf = "Gulf")

inputfiles <- read_delim("~/Github/pvdiv-phenology-gxe/analysis/gwas/inputkinship.txt", delim = " ", col_names = "SNPfiles")


for(i in seq_along(subpop_v)){
  snp <- snp_attach(inputfiles$SNPfiles[i])
  phe_df <- read_csv(file.path(inputdir, paste0("G_BLUP_phenotypes_",
                                                "for_gwas_GR_FL_",
                                                names(subpop_v)[i], ".csv")))
  pvdiv_standard_gwas(snp, df = phe_df, type = "linear", outputdir = outputdir,
                      savegwas = TRUE, saveannos = TRUE, txdb = txdb,
                      minphe = 130)
}
snp_v <- c("12\\.1M", "8\\.8M", "10\\.3M")
# Make output of top 1000 SNPs
for(i in seq_along(subpop_v)){
  snp <- snp_attach(inputfiles$SNPfiles[i])
  gwas_rds <- pvdiv_results_in_folder(path = outputdir, 
                                      pattern = paste0("GWAS_datatable*.*",
                                                       snp_v[i]))
  phenotypes <- str_sub(gwas_rds, start = 16, end = -43)
  out1 <- pvdiv_bigsnp2mashr(path = outputdir, snp = snp, gwas_rds = gwas_rds, 
                             phenotypes = phenotypes, numSNPs = 1000, 
                             clump = FALSE, scaled = FALSE, saveoutput = TRUE,
                             suffix = paste0("GR_FL_", names(subpop_v)[i]))
}

```
# examine top 1000 SNPs
```{r}
gulf_top1000 <- readRDS("~/Github/pvdiv-phenology-gxe/analysis/gwas/G_BLUP/Part-One-Output_Top-Effects-1000-SNPs_GR_FL_Gulf.rds")
gandm_top1000 <- readRDS("~/Github/pvdiv-phenology-gxe/analysis/gwas/G_BLUP/Part-One-Output_Top-Effects-1000-SNPs_GR_FL_Gulf_and_Midwest.rds")
midw_top1000 <- readRDS("~/Github/pvdiv-phenology-gxe/analysis/gwas/G_BLUP/Part-One-Output_Top-Effects-1000-SNPs_GR_FL_Midwest.rds")
log10p_gulf <- readRDS("~/Github/pvdiv-phenology-gxe/analysis/gwas/G_BLUP/log10p_strong_df_1000topSNPs_GR_FL_Gulf.rds")
log10p_gandm <- readRDS("~/Github/pvdiv-phenology-gxe/analysis/gwas/G_BLUP/log10p_strong_df_1000topSNPs_GR_FL_Gulf_and_Midwest.rds")
log10p_midw <- readRDS("~/Github/pvdiv-phenology-gxe/analysis/gwas/G_BLUP/log10p_strong_df_1000topSNPs_GR_FL_Midwest.rds")
```

```{r}
gulf_top1000 %>%
  arrange(desc(max_score_log10p)) %>%
  ungroup() %>%
  #top_n(100, max_score_log10p) %>%
  left_join(log10p_Gulf_top1000) %>%
  #select(-starts_with("GR50"), -starts_with("tmin"), -starts_with("cgdd_12c_10d"), -marker, -CHRN) %>%
  ggplot(aes(x = FL50_G_BLUP_log10p, y = cgdd_12c_10d_G_BLUP_log10p)) +
  geom_point(aes(shape = as_factor(n_cond_in_top_set), color = max_score_log10p),
             size = 2) +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_viridis(end = 0.9, option = "B") +
  scale_shape_manual(values = c(0, 1, 2, 5, 3)) +
  theme(legend.position = "right")
names(log10p_Gulf_top1000)

midw_top1000 %>%
  arrange(desc(max_score_log10p)) %>%
  ungroup() %>%
  #top_n(100, max_score_log10p) %>%
  left_join(log10p_midwest) %>%
  #select(-starts_with("GR50"), -starts_with("tmin"), -starts_with("cgdd_12c_10d"), -marker, -CHRN) %>%
  ggplot(aes(x = FL50_G_BLU_log10p, y = cgdd_12c_gr2fl_G_BLU_log10p)) +
  geom_point(aes(shape = as_factor(n_cond_in_top_set), color = max_score_log10p),
             size = 2) +
  geom_abline(slope = 1, intercept = 0) +
  scale_color_viridis(end = 0.9, option = "B") +
  scale_shape_manual(values = c(0, 1, 2, 5, 3)) +
  theme(legend.position = "right")
names(log10p_midwest)
midw_top1000 %>%
  arrange(desc(max_score_log10p)) %>%
  ungroup() %>%
  top_n(100, max_score_log10p)
```
Midwest has a lot more top hits for these G BLUPs that are at least on the same chromosomes as Li's QTLs. FWIW. GxE may be better, and need to get Li's intervals again and do a formal comparison, but just at first glance. top hits for Midwest are mostly for dyln_fl50...
```{r}
qtl <- read_csv("~/Github/pvdiv-phenology-gxe/data/QTLsWithFlankMarkers_FL50_GDD_7_SITES_Full_model.csv")
qtldf <- qtl %>%
  separate(flank_lo, into = c("CHR", "POS_lo"), sep = "_", convert = TRUE) %>%
  separate(flank_hi, into = c("CHR1", "POS_hi"), sep = "_", convert = TRUE) %>%
  dplyr::select(-chr, -CHR1, -X1, -lodindex) %>%
  mutate(POS_lo = POS_lo*1000000,
         POS_hi = POS_hi*1000000) %>%
  filter(lodcolumn != "D2F")

i=1
midw_qtl_overlap <- tibble()
gulf_qtl_overlap <- tibble()
gandm_qtl_overlap <- tibble()
for(i in 1:nrow(qtldf)){
  gam_qtl1 <- gandm_top1000 %>%
    filter(CHR == qtldf$CHR[i] & between(POS, qtldf$POS_lo[i], qtldf$POS_hi[i]))
  gulf_qtl1 <- gulf_top1000 %>%
    filter(CHR == qtldf$CHR[i] & between(POS, qtldf$POS_lo[i], qtldf$POS_hi[i]))
  mid_qtl1 <- midw_top1000 %>%
    filter(CHR == qtldf$CHR[i] & between(POS, qtldf$POS_lo[i], qtldf$POS_hi[i]))
  if(i == 1){
    midw_qtl_overlap <- mid_qtl1
    gulf_qtl_overlap <- gulf_qtl1
    gandm_qtl_overlap <- gam_qtl1
  } else {
    midw_qtl_overlap <- midw_qtl_overlap %>%
      full_join(mid_qtl1, by = c("CHR", "POS", "Condition", "max_score_log10p",
                                 "n_cond_in_top_set", "marker", "CHRN"))
    gandm_qtl_overlap <- gandm_qtl_overlap %>%
      full_join(gam_qtl1, by = c("CHR", "POS", "Condition", "max_score_log10p",
                                 "n_cond_in_top_set", "marker", "CHRN"))
    gulf_qtl_overlap <- gulf_qtl_overlap %>%
      full_join(gulf_qtl1, by = c("CHR", "POS", "Condition", "max_score_log10p",
                                 "n_cond_in_top_set", "marker", "CHRN"))
    
  }
} 

i=3
snp <- snp_attach(inputfiles$SNPfiles[i])
markers <- tibble(CHR = snp$map$chromosome, POS = snp$map$physical.pos)
marker_qtl_overlap <- tibble()
for(i in 1:nrow(qtldf)){
  qtl1 <- markers %>%
    filter(CHR == qtldf$CHR[i] & between(POS, qtldf$POS_lo[i], qtldf$POS_hi[i]))
  if(i == 1){
    marker_qtl_overlap <- qtl1
  } else {
    marker_qtl_overlap <- marker_qtl_overlap %>%
      full_join(qtl1, by = c("CHR", "POS"))
  }
} 
nrow(markers)
nrow(marker_qtl_overlap)

nrow(midw_top1000)
midw_qtl_overlap %>%
  arrange(desc(max_score_log10p))
nrow(gandm_top1000)
gandm_qtl_overlap %>%
  arrange(desc(max_score_log10p))
nrow(gulf_top1000)
gulf_qtl_overlap %>%
  arrange(desc(max_score_log10p)) %>%
  group_by(Condition, n_cond_in_top_set) %>% tally()

names(log10p_Gulf_top1000)
log10p_Gulf_top1000 %>%
  ungroup() %>%
  top_n(1000, cgdd_12c_gr2fl_G_BLUP_log10p) %>%
  inner_join(gulf_qtl_overlap)
snpmatrix <- matrix(c(49, 1000, 565807, 10320201), nrow = 2, dimnames = list(Observed = c("SNPs in QTL", "Total SNPs"), Expected = c("Sig SNPs", "Total SNPs")))
fisher.test(snpmatrix, alternative = "t")
```
FL50; dyln_fl50 in Gulf: 6.67 (52 / 1000)
dyln_change_sec in Gulf: 8.21 (64 / 1000)
cgdd_12c_10d_G_BLUP_log10p 37 / 1000 4.75

How many SNPs are in the QTL regions? And how many SNPs are in the full genome?
565807 in QTL region out of 10320201 SNPs in Gulf set. 241 / 5043   5.48%
650092 in QTL region out of 12083955 SNPs in Gulf and Midwest set. 276 / 5320 5.38%
475951 in QTL region out of 8824339 SNPs in Midwest set. Whoa. That's already a striking difference. ~5.39 %ofthegenome.
Therefore how many SNPs would we expect at random?

Fisher's exact test for SNPs above Bonferroni in QTL intervals, vs elsewhere.
```{r}
-log10(0.05/10320201)  # 8.31 Gulf
-log10(0.05/12083955)  # 8.38 GandM
-log10(0.05/8824339)   # 8.25 Midwest
subpop_v
bonferroni_v <- c(-log10(0.05/12083955), -log10(0.05/8824339), 
                  -log10(0.05/10320201))
log10p_list <- list(gandm = log10p_gandm, midwest = log10p_midw, gulf = log10p_gulf)

```

```{r}
sig_df <- tibble()

for(j in seq_along(bonferroni_v)){
  log10p_df <- log10p_list[[j]] %>%
    mutate_at(.vars = vars(ends_with("log10p")),
              .funs = ~ if_else(. >= bonferroni_v[j], ., NA_real_)) %>%
    ungroup %>%
    select(-(1)) 
  if(j == 1){
    sig_df <- enframe(colSums(!is.na(log10p_df[,-1])), name = "Column", 
                      value = paste0("Num_sig_", names(subpop_v)[j])) %>%
      mutate(Column = str_sub(Column, end = 10))
  } else {
    sig1 <- enframe(colSums(!is.na(log10p_df[,-1])), name = "Column", 
                    value = paste0("Num_sig_", names(subpop_v)[j])) %>%
      mutate(Column = str_sub(Column, end = 10))
    sig_df <- sig_df %>%
      full_join(sig1, by = "Column")
  }
}
sig_df
write_csv(sig_df, "G_BLUP_Num_SNPs_above_Bonferroni_GR_FL.csv")
```



```{r}
j = 1
i=4
names(log10p_gandm)[4:ncol(log10p_gandm)]
log10p_df <- log10p_gandm

marker_qtl_overlap <- tibble()

for(i in 4:ncol(log10p_df)){
  phe_use <- names(log10p_df)[i]
  sig_markers <- log10p_df %>%
    select(SNP:POS, eval(phe_use)) %>%
    ungroup() %>%
    filter(!! sym(phe_use) >= bonferroni_v[j])
  
  for(k in 1:nrow(qtldf)){
  qtl1 <- sig_markers %>%
    filter(CHR == qtldf$CHR[k] & between(POS, qtldf$POS_lo[k], qtldf$POS_hi[k]))
  if(k == 1){
    marker_qtl_overlap <- qtl1
  } else {
    marker_qtl_overlap <- marker_qtl_overlap %>%
      full_join(qtl1, by = c("CHR", "POS"))
  }
} 
nrow(sig_markers)
nrow(marker_qtl_overlap)
  
}
log10p_df <- log10p_df %>%
  mutate_at(.vars = vars(ends_with("log10p")), .funs = ~ if_else(. >= bonferroni_v[j], ., NA_real_)) %>%
  ungroup %>%
  select(-(1)) 

colSums(!is.na(log10p_df[,-1]))
marker_qtl_overlap <- tibble()

for(k in 1:nrow(qtldf)){
  phe_use <- names(log10p_df)[3]
  qtl1 <- log10p_df %>%
    filter(!is.na(!! sym(phe_use)) & CHR == qtldf$CHR[k] & between(POS, qtldf$POS_lo[k], qtldf$POS_hi[k]))
  if(k == 1){
    marker_qtl_overlap <- qtl1
  } else {
    marker_qtl_overlap <- marker_qtl_overlap %>%
      full_join(qtl1, by = c("CHR", "POS"))
  }
}
k=1

```


Um, so by Fisher's exact test, there are fewer Midwest SNPs in the sig set than we would expect at random. Rad. What the heck.
And no significant enrichment of Gulf SNPs in the sig set.
And no significant enrichment in the Gulf & Midwest set. So what does this mean?
```{r}
snpmatrix <- matrix(c(115, 4876, 475951, 8824339), nrow = 2, dimnames = list(Observed = c("SNPs in QTL", "Total SNPs"), Expected = c("Sig SNPs", "Total SNPs")))
matrix(c(3, 40, 500, 80278), nrow = 2, dimnames = list(Observed = c("GO in Sample", "Genes in Sample"), Expected = c("GO in Genome", "Genes in Genome")))
fisher.test(snpmatrix, alternative = "t")
(115/4876)/(475951/8824339)
snpmatrix <- matrix(c(241, 5043, 565807, 10320201), nrow = 2, dimnames = list(Observed = c("SNPs in QTL", "Total SNPs"), Expected = c("Sig SNPs", "Total SNPs")))
fisher.test(snpmatrix, alternative = "t")

snpmatrix <- matrix(c(276, 5320, 650092, 12083955), nrow = 2, dimnames = list(Observed = c("SNPs in QTL", "Total SNPs"), Expected = c("Sig SNPs", "Total SNPs")))
fisher.test(snpmatrix, alternative = "t")
```
genematrix <- matrix(c(3, 40, 500, 80278), nrow = 2, dimnames = list(Observed = c("GO in Sample", "Genes in Sample"), Expected = c("GO in Genome", "Genes in Genome")))
fisher.test(GOmatrix, alternative = "t")


# Make new environmental variables for Li to use? Did this in Data_Setup.

We next evaluated genetic effects of greenup as functions of GDD (12C) and minimum temperature 10 days prior to greenup, and genetic effects of flowering as functions of daylength at flowering and GDD between greenup and flowering. 

Can I estimate heritability for the fourway cross?
