---
title: "Plots_manuscript_G_BLUP"
author: "Alice MacQueen"
date: "9/30/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
```

# Load data

Xiaoyu's annotation file, QTL 
```{r}
workingdir <- file.path("~", "Github", "pvdiv-phenology-gxe")
metadata <- read_rds(file.path(workingdir, "data", "metadata.rds"))
sites <- read_rds(file.path(workingdir, "data", "sites.rds"))
ftanno <- read_csv(file.path(workingdir, "data/Pvirgatum_v5.1.annotation_FTGeneHomologs.csv"))
```

# Create dataframes to plot (usually end with "_plot")

## QTL
```{r}
qtldf <- read_csv("~/Github/pvdiv-phenology-gxe/phenology-fourway/5-QTLmappingForAlice/QTLsWithFlankMarkers_FL50_phenotypes_Alice_7_SITES_Full_model.csv")

qtldf <- qtldf %>%
  separate(flank_lo, into = c("CHR", "Pos_Mb_lo"), sep = "_", convert = TRUE) %>%
  separate(flank_hi, into = c("flank", "Pos_Mb_hi"), sep = "_", convert = TRUE) %>%
  mutate(Pos_Mb_lo = round(Pos_Mb_lo, digits = 2),
         Pos_Mb_hi = round(Pos_Mb_hi, digits = 2)) %>%
  select(CHR, Pos_Mb_lo, Pos_Mb_hi, lodcolumn, lod, everything()) %>%
  arrange(CHR, Pos_Mb_lo)

qtlplot <- qtldf %>%
  mutate(phe = case_when(lodcolumn == "FL50" ~ "50% flowering",
                              lodcolumn == "dyln_fl50" ~ "Daylength",
                              lodcolumn == "dyln_change_sec" ~ "daylength change (s)",
                              lodcolumn == "CGDD_12C" ~ "GDD, GR to FL",
                              lodcolumn == "crain_gr2fl" ~ "Rainfall, GR to FL")) %>%
  filter(phe %in% c("Daylength", "GDD, GR to FL")) %>%
  mutate(POS_lo = Pos_Mb_lo * 1000000,
         POS_hi = Pos_Mb_hi * 1000000)

qtlplot$phe <- factor(qtlplot$phe, 
                      levels = c("50% flowering", "Rainfall, GR to FL",
                                 "GDD, GR to FL", "Daylength",
                                          "daylength change (s)",
                                          "Rainfall, 1 day",
                                                  #"Rainfall, 2d sum",
                                                  "Rainfall, 3d sum",
                                                  "Rainfall, 5d sum"#,
                                                  #"Rainfall, 7d sum"
                                                  ))
```

## G BLUP GWAS P Values
```{r}
subpop_v <- list(Gulf_and_Midwest = c("Gulf", "Midwest"), Midwest = "Midwest", Gulf = "Gulf")
set_v <- list(eight_sites = "eight_sites", tx_sites = "tx_sites", north_sites = "north_sites", single_sites = "!sites")
j = 1
i = 1

logpdf <- tibble()
for(i in seq_along(subpop_v)){
  for(j in 1:3){
        logpdf1 <- readRDS(file.path(workingdir, "analysis", "gwas", "BLUPs",
                                  names(set_v)[j],
                                  paste0("log10p_strong_df_1000topSNPs_FL_",
                                         names(set_v)[j], "_", 
                                         names(subpop_v)[i], ".rds"))) %>%
        mutate(subpop = names(subpop_v)[i]) %>% ungroup()
        logpdf <- bind_rows(logpdf, logpdf1)
      }
  }


log10p_long <- logpdf %>%
  select(-starts_with("GR50"), -starts_with("cgdd_12c_10d")) %>%
  pivot_longer(cols = ends_with("log10p"), names_to = "File", values_to = "log10p") %>%
  mutate(phe = case_when(grepl("dyln_fl50", File) ~ "dyln_fl50",
                           grepl("cgdd_12c_gr2fl", File) ~ "cgdd_12c_gr2fl",
                           grepl("FL50", File) ~ "FL50",
                           grepl("GR50", File) ~ "GR50",
                           grepl("crain_gr2fl", File) ~ "crain_gr2fl",
                           grepl("crain_1d", File) ~ "crain_1d",
                           grepl("cgdd_12c_10d", File) ~ "cgdd_12c_10d"),
           sites = case_when(grepl("eight_site", File) ~ "eight_sites",
                             grepl("north_site", File) ~ "north_sites",
                             grepl("tx_site", File) ~ "tx_sites")) %>%
  select(SNP:POS, sites, subpop, phe, log10p) %>%
  filter(!is.na(log10p))
log10p_long$phe <- factor(log10p_long$phe, levels = c("crain_gr2fl", "cgdd_12c_gr2fl", "dyln_fl50", "crain_1d", "FL50"))

log10p_plot <- log10p_long %>%
  mutate(site = case_when(is.na(sites) ~ "all",
                          sites == "eight_sites" ~ "All",
                          sites == "tx_sites" ~ "Texas",
                          sites == "north_sites" ~ "North",
                          TRUE ~ sites),
         subpop_name = case_when(subpop == "Gulf_and_Midwest" ~ "Gulf and Midwest",
                                 subpop == "three_subpops" ~ "all",
                                 TRUE ~ subpop),
         phe = as.character(phe),
         phe_name = case_when(phe == "FL50" ~ "50% flowering",
                              phe == "dyln_fl50" ~ "Daylength",
                              phe == "dyln_change_sec" ~ "daylength change (s)",
                              phe == "cgdd_12c_gr2fl" ~ "GDD, GR to FL",
                              phe == "crain_gr2fl" ~ "Rainfall, GR to FL",
                              phe == "crain_1d" ~ "Rainfall, 1 day",
                              phe == "crain_2d" ~ "Rainfall, 2d sum",
                              phe == "crain_3d" ~ "Rainfall, 3d sum",
                              phe == "crain_5d" ~ "Rainfall, 5d sum",
                              phe == "crain_7d" ~ "Rainfall, 7d sum",
                              TRUE ~ phe)) %>%
  filter(!(phe %in% c("crain_1d", "FL50")) & !(subpop == "Midwest" & between(log10p, 9.242223, 9.242225)) & !(subpop == "Midwest" & between(log10p, 12.466318, 12.46632)))

log10p_plot$site <- factor(log10p_plot$site, levels = c("TX1", "TX2", "TX3", "OK", "MO", "NE", "IL", "MI", "SD", "All", "Texas", "North"))
log10p_plot$phe_name <- factor(log10p_plot$phe_name, 
                               levels = c("50% flowering", "Rainfall, GR to FL",
                                          "GDD, GR to FL", "Daylength",
                                          "daylength change (s)",
                                          "Rainfall, 1 day",
                                                  #"Rainfall, 2d sum",
                                                  "Rainfall, 3d sum",
                                                  "Rainfall, 5d sum"#,
                                                  #"Rainfall, 7d sum"
                                                  ))

log10p_lim <- log10p_plot %>%
  dplyr::group_by(CHR) %>%
  dplyr::summarise(maxP = max(POS)) %>%
  mutate(minP = 0, maxP = maxP+2000000, log10p = 2) %>%
  pivot_longer(names_to = "type", values_to = "POS", cols = minP:maxP)
  
```

## Top annotations from G_BLUP GWAS

```{r}
set_v <- list(eight_sites = "eight_sites", tx_sites = "tx_sites", north_sites = "north_sites", single_sites = "!sites")
j = 1
annos_df <- tibble()
for(j in seq_along(set_v)){
annos1 <- read_csv(file = file.path(workingdir, "analysis", "gwas", "BLUPs",
                                    paste0(names(set_v)[j],     
                                           "_top_500_SNPs_above_FDR",
                                           "_closest_annotations.csv")))
annos_df <- bind_rows(annos_df, annos1)
}

annos_plot <- annos_df %>%
  filter(sites %in% c("eight_sites", "tx_sites", "north_sites") & !(phe %in% c("GR50", "cgdd_12c_10d", "FL50", "crain_1d"))) %>%
  group_by(CHR, POS_Mb, sites, subpop, phe) %>% add_tally(name = "n_SNPs") %>%
  arrange(CHR, POS_Mb) %>%
  #filter(n_SNPs > 1) %>%
  slice(1) %>%
  group_by(CHR, POS_Mb) %>% add_tally(name = "n_Mb") %>%
  filter(n_Mb > 5) %>% arrange(CHR, POS_Mb) %>%
  select(n_SNPs, n_Mb, everything())
annos_plot1 <- annos_plot %>% group_by(CHR, POS_Mb) %>% tally() %>%
  mutate(POS_lo = POS_Mb*1000000,
         POS_hi = POS_Mb*1000000+10000)

annos_plot <- annos_df %>% 
  mutate(POS_bin_lo = floor(POS_Mb*100/2),
         POS_bin_hi = ceiling(POS_Mb*100/2)) %>%
  filter(sites %in% c("eight_sites", "tx_sites", "north_sites") & !(phe %in% c("GR50", "cgdd_12c_10d", "FL50", "crain_1d", "crain_gr2fl"))) %>%
  group_by(CHR, POS_bin_hi, sites, subpop, phe) %>% add_tally(name = "n_SNPs") %>%
  arrange(CHR, POS_bin_hi) %>%
  #filter(n_SNPs > 1) %>%
  slice(1) %>%
  group_by(CHR, POS_bin_hi) %>% add_tally(name = "n_Mb") %>%
  filter(n_Mb > 4) %>% arrange(CHR, POS_bin_hi) %>%
  select(n_SNPs, n_Mb, everything())
annos_plot %>% group_by(CHR, POS_bin_hi, POS_Mb) %>% tally() %>%
  mutate(POS_lo = POS_Mb*1000000,
         POS_hi = POS_Mb*1000000+10000)
```


# G_BLUP Manhattan with overlapping top hits

```{r}
k_subgenome <- log10p_plot %>%
  filter(grepl("K", CHR) & phe != "crain_gr2fl") %>%
  ggplot() +
  geom_rect(data = filter(qtlplot, grepl("K", CHR)), mapping = aes(ymin = 6, ymax = 15, xmin = POS_lo, xmax = POS_hi, fill = phe), alpha = 0.2, color = "grey70", size = 0.4) +
  geom_rect(data = filter(annos_plot1, grepl("K", CHR)), mapping = aes(ymin = 6, ymax = 15, xmin = POS_lo, xmax = POS_hi), fill = "grey70", color = "grey10", linetype = 2) +
  geom_point(aes(x = POS, y = log10p, color = phe_name, shape = subpop_name),
             size = 0.65, stroke = 0.75) +
  geom_blank(data = filter(log10p_lim, grepl("K", CHR)), aes(x = POS, y = log10p)) +
  facet_grid(site ~ CHR, scales = "free_x", space = "free_x", switch = "x") +
      theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), 
              legend.position = "top",
            panel.background = element_rect(fill = "grey90"), panel.spacing.x = unit(0, 'cm')) +
            labs(x = "Chromosome", y = "-log10(p value)") +
            scale_x_continuous(breaks = c(20000000, 40000000, 60000000,
                                          80000000), 
                               labels = c("20 Mb", "40 Mb", "60 Mb", "80 Mb"),
                               expand = c(0.1, 0.1)) +
      scale_color_viridis(discrete = TRUE, begin = 0.35, end = 0.75, option = "B") +
      scale_fill_viridis(discrete = TRUE, begin = 0.35, end = 0.75, option = "B") +
      scale_shape_manual(values = c(1, 4, 6, 4, 16, 4, 17, 18, 14, 3, 4), guide = FALSE) +
  ylim(c(6, 15)) + geom_hline(yintercept = c(8.3), linetype = 2) + labs(subtitle = "K subgenome", x = "")

n_subgenome <- log10p_plot %>%
  filter(grepl("N", CHR) & phe != "crain_gr2fl") %>%
  ggplot() +
  geom_rect(data = filter(qtlplot, grepl("N", CHR)), mapping = aes(ymin = 6, ymax = 15, xmin = POS_lo, xmax = POS_hi, fill = phe), alpha = 0.2, color = "grey70", size = 0.4) +
  geom_rect(data = filter(annos_plot1, grepl("N", CHR)), mapping = aes(ymin = 6, ymax = 15, xmin = POS_lo, xmax = POS_hi), fill = "grey70", color = "grey10", linetype = 2) +
  geom_point(aes(x = POS, y = log10p, color = phe_name, shape = subpop_name),
             size = 0.65, stroke = 0.75) +
  geom_blank(data = filter(log10p_lim, grepl("N", CHR)), aes(x = POS, y = log10p)) +
  facet_grid(site ~ CHR, scales = "free_x", space = "free_x", switch = "x") +
      theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), 
              legend.position = "top",
            panel.background = element_rect(fill = "grey90"), panel.spacing.x = unit(0, 'cm')) +
            labs(x = "Chromosome", y = "-log10(p value)") +
            scale_x_continuous(breaks = c(20000000, 40000000, 60000000,
                                          80000000), 
                               labels = c("20 Mb", "40 Mb", "60 Mb", "80 Mb"),
                               expand = c(0.1, 0.1)) +
      scale_color_viridis(discrete = TRUE, begin = 0.35, end = 0.75, option = "B", guide = FALSE) +
      scale_fill_viridis(discrete = TRUE, begin = 0.35, end = 0.75, option = "B", guide = FALSE) +
      scale_shape_manual(values = c(1, 4, 6, 4, 16, 4, 17, 18, 14, 3, 4)) +
  ylim(c(6, 15)) + geom_hline(yintercept = c(8.3), linetype = 2) + labs(subtitle = "N subgenome", x = "")

both_subgenomes <- plot_grid(k_subgenome, n_subgenome, nrow = 2)
save_plot(paste0("../manuscript/plots/", "G_BLUP_Combined_Manhattans_GDD_dyln_",
                 "subpops_sites_by_subgenome_with_QTL_and_20kb_",
                 "common_associations.png"), base_height = 7, 
          base_width = 7, plot = both_subgenomes)
```


# G_BLUP 20kb windows Upset plot

anno_df: All are above a 10% FDR. If there are more than 500 SNPs above the FDR, only returns the top 500 by p value. Only gets the closest gene to each SNP. Should be three site sets and three subpop sets, several (four?) flowering phenotypes.

ftanno: OsGene and AtGene columns of this df indicate if the gene is a FT homolog in rice (8544 rows) or A. thaliana  (865 rows). Key is the locusName column.

log10p_plot: Top 1000 SNPs for each GWAS. 

qtlplot: Significant QTL for flowering in fourway cross.
```{r}
annos_df
ft_annos <- ftanno %>%
  filter(!is.na(OsGene) | !is.na(AtGene)) %>%
  mutate(Pos_Mb = round(start/10000)/100,
         #Pos_Mb_e = floor(start/10000)/100,
         POS_bin = round(Pos_Mb*100/2))# ,
         #POS_bin_hi = round(Pos_Mb_e*100/2))
  
  # checked that gene start & end never fall into different 20kb bins.   filter(POS_bin_lo != POS_bin_hi)  filter(Pos_Mb_s != Pos_Mb_e)
log10p_plot
qtlplot
```

Are anno of the annos in annos_df in ftanno?
```{r}
annos_df %>%
  inner_join(ft_annos, by = c("Gene ID" = "locusName")) %>%
  group_by(`Gene ID`) %>% tally()
annos_df %>%
  group_by(`Gene ID`) %>% tally() %>%
  inner_join(ft_annos, by = c("Gene ID" = "locusName"))
```
2682 of 19963...

Are any of the 20kb regions in log10p_plot in the 20kb regions in ftanno?
```{r}
log10p_plot %>%
  mutate(Pos_Mb = round(POS/10000)/100) %>%
  inner_join(ft_annos, by = c("CHR" = "seqid", "Pos_Mb"))
```
11324 of 110822 rows...

Do the QTL overlap any of the genes in ftanno?
```{r}
ft_annos %>%
  filter(seqid %in% qtldf$CHR[1] & between(Pos_Mb, qtldf$Pos_Mb_lo[1], qtldf$Pos_Mb_hi[1]))

qtldf
i=27
ft_annos %>%
  filter(seqid %in% qtldf$CHR[i] & between(Pos_Mb, qtldf$Pos_Mb_lo[i], qtldf$Pos_Mb_hi[i])) %>%
  select(OsGene, AtGene, everything())
```

Yes. Lots. Very encouraging.
