---
title: "Plots_manuscript_G_BLUP"
author: "Alice MacQueen"
date: "9/30/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(ComplexUpset)
library(cowplot)

```

# Load data

Xiaoyu's annotation file, and FT keywords for rice from functionally annotated genes
```{r}
workingdir <- file.path("~", "Github", "pvdiv-phenology-gxe")
metadata <- read_rds(file.path(workingdir, "data", "metadata.rds"))
sites <- read_rds(file.path(workingdir, "data", "sites.rds"))
ftanno <- read_csv(file.path(workingdir, "data/Pvirgatum_v5.1.annotation_FTGeneHomologs.csv"))
ftkey <- read_delim(file.path(workingdir, "data", "OsGene_keyword_table.txt"), delim = "\t")
ftanno %>%
  filter(!is.na(OsGene)) %>%
  left_join(ftkey, by = c("OsGene" = "Symbol")) %>%
  filter(Keyword %in% c("flowering time", "heading date")) %>% unique()

osgene_keyword_df <- ftkey %>%
  select(-RAPdb, -MSU) %>%
  group_by(Symbol, Keyword) %>%
  slice(1) %>%
  pivot_wider(names_from = Keyword, values_from = Title)

osgene_ft_key <- osgene_keyword_df %>%
  select(Symbol, `flowering time`, `heading date`, anther, flower, `floral meristem`, `floral meristem determinacy`, flowering, `flower development`, `carpel differentiation`, stamen, `stamen number`) %>%
  pivot_longer(cols = -Symbol, names_to = "Keyword", values_to = "Title") %>%
  filter(!is.na(Title)) %>%
  pivot_wider(names_from = Keyword, values_from = Title)

ft_annos <- ftanno %>%
  filter(OsGene %in% osgene_ft_key$Symbol | !is.na(AtGene)) %>%
  mutate(Pos_Mb = round(start/10000)/100,
         #Pos_Mb_e = floor(start/10000)/100,
         POS_bin = round(Pos_Mb*100/2)) %>%# ,
         #POS_bin_hi = round(Pos_Mb_e*100/2))
  select(OsGene, AtGene, everything())   
1599 / 83260
```
~1600 virgatum genes have rice or thaliana homologs with roles in flowering.
1599 / 83260 total genes. 1.92% of genes.

# Create dataframes to plot (usually end with "_plot")

## QTL
```{r}
qtldf <- read_csv("~/Github/pvdiv-phenology-gxe/phenology-fourway/5-QTLmappingForAlice/QTLsWithFlankMarkers_FL50_phenotypes_Alice_7_SITES_Full_model.csv")

qtldf <- qtldf %>%
  separate(flank_lo, into = c("CHR", "Pos_Mb_lo"), sep = "_", convert = TRUE) %>%
  separate(flank_hi, into = c("flank", "Pos_Mb_hi"), sep = "_", convert = TRUE) %>%
  mutate(Pos_Mb_lo = round(Pos_Mb_lo, digits = 2),
         Pos_Mb_hi = round(Pos_Mb_hi, digits = 2)) %>%
  select(CHR, Pos_Mb_lo, Pos_Mb_hi, lodcolumn, lod, everything()) %>%
  arrange(CHR, Pos_Mb_lo)

qtlplot <- qtldf %>%
  mutate(phe = case_when(lodcolumn == "FL50" ~ "50% flowering",
                              lodcolumn == "dyln_fl50" ~ "Daylength",
                              lodcolumn == "dyln_change_sec" ~ "daylength change (s)",
                              lodcolumn == "CGDD_12C" ~ "GDD, GR to FL",
                              lodcolumn == "crain_gr2fl" ~ "Rainfall, GR to FL")) %>%
  filter(phe %in% c("Daylength", "GDD, GR to FL")) %>%
  mutate(POS_lo = Pos_Mb_lo * 1000000,
         POS_hi = Pos_Mb_hi * 1000000)

qtlplot$phe <- factor(qtlplot$phe, 
                      levels = c("50% flowering", "Rainfall, GR to FL",
                                 "GDD, GR to FL", "Daylength",
                                          "daylength change (s)",
                                          "Rainfall, 1 day",
                                                  #"Rainfall, 2d sum",
                                                  "Rainfall, 3d sum",
                                                  "Rainfall, 5d sum"#,
                                                  #"Rainfall, 7d sum"
                                                  ))
```

## G BLUP GWAS P Values
```{r}
subpop_v <- list(Gulf_and_Midwest = c("Gulf", "Midwest"), Midwest = "Midwest", Gulf = "Gulf")
set_v <- list(eight_sites = "eight_sites", tx_sites = "tx_sites", north_sites = "north_sites", single_sites = "!sites")
j = 1
i = 1

logpdf <- tibble()
for(i in seq_along(subpop_v)){
  for(j in 1:3){
        logpdf1 <- readRDS(file.path(workingdir, "analysis", "gwas", "BLUPs",
                                  names(set_v)[j],
                                  paste0("log10p_strong_df_1000topSNPs_FL_",
                                         names(set_v)[j], "_", 
                                         names(subpop_v)[i], ".rds"))) %>%
        mutate(subpop = names(subpop_v)[i]) %>% ungroup()
        logpdf <- bind_rows(logpdf, logpdf1)
      }
  }


log10p_long <- logpdf %>%
  select(-starts_with("GR50"), -starts_with("cgdd_12c_10d")) %>%
  pivot_longer(cols = ends_with("log10p"), names_to = "File", values_to = "log10p") %>%
  mutate(phe = case_when(grepl("dyln_fl50", File) ~ "dyln_fl50",
                           grepl("cgdd_12c_gr2fl", File) ~ "cgdd_12c_gr2fl",
                           grepl("FL50", File) ~ "FL50",
                           grepl("GR50", File) ~ "GR50",
                           grepl("crain_gr2fl", File) ~ "crain_gr2fl",
                           grepl("crain_1d", File) ~ "crain_1d",
                           grepl("cgdd_12c_10d", File) ~ "cgdd_12c_10d"),
           sites = case_when(grepl("eight_site", File) ~ "eight_sites",
                             grepl("north_site", File) ~ "north_sites",
                             grepl("tx_site", File) ~ "tx_sites")) %>%
  select(SNP:POS, sites, subpop, phe, log10p) %>%
  filter(!is.na(log10p))
log10p_long$phe <- factor(log10p_long$phe, levels = c("crain_gr2fl", "cgdd_12c_gr2fl", "dyln_fl50", "crain_1d", "FL50"))

log10p_plot <- log10p_long %>%
  mutate(site = case_when(is.na(sites) ~ "all",
                          sites == "eight_sites" ~ "All",
                          sites == "tx_sites" ~ "Texas",
                          sites == "north_sites" ~ "North",
                          TRUE ~ sites),
         subpop_name = case_when(subpop == "Gulf_and_Midwest" ~ "Gulf and Midwest",
                                 subpop == "three_subpops" ~ "all",
                                 TRUE ~ subpop),
         phe = as.character(phe),
         phe_name = case_when(phe == "FL50" ~ "50% flowering",
                              phe == "dyln_fl50" ~ "Daylength",
                              phe == "dyln_change_sec" ~ "daylength change (s)",
                              phe == "cgdd_12c_gr2fl" ~ "GDD, GR to FL",
                              phe == "crain_gr2fl" ~ "Rainfall, GR to FL",
                              phe == "crain_1d" ~ "Rainfall, 1 day",
                              phe == "crain_2d" ~ "Rainfall, 2d sum",
                              phe == "crain_3d" ~ "Rainfall, 3d sum",
                              phe == "crain_5d" ~ "Rainfall, 5d sum",
                              phe == "crain_7d" ~ "Rainfall, 7d sum",
                              TRUE ~ phe)) %>%
  filter(!(phe %in% c("crain_1d", "FL50")) & !(subpop == "Midwest" & between(log10p, 9.242223, 9.242225)) & !(subpop == "Midwest" & between(log10p, 12.466318, 12.46632)))

log10p_plot$site <- factor(log10p_plot$site, levels = c("TX1", "TX2", "TX3", "OK", "MO", "NE", "IL", "MI", "SD", "All", "Texas", "North"))
log10p_plot$phe_name <- factor(log10p_plot$phe_name, 
                               levels = c("50% flowering", "Rainfall, GR to FL",
                                          "GDD, GR to FL", "Daylength",
                                          "daylength change (s)",
                                          "Rainfall, 1 day",
                                                  #"Rainfall, 2d sum",
                                                  "Rainfall, 3d sum",
                                                  "Rainfall, 5d sum"#,
                                                  #"Rainfall, 7d sum"
                                                  ))

log10p_lim <- log10p_plot %>%
  dplyr::group_by(CHR) %>%
  dplyr::summarise(maxP = max(POS)) %>%
  mutate(minP = 0, maxP = maxP+2000000, log10p = 2) %>%
  pivot_longer(names_to = "type", values_to = "POS", cols = minP:maxP)

log10p_long %>% arrange(log10p) %>%
  group_by(sites, subpop, phe) %>%
  slice(1) %>%
  arrange(sites, subpop, phe) %>%
  mutate(pval = 10^-log10p) %>%
  arrange(pval)
```

## Top annotations from G_BLUP GWAS

```{r}
set_v <- list(eight_sites = "eight_sites", tx_sites = "tx_sites", north_sites = "north_sites", single_sites = "!sites")
j = 1
annos_df <- tibble()
for(j in seq_along(set_v)){
annos1 <- read_csv(file = file.path(workingdir, "analysis", "gwas", "BLUPs",
                                    paste0(names(set_v)[j],     
                                           "_top_500_SNPs_above_FDR",
                                           "_closest_annotations.csv")))
annos_df <- bind_rows(annos_df, annos1)
}

annos_plot <- annos_df %>%
  filter(sites %in% c("eight_sites", "tx_sites", "north_sites") & !(phe %in% c("GR50", "cgdd_12c_10d", "FL50", "crain_1d"))) %>%
  group_by(CHR, POS_Mb, sites, subpop, phe) %>% add_tally(name = "n_SNPs") %>%
  arrange(CHR, POS_Mb) %>%
  #filter(n_SNPs > 1) %>%
  slice(1) %>%
  group_by(CHR, POS_Mb) %>% add_tally(name = "n_Mb") %>%
  filter(n_Mb > 5) %>% arrange(CHR, POS_Mb) %>%
  select(n_SNPs, n_Mb, everything())
annos_plot1 <- annos_plot %>% group_by(CHR, POS_Mb) %>% tally() %>%
  mutate(POS_lo = POS_Mb*1000000,
         POS_hi = POS_Mb*1000000+10000)

annos_plot <- annos_df %>% 
  mutate(POS_bin_lo = floor(POS_Mb*100/2),
         POS_bin_hi = ceiling(POS_Mb*100/2)) %>%
  filter(sites %in% c("eight_sites", "tx_sites", "north_sites") & !(phe %in% c("GR50", "cgdd_12c_10d", "FL50", "crain_1d", "crain_gr2fl"))) %>%
  group_by(CHR, POS_bin_hi, sites, subpop, phe) %>% add_tally(name = "n_SNPs") %>%
  arrange(CHR, POS_bin_hi) %>%
  #filter(n_SNPs > 1) %>%
  slice(1) %>%
  group_by(CHR, POS_bin_hi) %>% add_tally(name = "n_Mb") %>%
  filter(n_Mb > 4) %>% arrange(CHR, POS_bin_hi) %>%
  select(n_SNPs, n_Mb, everything())
annos_plot %>% group_by(CHR, POS_bin_hi, POS_Mb) %>% tally() %>%
  mutate(POS_lo = POS_Mb*1000000,
         POS_hi = POS_Mb*1000000+10000)

annos_df %>% 
  arrange(desc(`FDR Adjusted p value`)) %>%
  group_by(sites, subpop, phe) %>%
  slice(1) %>%
  arrange(sites, subpop, phe) %>%
  filter(sites %in% c("eight_sites", "tx_sites", "north_sites") & !(phe %in% c("GR50", "cgdd_12c_10d", "FL50", "crain_1d", "crain_gr2fl")) & !subpop %in% c("Gulf_and_Midwest")) %>%
  arrange(`FDR Adjusted p value`)
  
  

```
# Get all SNPs above FDR from server.
phe %in% c("dyln_fl50", "cgdd_12c_gr2fl", "FL50")
sites %in% c("eight_sites", "tx_sites", "north_sites")
```{r}
phe_v <- c("dyln_fl50", "cgdd_12c_gr2fl", "FL50")
site_v <- c("eight_sites", "tx_sites", "north_sites")
i=1
j=1
gwas_FDR <- tibble()
for(i in seq_along(phe_v)){
  for(j in seq_along(site_v)){
  gwas_files <- list.files(path = paste0("./", site_v[j]), pattern = paste0("GWAS_datatable_", phe_v[i]))
  for(k in seq_along(gwas_files)){
    gwas1 <- readRDS(file.path(site_v[j], gwas_files[k])) %>%
      filter(FDR_adj < 0.1) %>%
      mutate(sites = site_v[j],
             phe = phe_v[i],
             gwas = gwas_files[k])
    gwas_FDR <- bind_rows(gwas_FDR, gwas1)
  }
}
}
saveRDS(gwas_FDR, file = "Univariate_G_BLUP_27_flowering_GWAS_SNPs_above_FDR10per.rds")
```

# FDR GWAS hits
```{r}
gwas_fdr <- read_rds("../analysis/gwas/BLUPs/Univariate_G_BLUP_27_flowering_GWAS_SNPs_above_FDR10per.rds")

gwas_fdr <- gwas_fdr %>%
  mutate(subpop = case_when(grepl("8\\.8M", gwas) ~ "Midwest",
                              grepl("10\\.3M", gwas) ~ "Gulf",
                              grepl("12\\.1M", gwas) ~ "Gulf_and_Midwest"),
         Pos_Mb = round(POS/10000)/100,
         POS_bin = round(Pos_Mb*100/2),
         Pos_Mb1 = round(Pos_Mb, digits = 1)) %>%
  filter(!(subpop == "Midwest" & between(log10p, 9.242223, 9.242225)) & !(subpop == "Midwest" & between(log10p, 12.466318, 12.46632)))

gwas_fdr %>%
  filter(subpop %in% c("Midwest", "Gulf") & phe %in% c("dyln_fl50", "cgdd_12c_gr2fl")) %>%
  group_by(sites, phe, subpop) %>%
  tally()

gwas_fdr %>%
  filter(subpop %in% c("Midwest", "Gulf") & phe %in% c("dyln_fl50", "cgdd_12c_gr2fl")) %>%
  group_by(CHR, POS) %>%
  tally() %>% arrange(desc(n))

gwas_fdr %>%
  filter(subpop %in% c("Midwest", "Gulf") & phe %in% c("dyln_fl50", "cgdd_12c_gr2fl")) %>%
  group_by(CHR, Pos_Mb1, POS_bin) %>%
  tally() %>% arrange(desc(n))
annos_plot1
ft_annos

gwas_fdr %>%
  filter(subpop %in% c("Midwest", "Gulf") & phe %in% c("dyln_fl50", "cgdd_12c_gr2fl")) %>%
  group_by(CHR, POS_bin) %>%
  tally() %>% arrange(desc(n))

gwas_fdr %>%
  group_by(log10p) %>%
  tally() %>% arrange(desc(n))
```
## Upset plot FDR
```{r}
fdr_upset <- gwas_fdr %>%
  filter(subpop %in% c("Midwest", "Gulf") & phe %in% c("dyln_fl50", "cgdd_12c_gr2fl")) %>%
  select(sites, subpop, phe, CHR, POS_bin) %>%
  unite(col = "GWAS", sites, subpop, phe) %>%
  unite(col = "CHR_POS", CHR, POS_bin) %>%
  mutate(Sig = TRUE) %>%
  complete(GWAS, CHR_POS, fill = list(Sig = FALSE)) %>%
  group_by(GWAS, CHR_POS, Sig) %>%
  slice(1) %>%
  pivot_wider(names_from = GWAS, values_from = Sig)
upset_v0 <- names(fdr_upset)[2:ncol(fdr_upset)]
freqplot_u <- upset(data = fdr_upset, intersect = upset_v0, 
      name="20kb region associations by univariate GWAS",
      base_annotations=list(
        'Intersection size'=intersection_size(text_aes=aes(label=paste0(round(intersection_size/nrow(fdr_upset) * 100), '%')))),
      min_size = 10,
      width_ratio = 0.125,
      themes=upset_modify_themes(
        list(
            'overall_sizes'=theme(axis.text.x=element_text(angle=45, hjust =1))
        ))) +
  labs(caption = "Number of 20kb windows with a significant GWAS hit.")
save_plot(filename = "../writeup/figures/Upset_plot_20kb_sig_allele_freq.png", plot = freqplot_u, base_height = 3)
```

Check out various smaller plots
```{r}
fdr_upset <- gwas_fdr %>%
  filter(subpop %in% c("Midwest", "Gulf") & !phe %in% c("FL50")) %>%
  select(sites, subpop, phe, CHR, POS_bin) %>%
  unite(col = "GWAS", sites, subpop, phe) %>%
  unite(col = "CHR_POS", CHR, POS_bin) %>%
  mutate(Sig = TRUE) %>%
  complete(GWAS, CHR_POS, fill = list(Sig = FALSE)) %>%
  group_by(GWAS, CHR_POS, Sig) %>%
  slice(1) %>%
  pivot_wider(names_from = GWAS, values_from = Sig)
upset_v0 <- names(fdr_upset)[2:ncol(fdr_upset)]
upset(data = fdr_upset, intersect = upset_v0, 
      name="20kb region associations by univariate GWAS",
      base_annotations=list(
        'Intersection size'=intersection_size()),
      min_size = 0,
      width_ratio = 0.125,
      themes=upset_modify_themes(
        list(
            'overall_sizes'=theme(axis.text.x=element_text(angle=45, hjust =1))
        ))) +
  labs(caption = "Number of 20kb windows with a significant GWAS hit.")


three_uni_gwas <- gwas_fdr %>%
  filter(subpop %in% c("Midwest", "Gulf") & !phe %in% c("FL50")) %>%
  group_by(sites, subpop, phe, CHR, POS_bin) %>%
  tally() %>%
  group_by(CHR, POS_bin) %>%
  summarise(count = n()) %>% arrange(desc(count)) %>%
  filter(count > 2)

three_uni_gwas %>%
  arrange(CHR, POS_bin) %>%
  left_join(ft_annos, by = c("CHR" = "seqid", "POS_bin")) %>%
  select(CHR, Pos_Mb, everything())

gwas_fdr %>%
  filter(log10p > 7) %>%
  filter(subpop %in% c("Gulf", "Midwest") & !phe %in% c("FL50")) %>%
  group_by(sites, subpop, phe, CHR, POS_bin) %>%
  tally() %>%
  group_by(CHR, POS_bin) %>%
  summarise(count = n()) %>%
  left_join(ft_annos, by = c("CHR" = "seqid", "POS_bin")) %>%
  select(CHR, Pos_Mb, everything()) %>%
  filter(!is.na(OsGene) | !is.na(AtGene)) %>%
  left_join(osgene_ft_key, by = c("OsGene" = "Symbol")) # 93 out of 3148 rows contain candidates
(9/223)/(1599/83260) # all regions log10p > 7   OR 2.1
(26/793)/(1599/83260) # all regions log10p > 6   OR 1.7
(65/2240)/(1599/83260) # all regions log10p > 5   OR 1.51
(10/280)/(1599/83260) # all regions FDR 1%
(43/1393)/(1599/83260) # all regions FDR 5%
(95/3137)/(1599/83260) # all regions FDR 10%

(10/268)/(1599/83260)
(81/2551)/(1599/83260) # OR north sites 1.65
(8/420)/(1599/83260) # OR tx sites 0.99! NS!
(16/463)/(1599/83260) # OR all sites 1.80
(95/3137)/(1599/83260)
fisher.test(matrix(c(95, 1599, 3137, 83260), nrow = 2))
fisher.test(matrix(c(16, 1599, 463, 83260), nrow = 2))
fisher.test(matrix(c(81, 1599, 2551, 83260), nrow = 2))
fisher.test(matrix(c(8, 1599, 420, 83260), nrow = 2))
```
For the Midwest subpop, there aren't sig assoc for dyln in TX. Just CGDD.
For the Gulf, there aren't sig assoc for CGDD in TX. Just dyln.
For G&M in Texas, 61% signals are just for dyln. 33% both dyln & CGDD. 6% CGDD.
For G&M in North, 76% signals are just for CGDD. 20% both. 2% dyln.

2K 52: FT
3K 19 FTL10

# Total # 20kb regions
Midwest: 8824339 SNPs       52797 20kb regions
Gulf: 10320201 SNPs         53025 20kb regions
Gulf_and_Midwest: 12083955  53086 20kb regions
53240 20kb regions total
```{r}
phe_v <- c("dyln_fl50", "cgdd_12c_gr2fl", "FL50")
site_v <- c("eight_sites", "tx_sites", "north_sites")
i=1
j=1
k=1

  gwas_files <- list.files(path = paste0("./", site_v[j]), pattern = paste0("GWAS_datatable_", phe_v[i]))
  
  gwas_files
  
  gwas_20kb <- tibble()
subpop_v <- c("Midwest", "Gulf", "Gulf_and_Midwest")
  
for(k in seq_along(gwas_files)){
  gwas1 <- readRDS(file.path(site_v[j], gwas_files[k]))
gwas1 <- gwas1 %>%
  mutate(Pos_Mb = round(POS/10000)/100,
         POS_bin = round(Pos_Mb*100/2)) %>%
  group_by(CHR, POS_bin) %>% 
  tally() %>%
  mutate(subpop = subpop_v[k])
gwas_20kb <- bind_rows(gwas_20kb, gwas1)
}
```


# Overlaps phe across subpops

Associations did or did not overlap significantly in the same phenotypes among subpopulations? What about in the same subpop between sets of sites? This is GxE at the continental scale, point out.
In both Gulf & Midwest:
All  dyln_fl50  Chr01N	229	2	   1 region 1/463     NS 0.08496942
All cgdd_12c_gr2fl  0 regions   0/2
All FL50  0 regions   0/18
North  dyln_fl50 39 regions  39/2530      sig 0.00688
North cgdd_12c_gr2fl  0 regions  0/59
North  FL50    6 regions 5Nx2 6Kx2 8N 9K   6/1097     NS 0.141
Texas  dyln_fl50         0 regions  0/232
Texas  cgdd_12c_gr2fl    0 regions  0/190
Texas  FL50             0 regions   0/64
```{r}
sitev <- "eight_sites"
phev <- "dyln_fl50"
popv <- list(Gulf = 53025, Midwest = 52797, Gulf_and_Midwest = 53086)
i=2

AandB <- gwas_fdr %>%
  filter(subpop %in% names(popv)[c(1,2)] & phe %in% phev & sites %in% sitev) %>%
    group_by(sites, subpop, phe, CHR, POS_bin) %>%
    tally() %>%
    group_by(CHR, POS_bin) %>%
    summarise(count = n()) %>%
    filter(count > 1)
inB <- gwas_fdr %>%
  filter(subpop %in% names(popv)[1] & phe %in% phev & sites %in% sitev) %>%
    group_by(sites, subpop, phe, CHR, POS_bin) %>%
    tally() %>%
    group_by(CHR, POS_bin) %>%
    summarise(count = n()) 
notinB <- popv[[i]] - nrow(inB)
inA <- gwas_fdr %>%
  filter(subpop %in% names(popv)[2] & phe %in% phev & sites %in% sitev) %>%
    group_by(sites, subpop, phe, CHR, POS_bin) %>%
    tally() %>%
    group_by(CHR, POS_bin) %>%
    summarise(count = n()) 

phyper(nrow(AandB), nrow(inB), notinB, nrow(inA), lower.tail = FALSE)
```

Eight & Texas:
In Gulf
dyln_fl50  22/274
FL50
cgdd_12c_gr2fl 0/2

## Hypergeometric test for significance of overlaps
phyper(10, 50, 200 - 50, 25, lower.tail = FALSE)
phyper(A&B, inB, notinB, inA, lower.tail = FALSE)
```{r}
A <- "north_sites"
B <- "tx_sites"
phev <- "cgdd_12c_gr2fl"
popv <- list(Gulf = 53025, Midwest = 52797, Gulf_and_Midwest = 53086)
i=2

AandB <- gwas_fdr %>%
  filter(subpop %in% names(popv)[i] & phe %in% phev & sites %in% c(A,B)) %>%
    group_by(sites, subpop, phe, CHR, POS_bin) %>%
    tally() %>%
    group_by(CHR, POS_bin) %>%
    summarise(count = n()) %>%
    filter(count > 1)
inB <- gwas_fdr %>%
  filter(subpop %in% names(popv)[i] & phe %in% phev & sites %in% c(B)) %>%
    group_by(sites, subpop, phe, CHR, POS_bin) %>%
    tally() %>%
    group_by(CHR, POS_bin) %>%
    summarise(count = n()) 
notinB <- popv[[i]] - nrow(inB)
inA <- gwas_fdr %>%
  filter(subpop %in% names(popv)[i] & phe %in% phev & sites %in% c(A)) %>%
    group_by(sites, subpop, phe, CHR, POS_bin) %>%
    tally() %>%
    group_by(CHR, POS_bin) %>%
    summarise(count = n()) 

phyper(nrow(AandB), nrow(inB), notinB, nrow(inA), lower.tail = FALSE)
8/232
```
tx & north dyln Gulf 0.0078 (8 overlaps)
eight & north cgdd Gulf 1.13e-06
eight & north dyln Gulf 5.36e-31 (25 overlaps)
eight & tx dyln Gulf 2.2e-38 (22 overlaps)

eight & north dyln Midwest 4.42e-242
Very few overlaps between site sets for the Midwest subpopulation.
Lots of overlaps & sig for Gulf_and_Midwest, if we trust this.

# Manhattan with overlapping top hits (G_BLUP)

```{r}
k_subgenome <- log10p_plot %>%
  filter(grepl("K", CHR) & phe != "crain_gr2fl") %>%
  ggplot() +
  geom_rect(data = filter(qtlplot, grepl("K", CHR)), mapping = aes(ymin = 6, ymax = 15, xmin = POS_lo, xmax = POS_hi, fill = phe), alpha = 0.2, color = "grey70", size = 0.4) +
  geom_rect(data = filter(annos_plot1, grepl("K", CHR)), mapping = aes(ymin = 6, ymax = 15, xmin = POS_lo, xmax = POS_hi), fill = "grey70", color = "grey10", linetype = 2) +
  geom_point(aes(x = POS, y = log10p, color = phe_name, shape = subpop_name),
             size = 0.65, stroke = 0.75) +
  geom_blank(data = filter(log10p_lim, grepl("K", CHR)), aes(x = POS, y = log10p)) +
  facet_grid(site ~ CHR, scales = "free_x", space = "free_x", switch = "x") +
      theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), 
              legend.position = "top",
            panel.background = element_rect(fill = "grey90"), panel.spacing.x = unit(0, 'cm')) +
            labs(x = "Chromosome", y = "-log10(p value)") +
            scale_x_continuous(breaks = c(20000000, 40000000, 60000000,
                                          80000000), 
                               labels = c("20 Mb", "40 Mb", "60 Mb", "80 Mb"),
                               expand = c(0.1, 0.1)) +
      scale_color_viridis_d(begin = 0.35, end = 0.75, option = "B") +
      scale_fill_viridis_d(begin = 0.35, end = 0.75, option = "B") +
      scale_shape_manual(values = c(1, 4, 6, 4, 16, 4, 17, 18, 14, 3, 4), guide = FALSE) +
  ylim(c(6, 15)) + geom_hline(yintercept = c(8.3), linetype = 2) + labs(subtitle = "K subgenome", x = "")

n_subgenome <- log10p_plot %>%
  filter(grepl("N", CHR) & phe != "crain_gr2fl") %>%
  ggplot() +
  geom_rect(data = filter(qtlplot, grepl("N", CHR)), mapping = aes(ymin = 6, ymax = 15, xmin = POS_lo, xmax = POS_hi, fill = phe), alpha = 0.2, color = "grey70", size = 0.4) +
  geom_rect(data = filter(annos_plot1, grepl("N", CHR)), mapping = aes(ymin = 6, ymax = 15, xmin = POS_lo, xmax = POS_hi), fill = "grey70", color = "grey10", linetype = 2) +
  geom_point(aes(x = POS, y = log10p, color = phe_name, shape = subpop_name),
             size = 0.65, stroke = 0.75) +
  geom_blank(data = filter(log10p_lim, grepl("N", CHR)), aes(x = POS, y = log10p)) +
  facet_grid(site ~ CHR, scales = "free_x", space = "free_x", switch = "x") +
      theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), 
              legend.position = "top",
            panel.background = element_rect(fill = "grey90"), panel.spacing.x = unit(0, 'cm')) +
            labs(x = "Chromosome", y = "-log10(p value)") +
            scale_x_continuous(breaks = c(20000000, 40000000, 60000000,
                                          80000000), 
                               labels = c("20 Mb", "40 Mb", "60 Mb", "80 Mb"),
                               expand = c(0.1, 0.1)) +
      scale_color_viridis(discrete = TRUE, begin = 0.35, end = 0.75, option = "B", guide = FALSE) +
      scale_fill_viridis(discrete = TRUE, begin = 0.35, end = 0.75, option = "B", guide = FALSE) +
      scale_shape_manual(values = c(1, 4, 6, 4, 16, 4, 17, 18, 14, 3, 4)) +
  ylim(c(6, 15)) + geom_hline(yintercept = c(8.3), linetype = 2) + labs(subtitle = "N subgenome", x = "")

both_subgenomes <- plot_grid(k_subgenome, n_subgenome, nrow = 2)
save_plot(paste0("../manuscript/plots/", "G_BLUP_Combined_Manhattans_GDD_dyln_",
                 "subpops_sites_by_subgenome_with_QTL_and_20kb_",
                 "common_associations.png"), base_height = 7, 
          base_width = 7, plot = both_subgenomes)
```

# Definitions
# Rice/thaliana FT

anno_df: All are above a 10% FDR. If there are more than 500 SNPs above the FDR, only returns the top 500 by p value. Only gets the closest gene to each SNP. Should be three site sets and three subpop sets, several (four?) flowering phenotypes.

ftanno: OsGene and AtGene columns of this df indicate if the gene is a FT homolog in rice (8544 rows) or A. thaliana  (865 rows). Key is the locusName column.

log10p_plot: Top 1000 SNPs for each GWAS. 

qtlplot: Significant QTL for flowering in fourway cross.
```{r}
annos_df %>%
  arrange(desc(`FDR Adjusted p value`))
ft_annos
osgene_ft_key
  
  # checked that gene start & end never fall into different 20kb bins.   filter(POS_bin_lo != POS_bin_hi)  filter(Pos_Mb_s != Pos_Mb_e)
log10p_plot
qtlplot
```

Are anno of the annos in annos_df in ftanno?
```{r}
annos_df %>%
  inner_join(ft_annos, by = c("Gene ID" = "locusName")) %>%
  group_by(`Gene ID`) %>% tally()
annos_df %>%
  group_by(`Gene ID`) %>% tally() %>%
  inner_join(ft_annos, by = c("Gene ID" = "locusName"))

ft_annos <- ft_annos %>%
  mutate(Pos_Mb1 = round(Pos_Mb, digits = 1))
annos_plot %>%
  mutate(Pos_Mb1 = round(POS_Mb, digits = 1)) %>%
  inner_join(ft_annos, by = c("CHR" = "seqid", "Pos_Mb1"))
```
528 of 19963...
528 of 1599

Are any of the 20kb regions in log10p_plot in the 20kb regions in ftanno?
```{r}
log10p_plot %>%
  mutate(Pos_Mb = round(POS/10000)/100) %>%
  inner_join(ft_annos, by = c("CHR" = "seqid", "Pos_Mb"))
```
1780 of 110822 rows...

Do the QTL overlap any of the genes in ftanno?
```{r}
ft_annos %>%
  filter(seqid %in% gtldf3$CHR[1] & between(Pos_Mb, gtldf3$Pos_Mb_lo[1], gtldf3$Pos_Mb_hi[1]))

gtldf3 <- qtldf %>% filter(lod > 30 & lodcolumn == "dyln_fl50")
i=4
ft_annos %>%
  filter(seqid %in% gtldf3$CHR[i] & between(Pos_Mb, gtldf3$Pos_Mb_lo[i], gtldf3$Pos_Mb_hi[i])) %>%
  select(OsGene, AtGene, everything()) %>%
  left_join(osgene_ft_key, by = c("OsGene" = "Symbol"))
```
## start here QTL candidate genes
Chr02N	57.23	58.81	   SIP1|OsSUF4	SUF4
SIP1 participates in regulation of flowering time in rice by recruiting OsTrx1 to Ehd1.

Chr04K	4.23	5.50     6 genes including Hd3a	FT
	Hd3a, a rice ortholog of the Arabidopsis FT gene, promotes transition to flowering downstream of Hd1 under short-day conditions
	Phytochrome dependent quantitative control of Hd3a transcription is the basis of the night break effect in rice flowering
	Variations in Hd1 proteins, Hd3a promoters, and Ehd1 expression levels contribute to diversity of flowering time in cultivated rice
	
Chr04K	9.92	13.93    3 genes including Hd3a	FT ( a different copy)
Hd3a, a rice ortholog of the Arabidopsis FT gene, promotes transition to flowering downstream of Hd1 under short-day conditions
	Phytochrome dependent quantitative control of Hd3a transcription is the basis of the night break effect in rice flowering
	OsMFT1 increases spikelets per panicle and delays heading date in rice by suppressing Ehd1, FZP and SEPALLATA-like genes.

Chr05N	2.04	3.44     5 genes including FT and OsFTL1	FT &&&&& OSMADS3:
Genetic Enhancer Analysis Reveals that FLORAL ORGAN NUMBER2 and OsMADS3 Cooperatively Regulate Maintenance and Determinacy of the Flower Meristem in Rice.

Chr05N	64.27	65.98	  3 genes including OsFTL10 and OsABF1|OsABI5|OREB1|OsbZIP10: 	A Drought-inducible bZIP Transcription Factor OsABF1 Delays Reproductive Timing in Rice.

Yes. Lots. Very encouraging.

# ----------------------------
# Upset plots
## 20kb windows Upset plot (G_BLUP)

```{r}
#annos_upset <- 
annos_df %>% 
  mutate(POS_bin_lo = floor(POS_Mb*100/2),
         POS_bin_hi = ceiling(POS_Mb*100/2)) %>%
  filter(sites %in% c("eight_sites", "tx_sites", "north_sites") & !(phe %in% c("GR50", "cgdd_12c_10d", "FL50", "crain_1d", "crain_gr2fl"))) %>%
  select(sites, subpop, phe, CHR, POS_Mb) %>%
  mutate(All = case_when(sites == "eight_sites" ~ TRUE,
                         TRUE ~ FALSE),
         North = case_when(sites == "north_sites" ~ TRUE,
                         TRUE ~ FALSE),
         Texas = case_when(sites == "tx_sites" ~ TRUE,
                         TRUE ~ FALSE),
         Midwest = case_when(subpop == "Midwest" ~ TRUE,
                         TRUE ~ FALSE),
         Gulf = case_when(subpop == "Gulf" ~ TRUE,
                         TRUE ~ FALSE),
         Gulf_and_Midwest = case_when(subpop == "Gulf_and_Midwest" ~ TRUE,
                         TRUE ~ FALSE),
         )

upsetdf1 <- annos_df %>% 
  mutate(POS_bin_lo = floor(POS_Mb*100/2),
         POS_bin_hi = ceiling(POS_Mb*100/2)) %>%
  filter(sites %in% c("eight_sites", "tx_sites", "north_sites") & !(phe %in% c("GR50", "cgdd_12c_10d", "FL50", "crain_1d", "crain_gr2fl"))) %>%
  select(sites, subpop, phe, CHR, POS_Mb) %>%
  unite(col = "Group", sites, subpop, phe) %>%
  unite(col = "CHR_POS", CHR, POS_Mb) %>%
  mutate(Sig = TRUE) %>%
  complete(Group, CHR_POS, fill = list(Sig = FALSE)) %>%
  group_by(Group, CHR_POS, Sig) %>%
  slice(1) %>%
  pivot_wider(names_from = Group, values_from = Sig)
upset_v1 <- names(upsetdf1)[2:16]
```


```{r}
freqplot_u <- upset(data = upsetdf1, intersect = upset_v1, 
      name="20kb region frequency changes by sites/subpops/phenotype",
      base_annotations=list(
        'Intersection size'=intersection_size(text_aes=aes(label=paste0(round(intersection_size/nrow(upsetdf1) * 100), '%')))),
      min_size = 10,
      width_ratio = 0.125,
      themes=upset_modify_themes(
        list(
            'overall_sizes'=theme(axis.text.x=element_text(angle=45, hjust =1))
        ))) +
  labs(caption = "Number of 20kb windows with a significant GWAS hit.")
save_plot(filename = "../writeup/figures/Upset_plot_20kb_sig_allele_freq.png", plot = freqplot_u, base_height = 3)
```

