---
title: "Plots_manuscript_v0.3"
author: "Alice MacQueen"
date: "7/22/2020"
output: html_document
---

Add any libraries/scripts that need to be loaded for these analyses here.
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(cowplot)
library(viridis)
library(lubridate)
library(switchgrassGWAS)
library(GGally)
library(maps)
library(ggmap)
usa <- map_data("usa")
source("~/Github/Functions_ggplot-theme-adjustments_2018-01-03.R")
```

# Load data
Add any data that needs to be loaded for these analyses here.
```{r}
workingdir <- file.path("~", "Github", "pvdiv-phenology-gxe")
metadata <- read_rds(file.path(workingdir, "data", "metadata.rds"))
sites <- read_rds(file.path(workingdir, "data", "sites.rds"))
subpops <- read_rds(file.path(workingdir, "data", "subpop_color_coding.rds"))
phe_gr_df <- read_rds(file.path(workingdir, "data",
                                "Weather_related_phe_for_asreml_h2_GR50.rds"))
phe_fl_df <- read_rds(file.path(workingdir, "data",
                                "Weather_related_phe_for_asreml_h2_FL50.rds"))
phenotypes <- readRDS("~/Github/pvdiv-phenology-gxe/data/Julian_date_GDD_and_rainfall_phenotypes_4wcr_and_gwas.rds")
h2gr <- read_csv(file.path(workingdir, "analysis", "heritability", "kinship_subpop", "h2_df_GR50_related_functions_of_weather_v1.csv"))
h2fl <- read_csv(file.path(workingdir, "analysis", "heritability", "kinship_subpop", "h2_df_FL50_related_functions_of_weather_v1.csv"))
h2gr_gxe <- read_csv(file.path(workingdir, "analysis", "heritability", "kinship_subpop", "gxe_and_h2_df_GR50_related_functions_of_weather_v1.csv"))
h2fl_gxe <- read_csv(file.path(workingdir, "analysis", "heritability", "kinship_subpop", "gxe_and_h2_df_FL50_related_functions_of_weather_v1.csv"))

h2gr <- h2gr %>%
  full_join(h2gr_gxe)
h2fl <- h2fl %>%
  full_join(h2fl_gxe)
```a the code for any new analyses that I'm running to do new writeup for the manuscript v0.3.

v0.2 has the ASReml h2 analysis, has BLUPs for G (which as of v0.3 probably needs a GxE term... define some possibilities below.)

Results: h2 at single sites. Then h2 at all eight sites. And so there must be GxE, and there must be a lot of rank-changing GxE, in that we canâ€™t predict ranks across sites.

To show this, make barplots of the different variables and facet them by site. Then, show heritability across all eight sites (in a model with a random effect of site, interacted with some number of eigenvectors from a SVD of the kinship matrix.)

# Map Fig 1A

```{r}
subpops
library(ggrepel)

metaplot <- metadata %>%
  filter(SUBPOP %in% c("Gulf", "Midwest") & !is.na(LATITUDE)) %>%
  mutate(LONGITUDE = -LONGITUDE)
metaplot$ECOTYPE_NNET <- factor(metaplot$ECOTYPE_NNET, levels = c("Coastal", "Lowland", "Upland", "Unknown"))
siteplot <- sites %>%
  filter(!(SITE %in% c("FRMI", "OVTN")))

ggplot() + 
  geom_polygon(data = usa, aes(x=long, y = lat, group = group), 
               fill = "white", color = "black") +
  coord_quickmap(xlim = c(-105, -66), ylim = c(25, 50)) +
  geom_jitter(data = metaplot, aes(x = LONGITUDE, y = LATITUDE, 
                                   color = SUBPOP, shape = ECOTYPE_NNET),
              width = 0.45, height = 0.45) + 
  geom_point(data = siteplot, aes(x = Longitude, y = Latitude), shape = 21, size = 4, stroke = 1.5, fill = "white") +
  geom_point(data = siteplot, aes(x = Longitude, y = Latitude), shape = 4, size = 3, stroke = 1.5) +
  geom_label_repel(data = siteplot, aes(x = Longitude, y = Latitude, label = manu_site), alpha = 0.7) +
  labs(x = "Longitude", y = "Latitude") + 
  theme(legend.position = "right") +
  scale_color_manual(values = c("#F47F72", "#442C83")) +
  scale_shape_manual(values = c(0, 6, 1, 4)) 
  #scale_fill_manual(values = c("#280B54FF","#FAC127FF", "#D44842FF", "white")) +
  #scale_color_manual(values = c(rep("black", 10), "#280B54FF","#FAC127FF", "#D44842FF", "white", "#280B54FF","#FAC127FF", "#D44842FF", "white"), guide = FALSE) +
  #scale_alpha_manual(values = c(0.7, 0.7, 0.7, 0.7), guide = FALSE)
save_plot(filename = "../manuscript/plots/Map_subpops_and_Ecotypes.png", plot = last_plot(), base_height = 3)
?geom_point
```

# -- Figure 1 B
# Phenotype correlations
can you easily look at the correlation between greenup ranks at OK versus NE?
```{r}
phe_gr_df %>%
  filter(SITE %in% c("LINC", "STIL")) %>%
  ungroup() %>%
  select(PLANT_ID:GR50, -PLOT_GL) %>%
  pivot_wider(names_from = "SITE", values_from = "GR50", values_fn = mean) %>%
  ggplot(aes(x = STIL, y = LINC)) +
  geom_hex() + #aes(color = SUBPOP), height = 1, width = 1) +
  scale_fill_viridis(option = "B",
                     end = 0.9, direction = -1)

phe_gr_df %>%
  filter(SITE %in% c("STIL", "PKLE")) %>%
  ungroup() %>%
  select(PLANT_ID:GR50, -PLOT_GL) %>%
  pivot_wider(names_from = "SITE", values_from = "GR50", values_fn = mean) %>%
  ggplot(aes(x = STIL, y = Lat_origin)) +
  geom_hex() + #aes(color = SUBPOP), height = 1, width = 1) +
  scale_fill_viridis(option = "B",
                     end = 0.9, direction = -1)


cor_gr <- phe_gr_df %>%
  left_join(select(sites, SITE, manu_site)) %>%
  ungroup() %>%
  select(PLANT_ID:GR50, manu_site, -PLOT_GL, -SITE) %>%
  pivot_wider(names_from = "manu_site", values_from = "GR50", values_fn = mean)
cor_fl <- phe_fl_df %>%
  left_join(select(sites, SITE, manu_site)) %>%
  ungroup() %>%
  select(PLANT_ID:FL50, manu_site, -PLOT_GL, -SITE) %>%
  pivot_wider(names_from = "manu_site", values_from = "FL50", values_fn = mean)

cor_north <- phe_gr_df %>%
  left_join(select(sites, SITE, manu_site)) %>%
  ungroup() %>%
  filter(Lat_origin > 40) %>%
  select(PLANT_ID:GR50, manu_site, -PLOT_GL, -SITE) %>%
  pivot_wider(names_from = "manu_site", values_from = "GR50", values_fn = mean)
cor_south <- phe_gr_df %>%
  left_join(select(sites, SITE, manu_site)) %>%
  ungroup() %>%
  filter(Lat_origin < 32) %>%
  select(PLANT_ID:GR50, manu_site, -PLOT_GL, -SITE) %>%
  pivot_wider(names_from = "manu_site", values_from = "GR50", values_fn = mean)
cor(cor_north[,5:13], use = "pairwise") %>%
  switchgrassGWAS:::reorder_cormat(.) %>%
  ggcorr(data = NULL, cor_matrix = ., label = TRUE)
cor(cor_south[,5:13], use = "pairwise") %>%
  switchgrassGWAS:::reorder_cormat(.) %>%
  ggcorr(data = NULL, cor_matrix = ., label = TRUE)

cor(cor_gr$STIL, cor_gr$LINC, use = "pairwise")
cor(cor_gr[,5:13], use = "pairwise") %>%
  ggcorr(data = NULL, cor_matrix = ., label = TRUE)
cor(cor_fl[,5:13], use = "pairwise") %>%
  ggcorr(data = NULL, cor_matrix = ., label = TRUE)

# two different ways. Think I like the second more. Force latitudinal order.
cor_gr %>%
  filter(SUBPOP %in% c("Midwest", "Gulf")) %>%
  select(TX1, TX2, TX3, OK, MO, NE, MI, SD) %>%
  cor(use = "pairwise") %>%
  switchgrassGWAS:::reorder_cormat(.) %>%
  ggcorr(data = NULL, cor_matrix = ., label = TRUE)
```

```{r}
pand <- cor_fl %>%
  filter(SUBPOP %in% c("Gulf")) %>%
  select(TX1, TX2, TX3, OK, MO, NE, MI, SD) %>%
  ggcorr(method = "pairwise", label = FALSE, layout.exp = 0.2, hjust = 0.95) +
  scale_fill_gradientn(colors = c("#440154FF", "#3B528BFF", "#2C728EFF",
                                      "white", "#27AD81FF", "#5DC863FF",
                                      "#FDE725FF"), limits = c(-1,1)) +
  theme(legend.position = "none")
pane <- cor_fl %>%
  filter(SUBPOP %in% c("Midwest")) %>%
  select(TX1, TX2, TX3, OK, MO, NE, MI, SD) %>%
  ggcorr(method = "pairwise", label = FALSE, layout.exp = 0.2, hjust = 0.95) +
  scale_fill_gradientn(colors = c("#440154FF", "#3B528BFF", "#2C728EFF",
                                      "white", "#27AD81FF", "#5DC863FF",
                                      "#FDE725FF"), limits = c(-1,1)) +
  theme(legend.position = "none")
panf <- cor_fl %>%
  filter(SUBPOP %in% c("Gulf", "Midwest")) %>%
  select(TX1, TX2, TX3, OK, MO, NE, MI, SD) %>%
  ggcorr(method = "pairwise", label = FALSE, layout.exp = 0.2, hjust = 0.95) +
  scale_fill_gradientn(colors = c("#440154FF", "#3B528BFF", "#2C728EFF",
                                      "white", "#27AD81FF", "#5DC863FF",
                                      "#FDE725FF"), limits = c(-1,1)) +
  theme(legend.position = "none")
pana <- cor_gr %>%
  filter(SUBPOP %in% c("Gulf")) %>%
  select(TX1, TX2, TX3, OK, MO, NE, MI, SD) %>%
  ggcorr(method = "pairwise", label = FALSE, layout.exp = 0.2, hjust = 0.95) +
  scale_fill_gradientn(colors = c("#440154FF", "#3B528BFF", "#2C728EFF",
                                      "white", "#27AD81FF", "#5DC863FF",
                                      "#FDE725FF"), limits = c(-1,1)) +
  theme(legend.position = c(0.4, 0.85))
panb <- cor_gr %>%
  filter(SUBPOP %in% c("Midwest")) %>%
  select(TX1, TX2, TX3, OK, MO, NE, MI, SD) %>%
  ggcorr(method = "pairwise", label = FALSE, layout.exp = 0.2, hjust = 0.95) +
  scale_fill_gradientn(colors = c("#440154FF", "#3B528BFF", "#2C728EFF",
                                      "white", "#27AD81FF", "#5DC863FF",
                                      "#FDE725FF"), limits = c(-1,1)) +
  theme(legend.position = "none")
panc <- cor_gr %>%
  filter(SUBPOP %in% c("Gulf", "Midwest")) %>%
  select(TX1, TX2, TX3, OK, MO, NE, MI, SD) %>%
  ggcorr(method = "pairwise", label = FALSE, layout.exp = 0.2, hjust = 0.95) +
  scale_fill_gradientn(colors = c("#440154FF", "#3B528BFF", "#2C728EFF",
                                      "white", "#27AD81FF", "#5DC863FF",
                                      "#FDE725FF"), limits = c(-1,1)) +
  theme(legend.position = "none")

pheno_corr <- plot_grid(pana, panb, panc, pand, pane, panf, nrow = 2)
save_plot(filename = "../manuscript/plots/Phenotypic_correlations_GR50_FL50_2.svg", plot = pheno_corr, base_width = 6.5)
```


# --- Fig 1 C & D
# Heritability Plots

Within common gardens, h2 were typically quite high: 62% on average for greenup date, and 87% for flowering date. However, h2 were not consistently high, notably for greenup at our OK and NE common gardens.
```{r}
h2all <- h2fl %>%
  full_join(h2gr)

h2all %>%
  filter(phe %in% c("FL50", "GR50") & site %in% c("TX1", "TX2", "TX3", "OK", "MO", "NE", "IL", "MI", "SD")) %>%
  group_by(phe) %>%
  summarise(meanh2 = mean(Estimate))

h2all %>%
  filter(phe %in% c("FL50", "GR50") & site %in% c("TX1", "TX2", "TX3", "OK", "MO", "NE", "IL", "MI", "SD")) %>%
  filter(Estimate < 0.2)

h2all %>%
  filter(phe %in% c("FL50", "GR50") & site %in% c("eight_sites")) %>%
  group_by(phe) %>%
  summarise(meanh2 = mean(Estimate))

h2all %>%
  filter(phe %in% c("GR50") & site %in% c("tx_sites", "north_sites")) %>%
  mutate(GandGxE = Estimate + GxE_Estimate) %>%
  select(GandGxE, Estimate:phe, GxE_Estimate) %>%
  group_by(site) %>%
  summarise(meanh2 = mean(Estimate + GxE_Estimate))

h2all %>%
  filter(site %in% c("eight_sites")) %>%
  mutate(GandGxE = Estimate + GxE_Estimate)  %>%
  select(GandGxE, site:phe, Estimate, SE, GxE_Estimate, GxE_SE)
```


## h2 as functions of GR50 and FL50
```{r}
h2all <- h2fl %>%
  full_join(h2gr)
unique(h2all$site)
f1h2 <- h2all %>%
  filter(phe %in% c("FL50", "GR50")) %>%
  mutate(site = case_when(is.na(site) ~ "all",
                          site == "eight_sites" ~ "All",
                          site == "tx_sites" ~ "Texas",
                          site == "north_sites" ~ "North",
                          TRUE ~ site),
         subpop_name = case_when(subpop == "Gulf_and_Midwest" ~ "Gulf and Midwest",
                                 subpop == "three_subpops" ~ "all",
                                 TRUE ~ subpop),
         phe_name = case_when(phe == "GR50" ~ "50% greenup",
                              phe == "FL50" ~ "50% flowering",
                              TRUE ~ phe))
f1h2$site <- factor(f1h2$site, levels = c("TX1", "TX2", "TX3", "OK", "MO", "NE", "IL", "MI", "SD", "All"))#, "Texas", "North"))
f1h2$phe_name <- factor(f1h2$phe_name, levels = c("50% greenup", "50% flowering"))
f1h2$subpop_name <- factor(f1h2$subpop_name, levels = c("Gulf", "Midwest", "Gulf and Midwest", "Atlantic", "all"))
ggh21 <- f1h2 %>%
  filter(!(site %in% c("nine_sites", "IL")) & !is.na(subpop_name) & !is.na(phe_name) & !is.na(site)) %>%
  ggplot(aes(x = site, y = Estimate)) +
  geom_bar(aes(fill = site %in% c("All", "Texas", "North")), stat = "identity") +
  geom_errorbar(aes(ymin = Estimate - SE, ymax = Estimate + SE), width = 0.3) +
  geom_linerange(aes(ymin = Estimate, ymax = Estimate + SE)) +
  geom_linerange(aes(ymin = Estimate - SE, ymax = Estimate)) +
  facet_grid(phe_name ~ subpop_name) +
  theme(panel.spacing.x = unit(0.1, 'cm'), 
        axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
        legend.position = "none") + 
  ylim(c(0, 1)) +
  xlab("Common Gardens") + 
  ylab(bquote('h'^2)) +
  scale_fill_viridis(discrete = TRUE, end = 0.8, direction = 1)
  #
save_plot("../manuscript/plots/h2_GR_FL_site_by_subpop_k_specific.png", plot = ggh21, base_width = 5.5, base_height = 3)
```

## h2 flowering weather
```{r}
flh2 <- h2fl %>%
  filter(phe %in% c("FL50", "dyln_fl50", "dyln_change_sec", "cgdd_12c_gr2fl", "crain_gr2fl", "crain_1d", "crain_2d", "crain_3d", "crain_5d", "crain_7d")) %>%
  mutate(site = case_when(is.na(site) ~ "all",
                          site == "eight_sites" ~ "All",
                          site == "tx_sites" ~ "Texas",
                          site == "north_sites" ~ "North",
                          TRUE ~ site),
         subpop_name = case_when(subpop == "Gulf_and_Midwest" ~ "Gulf and Midwest",
                                 subpop == "three_subpops" ~ "all",
                                 TRUE ~ subpop),
         phe_name = case_when(phe == "FL50" ~ "50% flowering",
                              phe == "dyln_fl50" ~ "daylength",
                              phe == "dyln_change_sec" ~ "daylength change (s)",
                              phe == "cgdd_12c_gr2fl" ~ "GDD, GR to FL",
                              phe == "crain_gr2fl" ~ "Rainfall, GR to FL",
                              phe == "crain_1d" ~ "Rainfall, 1 day",
                              phe == "crain_2d" ~ "Rainfall, 2d sum",
                              phe == "crain_3d" ~ "Rainfall, 3d sum",
                              phe == "crain_5d" ~ "Rainfall, 5d sum",
                              phe == "crain_7d" ~ "Rainfall, 7d sum",
                              TRUE ~ phe))
flh2$site <- factor(flh2$site, levels = rev(c("TX1", "TX2", "TX3", "OK", "MO", "NE", "IL", "MI", "SD", "All", "Texas", "North")))
flh2$phe_name <- factor(flh2$phe_name, levels = c("50% flowering", "daylength",
                                                  "daylength change (s)",
                                                  "GDD, GR to FL",
                                                  "Rainfall, GR to FL",
                                                  "Rainfall, 1 day",
                                                  #"Rainfall, 2d sum",
                                                  "Rainfall, 3d sum",
                                                  "Rainfall, 5d sum"#,
                                                  #"Rainfall, 7d sum"
                                                  ))
flh2$subpop_name <- factor(flh2$subpop_name, levels = c("Gulf", "Midwest", "Gulf and Midwest", "Atlantic", "all"))
site_plot <- rev(c("TX1", "TX2", "TX3", "OK", "MO", "NE", "MI", "SD", "All", "Texas", "North"))
subpop_v <- list(Gulf = "Gulf", Midwest = "Midwest", Gulf_and_Midwest = c("Gulf", "Midwest"), Atlantic = "Atlantic", three_subpops = c("Atlantic", "Gulf", "Midwest"))

flh2_p1 <- flh2 %>%
  filter(subpop_name %in% c("Gulf and Midwest", subpop_v[[1]], subpop_v[[2]]) & (site %in% site_plot) & !is.na(site) & !is.na(phe_name)) %>% # subpop == subpop_v[[i]] & 
  ggplot(aes(x = phe_name, y = Estimate)) +
  geom_bar(stat = "identity", aes(fill = phe_name == "50% flowering")) +
  geom_errorbar(aes(ymin = Estimate - SE, ymax = Estimate + SE), width = 0.3) +
  geom_linerange(aes(ymin = Estimate, ymax = Estimate + SE)) +
  geom_linerange(aes(ymin = Estimate - SE, ymax = Estimate)) +
  facet_grid(site ~ subpop_name) +
  geom_hline(data = filter(flh2, subpop_name %in% c("Gulf and Midwest", subpop_v[[1]], subpop_v[[2]]) & (site %in% site_plot) & !is.na(site) & phe == "FL50"), aes(yintercept = Estimate), linetype = 2) +
  theme(panel.spacing.x = unit(0.1, 'cm'), axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), legend.position = "none") + ylim(c(0, 1.1)) +
  scale_fill_viridis(discrete = TRUE, end = 0.8, direction = 1)

save_plot("../manuscript/plots/h2_FL_site_weather_functions_single_sites_k_specific.png", plot = flh2_p1, base_height = 7, base_asp = 1.1)

site_plot <- c("All", "Texas", "North")

flh2_p2 <- flh2 %>%
  filter(subpop_name %in% c("Gulf and Midwest", subpop_v[[1]], subpop_v[[2]]) & (site %in% site_plot) & !is.na(site) & !is.na(phe_name)) %>% # subpop == subpop_v[[i]] & 
  ggplot(aes(x = phe_name, y = Estimate)) +
  geom_bar(stat = "identity", aes(fill = phe_name == "50% flowering")) +
  geom_errorbar(aes(ymin = Estimate - SE, ymax = Estimate + SE), width = 0.3) +
  geom_linerange(aes(ymin = Estimate, ymax = Estimate + SE)) +
  geom_linerange(aes(ymin = Estimate - SE, ymax = Estimate)) +
  facet_grid(site ~ subpop_name) +
  geom_hline(data = filter(flh2, subpop_name %in% c("Gulf and Midwest", subpop_v[[1]], subpop_v[[2]]) & (site %in% site_plot) & !is.na(site) & phe == "FL50"), aes(yintercept = Estimate), linetype = 2) +
  theme(panel.spacing.x = unit(0.1, 'cm'), axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), legend.position = "none") + ylim(c(0, 0.8)) +
  scale_fill_viridis(discrete = TRUE, end = 0.8, direction = 1) + xlab("")

save_plot("../manuscript/plots/h2_FL_site_weather_functions_site_groups_k_specific.png", plot = flh2_p2, base_height = 4, base_asp = 1.5)

site_plot <- c("All")

flh2_p3 <- flh2 %>%
  filter(subpop_name %in% c("Gulf and Midwest", subpop_v[[1]], subpop_v[[2]]) & (site %in% site_plot) & !is.na(site) & !is.na(phe_name)) %>% # subpop == subpop_v[[i]] & 
  ggplot(aes(x = phe_name, y = Estimate)) +
  geom_bar(stat = "identity", aes(fill = phe_name == "50% flowering")) +
  geom_errorbar(aes(ymin = Estimate - SE, ymax = Estimate + SE), width = 0.3) +
  geom_linerange(aes(ymin = Estimate, ymax = Estimate + SE)) +
  geom_linerange(aes(ymin = Estimate - SE, ymax = Estimate)) +
  facet_grid(site ~ subpop_name) +
  geom_hline(data = filter(flh2, subpop_name %in% c("Gulf and Midwest", subpop_v[[1]], subpop_v[[2]]) & (site %in% site_plot) & !is.na(site) & phe == "FL50"), aes(yintercept = Estimate), linetype = 2) +
  theme(panel.spacing.x = unit(0.1, 'cm'), axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), legend.position = "none") + ylim(c(0, 0.5)) +
  scale_fill_viridis(discrete = TRUE, end = 0.8, direction = 1) + xlab("")
save_plot("../manuscript/plots/h2_FL_site_weather_functions_All_Sites_k_specific.png", plot = flh2_p3, base_height = 3.5)
```
## gxe flowering all, north, texas
```{r}
site_plot <- c("All", "Texas", "North")

fl_barplot2 <- flh2 %>%
  filter(subpop_name %in% c("Gulf and Midwest", subpop_v[[1]], subpop_v[[2]]) & (site %in% site_plot) & !is.na(site) & !is.na(phe_name)) %>%
  select(site:phe, subpop_name, phe_name, ends_with("SE")) %>%
  pivot_longer(cols = ends_with("SE"), names_to = "Type", values_to = "SE") %>%
  mutate(type_name = case_when(Type == "SE" ~ "G",
                               Type == "GxE_SE" ~ "GxE",
                               Type == "error_SE" ~ "error",
                               Type == "Env_SE" ~ "E")) %>%
  select(-Type)
fl_barplot <- flh2 %>%
  filter(subpop_name %in% c("Gulf and Midwest", subpop_v[[1]], subpop_v[[2]]) & (site %in% site_plot) & !is.na(site) & !is.na(phe_name)) %>%
  select(site:phe, subpop_name, phe_name, ends_with("Estimate")) %>%
  pivot_longer(cols = ends_with("Estimate"), names_to = "Type", values_to = "Estimate") %>%
  mutate(type_name = case_when(Type == "Estimate" ~ "G",
                               Type == "GxE_Estimate" ~ "GxE",
                               Type == "error_Estimate" ~ "error",
                               Type == "Env_Estimate" ~ "E"))
fl_barplot <- fl_barplot %>%
  left_join(fl_barplot2)
fl_barplot$type_name <- factor(fl_barplot$type_name, levels = rev(c("G", "GxE", "E", "error")))

fl_bar_gxe <- fl_barplot %>%
  ggplot(aes(x = phe_name, y = Estimate)) +
  geom_bar(stat = "identity", aes(fill = type_name)) + 
  geom_errorbar(data = filter(fl_barplot, type_name == "G"), aes(ymin = Estimate - SE, ymax = Estimate + SE), width = 0.3) +
  facet_grid(site ~ subpop_name)+
  scale_fill_viridis(discrete = TRUE, direction = -1) +
  theme(panel.spacing.x = unit(0.1, 'cm'), axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), legend.position = "none") +
  geom_hline(data = filter(flh2, subpop_name %in% c("Gulf and Midwest", subpop_v[[1]], subpop_v[[2]]) & (site %in% site_plot) & !is.na(site) & phe == "FL50"), aes(yintercept = GxE_Estimate+Estimate), linetype = 2) + xlab("") +
  ylab("Variance Explained")
  
save_plot("../manuscript/plots/Var_Comp_flowering_site_groups.png", plot = fl_bar_gxe, base_width = 5.5, base_height = 4)

flh2_p4 <- flh2 %>%
  filter(subpop_name %in% c("Gulf and Midwest", subpop_v[[1]], subpop_v[[2]]) & (site %in% site_plot) & !is.na(site) & !is.na(phe_name)) %>% # subpop == subpop_v[[i]] & 
  ggplot(aes(x = phe_name, y = GxE_Estimate)) +
  geom_bar(stat = "identity", aes(fill = phe_name == "50% flowering")) +
  geom_errorbar(aes(ymin = GxE_Estimate - GxE_SE, ymax = GxE_Estimate + GxE_SE), width = 0.3) +
  geom_linerange(aes(ymin = GxE_Estimate, ymax = GxE_Estimate + GxE_SE)) +
  geom_linerange(aes(ymin = GxE_Estimate - GxE_SE, ymax = GxE_Estimate)) +
  facet_grid(site ~ subpop_name) +
  geom_hline(data = filter(flh2, subpop_name %in% c("Gulf and Midwest", subpop_v[[1]], subpop_v[[2]]) & (site %in% site_plot) & !is.na(site) & phe == "FL50"), aes(yintercept = GxE_Estimate), linetype = 2) +
  theme(panel.spacing.x = unit(0.1, 'cm'), axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), legend.position = "none") + ylim(c(0, 1)) +
  scale_fill_viridis(discrete = TRUE, end = 0.8, direction = 1) + xlab("")

save_plot("../manuscript/plots/h2_FL_site_weather_functions_site_groups_k_specific.png", plot = flh2_p2, base_height = 4, base_asp = 1.5)

library(scales)
show_col(viridis(4)) # #440154FF   #31688EFF
```


## h2 greenup weather

```{r}
unique(h2gr$site)
grh2 <- h2gr %>%
  filter(!(phe %in% c("FL50", "dyln_fl50", "dyln_change_sec", "cgdd_12c_gr2fl", "crain_gr2fl", "crain_1d", "crain_2d", "crain_3d", "crain_5d", "crain_7d", "DYLN"))) %>%
  mutate(site = case_when(is.na(site) ~ "all",
                          site == "eight_sites" ~ "All",
                          site == "tx_sites" ~ "Texas",
                          site == "north_sites" ~ "North",
                          TRUE ~ site),
         subpop_name = case_when(subpop == "Gulf_and_Midwest" ~ "Gulf and Midwest",
                                 subpop == "three_subpops" ~ "All",
                                 TRUE ~ subpop),
         phe_name = case_when(phe == "GR50" ~ "50% greenup",
                              phe == "FL50" ~ "50% flowering",
                              phe == "cgdd_12c_jan2gr" ~ "GDD 01-01 to GR",
                              phe == "cgdd_8c_jan2gr" ~ "GDD 8C 01-01 to greenup",
                              phe == "tmax_18d" ~ "Max T, 18d mean",
                              phe == "tmin_18d" ~ "Min T, 18d mean",
                              phe == "tave_18d" ~ "Mean T, 18d mean",
                              phe == "tmax_10d" ~ "Max T, 10d mean",
                              phe == "tmin_10d" ~ "Min T, 10d mean",
                              phe == "tave_10d" ~ "Mean T, 10d mean",
                              phe == "tmax_5d" ~ "Max T, 5d mean",
                              phe == "tmin_5d" ~ "Min T, 5d mean",
                              phe == "tave_5d" ~ "Mean T, 5d mean",
                              phe == "cgdd_12c_10d" ~ "GDD, 10d",
                              phe == "cgdd_12c_18d" ~ "GDD, 18d",
                              phe == "cgdd_12c_5d" ~ "GDD, 5d",
                              TRUE ~ phe))
grh2$site <- factor(grh2$site, levels = rev(c("TX1", "TX2", "TX3", "OK", "MO", "NE", "IL", "MI", "SD", "All", "Texas", "North", "nine_sites")))
grh2$phe_name <- factor(grh2$phe_name, levels = c("50% greenup", "GDD, 5d",
                                                  "Min T, 5d mean",
                                                  #"Mean T, 5d mean",
                                                  #"Max T, 5d mean",
                                                  "GDD, 10d",
                                                  "Min T, 10d mean",
                                                  #"Mean T, 10d mean",
                                                  #"Max T, 10d mean",
                                                  "GDD, 18d",
                                                  "Min T, 18d mean",
                                                  #"Mean T, 18d mean",
                                                  #"Max T, 18d mean",
                                                  #"GDD 8C 01-01 to greenup",
                                                  "GDD 01-01 to GR"))
grh2$subpop_name <- factor(grh2$subpop_name, levels = c("Gulf", "Midwest", "Gulf and Midwest", "Atlantic", "All"))


subpop_v <- list(Gulf = "Gulf", Midwest = "Midwest", Gulf_and_Midwest = c("Gulf", "Midwest"), Atlantic = "Atlantic", three_subpops = c("Atlantic", "Gulf", "Midwest"))

site_plot <- c("TX1", "TX2", "TX3", "OK", "MO", "NE", "IL", "MI", "SD", "All", "Texas", "North")

grh2_p1 <- grh2 %>%
  filter(subpop_name %in% c("Gulf and Midwest", subpop_v[[1]], subpop_v[[2]]) & (site %in% site_plot) & !is.na(site) & !is.na(phe_name)) %>% # subpop == subpop_v[[i]] & 
  ggplot(aes(x = phe_name, y = Estimate)) +
  geom_bar(stat = "identity", aes(fill = phe_name == "50% greenup")) +
  geom_errorbar(aes(ymin = Estimate - SE, ymax = Estimate + SE), width = 0.3) +
  geom_linerange(aes(ymin = Estimate, ymax = Estimate + SE)) +
  geom_linerange(aes(ymin = Estimate - SE, ymax = Estimate)) +
  facet_grid(site ~ subpop_name) +
  geom_hline(data = filter(grh2, subpop_name %in% c("Gulf and Midwest", subpop_v[[1]], subpop_v[[2]]) & (site %in% site_plot) & !is.na(site) & phe == "GR50"), aes(yintercept = Estimate), linetype = 2) +
  theme(panel.spacing.x = unit(0.1, 'cm'), axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), legend.position = "none") + ylim(c(0, 1.1)) +
  scale_fill_viridis(discrete = TRUE, end = 0.8, direction = 1)
save_plot("../manuscript/plots/h2_GR_site_weather_functions_single_sites_k_specific.png", plot = grh2_p1, base_height = 7.5, base_asp = 1.1)

site_plot <- c( "All", "Texas", "North")

grh2_p2 <- grh2 %>%
  filter(subpop_name %in% c("Gulf and Midwest", subpop_v[[1]], subpop_v[[2]]) & (site %in% site_plot) & !is.na(site) & !is.na(phe_name)) %>% # subpop == subpop_v[[i]] & 
  ggplot(aes(x = phe_name, y = Estimate)) +
  geom_bar(stat = "identity", aes(fill = phe_name == "50% greenup")) +
  geom_errorbar(aes(ymin = Estimate - SE, ymax = Estimate + SE), width = 0.3) +
  geom_linerange(aes(ymin = Estimate, ymax = Estimate + SE)) +
  geom_linerange(aes(ymin = Estimate - SE, ymax = Estimate)) +
  facet_grid(site ~ subpop_name) +
  geom_hline(data = filter(grh2, subpop_name %in% c("Gulf and Midwest", subpop_v[[1]], subpop_v[[2]]) & (site %in% site_plot) & !is.na(site) & phe == "GR50"), aes(yintercept = Estimate), linetype = 2) +
  theme(panel.spacing.x = unit(0.1, 'cm'), axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), legend.position = "none") + ylim(c(0, 0.6)) +
  scale_fill_viridis(discrete = TRUE, end = 0.8, direction = 1)
save_plot("../manuscript/plots/h2_GR_site_weather_functions_site_groups_k_specific.png", plot = grh2_p2, base_height = 5, base_asp = 1.3)

site_plot <- c( "All")

grh2_p3 <- grh2 %>%
  filter(subpop_name %in% c("Gulf and Midwest", subpop_v[[1]], subpop_v[[2]]) & (site %in% site_plot) & !is.na(site) & !is.na(phe_name)) %>% # subpop == subpop_v[[i]] & 
  ggplot(aes(x = phe_name, y = Estimate)) +
  geom_bar(stat = "identity", aes(fill = phe_name == "50% greenup")) +
  geom_errorbar(aes(ymin = Estimate - SE, ymax = Estimate + SE), width = 0.3) +
  geom_linerange(aes(ymin = Estimate, ymax = Estimate + SE)) +
  geom_linerange(aes(ymin = Estimate - SE, ymax = Estimate)) +
  facet_grid(site ~ subpop_name) +
  geom_hline(data = filter(grh2, subpop_name %in% c("Gulf and Midwest", subpop_v[[1]], subpop_v[[2]]) & (site %in% site_plot) & !is.na(site) & phe == "GR50"), aes(yintercept = Estimate), linetype = 2) +
  theme(panel.spacing.x = unit(0.1, 'cm'), axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), legend.position = "none") + ylim(c(0, 0.11)) +
  scale_fill_viridis(discrete = TRUE, end = 0.8, direction = 1) + xlab("")
save_plot("../manuscript/plots/h2_GR_site_weather_functions_All_Sites_k_specific.png", plot = grh2_p3, base_height = 3.5)

gr_fl_p1 <- plot_grid(grh2_p3, flh2_p3, nrow = 2)

save_plot("../manuscript/plots/h2_GR_and_FL_site_weather_functions_All_Sites_k_specific.png", plot = gr_fl_p1, base_height = 4.6)
```
## gxe greenup

```{r}
site_plot <- c("All", "Texas", "North")

gr_barplot2 <- grh2 %>%
  filter(subpop_name %in% c("Gulf and Midwest", subpop_v[[1]], subpop_v[[2]]) & (site %in% site_plot) & !is.na(site) & !is.na(phe_name)) %>%
  select(site:phe, subpop_name, phe_name, ends_with("SE")) %>%
  pivot_longer(cols = ends_with("SE"), names_to = "Type", values_to = "SE") %>%
  mutate(type_name = case_when(Type == "SE" ~ "G",
                               Type == "GxE_SE" ~ "GxE",
                               Type == "error_SE" ~ "error",
                               Type == "Env_SE" ~ "E")) %>%
  select(-Type)
gr_barplot <- grh2 %>%
  filter(subpop_name %in% c("Gulf and Midwest", subpop_v[[1]], subpop_v[[2]]) & (site %in% site_plot) & !is.na(site) & !is.na(phe_name)) %>%
  select(site:phe, subpop_name, phe_name, ends_with("Estimate")) %>%
  pivot_longer(cols = ends_with("Estimate"), names_to = "Type", values_to = "Estimate") %>%
  mutate(type_name = case_when(Type == "Estimate" ~ "G",
                               Type == "GxE_Estimate" ~ "GxE",
                               Type == "error_Estimate" ~ "error",
                               Type == "Env_Estimate" ~ "E"))
gr_barplot <- gr_barplot %>%
  left_join(gr_barplot2)
gr_barplot$type_name <- factor(gr_barplot$type_name, levels = rev(c("G", "GxE", "E", "error")))

gr_bar_gxe <- gr_barplot %>%
  ggplot(aes(x = phe_name, y = Estimate)) +
  geom_bar(stat = "identity", aes(fill = type_name)) + 
  geom_errorbar(data = filter(gr_barplot, type_name == "G"), aes(ymin = Estimate - SE, ymax = Estimate + SE), width = 0.3) +
  facet_grid(site ~ subpop_name)+
  scale_fill_viridis(discrete = TRUE, direction = -1) +
  theme(panel.spacing.x = unit(0.1, 'cm'), axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1), legend.position = "right") +
  geom_hline(data = filter(grh2, subpop_name %in% c("Gulf and Midwest", subpop_v[[1]], subpop_v[[2]]) & (site %in% site_plot) & !is.na(site) & phe == "GR50"), aes(yintercept = GxE_Estimate+Estimate), linetype = 2)
  
save_plot("../manuscript/plots/Var_Comp_greenup_site_groups.png", plot = gr_bar_gxe, base_height = 4, base_asp = 1.5)
```


# --------------
# SVD of K

## Plot eigenvectors to capture predominant patterns in K matrix.
```{r}
k_full <- read_rds("../data/Kinship_van_Raden_630_individuals_SNPs_r2_20percent.rds")
k_gandm <- read_rds("../data/Kinship_van_Raden_method_363g_Gulf_and_Midwest_subset_maf0.05.rds")
k_midw <- read_rds("../data/Kinship_van_Raden_method_134g_Midwest_subset_maf0.05.rds")
k_gulf <- read_rds("../data/Kinship_van_Raden_method_229g_Gulf_subset_maf0.05.rds")
```

```{r}
b1 <- svd(k_full)
b1$d[1] / sum(b1$d) # Proportion of variance explained by first eigenvector 11.5
b1$d[2] / sum(b1$d) # 7.1
b1$d[3] / sum(b1$d) # 6.1
b1$d[4] / sum(b1$d) # 1.7
b1$d[5] / sum(b1$d) # 1.3
sum(b1$d[1:5]) / sum(b1$d) # First 5 eigenvectors explain 1% or more variance, and 27.7% of total variance.
for(g in 1:5){
barplot(b1$v[,g]/k_full[which.max(abs(k_full[,g])),g])
}
order <- enframe(colnames(k_full), name = NULL, value = "PLANT_ID")
ggb1 <- tibble(`First eigenvector (PVE = 11.5%)` = b1$v[,1]/k_full[which.max(abs(k_full[,1])),1],
       `Second eigenvector (PVE = 7.1%)` = b1$v[,2]/k_full[which.max(abs(k_full[,2])),2],
       `Third eigenvector (PVE = 6.1%)` = b1$v[,3]/k_full[which.max(abs(k_full[,3])),3],
       `Fourth eigenvector (PVE = 1.7%)` = b1$v[,4]/k_full[which.max(abs(k_full[,4])),4],
       `Fifth eigenvector (PVE = 1.3%)` = b1$v[,5]/k_full[which.max(abs(k_full[,5])),5],
       `Sixth eigenvector (PVE = 0.9%)` = b1$v[,6]/k_full[which.max(abs(k_full[,6])),6],
       PLANT_ID = order$PLANT_ID) %>%
  left_join(metadata, by = "PLANT_ID") %>%
  dplyr::select(PLANT_ID, SUBPOP, ECOTYPE_NNET, LATITUDE, ends_with("%)"))

ggb1 <- ggb1 %>%
  arrange(LATITUDE) %>%
  mutate(rank = row_number()) %>%
  pivot_longer(cols = ends_with("%)"), names_to = "Eigenvector", values_to = "value")
ggb1$Eigenvector <- factor(ggb1$Eigenvector, 
                           levels = c("First eigenvector (PVE = 11.5%)", 
                                      "Second eigenvector (PVE = 7.1%)", 
                                      "Third eigenvector (PVE = 6.1%)", 
                                      "Fourth eigenvector (PVE = 1.7%)",
                                      "Fifth eigenvector (PVE = 1.3%)",
                                      "Sixth eigenvector (PVE = 0.9%)"
                                      ))
ggb1_plot <- ggb1 %>%
  ggplot(aes(x = rank, y = value)) +
  geom_bar(aes(color = SUBPOP, fill = SUBPOP), stat = "identity") +
  facet_wrap(~ Eigenvector) +
  theme(legend.position = "right", axis.text.x = element_blank(), panel.spacing.x = unit(0.1, 'cm')) +
  scale_color_viridis(discrete = TRUE, end = 0.9, direction = -1) +
  scale_fill_viridis(discrete = TRUE, end = 0.9, direction = -1) +
  xlab("Latitude of Origin (Rank Order)") + ylab("") +
  geom_vline(xintercept = 607, linetype = 2)
save_plot("../manuscript/plots/SVD_of_Kinship_matrix_first_six_eigenvectors.png", base_height = 5, base_asp = 2, plot = ggb1_plot)

ggb1_plot2 <- ggb1 %>%
  ggplot(aes(x = rank, y = value)) +
  geom_bar(aes(color = ECOTYPE_NNET, fill = ECOTYPE_NNET), stat = "identity") +
  facet_wrap(~ Eigenvector) +
  theme(legend.position = "right", axis.text.x = element_blank(), panel.spacing.x = unit(0.1, 'cm')) +
  scale_color_viridis(discrete = TRUE, option = "B", end = 0.9, 
                      direction = -1) +
  scale_fill_viridis(discrete = TRUE, option = "B", end = 0.9, direction = -1) +
  xlab("Latitude of Origin (Rank Order)") + ylab("") +
  geom_vline(xintercept = 607, linetype = 2)
save_plot("../manuscript/plots/SVD_of_Kinship_matrix_first_six_eigenvectors_ecotype.png", base_height = 5, base_asp = 2, plot = ggb1_plot2)
```

## k_gandm

```{r}
b1 <- svd(k_gandm)
b1$d[1] / sum(b1$d) # Proportion of variance explained by first eigenvector 29.2
b1$d[2] / sum(b1$d) # 18.2
b1$d[3] / sum(b1$d) # 3.7
b1$d[4] / sum(b1$d) # 1.4
b1$d[5] / sum(b1$d) # 0.008
sum(b1$d[1:4]) / sum(b1$d) # First 4 eigenvectors explain 1% or more variance, and 52.5% of total variance.

for(g in 1:5){
barplot(b1$v[,g]/k_gandm[which.max(abs(k_gandm[,g])),g])
}
order <- enframe(colnames(k_gandm), name = NULL, value = "PLANT_ID")
ggb1 <- tibble(`First eigenvector (PVE = 29.2%)` = b1$v[,1]/k_gandm[which.max(abs(k_gandm[,1])),1],
       `Second eigenvector (PVE = 18.2%)` = b1$v[,2]/k_gandm[which.max(abs(k_gandm[,2])),2],
       `Third eigenvector (PVE = 3.7%)` = b1$v[,3]/k_gandm[which.max(abs(k_gandm[,3])),3],
       `Fourth eigenvector (PVE = 1.4%)` = b1$v[,4]/k_gandm[which.max(abs(k_gandm[,4])),4],
       #`Fifth eigenvector (PVE = 0.8%)` = b1$v[,5]/k_gandm[which.max(abs(k_gandm[,5])),5],
       PLANT_ID = order$PLANT_ID) %>%
  left_join(metadata, by = "PLANT_ID") %>%
  dplyr::select(PLANT_ID, SUBPOP, ECOTYPE_NNET, LATITUDE, ends_with("%)"))

ggb1 <- ggb1 %>%
  arrange(LATITUDE) %>%
  mutate(rank = row_number()) %>%
  pivot_longer(cols = ends_with("%)"), names_to = "Eigenvector", values_to = "value")
ggb1$Eigenvector <- factor(ggb1$Eigenvector, 
                           levels = c("First eigenvector (PVE = 29.2%)", 
                                      "Second eigenvector (PVE = 18.2%)", 
                                      "Third eigenvector (PVE = 3.7%)", 
                                      "Fourth eigenvector (PVE = 1.4%)",
                                      "Fifth eigenvector (PVE = 0.8%)"
                                      ))
ggb1$ECOTYPE_NNET <- factor(ggb1$ECOTYPE_NNET, levels = c("Coastal", "Lowland", "Upland", "Unknown"))
ggb1_plot <- ggb1 %>%
  ggplot(aes(x = rank, y = value)) +
  geom_bar(aes(color = SUBPOP, fill = SUBPOP), stat = "identity") +
  facet_wrap(~ Eigenvector) +
  theme(legend.position = "right", axis.text.x = element_blank(), panel.spacing.x = unit(0.1, 'cm')) +
  scale_color_manual(values = c("#F47F72", "#442C83")) +
  scale_fill_manual(values = c("#F47F72", "#442C83")) +
  xlab("Latitude of Origin (Rank Order)") + ylab("") +
  geom_vline(xintercept = 342, linetype = 2)
save_plot("../manuscript/plots/SVD_of_Kinship_matrix_Gulf_and_Midwest_first_four_eigenvectors.png", base_height = 5, base_asp = 2, plot = ggb1_plot)

ggb1_plot2 <- ggb1 %>%
  ggplot(aes(x = rank, y = value)) +
  geom_bar(aes(color = ECOTYPE_NNET, fill = ECOTYPE_NNET), stat = "identity") +
  facet_wrap(~ Eigenvector) +
  theme(legend.position = "right", axis.text.x = element_blank(), panel.spacing.x = unit(0.1, 'cm')) +
  scale_color_viridis(discrete = TRUE, option = "B", end = 0.9, 
                      direction = -1) +
  scale_fill_viridis(discrete = TRUE, option = "B", end = 0.9, direction = -1) +
  xlab("Latitude of Origin (Rank Order)") + ylab("") +
  geom_vline(xintercept = 342, linetype = 2)
save_plot("../manuscript/plots/SVD_of_Kinship_matrix_Gulf_and_Midwest_first_four_eigenvectors_ecotype.png", base_height = 5, base_asp = 2, plot = ggb1_plot2)
```

## k_midw

```{r}
b1 <- svd(k_midw)
b1$d[1] / sum(b1$d) # Proportion of variance explained by first eigenvector 10.2
b1$d[2] / sum(b1$d) # 4.8
b1$d[3] / sum(b1$d) # 2.4
b1$d[4] / sum(b1$d) # 2.0
b1$d[5] / sum(b1$d) # 1.7
b1$d[6] / sum(b1$d) # 1.3
b1$d[7] / sum(b1$d) # 1.3
b1$d[8] / sum(b1$d) # 1.1
b1$d[9] / sum(b1$d) # 1.1
b1$d[10] / sum(b1$d) # 1.1
b1$d[11] / sum(b1$d) # 1.1
b1$d[12] / sum(b1$d) # 1.0
b1$d[13] / sum(b1$d) # 1.0
sum(b1$d[1:13]) / sum(b1$d) # First 13 eigenvectors explain 1% or more variance, and 30% of total variance. Phew! Exhausting.
```

```{r}
k_use <- k_midw
for(g in 1:5){
barplot(b1$v[,g]/k_use[which.max(abs(k_use[,g])),g])
}
order <- enframe(colnames(k_use), name = NULL, value = "PLANT_ID")
ggb1 <- tibble(`First eigenvector (PVE = 10.2%)` = b1$v[,1]/k_use[which.max(abs(k_use[,1])),1],
       `Second eigenvector (PVE = 4.8%)` = b1$v[,2]/k_use[which.max(abs(k_use[,2])),2],
       `Third eigenvector (PVE = 2.4%)` = b1$v[,3]/k_use[which.max(abs(k_use[,3])),3],
       `Fourth eigenvector (PVE = 2.0%)` = b1$v[,4]/k_use[which.max(abs(k_use[,4])),4],
       `Fifth eigenvector (PVE = 1.7%)` = b1$v[,5]/k_use[which.max(abs(k_use[,5])),5],
       `Sixth eigenvector (PVE = 1.3%)` = b1$v[,6]/k_use[which.max(abs(k_use[,6])),6],
       PLANT_ID = order$PLANT_ID) %>%
  left_join(metadata, by = "PLANT_ID") %>%
  dplyr::select(PLANT_ID, SUBPOP, ECOTYPE_NNET, LATITUDE, ends_with("%)"))

ggb1 <- ggb1 %>%
  arrange(LATITUDE) %>%
  mutate(rank = row_number()) %>%
  pivot_longer(cols = ends_with("%)"), names_to = "Eigenvector", values_to = "value")
ggb1$Eigenvector <- factor(ggb1$Eigenvector, 
                           levels = c("First eigenvector (PVE = 10.2%)", 
                                      "Second eigenvector (PVE = 4.8%)", 
                                      "Third eigenvector (PVE = 2.4%)", 
                                      "Fourth eigenvector (PVE = 2.0%)",
                                      "Fifth eigenvector (PVE = 1.7%)",
                                      "Sixth eigenvector (PVE = 1.3%)"
                                      ))
ggb1$ECOTYPE_NNET <- factor(ggb1$ECOTYPE_NNET, levels = c("Coastal", "Lowland", "Upland", "Unknown"))

ggb1_plot2 <- ggb1 %>%
  ggplot(aes(x = rank, y = value)) +
  geom_bar(aes(color = ECOTYPE_NNET, fill = ECOTYPE_NNET), stat = "identity") +
  facet_wrap(~ Eigenvector) +
  theme(legend.position = "right", axis.text.x = element_blank(), panel.spacing.x = unit(0.1, 'cm')) +
  scale_color_viridis(discrete = TRUE, option = "B", end = 0.9, 
                      direction = -1) +
  scale_fill_viridis(discrete = TRUE, option = "B", end = 0.9, direction = -1) +
  xlab("Latitude of Origin (Rank Order)") + ylab("") +
  geom_vline(xintercept = 133, linetype = 2)
save_plot("../manuscript/plots/SVD_of_Kinship_matrix_Midwest_first_six_eigenvectors_ecotype.png", base_height = 5, base_asp = 2, plot = ggb1_plot2)
```
## k_gulf

```{r}
b1 <- svd(k_gulf)
b1$d[1] / sum(b1$d) # Proportion of variance explained by first eigenvector 22.5
b1$d[2] / sum(b1$d) # 9.4
b1$d[3] / sum(b1$d) # 3.1
b1$d[4] / sum(b1$d) # 1.8
b1$d[5] / sum(b1$d) # 1.6
b1$d[6] / sum(b1$d) # 1.3
b1$d[7] / sum(b1$d) # 1.1
b1$d[8] / sum(b1$d) # 1.0
sum(b1$d[1:8]) / sum(b1$d) # First 8 eigenvectors explain 1% or more variance, and 41.8% of total variance. 
```

```{r}
k_use <- k_gulf
for(g in 1:8){
barplot(b1$v[,g]/k_use[which.max(abs(k_use[,g])),g])
}
order <- enframe(colnames(k_use), name = NULL, value = "PLANT_ID")
ggb1 <- tibble(`First eigenvector (PVE = 22.5%)` = b1$v[,1]/k_use[which.max(abs(k_use[,1])),1],
       `Second eigenvector (PVE = 9.4%)` = b1$v[,2]/k_use[which.max(abs(k_use[,2])),2],
       `Third eigenvector (PVE = 3.1%)` = b1$v[,3]/k_use[which.max(abs(k_use[,3])),3],
       `Fourth eigenvector (PVE = 1.8%)` = b1$v[,4]/k_use[which.max(abs(k_use[,4])),4],
       `Fifth eigenvector (PVE = 1.6%)` = b1$v[,5]/k_use[which.max(abs(k_use[,5])),5],
       `Sixth eigenvector (PVE = 1.3%)` = b1$v[,6]/k_use[which.max(abs(k_use[,6])),6],
       PLANT_ID = order$PLANT_ID) %>%
  left_join(metadata, by = "PLANT_ID") %>%
  dplyr::select(PLANT_ID, SUBPOP, ECOTYPE_NNET, LATITUDE, ends_with("%)"))

ggb1 <- ggb1 %>%
  arrange(LATITUDE) %>%
  mutate(rank = row_number()) %>%
  pivot_longer(cols = ends_with("%)"), names_to = "Eigenvector", values_to = "value")
ggb1$Eigenvector <- factor(ggb1$Eigenvector, 
                           levels = c("First eigenvector (PVE = 22.5%)", 
                                      "Second eigenvector (PVE = 9.4%)", 
                                      "Third eigenvector (PVE = 3.1%)", 
                                      "Fourth eigenvector (PVE = 1.8%)",
                                      "Fifth eigenvector (PVE = 1.6%)",
                                      "Sixth eigenvector (PVE = 1.3%)"
                                      ))
ggb1$ECOTYPE_NNET <- factor(ggb1$ECOTYPE_NNET, levels = c("Coastal", "Lowland", "Upland", "Unknown"))

ggb1_plot2 <- ggb1 %>%
  ggplot(aes(x = rank, y = value)) +
  geom_bar(aes(color = ECOTYPE_NNET, fill = ECOTYPE_NNET), stat = "identity") +
  facet_wrap(~ Eigenvector) +
  theme(legend.position = "right", axis.text.x = element_blank(), panel.spacing.x = unit(0.1, 'cm')) +
  scale_color_viridis(discrete = TRUE, option = "B", end = 0.9, 
                      direction = -1) +
  scale_fill_viridis(discrete = TRUE, option = "B", end = 0.9, direction = -1) +
  xlab("Latitude of Origin (Rank Order)") + ylab("") +
  geom_vline(xintercept = 210, linetype = 2)
save_plot("../manuscript/plots/SVD_of_Kinship_matrix_Gulf_first_six_eigenvectors_ecotype.png", base_height = 5, base_asp = 2, plot = ggb1_plot2)
```

The eigenvectors capture the predominant patterns in the Uk3 covariance matrix.
```
v <- m$fitted_g$Ulist$ED_PCA_1
b1=svd(v)
for(g in 1){
barplot(b1$v[,g]/v[which.max(abs(v[,g])),g])
}

b1$d # the top eigenvector captures the predominant pattern in the ED_PCA_1 covariance matrix. 
```

WAIT. I could also plot this for the svd from the whole SNP set. That might make more sense than doing a svd on the kinship matrix... 


```{r}
snp <- snp_attach("~/Github/pvdiv-genome/genetic_subpops/kinship/Pvirgatum_V5_GWAS_134g_Midwest_K_subset_maf0.05.rds")

```


# -------------
# ASReml models for eight sites including GxE
```{r}

```

## ASReml h2 greenup analysis
```{r}
library(tidyverse)
library(asreml)
library(switchgrassGWAS)
library(cowplot)

source("~/Github/Functions_ggplot-theme-adjustments_2018-01-03.R")
workingdir <- file.path("~", "Github", "pvdiv-phenology-gxe")
outputdir <- file.path(workingdir, "analysis", "heritability")
metadata <- read_rds(file.path(workingdir, "data", "metadata.rds"))
sites <- read_rds(file.path(workingdir, "data", "sites.rds"))
subpops <- read_rds(file.path(workingdir, "data", "subpop_color_coding.rds"))
phe_gr_df <- read_rds(file.path(workingdir, "data",
                                "Weather_related_phe_for_asreml_h2_GR50.rds"))
phe_fl_df <- read_rds(file.path(workingdir, "data",
                                "Weather_related_phe_for_asreml_h2_FL50.rds"))
k_full <- read_rds("~/Github/pvdiv-genome/tensite_twoyear/Kinship_van_Raden_630_individuals_SNPs_r2_20percent.rds")

```

### loop to make heritability table using ASReml mixed models and vpredict.

## Include just texas sites for greenup and just northern four sites.
e.g.:
for(p in 1){#seq_along(phe_dfs)){
  phe_v <- phe_lists[[p]]
  phe <- phe_dfs[[p]]
  
  for(i in 12:13){#seq_along(site_v)){ # may need to adjust below for h2_table
    for(j in 1:3){#seq_along(subpop_v)){
      for(k in seq_along(phe_v)){
      
## Model eight, texas, and north sites with a kinship : site interaction.
Then, find GxE and G proportions of variance.
```{r}
site_v <- list(TX1 = "TX1", TX2 = "TX2", TX3 = "TX3", OK = "OK", MO = "MO",
               NE = "NE", IL = "IL", MI = "MI", SD = "SD", eight_sites = c("TX1", "TX2", "TX3", "OK", "MO", "NE", "MI", "SD"), nine_sites = c("TX1", "TX2", "TX3", "OK", "MO", "NE", "IL", "MI", "SD"), tx_sites = c("TX1", "TX2", "TX3"), north_sites = c("MO", "NE", "MI", "SD"))
subpop_v <- list(Gulf = "Gulf", Midwest = "Midwest", Gulf_and_Midwest = c("Gulf", "Midwest"), Atlantic = "Atlantic", three_subpops = c("Atlantic", "Gulf", "Midwest"))
phe_fl <- c("FL50", "dyln_fl50", "dyln_change_sec", "cgdd_12c_gr2fl",
           "crain_gr2fl", "crain_1d", "crain_2d", "crain_3d", "crain_5d",
           "crain_7d")
phe_gr <- c("GR50", "DYLN", "cgdd_12c_jan2gr", "cgdd_8c_jan2gr", "tmax_18d", 
            "tmin_18d", "tave_18d", "cgdd_12c_18d", "tmax_10d", "tmin_10d",
            "tave_10d", "cgdd_12c_10d", "tmax_5d", "tmin_5d", "tave_5d",
            "cgdd_12c_5d")
phe_dfs <- list(phe_gr_df, phe_fl_df)
phe_lists <- list(phe_gr, phe_fl)
h2_table <- tibble()

for(p in seq_along(phe_dfs)){
  phe_v <- phe_lists[[p]]
  phe <- phe_dfs[[p]]
  
  for(i in c(10,12:13)){#seq_along(site_v)){ # may need to adjust below for h2_table
    for(j in 1:3){#seq_along(subpop_v)){
      for(k in seq_along(phe_v)){
        phe_single <- phe %>%
    left_join(sites, by = "SITE") %>%
    filter(manu_site %in% site_v[[i]] & SUBPOP %in% subpop_v[[j]]) %>%
    rename(phenotype = eval(phe_v[k])) %>%
    mutate(PLANT_ID = as_factor(PLANT_ID),
           SITE = as_factor(SITE),
           PLOT_GL = as_factor(PLOT_GL)) %>%
    as.data.frame()
  
  if(length(site_v[[i]]) > 1){  # need to add site as a random factor if 
    # there is more than one site included in the dataset.
    tryCatch({
     asr_out <- asreml(phenotype ~ 1,
                       random = ~vm(PLANT_ID, k_full) +
                         ~idv(SITE):vm(PLANT_ID, k_full),
                       residual = ~idv(units),
                       data = phe_single, 
                       workspace = "3gb")
  
    h2_est <- vpredict(asr_out, h2 ~ V2/(V1+V2+V3)) %>% as_tibble() %>%
      mutate(site = names(site_v)[i],
             subpop = names(subpop_v)[j],
             phe = phe_v[k],
             bic = summary(asr_out)$bic[1],
             loglik = summary(asr_out)$loglik)
    }, error=function(e){cat("ERROR :",conditionMessage(e), "\n")})
  } else {
    tryCatch({
    asr_out <- asreml(phenotype ~ 1,
                       random = ~vm(PLANT_ID, k_full),
                       residual = ~idv(units),
                       data = phe_single, 
                       workspace = "3gb")
    
    h2_est <- vpredict(asr_out, h2 ~ V1/(V1+V2)) %>% as_tibble() %>%
      mutate(site = names(site_v)[i],
             subpop = names(subpop_v)[j],
             phe = phe_v[k],
             bic = summary(asr_out)$bic[1],
             loglik = summary(asr_out)$loglik)
  }, error=function(e){cat("ERROR :",conditionMessage(e), "\n")})
  }
  
  if(i == 12 & j == 1 & k == 1){
    h2_table <- h2_est
  } else {
    h2_table <- add_row(h2_table, h2_est)
  }
  tryCatch({
      save_plot(filename=file.path(outputdir, paste0("ASReml_performance_",
                                                 names(site_v)[i], "_",
                                                 names(subpop_v)[j],
                                                 "_", phe_v[k], ".png")), 
            plot = plot(asr_out))
    }, error=function(e){cat("ERROR :",conditionMessage(e), "\n")})
  
      }
    }
  }
  write_csv(h2_table, file.path(outputdir, paste0("h2_df_", phe_v[1],
                                                  "_related_functions_", 
                                                  "of_weather_v4.csv")))
}
```

# ASReml 07-29

## Model eight, texas, and north sites with a kinship : site interaction.
Then, find GxE and G proportions of variance.
## Use Midwest-specific kinship matrix for Midwest-only, and Gulf-specific kinship matrix for Gulf-only comparisons. And use (and visualize eigenvectors of? Midwest and Gulf kinship matrix for the comparisons with both.)
```{r}
library(tidyverse)
library(asreml)
library(switchgrassGWAS)
library(cowplot)

source("~/Github/Functions_ggplot-theme-adjustments_2018-01-03.R")
workingdir <- file.path("~", "Github", "pvdiv-phenology-gxe")
outputdir <- file.path(workingdir, "analysis", "heritability", "kinship_subpop")
metadata <- read_rds(file.path(workingdir, "data", "metadata.rds"))
sites <- read_rds(file.path(workingdir, "data", "sites.rds"))
subpops <- read_rds(file.path(workingdir, "data", "subpop_color_coding.rds"))
phe_gr_df <- read_rds(file.path(workingdir, "data",
                                "Weather_related_phe_for_asreml_h2_GR50.rds"))
phe_fl_df <- read_rds(file.path(workingdir, "data",
                                "Weather_related_phe_for_asreml_h2_FL50.rds"))

kinshipdir <- file.path(workingdir, "data")
k_gandm <- read_rds(file.path(kinshipdir, "Kinship_van_Raden_method_363g_Gulf_and_Midwest_subset_maf0.05.rds"))
k_midw <- read_rds(file.path(kinshipdir,
                "Kinship_van_Raden_method_134g_Midwest_subset_maf0.05.rds"))
k_gulf <- read_rds(file.path(kinshipdir, "Kinship_van_Raden_method_229g_Gulf_subset_maf0.05.rds"))

```

```{r}
site_v <- list(TX1 = "TX1", TX2 = "TX2", TX3 = "TX3", OK = "OK", MO = "MO",
               NE = "NE", MI = "MI", SD = "SD", eight_sites = c("TX1", "TX2", "TX3", "OK", "MO", "NE", "MI", "SD"), tx_sites = c("TX1", "TX2", "TX3"), north_sites = c("MO", "NE", "MI", "SD"))
subpop_v <- list(Gulf = "Gulf", Midwest = "Midwest", Gulf_and_Midwest = c("Gulf", "Midwest"))
phe_fl <- c("FL50", "dyln_fl50", "dyln_change_sec", "cgdd_12c_gr2fl",
           "crain_gr2fl", "crain_1d", "crain_3d", "crain_5d")
phe_gr <- c("GR50", "DYLN", "cgdd_12c_jan2gr", "tmin_18d", "cgdd_12c_18d",
            "tmin_10d", "cgdd_12c_10d", "tmin_5d", "cgdd_12c_5d")
phe_dfs <- list(phe_gr_df, phe_fl_df)
phe_lists <- list(phe_gr, phe_fl)
k_list <- list(Gulf = k_gulf, Midwest = k_midw, Gulf_and_Midwest = k_gandm)
h2_table <- tibble()
gxe_table <- tibble()

for(p in seq_along(phe_dfs)){
  phe_v <- phe_lists[[p]]
  phe <- phe_dfs[[p]]
  
  for(i in seq_along(site_v)){ # may need to adjust below for h2_table
    dir.create(file.path(outputdir, names(site_v)[i]), showWarnings = FALSE)
    for(j in seq_along(subpop_v)){
      k_use <- k_list[[j]]
      for(k in seq_along(phe_v)){
        phe_single <- phe %>%
          left_join(sites, by = "SITE") %>%
          filter(manu_site %in% site_v[[i]] & SUBPOP %in% subpop_v[[j]]) %>%
          rename(phenotype = eval(phe_v[k])) %>%
          mutate(PLANT_ID = as_factor(PLANT_ID),
                 SITE = as_factor(SITE),
                 PLOT_GL = as_factor(PLOT_GL)) %>%
          as.data.frame()
  
  if(length(site_v[[i]]) > 1){  # need to add site as a random factor if 
    # there is more than one site included in the dataset.
    tryCatch({
     asr_out <- asreml(phenotype ~ 1,
                       random = ~idv(SITE)*vm(PLANT_ID, k_use),
                       residual = ~idv(units),
                       data = phe_single, 
                       workspace = "3gb")
  
    Env_est <- vpredict(asr_out, E ~ V1/(V1+V2+V3+V4))
    GxE_est <- vpredict(asr_out, GxE_h2 ~ V3/(V1+V2+V3+V4))
    err_est <- vpredict(asr_out, err ~ V4/(V1+V2+V3+V4))
    h2_est <- vpredict(asr_out, G_h2 ~ V2/(V1+V2+V3+V4)) %>% as_tibble() %>%
      mutate(site = names(site_v)[i],
             subpop = names(subpop_v)[j],
             phe = phe_v[k],
             bic = summary(asr_out)$bic[1],
             loglik = summary(asr_out)$loglik,
             Env_Estimate = Env_est$Estimate[1],
             Env_SE = Env_est$SE[1],
             GxE_Estimate = GxE_est$Estimate[1],
             GxE_SE = GxE_est$SE[1],
             error_Estimate = err_est$Estimate[1],
             error_SE = err_est$SE[1])
    
    blup <- summary(asr_out, coef=TRUE)$coef.random %>% 
    as_tibble(rownames = "Effect")
    write_csv(blup, file.path(outputdir, names(site_v)[i], 
                              paste0("blups_", names(site_v)[i], "_",
                                     names(subpop_v)[j], "_", phe_v[k],
                                     ".csv")))
    }, error=function(e){cat("ERROR :",conditionMessage(e), "\n")})
  } else {
    tryCatch({
    asr_out <- asreml(phenotype ~ 1,
                       random = ~vm(PLANT_ID, k_use),
                       residual = ~idv(units),
                       data = phe_single, 
                       workspace = "3gb")
    
    h2_est <- vpredict(asr_out, h2 ~ V1/(V1+V2)) %>% as_tibble() %>%
      mutate(site = names(site_v)[i],
             subpop = names(subpop_v)[j],
             phe = phe_v[k],
             bic = summary(asr_out)$bic[1],
             loglik = summary(asr_out)$loglik)
    blup <- summary(asr_out, coef=TRUE)$coef.random %>% 
    as_tibble(rownames = "Effect")
    write_csv(blup, file.path(outputdir, names(site_v)[i],
                              paste0("blups_", names(site_v)[i], "_",
                                     names(subpop_v)[j], "_", phe_v[k],
                                     ".csv")))
  }, error=function(e){cat("ERROR :",conditionMessage(e), "\n")})
  }
  
  if(i == 1 & j == 1 & k == 1){
    h2_table <- h2_est
  } else if(i == 9 & j == 1 & k == 1){
    gxe_table <- h2_est
  } else if(length(site_v[[i]]) > 1){
    gxe_table <- add_row(gxe_table, h2_est)
  } else {
    h2_table <- add_row(h2_table, h2_est)
  }
  tryCatch({
      save_plot(filename=file.path(outputdir, names(site_v)[i],
                                   paste0("ASReml_performance_",
                                                 names(site_v)[i], "_",
                                                 names(subpop_v)[j],
                                                 "_", phe_v[k], ".png")), 
            plot = plot(asr_out))
    }, error=function(e){cat("ERROR :",conditionMessage(e), "\n")})
  
      }
    }
  }
  write_csv(h2_table, file.path(outputdir, paste0("h2_df_", phe_v[1],
                                                  "_related_functions_", 
                                                  "of_weather_v1.csv")))
  write_csv(gxe_table, file.path(outputdir, paste0("gxe_and_h2_df_", phe_v[1],
                                                  "_related_functions_", 
                                                  "of_weather_v1.csv")))
}
```

# ----
# D2F 

If flowering date frequently varies as a function of GDD in switchgrass, this explains observations that moving southern populations northwards delays flowering, and moving northern populations south hastens flowering (Sanderson et al 1996). < Indeed, we observed D2F for Midwest plants was shorter at TX than BRKG, and vice versa?>

```{r}
d2f_lat <- phenotypes %>%
  mutate(D2F = FL50 - GR50) %>%
  left_join(select(metadata, PLANT_ID, SUBPOP)) %>%
  filter(!(SITE %in% c("OVTN", "FRMI")) & WHERE == "GWAS" & SUBPOP %in% c("Gulf", "Midwest")) %>%
  group_by(SUBPOP, SITE) %>%
  summarise(mean_D2F = mean(D2F, na.rm = TRUE)) %>%
  left_join(select(sites, SITE, Latitude, manu_site)) %>%
  arrange(SUBPOP, Latitude) 
d2f_lat %>%
  ggplot(aes(x = Latitude, y = mean_D2F)) +
  geom_point(aes(color = SUBPOP))

summary(lm(mean_D2F ~ Latitude*SUBPOP, data = d2f_lat))
82-55
```

# ----------------------------
# G_BLUP GWAS

The plan is to do GWAS on G_BLUP from an eight site model with GxE modeled as site:kinship for each subpopulation and both combined. The traits I will do GWAS on for G_BLUPs are as follows:

eight_sites
FL50 (to show other assoc are stronger)
dyln_fl50
cgdd_12c_gr2fl

Maybe also do GWAS on G_BLUP from the North and Texas models as above, for the traits:
tx_sites & north_sites
FL50 (to show other assoc are stronger)
dyln_fl50
cgdd_12c_gr2fl
crain_gr2fl
crain_1d
GR50
cgdd_12c_10d

All I really want to show are that there are stronger G associations for the weather-derived variables than for FL50 and GR50. And possibly exploring G associations at subpopulation home and away sites, for the second bit. 
That might be too complicated. Eh.

It's possible I might want to confirm some of these G candidates with the fourway - how we see similar patterns in this cross for FL50 vs dyln/GDD, and there are some similar associations.

THEN, for the next section, do GWAS on the BLUPs from the eight single site models. For these traits:
all single sites...
FL50 (actually, maybe not)
dyln_fl50
cgdd_12c_gr2fl
crain_gr2fl
crain_1d
--
GR50 (or maybe not)
cgdd_12c_10d

Then do mash runs. First for single traits across all eight sites, to make U_hyp for the next step - take the U_ed with large percentage variance explained. Make U_hyp with those covariances but no covariance with anything else... 
Then run all the flowering-related traits together. Talk about how then SNPs can have effects that are strongest in response to one type of environmental cue. And how cues can be present or absent at different sites. And how this kind of analysis can help reveal all the patterns at play, because it doesn't make assumptions about how effects are structured.

Maybe bring back the QTL maps after this and talk about how patterns of effects differ across sites in these models, too. Though the Genstat model is less sensitive...

```{r}
h2all %>%
  filter(site %in% c("eight_sites"))
```

## Make phenotype df for GWAS
```{r}
workingdir <- file.path("~", "Github", "pvdiv-phenology-gxe")
outputdir <- file.path(workingdir, "analysis", "heritability", "kinship_subpop")
site_v <- list(TX1 = "TX1", TX2 = "TX2", TX3 = "TX3", OK = "OK", MO = "MO",
               NE = "NE", MI = "MI", SD = "SD", eight_sites = c("TX1", "TX2", "TX3", "OK", "MO", "NE", "MI", "SD"), tx_sites = c("TX1", "TX2", "TX3"), north_sites = c("MO", "NE", "MI", "SD"))
subpop_v <- list(Gulf = "Gulf", Midwest = "Midwest", Gulf_and_Midwest = c("Gulf", "Midwest"))
phe_eight <- c("FL50", "dyln_fl50", "cgdd_12c_gr2fl")
phe_txano <- c("FL50", "dyln_fl50", "cgdd_12c_gr2fl", "crain_gr2fl",
               "crain_1d", "GR50", "cgdd_12c_10d")
phe_single <- c("FL50", "dyln_fl50", "cgdd_12c_gr2fl", "crain_gr2fl",
               "crain_1d", "GR50", "cgdd_12c_10d")
phe_lists <- list(TX1 = phe_single, TX2 = phe_single, TX3 = phe_single, OK = phe_single, MO = phe_single,
               NE = phe_single, MI = phe_single, SD = phe_single, eight_sites = phe_eight, tx_sites = phe_txano, north_sites = phe_txano)

i = 9
j = 1
k = 1
phe_gwas <- tibble()

for(j in seq_along(subpop_v)){
    for(i in seq_along(site_v)){
      phe_v <- phe_lists[[i]]
      for(k in seq_along(phe_v)){
        phe_raw <- read_csv(file.path(outputdir, names(site_v)[i],
                                      paste0("blups_", names(site_v)[i], "_",
                                             names(subpop_v)[j], "_", phe_v[k],
                                             ".csv")))
        names(phe_raw)[2] <- paste0(phe_v[k], "_", names(site_v)[i])
        phe_proc <- phe_raw %>%
          separate(Effect, into = c("factor", "PLANT_ID"), sep = "\\)_") %>%
          filter(!is.na(PLANT_ID) & !grepl("^SITE", factor)) %>%
          left_join(metadata, by = "PLANT_ID") %>%
          filter(SUBPOP %in% subpop_v[[j]]) %>%
          select(PLANT_ID, 3)
        if(i == 1 & k == 1){
          phe_gwas <- phe_proc
        } else {
          phe_gwas <- phe_gwas %>%
            left_join(phe_proc, by = "PLANT_ID")
        }
      }
    }
  phe_gwas <- phe_gwas %>%
  select(PLANT_ID, ends_with("eight_sites"), ends_with("tx_sites"), ends_with("north_sites"), everything())
  write_csv(phe_gwas, path = file.path(workingdir, "analysis", "gwas", "BLUPs",
                                       paste0("BLUP_phenotypes_",
                                              "for_gwas_GR_FL_",
                                              names(subpop_v)[j],
                                              ".csv")))
}

ggplot(data = phe_gwas, aes(x = phe_gwas$cgdd_12c_gr2fl_eight_sites)) + geom_histogram()

sites %>% arrange(manu_site)
```


## Run GWAS 
```{r}
library(bigsnpr)
library(tidyverse)
library(switchgrassGWAS)
library(AnnotationDbi)
txdb <- loadDb(file = file.path("~", "Github", "pvdiv-genome",
                                "Pvirgatum_516_v5.1.gene.txdb.sqlite"))
workingdir <- file.path("~", "Github", "pvdiv-phenology-gxe")
inputdir <- file.path(workingdir, "analysis", "gwas", "BLUPs")
outputdir <- file.path(workingdir, "analysis", "gwas", "BLUPs")
subpop_v <- list(Gulf_and_Midwest = c("Gulf", "Midwest"), Midwest = "Midwest", Gulf = "Gulf")
snp_v <- c("12\\.1M", "8\\.8M", "10\\.3M")

inputfiles <- read_delim(file.path(workingdir, "analysis", "gwas",
                                   "inputkinship.txt"),  delim = " ", 
                                   col_names = "SNPfiles")
set_v <- list(eight_sites = "eight_sites", tx_sites = "tx_sites", north_sites = "north_sites", single_sites = "!sites")


for(j in 4){#seq_along(set_v)){
  for(i in seq_along(subpop_v)){
    dir.create(file.path(outputdir, names(set_v)[j]), showWarnings = FALSE)
    snp <- snp_attach(inputfiles$SNPfiles[i])
    phe_df <- read_csv(file.path(inputdir, paste0("BLUP_phenotypes_",
                                                  "for_gwas_GR_FL_",
                                                  names(subpop_v)[i],
                                                  ".csv"))) 
  if(names(set_v)[j] != "single_sites"){
    phe_df <- phe_df %>%
    dplyr::select(PLANT_ID, ends_with(names(set_v)[j]))
  } else {
    phe_df <- phe_df %>%
    dplyr::select(PLANT_ID, !ends_with("sites"))
    if(names(subpop_v)[i] != "Gulf"){
      phe_df <- phe_df %>%
    dplyr::select(PLANT_ID, ends_with("NE"), ends_with("MI"), ends_with("SD"))
    }
  }
  pvdiv_standard_gwas(snp, df = phe_df, type = "linear", 
                      outputdir = file.path(outputdir, names(set_v)[j]),
                      savegwas = TRUE, saveannos = TRUE, txdb = txdb,
                      minphe = 130)

  gwas_rds <- pvdiv_results_in_folder(path = file.path(outputdir,
                                                       names(set_v)[j]), 
                                      pattern = paste0("GWAS_datatable*.*",
                                                       snp_v[i]))
  phenotypes <- str_sub(gwas_rds, start = 16, end = -42) %>%
    str_replace(., "_$", "")
  out1 <- pvdiv_bigsnp2mashr(path = file.path(outputdir, names(set_v)[j]), 
                             snp = snp, gwas_rds = gwas_rds, 
                             phenotypes = phenotypes, numSNPs = 1000, 
                             clump = FALSE, scaled = FALSE, saveoutput = TRUE,
                             suffix = paste0("FL_", names(set_v)[j], "_",
                                             names(subpop_v)[i]))
  }
}


```


gwasdir <- file.path(workingdir, "analysis", "gwas", "G_BLUP")
log10p_gulf <- readRDS(file.path(gwasdir,                                 "log10p_strong_df_1000topSNPs_GR_FL_Gulf.rds"))
log10p_midw <- readRDS(file.path(gwasdir,                                 "log10p_strong_df_1000topSNPs_GR_FL_Midwest.rds"))
log10p_gandm <- readRDS(file.path(gwasdir,                                 "log10p_strong_df_1000topSNPs_GR_FL_Gulf_and_Midwest.rds"))

```{r}
j=1
i=3

names(set_v)[j]
names(subpop_v)[i]
bonferroni_v <- c(-log10(0.05/12100000), -log10(0.05/8800000), -log10(0.05/10300000))

gwasdir <- file.path(workingdir, "analysis", "gwas", "BLUPs", names(set_v)[j])
log10p_gulf <- readRDS(file.path(gwasdir, 
                                 paste0("log10p_strong_df_1000topSNPs_",
                                        "FL_", names(set_v)[j], "_Gulf",
                                        ".rds")))
log10p_midw <- readRDS(file.path(gwasdir,
                                 paste0("log10p_strong_df_1000topSNPs_", "FL_",
                                 names(set_v)[j], "_", "Midwest.rds")))
log10p_gandm <- readRDS(file.path(gwasdir,
                                  paste0("log10p_strong_df_1000topSNPs_",
                                         "FL_", names(set_v)[j], "_",
                                         "Gulf_and_Midwest.rds")))
log10p_df <- readRDS(file.path(gwasdir,
                                  paste0("log10p_strong_df_1000topSNPs_",
                                         "FL_", names(set_v)[j], "_",
                                         names(subpop_v)[i], ".rds")))
```

```{r}
log10p_gulf %>%
  pivot_longer(cols = -(SNP:POS), names_to = "PHE", values_to = "log10p") %>%
  ungroup() %>%
  filter(log10p > -log10(0.05/10300000)) %>%
  group_by(PHE) %>%
  tally() #%>%
  #write_csv(., paste0("../manuscript/tables/Gulf_", names(set_v)[j], "_SNPs_above_Bonferroni_by_trait.csv"))

log10p_midw %>%
  pivot_longer(cols = -(SNP:POS), names_to = "PHE", values_to = "log10p") %>%
  ungroup() %>%
  filter(log10p > -log10(0.05/8800000)) %>%
  group_by(PHE) %>%
  tally() #%>%
  #write_csv(., paste0("../manuscript/tables/Midwest_", names(set_v)[j], "_SNPs_above_Bonferroni_by_trait.csv"))

log10p_gandm %>%
  pivot_longer(cols = -(SNP:POS), names_to = "PHE", values_to = "log10p") %>%
  ungroup() %>%
  filter(log10p > -log10(0.05/12100000)) %>%
  group_by(PHE) %>%
  tally() #%>%
  #write_csv(., paste0("../manuscript/tables/Gulf_and_Midwest_", names(set_v)[j], "__SNPs_above_Bonferroni_by_trait.csv"))
  #
log10p_gandm %>%
  pivot_longer(cols = -(SNP:POS), names_to = "PHE", values_to = "log10p") %>%
  ungroup() %>%
  filter(log10p > -log10(0.05/12100000)) %>%
  mutate(Pos_Mb = floor(POS/100000)/10) %>%
  group_by(CHR, Pos_Mb) %>% tally()
bonferroni <- -log10(0.05/12100000)
log10p_gandm %>%
  filter(cgdd_12c_gr2fl_eight_sites_log10p > -log10(0.05/12100000) |
         FL50_eight_sites_log10p > -log10(0.05/12100000)) %>%
  group_by(cgdd_12c_gr2fl_eight_sites_log10p > FL50_eight_sites_log10p) %>%
  tally()
log10p_gandm %>%
  filter(dyln_fl50_eight_sites_log10p > -log10(0.05/12100000)  |
         FL50_eight_sites_log10p > -log10(0.05/12100000)) %>%
  group_by(dyln_fl50_eight_sites_log10p > FL50_eight_sites_log10p) %>%
  tally()

log10p_gandm %>%
  ggplot(aes(x = dyln_fl50_eight_sites_log10p, y = FL50_eight_sites_log10p)) + 
  geom_point(alpha = 0.5) + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) +
  geom_vline(xintercept = bonferroni, color = "red", linetype = 2) + 
  geom_hline(yintercept = bonferroni, color = "red", linetype = 2)
```
```{r}
i=1 # subpop
j=3 # site set 


names(set_v)[j]
names(subpop_v)[i]
bonferroni_v <- c(-log10(0.05/12100000), -log10(0.05/8800000), -log10(0.05/10300000))

gwasdir <- file.path(workingdir, "analysis", "gwas", "BLUPs", names(set_v)[j])
log10p_df <- readRDS(file.path(gwasdir,
                                  paste0("log10p_strong_df_1000topSNPs_",
                                         "FL_", names(set_v)[j], "_",
                                         names(subpop_v)[i], ".rds")))
top_set <- readRDS(file.path(gwasdir, 
                                paste0("Part-One-Output_Top-Effects-1000-SNPs_",
                                        "FL_", names(set_v)[j], "_",
                                        names(subpop_v)[i], ".rds")))

log10p_df %>%
  pivot_longer(cols = -(SNP:POS), names_to = "PHE", values_to = "log10p") %>%
  ungroup() %>%
  filter(log10p > bonferroni_v[i]) %>%
  group_by(CHR, POS) %>% tally()
log10p_df %>%
  pivot_longer(cols = -(SNP:POS), names_to = "PHE", values_to = "log10p") %>%
  ungroup() %>%
  filter(log10p > bonferroni_v[i]) %>%
  mutate(Pos_Mb = floor(POS/100000)/10) %>%
  group_by(CHR, Pos_Mb) %>% tally()

top50 <- top_set %>% ungroup() %>%
  top_n(100, max_score_log10p) %>%
  mutate(SNP = paste(CHR, POS, sep = "_"))
dyln_binom <- log10p_df %>%
  filter(SNP %in% top50$SNP) %>%
  group_by(dyln_fl50_north_sites_log10p > FL50_north_sites_log10p) %>%
  tally()
binom.test(x = dyln_binom$n[2], n = 100, alternative = "greater")

cgdd_binom <- log10p_df %>%
  filter(SNP %in% top50$SNP) %>%
  group_by(cgdd_12c_gr2fl_north_sites_log10p > FL50_north_sites_log10p) %>%
  tally()
binom.test(x = cgdd_binom$n[2], n = 100, alternative = "greater")

log10p_df %>%
  ggplot(aes(x = dyln_fl50_north_sites_log10p, y = FL50_north_sites_log10p)) + 
  geom_point(alpha = 0.5) + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) +
  geom_vline(xintercept = bonferroni_v[i], color = "red", linetype = 2) + 
  geom_hline(yintercept = bonferroni_v[i], color = "red", linetype = 2)
```


## Plot top GWAS hits
```{r}
for(i in seq_along(subpop_v)){
  for(j in 1:3){
    gwasdir <- file.path(workingdir, "analysis", "gwas", "BLUPs",
                         names(set_v)[j])
    top_set <- readRDS(file.path(gwasdir, 
                                paste0("Part-One-Output_Top-Effects-1000-SNPs_",
                                        "FL_", names(set_v)[j], "_",
                                        names(subpop_v)[i], ".rds")))
    top_set %>%
      group_by(n_cond_in_top_set) %>%
      tally()
    
    top_manh2 <- top_set %>%
      filter(max_score_log10p > 5.897) %>%
      ggplot(aes(x = POS, y = max_score_log10p)) + 
      geom_point(aes(color = Condition, shape = as_factor(n_cond_in_top_set))) + 
      #geom_line(aes(group = Condition)) +
      facet_wrap( ~ CHR, nrow = 1, scales = "free_x", 
                  strip.position = "bottom") +
      theme(legend.position = "top",
            axis.text.x = element_blank(), axis.ticks.x = element_blank(),
            panel.background = element_rect(fill=NA)) +
            labs(x = "Chromosome", y = "-log10(p value)") +
            scale_x_continuous(expand = c(0.18, 0.18)) +
      scale_color_viridis(discrete = TRUE, end = 0.97) +
      scale_shape_manual(values = c(4, 16, 17, 18, 14, 3, 4))
    save_plot(paste0("../manuscript/plots/", "FL_", names(set_v)[j], "_",
                     names(subpop_v)[i], "_top_500_Manhattan.png"), 
              base_height = 5, base_asp = 2.5, plot = top_manh2)
        
  }
}
```

## Overlap with QTL and between subpops

All I really want to show are that there are stronger G associations for the weather-derived variables than for FL50 and GR50. And possibly exploring G associations at subpopulation home and away sites, for the second bit. 
That might be too complicated. Eh.

It's possible I might want to confirm some of these G candidates with the fourway - how we see similar patterns in this cross for FL50 vs dyln/GDD, and there are some similar associations.


Comparing the top 100 hits within and between subpopulations, X of Y were found in the same 100kb regions.
```{r}
i=3 # subpop
j=1 # site set 


names(set_v)[j]
names(subpop_v)[i]
bonferroni_v <- c(-log10(0.05/12100000), -log10(0.05/8800000), -log10(0.05/10300000))

top_pos <- list()

gwasdir <- file.path(workingdir, "analysis", "gwas", "BLUPs", names(set_v)[j])
log10p_df <- readRDS(file.path(gwasdir,
                                  paste0("log10p_strong_df_1000topSNPs_",
                                         "FL_", names(set_v)[j], "_",
                                         names(subpop_v)[i], ".rds")))
for(i in 1:3){
top_set <- readRDS(file.path(gwasdir, 
                                paste0("Part-One-Output_Top-Effects-1000-SNPs_",
                                        "FL_", names(set_v)[j], "_",
                                        names(subpop_v)[i], ".rds")))

top_pos[[i]] <- top_set %>%
  ungroup() %>%
  top_n(n = 100, wt = max_score_log10p) %>%
  mutate(Pos_Mb = floor(POS/100000)/10) %>%
  group_by(CHR, Pos_Mb) %>%
  tally()
}
  
gandm_m <- top_pos[[1]] %>%
  inner_join(top_pos[[2]])
gandm_g <- top_pos[[1]] %>%
  inner_join(top_pos[[3]])
m_g <- top_pos[[2]] %>%
  inner_join(top_pos[[3]])
c(nrow(gandm_m)/nrow(top_pos[[1]]), nrow(gandm_m)/nrow(top_pos[[2]]),
  nrow(gandm_g)/nrow(top_pos[[1]]), nrow(gandm_g)/nrow(top_pos[[3]]),
  nrow(m_g)/nrow(top_pos[[2]]), nrow(m_g)/nrow(top_pos[[3]])
  )

top50 <- top_set %>% ungroup() %>%
  top_n(100, max_score_log10p) %>%
  mutate(SNP = paste(CHR, POS, sep = "_"))
dyln_binom <- log10p_df %>%
  filter(SNP %in% top50$SNP) 
```

```{r}
i=3 # subpop
j=1 # site set 


names(set_v)[j]
names(subpop_v)[i]
bonferroni_v <- c(-log10(0.05/12100000), -log10(0.05/8800000), -log10(0.05/10300000))

top_pos <- list()


log10p_df <- readRDS(file.path(gwasdir,
                                  paste0("log10p_strong_df_1000topSNPs_",
                                         "FL_", names(set_v)[j], "_",
                                         names(subpop_v)[i], ".rds")))
for(j in 1:3){
  gwasdir <- file.path(workingdir, "analysis", "gwas", "BLUPs", names(set_v)[j])
  top_set <- readRDS(file.path(gwasdir, 
                                paste0("Part-One-Output_Top-Effects-1000-SNPs_",
                                        "FL_", names(set_v)[j], "_",
                                        names(subpop_v)[i], ".rds")))

top_pos[[j]] <- top_set %>%
  ungroup() %>%
  top_n(n = 500, wt = max_score_log10p) %>%
  mutate(Pos_Mb = floor(POS/100000)/10) %>%
  group_by(CHR, Pos_Mb) %>%
  tally()
}
  
gandm_m <- top_pos[[1]] %>%
  inner_join(top_pos[[2]])
gandm_g <- top_pos[[1]] %>%
  inner_join(top_pos[[3]])
m_g <- top_pos[[2]] %>%
  inner_join(top_pos[[3]])
c(nrow(gandm_m)/nrow(top_pos[[1]]), nrow(gandm_m)/nrow(top_pos[[2]]),
  nrow(gandm_g)/nrow(top_pos[[1]]), nrow(gandm_g)/nrow(top_pos[[3]]),
  nrow(m_g)/nrow(top_pos[[2]]), nrow(m_g)/nrow(top_pos[[3]])
  )

top50 <- top_set %>% ungroup() %>%
  top_n(100, max_score_log10p) %>%
  mutate(SNP = paste(CHR, POS, sep = "_"))
dyln_binom <- log10p_df %>%
  filter(SNP %in% top50$SNP) 
```
for gulf and midwest:
4.3-6.5% overlap in top 100 between eight and tx; no overlap in top 100 between eight and north or north and texas.
500: 8% eight and texas; 1.6% eight and north. 0.9% north and texas.

for midwest:
5% overlap in top 500 eight and texas; 18% overlap eight and north; 1.7% north and texas.

for gulf:
6.2% overlap eight and texas in top 500; 5.2% overlap eight and north; 2.7% overlap north and texas.

I think the top 500 SNPs are good here - give you large enough numbers.
