---
title: "SI_Appendix_Datasets"
author: "Alice MacQueen"
date: 2021-06-24
output: html_document
---

Edited 2024-05-17+ for revision2.

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(here)
library(cowplot)
library(switchgrassGWAS)
library(ashr)
```

# 

Make all these datasets:

```         
Dataset 1. SNP-associated effects and standard errors for green-up estimated using the Gulf genetic subpopulation.
Dataset 2. SNP-associated effects and standard errors for green-up estimated using the Midwest genetic subpopulation.
Dataset 3. SNP-associated effects and standard errors for green-up estimated using Both genetic subpopulations.
Dataset 4. SNP-associated effects and standard errors for flowering estimated using the Gulf genetic subpopulation.
Dataset 5. SNP-associated effects and standard errors for flowering estimated using the Midwest genetic subpopulation.
Dataset 6. SNP-associated effects and standard errors for flowering estimated using Both genetic subpopulations.
Dataset 7. Genes within QTL that have functionally validated roles in flowering in rice.
Dataset 8. Overlap between quantitative trait loci for flowering measured using an outbred pseudo-F2 mapping population, and significant mash effect estimates (with a log10Bayes Factor > 2, or in the 1% tail of most significant mash results) found using three overlapping diversity panels. Genes within 20kb with a functionally validated role in flowering in rice are  indicated. 
```

# Dataset 1-6: mash objects

```{r}
mashdir <- here("analysis", "mash_greedy_algorithm", "mash_h2_diag_covar")

numSNPs_v <- c(19000, 19000, 19000, 19000, 24000, 33000)
suffix_v <- c("FL50_Gulf_and_Midwest", "FL50_Gulf", "GR50_Gulf_and_Midwest",
              "GR50_Gulf", "GR50_Midwest", "FL50_Midwest")
```

```{r}
m_gulf_gr_hyp <- readRDS(file.path(mashdir,  
                                   paste0("Strong_mash_posterior_",
                                          "est_w_ahm_random_greedy_h2_diag_Ulist", 
                                          numSNPs_v[4], "_SNPs_",
                                          suffix_v[4], ".rds")))
m_gulf_fl_hyp <- readRDS(file.path(mashdir,  
                                   paste0("Strong_mash_posterior_",
                                          "est_w_ahm_random_greedy_h2_diag_Ulist", 
                                          numSNPs_v[2], "_SNPs_",
                                          suffix_v[2], ".rds")))

m_midw_gr_hyp <- readRDS(file.path(mashdir,  
                                   paste0("Strong_mash_posterior_",
                                          "est_w_ahm_random_greedy_h2_diag_Ulist", 
                                          numSNPs_v[5], "_SNPs_",
                                          suffix_v[5], ".rds")))
m_midw_fl_hyp <- readRDS(file.path(mashdir,  
                                   paste0("Strong_mash_posterior_",
                                          "est_w_ahm_random_greedy_h2_diag_Ulist", 
                                          numSNPs_v[6], "_SNPs_",
                                          suffix_v[6], ".rds")))

m_gam_gr_hyp <- readRDS(file.path(mashdir,  
                                   paste0("Strong_mash_posterior_",
                                          "est_w_ahm_random_greedy_h2_diag_Ulist", 
                                          numSNPs_v[3], "_SNPs_",
                                          suffix_v[3], ".rds")))
m_gam_fl_hyp <- readRDS(file.path(mashdir,  
                                   paste0("Strong_mash_posterior_",
                                          "est_w_ahm_random_greedy_h2_diag_Ulist", 
                                          numSNPs_v[1], "_SNPs_",
                                          suffix_v[1], ".rds")))
```




## SI Appendix Table S1 Info

#### random SNP inputs

```{r}
popkey = list(all = "Gulf_and_Midwest", midwest = "Midwest", gulf = "Gulf")
phekey = list(greenup = "GR50", flowering = "FL50")
# i=2; j=1

for (i in seq_along(popkey)) {
  for (j in seq_along(phekey)) {
    
    if (i == 2 & j == 1) {
      betas <- readRDS(file = here("analysis", "mash", popkey[[i]], "inputs", 
                                 paste0("B_hat_random_df_24000topSNPs_",
                                        phekey[[j]], "_", popkey[[i]], ".rds")))
      stderr <- readRDS(file = here("analysis", "mash", popkey[[i]], "inputs", 
                                 paste0("S_hat_random_df_24000topSNPs_",
                                        phekey[[j]], "_", popkey[[i]], ".rds")))
    } else if (i == 2 & j == 2) {
      betas <- readRDS(file = here("analysis", "mash", popkey[[i]], "inputs", 
                                 paste0("B_hat_random_df_33000topSNPs_",
                                        phekey[[j]], "_", popkey[[i]], ".rds")))
      stderr <- readRDS(file = here("analysis", "mash", popkey[[i]], "inputs", 
                                 paste0("S_hat_random_df_33000topSNPs_",
                                        phekey[[j]], "_", popkey[[i]], ".rds")))
    } else {
      betas <- readRDS(file = here("analysis", "mash", popkey[[i]], "inputs", 
                                 paste0("B_hat_random_df_19000topSNPs_",
                                        phekey[[j]], "_", popkey[[i]], ".rds")))
      stderr <- readRDS(file = here("analysis", "mash", popkey[[i]], "inputs", 
                                 paste0("S_hat_random_df_19000topSNPs_",
                                        phekey[[j]], "_", popkey[[i]], ".rds")))
    }
    print(paste(names(popkey)[i], names(phekey)[j], "random SNP set (input into greedy mash algorithm) has", nrow(betas), "SNPs"))
  }
}

```

So random SNP sets have 71-127K SNPs.

#### strong SNP inputs

Numbers for this section: We then selected 19K relatively unlinked (r2 < 0.2) SNPs from each univariate GWAS that had the largest effect estimates in each GWAS (hereafter "strong" effects), which gave sets of 48K - 80K SNPs, and jointly re-estimated these effect size estimates at all eight sites using the set of significant covariance matrices that most improved the model fit on the random effects. Notably, the majority of these "strong" effects were not significant in univariate GWAS.

```{r}
switchgrassGWAS:::get_marker_df(m_gulf_gr_hyp) # 45931
switchgrassGWAS:::get_marker_df(m_gulf_fl_hyp) # 50490
switchgrassGWAS:::get_marker_df(m_midw_gr_hyp) # 49030
switchgrassGWAS:::get_marker_df(m_midw_fl_hyp) # 63367
switchgrassGWAS:::get_marker_df(m_gam_gr_hyp)  # 44765
switchgrassGWAS:::get_marker_df(m_gam_fl_hyp)  # 35735
```

```{r}
m <- m_gulf_gr_hyp
dataset_creation_function <- function(m, filename){

  all_effects <- switchgrassGWAS:::get_marker_df(m)
  BF <- switchgrassGWAS:::get_log10bf(m) |> 
    as.vector()
  all_effects$log10BF <- BF
  
  Means <- get_pm(m)
  StdErr <- get_psd(m)
  lfsr <- get_lfsr(m)
  # increase clarity of column names for Supplement data
  colnames(StdErr) <- colnames(StdErr) %>%
    str_replace(., "Stand_Bhat", "Effect_StandardError_") %>%
    str_replace(., "Gulf_and_Midwest", "Both") %>%
    str_replace(., "GR50", "GreenupDate") %>%
    str_replace(., "FL50", "FloweringDate")
  colnames(Means) <- colnames(Means) %>%
    str_replace(., "Stand_Bhat", "Effect_Mean_") %>%
    str_replace(., "Gulf_and_Midwest", "Both") %>%
    str_replace(., "GR50", "GreenupDate") %>%
    str_replace(., "FL50", "FloweringDate")
  colnames(lfsr) <- colnames(lfsr) %>%
    str_replace(., "Stand_Bhat", "Effect_lfsr_") %>%
    str_replace(., "Gulf_and_Midwest", "Both") %>%
    str_replace(., "GR50", "GreenupDate") %>%
    str_replace(., "FL50", "FloweringDate")
  
  all_effects <- cbind(all_effects, Means, StdErr, lfsr) %>%
    select(starts_with("Marker"), log10BF, ends_with("TX1"), ends_with("TX2"),
           ends_with("TX3"), ends_with("OK"), ends_with("MO"), ends_with("NE"),
           ends_with("MI"), ends_with("SD"))
  
  all_effects |> 
    write_csv(filename)
  
}


dataset_creation_function(m_gulf_gr_hyp, filename = here("manuscript", "revision2", "SI_Appendix", "Dataset_1_Gulf_vegetative_growth_date.csv"))
dataset_creation_function(m_gulf_fl_hyp, filename = here("manuscript", "revision2", "SI_Appendix", "Dataset_4_Gulf_reproductive_growth_date.csv"))

dataset_creation_function(m_gam_gr_hyp, filename = here("manuscript", "revision2", "SI_Appendix", "Dataset_3_Both_subpopulations_vegetative_growth_date.csv"))
dataset_creation_function(m_gam_fl_hyp, filename = here("manuscript", "revision2", "SI_Appendix", "Dataset_6_Both_subpopulations_reproductive_growth_date.csv"))

dataset_creation_function(m_midw_fl_hyp, filename = here("manuscript", "revision2", "SI_Appendix", "Dataset_5_Midwest_reproductive_growth_date.csv"))
dataset_creation_function(m_midw_gr_hyp, filename = here("manuscript", "revision2", "SI_Appendix", "Dataset_2_Midwest_vegetative_growth_date.csv"))
```


### Overlap with Hd1 from switchgrass functional validation study

```{r}
df_gam_fl <- read_csv(here("manuscript", "revision2", "SI_Appendix", "Dataset_6_Both_subpopulations_reproductive_growth_date.csv"))
df_gam_fl <- read_csv(here("manuscript", "revision2", "SI_Appendix", "Dataset_4_Gulf_reproductive_growth_date.csv"))
df_gam_fl <- read_csv(here("manuscript", "revision2", "SI_Appendix", "Dataset_5_Midwest_reproductive_growth_date.csv"))

df_gam_fl |> 
  separate(Marker, into = c("Chr", "Pos"), sep = "_", 
           remove = FALSE, convert = TRUE) |> 
  filter(Chr == "Chr04K", between(Pos, 23100000, 23200000)) |> 
  pivot_longer(cols = matches("lfsr"), names_to = "Garden", values_to = "lfsr")

get_significant_results(m_gam_fl_hyp) |> 
  as.data.frame() |> 
  rownames_to_column(var = "Marker") |> 
  separate(Marker, into = c("Chr", "Pos"), sep = "_", 
           remove = FALSE, convert = TRUE) |> 
  filter(Chr == "Chr04K", between(Pos, 23100000, 23200000))

mash_plot_effects(m = m_gam_fl_hyp, i = 13819)
```
Chr04K_23164108 in Both has log10BF ~ 2.3
Chr04K_23179015 in Gulf has log10BF ~ 1.86
Nearby SNPs _not_ significant in Midwest. log10BF ~ 0.1

# Dataset 7

```{r}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("AnnotationDbi")
BiocManager::install("GenomicFeatures")
BiocManager::install("VariantAnnotation")
```

Find: `qtlplot` in Analysis_v0.6_QTL_interval_plotting. Really, find code to remake this with new QTL from revision2.

Find this file on NAS: "\~/Github/pvdiv-genome/Pvirgatum_516_v5.1.gene.txdb.sqlite"

Couldn't find it on the NAS, scp'd it from the Juenger Lab Server: `scp -i ~/Documents/server_keys/jls_alice alice@popgen.biosci.utexas.edu:/home/alice/Github/pvdiv-genome/Pvirgatum_516_v5.1.gene.txdb.sqlite ~/Documents/Github/pvdiv-phenology-gxe/data-raw/ignore/`

It's too big to commit; had to add it to data-raw/ignore files ignored by .gitignore.

```{r}
qtldf <- read_csv(here("phenology-fourway/5-QTLmappingForAlice/QTLsWithFlankMarkers_FL50_phenotypes_Alice_7_SITES_Full_model.csv"))

qtldf <- qtldf %>%
  separate(flank_lo, into = c("CHR", "Pos_Mb_lo"), sep = "_", convert = TRUE) %>%
  separate(flank_hi, into = c("flank", "Pos_Mb_hi"), sep = "_", convert = TRUE) %>%
  mutate(Pos_Mb_lo = round(Pos_Mb_lo, digits = 2),
         Pos_Mb_hi = round(Pos_Mb_hi, digits = 2)) %>%
  dplyr::select(CHR, Pos_Mb_lo, Pos_Mb_hi, lodcolumn, lod, everything()) %>%
  arrange(CHR, Pos_Mb_lo)

qtlplot <- qtldf %>%
  mutate(phe = case_when(lodcolumn == "FL50" ~ "Flowering Date",
                         lodcolumn == "dyln_fl50" ~ "Daylength",
                         lodcolumn == "dyln_change_sec" ~ 
                           "daylength change (s)",
                         lodcolumn == "dyln_change_sec_2d_prior" ~ 
                           "daylength change, 2 days prior (s)",
                         lodcolumn == "GR50" ~ "Greenup Date",
                         lodcolumn == "crain_gr2fl" ~ "Rainfall, GR to FL")) %>%
  #filter(phe %in% c("Daylength", "GDD, GR to FL")) %>%
  mutate(POS_lo = Pos_Mb_lo * 1000000,
         POS_hi = Pos_Mb_hi * 1000000)

qtlplot$phe <- factor(qtlplot$phe, 
                      levels = c("Flowering Date", "Rainfall, GR to FL",
                                 "GDD, GR to FL", "Daylength",
                                 "daylength change (s)",
                                 "daylength change, 2 days prior (s)",
                                 "Rainfall, 1 day",
                                 "Greenup Date",
                                                  #"Rainfall, 2d sum",
                                                  "Rainfall, 3d sum",
                                                  "Rainfall, 5d sum"#,
                                                  #"Rainfall, 7d sum"
                                                  ))
```

```{r}
library(AnnotationDbi)
library(GenomicFeatures)
library(switchgrassGWAS)
qtl_anno_df <- qtlplot %>%
  dplyr::rename(start = POS_lo, end = POS_hi)
not_all_na <- function(x) any(!is.na(x))
txdb <- loadDb("~/Documents/Github/pvdiv-phenology-gxe/data-raw/ignore/Pvirgatum_516_v5.1.gene.txdb.sqlite")
# source("~/Github/Functions_ggplot-theme-adjustments_2018-01-03.R")

ftanno <- read_csv(here("data/Pvirgatum_v5.1.annotation_FTGeneHomologs.csv"))
ftkey <- read_delim(here("data", "OsGene_keyword_table.txt"),
                    delim = "\t")
ftanno %>%
  filter(!is.na(OsGene)) %>%
  left_join(ftkey, by = c("OsGene" = "Symbol")) %>%
  filter(Keyword %in% c("flowering time", "heading date")) %>% unique()

osgene_keyword_df <- ftkey %>%
  dplyr::select(-RAPdb, -MSU) %>%
  group_by(Symbol, Keyword) %>%
  dplyr::slice(1) %>%
  pivot_wider(names_from = Keyword, values_from = Title)

osgene_key <- osgene_keyword_df %>%
  dplyr::select(Symbol, `flowering time`, `heading date`, anther, flower,
                `floral meristem`, `floral meristem determinacy`, flowering,
                `flower development`, `carpel differentiation`, stamen, 
                `stamen number`, vegetative, `plant growth`, `chloroplast development`, chloroplast) %>%
  pivot_longer(cols = -Symbol, names_to = "Keyword", values_to = "Title") %>%
  filter(!is.na(Title)) %>%
  pivot_wider(names_from = Keyword, values_from = Title)

os_annos <- ftanno %>%
  filter(OsGene %in% osgene_key$Symbol | !is.na(AtGene)) %>%
  mutate(Pos_Mb = round(start/10000)/100,
         #Pos_Mb_e = floor(start/10000)/100,
         POS_bin = round(Pos_Mb*100/2)) %>%# ,
         #POS_bin_hi = round(Pos_Mb_e*100/2))
  dplyr::select(OsGene, AtGene, locusName, everything()) %>%
  dplyr::rename(`Gene ID` = locusName)

qtl_annos <- switchgrassGWAS::pvdiv_table_topsnps(df = qtl_anno_df, type = "table", txdb = txdb) %>%
    inner_join(os_annos, by = c("Gene ID")) %>%
    left_join(osgene_key, by = c("OsGene" = "Symbol")) %>%
    dplyr::select(where(not_all_na)) %>%
    group_by(CHR, pos, `Gene ID`) %>%
    dplyr::slice(1) 

names(qtl_annos)
all_na_gr_fl <- rowSums(is.na(qtl_annos[,59:70]))
add_column(qtl_annos, GR_FL_Osgene_NA = all_na_gr_fl, .before = TRUE) %>%
  group_by(CHR, pos, GR_FL_Osgene_NA < 12) %>%
  tally()
flowering_qtl_annos <- add_column(qtl_annos, GR_FL_Osgene_NA = all_na_gr_fl, .before = TRUE) %>%
  filter(GR_FL_Osgene_NA < 12) %>%
  dplyr::select(CHR, pos, phe, lod, Pos_Mb_lo, Pos_Mb_hi, ci_lo, ci_hi, `Gene ID`,  gene_start:gene_width, `Arabidopsis thaliana homolog`:`Rice gene annotation`, OsGene, anther:`floral meristem determinacy`) |> 
  mutate(
    Phenology_type = case_when(phe %in% 
                                 c("Flowering Date", 
                                   "daylength change, 2 days prior (s)") ~
                                 "Flowering Date",
                               phe %in% c("Greenup Date", 
                                          "daylength change (s)") ~ 
                                 "Greenup Date",
                               TRUE ~ NA_character_
                               ),
    .before = 3)

flowering_qtl_annos |> 
  filter(Phenology_type == "Greenup Date")
  
flowering_qtl_annos |> 
  arrange(Phenology_type) |> 
  write_csv(here("manuscript/revision2/SI_Appendix/Dataset_7_QTL_intervals_with_functionally_validated_homologs.csv"))
```

# Dataset 8

All flowering and green-up QTL intervals contained at least one SNP significant in at least one mash run at a log10-transformed Bayes Factor \> 2, or in the 1% tail of significance, whichever was stricter.

## Objects generated earlier

Get `qtlplot` object by running code starting at about line 233 (for Dataset 7). Get `m_gulf_fl_hyp` and other objects by running code starting at about line 35 (for Datasets 1-6).

## Thresholds

Decide on cutoff to use for rug plots and enrichment analysis: SNPs sig in 1% tail or BF\>2, whichever is stricter

1% tail is stricter in all cases except Midwest flowering, where BF\>2 is stricter, equivalent to a 0.49% tail (0.995119 quantile).

What is the Bayes Factor at the 99th quantile? If it's not above 2, what is the quantile where the BF \> 2?

```{r}
#quantile(pw_gf$log10BayesFactor, 0.99)  # Gulf: Use 2.56 BF
#quantile(pw_mf$log10BayesFactor, 0.99)  # Midwest: Use 2 BF
#quantile(pw_gamf$log10BayesFactor, 0.99)  # GAM: Use 2.65 BF
#quantile(pw_mf$log10BayesFactor, 0.995119)

#quantile(pw_gg$log10BayesFactor, 0.99)  # Gulf: Use 2.0989 BF
#quantile(pw_mg$log10BayesFactor, 0.99)  # Midwest: Use 4.6831 BF
#quantile(pw_gamg$log10BayesFactor, 0.99)  # GAM: Use 3.0865 BF
```

Now there are significant greenup date QTL, separate QTL & mash into greenup and flowering associations, and look for overlaps just within each phenology type.

```{r}
m <- m_gulf_fl_hyp
qtl_df <- qtlplot |>   # split by Phenology type. Define based on Figure 4.
  mutate(
    Phenology_type = case_when(phe %in% 
                                 c("Flowering Date", 
                                   "daylength change, 2 days prior (s)") ~
                                 "Flowering Date",
                               phe %in% c("Greenup Date", 
                                          "daylength change (s)") ~ 
                                 "Greenup Date",
                               TRUE ~ NA_character_
                               ),
    .before = 3)
qtl_fl <- qtl_df |> 
  filter(Phenology_type == "Flowering Date")
qtl_gr <- qtl_df |> 
  filter(Phenology_type == "Greenup Date")

subpop = "Gulf"
phe = "Flowering Date"

qtl_mash_overlap_function <- function(qtl, m, subpop, phe, onepcttail = 2){

  mash_df <- switchgrassGWAS:::get_log10bf(m = m) %>%
      as.data.frame() %>%
      dplyr::mutate(value = row_number()) %>%
      as_tibble() %>%
      left_join(switchgrassGWAS:::get_marker_df(m = m), by = "value") %>%
      dplyr::rename(log10BayesFactor = .data$V1) %>%
      #dplyr::select(-.data$value) |> 
  ## TODO: FIX list/matrix column here
    tidyr::separate(.data$Marker, into = c("Chr", "Pos"), remove = FALSE, 
                    sep = "_", extra = "merge", convert = TRUE) %>%
    dplyr::arrange(.data$Chr, .data$Pos)

  for(i in 1:nrow(qtl)){  # from 1 to 16
    ## In block & > 2
    qtl_sub <- mash_df %>%
      filter(Chr %in% qtl$CHR[i] & between(Pos, qtl$POS_lo[i],
                                               qtl$POS_hi[i])) %>% 
      filter(log10BayesFactor > 2)
  
    if(i == 1){
    outputdf <- tibble(Chr = qtl$CHR[i], QTL_Pos_lo = qtl$POS_lo[i], 
                       QTL_Pos_hi = qtl$POS_hi[i], 
                       QTL_phenotype = qtl$lodcolumn[i], 
                       QTL_LOD = qtl$lod[i], mash_Pos = qtl_sub$Pos, 
                       mash_idx = qtl_sub$value,
                       mash_log10BayesFactor = qtl_sub$log10BayesFactor, mash_subpop = subpop, 
                       mash_phenotype = phe)
    } else {
      outputdf1 <- tibble(Chr = qtl$CHR[i], QTL_Pos_lo = qtl$POS_lo[i], 
                          QTL_Pos_hi = qtl$POS_hi[i], 
                          QTL_phenotype = qtl$lodcolumn[i], 
                          QTL_LOD = qtl$lod[i], mash_Pos = qtl_sub$Pos, 
                          mash_idx = qtl_sub$value,
                          mash_log10BayesFactor = qtl_sub$log10BayesFactor,
                          mash_subpop = subpop, 
                          mash_phenotype = phe)
      outputdf <- outputdf %>%
        full_join(outputdf1)
    }
  }
  outputdf <- outputdf |> 
    mutate(
      Marker_in_mash_1pct_tail = 
        case_when(mash_log10BayesFactor >= onepcttail ~ TRUE,
                  mash_log10BayesFactor < onepcttail ~ FALSE,
                  TRUE ~ NA)
    )
  return(outputdf)
}

overlap1 <- qtl_mash_overlap_function(qtl_fl, m_gulf_fl_hyp, 
                                      subpop = "Gulf", 
                                      phe = "Flowering Date",
                                      onepcttail = 2.56) 
overlap2 <- qtl_mash_overlap_function(qtl_fl, m_midw_fl_hyp, 
                                      subpop = "Midwest", 
                                      phe = "Flowering Date",
                                      onepcttail = 2) 
overlap3 <- qtl_mash_overlap_function(qtl_fl, m_gam_fl_hyp, 
                                      subpop = "Both", 
                                      phe = "Flowering Date",
                                      onepcttail = 2.65) 
overlap4 <- qtl_mash_overlap_function(qtl_gr, m_gulf_gr_hyp, 
                                      subpop = "Gulf", 
                                      phe = "Greenup Date",
                                      onepcttail = 2.0989) 
overlap5 <- qtl_mash_overlap_function(qtl_gr, m_midw_gr_hyp, 
                                      subpop = "Midwest", 
                                      phe = "Greenup Date",
                                      onepcttail = 4.6831) 
overlap6 <- qtl_mash_overlap_function(qtl_gr, m_gam_gr_hyp, 
                                      subpop = "Both", 
                                      phe = "Greenup Date",
                                      onepcttail = 3.0865) 

overlap4 %>%
  full_join(overlap5) %>%
  full_join(overlap6) |> 
  arrange(Chr, mash_Pos)
dataset_8 <- overlap1 %>%
  full_join(overlap2) %>%
  full_join(overlap3) %>%
  full_join(overlap4) %>%
  full_join(overlap5) %>%
  full_join(overlap6) |> 
  arrange(mash_phenotype, Chr, mash_Pos)

write_csv(dataset_8, file = here("manuscript", "revision2", "SI_Appendix",
                                 "Dataset_8_QTL_mash_overlap.csv"))
```


### Exploratory effect plots for mash QTL overlaps

2N 59.5Mb: biggest effect at TX2, mostly opposite to other gardens or other gardens have NSE
2N 60.9Mb: biggest effect at TX2, AP between TX1 and TX2
2 Gulf effects I would say look nothing like that.
```{r}
overlap <- dataset_8 %>%
  select(mash_idx, mash_subpop, everything()) |> 
  distinct(mash_idx, .keep_all = TRUE)
overlap
```

```{r}
refactor_conditions <- function(phe, subpop, idx) {
  if (phe == "Flowering Date" & subpop == "Both") {
    plot <- mash_plot_effects(m_gam_fl_hyp, i = idx)
    plot$effect_df$Conditions <- 
      factor(plot$effect_df$Conditions, 
             levels = c(paste0("Gulf_and_Midwest_FL50_", c("TX1", "TX2", "TX3",
                                                           "OK", "MO", "NE",
                                                           "MI", "SD"))))
  } else if (phe == "Flowering Date" & subpop == "Gulf") {
    plot <- mash_plot_effects(m_gulf_fl_hyp, i = idx)
    plot$effect_df$Conditions <- 
      factor(plot$effect_df$Conditions, 
             levels = c(paste0("Gulf_FL50_", c("TX1", "TX2", "TX3",
                                               "OK", "MO", "NE",
                                               "MI", "SD"))))
  } else if (phe == "Flowering Date" & subpop == "Midwest") {
    plot <- mash_plot_effects(m_midw_fl_hyp, i = idx)
    plot$effect_df$Conditions <- 
      factor(plot$effect_df$Conditions, 
             levels = c(paste0("Midwest_FL50_", c("TX1", "TX2", "TX3",
                                               "MO", "M", "SD"))))
    } else if (phe == "Greenup Date" & subpop == "Both") {
    plot <- mash_plot_effects(m_gam_gr_hyp, i = idx)
    plot$effect_df$Conditions <- 
      factor(plot$effect_df$Conditions, 
             levels = c(paste0("Gulf_and_Midwest_GR50_", c("TX1", "TX2", "TX3",
                                                           "OK", "MO", "NE",
                                                           "MI", "SD"))))
  } else if (phe == "Greenup Date" & subpop == "Gulf") {
    plot <- mash_plot_effects(m_gulf_gr_hyp, i = idx)
    plot$effect_df$Conditions <- 
      factor(plot$effect_df$Conditions, 
             levels = c(paste0("Gulf_GR50_", c("TX1", "TX2", "TX3",
                                               "OK", "MO", "NE",
                                               "MI", "SD"))))
  } else if (phe == "Greenup Date" & subpop == "Midwest") {
    plot <- mash_plot_effects(m_midw_gr_hyp, i = idx)
    plot$effect_df$Conditions <- 
      factor(plot$effect_df$Conditions, 
             levels = c(paste0("Midwest_GR50_", c("TX1", "TX2", "TX3", "OK",
                                               "MO", "NE", "MI", "SD"))))
  } else stop("not built in yet")
  plot$ggobject <- ggplot(data = plot$effect_df) + 
      geom_point(mapping = aes(x = as.factor(.data$Conditions), 
                               y = .data$mn)) + 
      switchgrassGWAS::theme_oeco + 
      geom_errorbar(mapping = aes(ymin = .data$mn - 
                                    .data$se, ymax = .data$mn + .data$se, 
                                  x = .data$Conditions), 
                    width = 0.3) + geom_hline(yintercept = 0, lty = 2) + 
        labs(x = "Conditions", y = "Effect Size") + 
      theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
      coord_flip() +
      ggtitle(plot$marker)
  return(plot)
}
```

```{r}
gr_qtl_overlap <- overlap |> 
  mutate(row_n = row_number(),
         .before = 1) |> 
  filter(mash_phenotype == "Greenup Date") |> 
  arrange(Chr, desc(mash_log10BayesFactor)) |> 
  select(row_n, mash_subpop, mash_log10BayesFactor, Chr, QTL_Pos_lo,
         Marker_in_mash_1pct_tail, everything())

fl_qtl_overlap <- overlap |> 
  mutate(row_n = row_number(),
         .before = 1) |> 
  filter(mash_phenotype == "Flowering Date") |> 
  arrange(Chr, desc(mash_log10BayesFactor)) |> 
  select(row_n, mash_subpop, mash_log10BayesFactor, Chr, QTL_Pos_lo,
         Marker_in_mash_1pct_tail, everything())
```
row_n 117: No match to QTL effects, just large Midwest mash effect at MO

9N QTL: (-) effects at OK & MI. 

For greenup, all but 3 significant mash SNPs (in the Gulf subpop) have just a single significant effect at MO. No QTL are like this, the only strong effect at MO is on Chr3K in the QTL. The 3 Gulf QTL don't really look like the 9N QTL either.

```{r}
i = 142 # 152
for (i in 1:nrow(gr_qtl_overlap)) {
  i <- gr_qtl_overlap$row_n[i]
  replot <- refactor_conditions(phe = overlap$mash_phenotype[i],
                    subpop = overlap$mash_subpop[i],
                    idx = overlap$mash_idx[i])
  save_plot(filename = here("manuscript", "revision2", "intermediate_files", "mash_effect_plots", "greenup", paste0(str_replace_all(replot$marker, pattern = " ", "_"), "_effect.png")), plot = replot$ggobject)
}

for (i in 1:nrow(fl_qtl_overlap)) {
  i <- fl_qtl_overlap$row_n[i]
  replot <- refactor_conditions(phe = overlap$mash_phenotype[i],
                    subpop = overlap$mash_subpop[i],
                    idx = overlap$mash_idx[i])
  save_plot(filename = here("manuscript", "revision2", "intermediate_files", "mash_effect_plots", "flowering", paste0(str_replace_all(replot$marker, pattern = " ", "_"), "_effect.png")), plot = replot$ggobject)
}
```


## Figure S1

h^2 analyses that was originally figure 1C. Try moving to the supplement. 

## Figure S2 & S3

Figure 3 is North x Texas GxE effects. Figure S2 is garden x garden GxE effects of individual loci for green-up; Figure S3 is for flowering date.

Most differences in sign were between TX1, the southernmost garden, and all other gardens (SI Appendix, Figure S1). Similarly, more effects that differed in magnitude included gardens in the Texas region (52.3-55.9%), and most effect pairs in the North region were not distinguishable (91.5%).

## Figure S4


QTL x Env effects

We conducted quantitative trait loci (QTL) mapping of flowering as functions of five environmental cues that we also used as covariance matrices in mash, and identified eight QTL for flowering date, eight QTL for flowering as a function of day length change two days prior, one QTL for the start of vegetative growth, and two QTL for vegetative growth as a function of daylength change one day prior, all of which showed QTL by environment interactions (SI Appendix, Figure S4).

```{r}
library(qtl)
library(qtl2)
library(qtl2convert)
```


#

```{r prep QTL_Eff df}
###QTL Effect plot
###load the original cross and reduce the markers from 4700 to 1185
#load('../phenology-fourway/5-QTLmappingForAlice/sexAveraged.cross.rda', verbose = T)
workingdir <- here("phenology-fourway/5-QTLmappingForAlice/")
load(file.path(workingdir, 'cross_reduced_marker_FL50_phenotypes_Alice_7_SITES.RData'))
map  =  lapply(pull.map(cross), function(x) x[1,])
# reduced_markers  =  unlist(
#   lapply(
#     reduce_markers(
#       lapply(pull.map(cross), function(x)
#         x[1,]), min_distance = 1), names))
#
# mars2drop  =  markernames(cross)[!markernames(cross) %in% reduced_markers]
# cross  =  drop.markers(cross, mars2drop)
cross = calc.genoprob(cross, step=1, error.prob = 1e-3,map.function = "kosambi",stepwidth = "max")
######### Converting probalibity from qtl1 to qtl2
probs  =  probs_qtl_to_qtl2(cross)$probs
##calculate the kinship matrix for single site
kinship_loco_Single = calc_kinship(probs,use_allele_probs = F,type = "loco")

###extract the genotype ID from the original cross file, so we can align the phenology file with genotype ID
ID <- data.frame(ID=cross$pheno$id)
ID <- ID %>% 
  mutate_at(vars('ID'), list(as.character)) %>%
  mutate_at(vars('ID'), list(as.numeric))

######get the phenotype at each site
phenology = read.csv('../phenology-fourway/5-QTLmappingForAlice/FL50_phenotypes_Alice.csv', header = TRUE)
colnames(phenology)[1:2] = c('Covariate','ID')
str(phenology) ###make sure your phenology traits are numeric, if not, change them into numeric
phenos <- phenology %>% 
  mutate_at(vars('Covariate'), list(as.character)) %>%
  mutate(ID = as.numeric(ID)) %>% 
  group_by(Covariate,ID) %>%
  summarize_all(mean, na.rm=T) %>% 
  ungroup() %>% 
  as.data.frame()
```

```{r setup QTL effects for all traits}
###QTL effect for each trait
EffectMatrix = vector('list', length=0)
t=0
for (envi in unique(phenos$Covariate)){
    t=t+1
    pheno = phenos %>% 
      filter(Covariate==envi) %>% 
      dplyr::select(-Covariate) %>%
      select(ID, FL50) %>% 
      right_join(ID) %>%  column_to_rownames('ID') %>% as.matrix()

    ##ran effect scan for each chromosome
    eff_out = vector('list',length = 0)
    se_out = vector('list',length = 0)
    add_contrasts=cbind(mu=c(1,1,1,1), add1=c(-1, 1, -1, 1),add2=c(-1, -1, 1, 1), dom=c(-1, 1, 1, -1))

    z = 0
    # chr = "1K"
    for (chr in chrnames(cross)){
      print(c(chr, envi,t))
      z= z+1
      eff = scan1coef(probs[,chr],pheno,
                      kinship = kinship_loco_Single[chr], 
                      se=T, contrasts = add_contrasts)
      se = attr(eff, 'SE')
      eff_out[[z]] =  eff %>% as.data.frame() %>% rownames_to_column('MARKER')
      se_out[[z]] =  se %>% as.data.frame() %>% rownames_to_column('MARKER')
    }
    eff_out = bind_rows(eff_out)
    se_out = bind_rows(se_out)

    se_out2 = se_out %>% dplyr::select(-mu) %>% gather('CROSS','SE',-MARKER)
    eff_out2 = eff_out %>% dplyr::select(-mu) %>% gather('CROSS','EFF',-MARKER) %>% mutate(SE=se_out2$SE, Covariate=envi)

    EffectMatrix[[t]] = eff_out2
  }

EffectMatrix2 = bind_rows(EffectMatrix)

###unlist nmap and merge with Effect file
nmap = data.frame(pos = unlist(map))
nnmap = nmap %>% rownames_to_column('MARKER') %>% rowwise() %>%
  mutate(chr = str_sub(MARKER, 1,2), MARKER=str_sub(MARKER, 4,nchar(MARKER)))%>%
  ungroup() %>% mutate(pos=round(pos,4))

flank_marker_full = read.csv('../phenology-fourway/5-QTLmappingForAlice/QTLsWithFlankMarkers_FL50_phenotypes_Alice_7_SITES_Full_model.csv')
flank_marker_full = flank_marker_full %>% mutate_at(vars(chr),list(as.character))

for (i in 1:nrow(flank_marker_full)){
  tmp = nnmap %>% filter(chr==flank_marker_full$chr[i] & pos==round(flank_marker_full$pos[i],4))
  flank_marker_full$MARKER[i] = tmp$MARKER
}

QTL_full_model = flank_marker_full %>% dplyr::rename(trait=lodcolumn)



```

```{r ggplots of QTL effects}
QTLdf <- QTL_full_model %>% left_join(EffectMatrix2)%>%
  mutate(CROSS = str_replace(CROSS,'add2','C x D')) %>%
  mutate(CROSS = str_replace(CROSS,'add1','A x B'),
         Cross = case_when(CROSS == "A x B" ~ "AP13 x DAC",
                           CROSS == "C x D" ~ "VS16 x WBC",
                           CROSS == "dom" ~ "Dominance",
                           TRUE ~ "Other"),
         Trait = case_when(trait == "dyln_change_sec" ~ "Daylength change (s), on Green-up Date",
                           trait == "dyln_change_sec_2d_prior" ~ "Daylength change (s), 2d prior to Flowering Date",
                           trait == "FL50" ~ "Flowering Date",
                           trait == "GR50" ~ "Green-up Date",
                           TRUE ~ "Other")
         ) %>%
  separate(MARKER, into = c("Chr", "Pos"), sep = "_", convert = TRUE, 
           remove = FALSE) %>%
  mutate(Pos_plot = paste0(Chr, "\n", round(Pos, 1), " Mb"))

QTLdf$Covariate = factor(QTLdf$Covariate, levels = c("TX1", "TX2", "OK", "MO", 
                                                   "NE", "MI", "SD"))

QTLdf %>% 
  #filter(Cross != "Dominance") %>% # trait == 'FL50' &
  ggplot(aes(x = Covariate, y = EFF, group = Cross)) +
  geom_point(size = 2) + facet_grid(Cross + trait ~ Pos_plot) +
  geom_hline(yintercept = 0, linetype=2) + 
  geom_errorbar(aes(ymin=EFF-SE, ymax=EFF+SE), size=0.6, width=0.5) +
  xlab("") +
  ylab('QTL Effect') + 
  coord_flip() + 
  theme_bw()+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        text=element_text(size=12, face="bold"),
        axis.title=element_text(size=12,face="bold"),
        axis.text.x = element_text(size = 10, angle = 45, hjust = 1, vjust = 1))
# shared QTL for more than one trait differ in LOD score, but not meaningfully in effect patterns across gardens. Useful to know.

QTLplot <- QTLdf %>%
  arrange(desc(lod)) %>%
  group_by(Chr, Pos, Covariate, Cross) %>%
  slice(1) %>% 
  filter(Cross != "Dominance") # trait == 'FL50' &
 
QTLplot %>%
  ggplot(aes(x = Covariate, y = EFF, group = Cross)) +
  geom_col(size = 1, aes(fill = Trait)) + 
  facet_grid(Cross ~ Pos_plot) +
  geom_hline(yintercept = 0, linetype=2) + 
  geom_errorbar(aes(ymin=EFF-SE, ymax=EFF+SE, color = Trait), size=0.6, width=0.5) +
  xlab("") + scale_y_continuous(limits = c(-2.5, 4), breaks = c(-2, 0, 2)) +
  ylab('QTL Effect') + 
  coord_flip() + 
  switchgrassGWAS::theme_oeco + 
  scale_color_manual(values = c("#CC0066", "#FFCC00", "darkgrey", "darkgreen")) +
  scale_fill_manual(values = c("#CC0066", "#FFCC00", "darkgrey", "darkgreen")) +
  theme(legend.position = "top",
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        text = element_text(size = 10, face="bold"),
        strip.text = element_text(size = 10, face="bold"),
        axis.title = element_text(size = 10,face="bold"),
        axis.text.x = element_text(size = 10, angle = 45, hjust = 1, vjust = 1),
        )# +
    #scale_y_continuous(expand = c(0.2, 0.2)) 
save_plot(filename = here("manuscript/revision2/intermediate_files/ Figure_S4_QTL_effects_by_trait.svg"), base_height = 4.5, base_width = 9, plot = last_plot())

QTLdf %>%
  filter(trait == 'dyln_change_sec_2d_prior') 
QTLdf %>% 
  filter(trait == 'FL50')
QTLdf %>% 
  filter(trait == 'dyln_change_sec') 

unique(QTLdf$trait)
```
