---
title: "Plots_for_DOE_poster"
author: "Alice MacQueen"
date: "2/19/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(PopGenome)
library(geosphere)
library(cowplot)
library(viridis)
library(maps)
library(ggmap)
library(XLConnect)
source("~/Github/Functions_ggplot-theme-adjustments_2018-01-03.R")
theme_set(theme_poster)
usa <- map_data("usa")
load("~/Github/pvdiv-phenology-gxe/data/sites.rda")
```

```{r}
metadata <- read_csv("~/Github/pvdiv-phenology-gxe/data/metadata_hierarchical_kof3.csv")

```


## 





# Map of Diversity Panel

```{r}
metaplot <- metadata %>%
  dplyr::select(PLANT_ID:CO_ecotype, LATITUDE, LONGITUDE) %>%
  filter(LIB_GROWN == "Y" | LIB_BIOCLIM == "Y") %>%
  mutate(GENETIC_SUBPOP = case_when(is.na(Genetic_subpop_50per) ~ "Unknown",
                                    Genetic_subpop_50per == "Eastcoast" ~ "Atlantic",
                                    Genetic_subpop_50per == "Texas" ~ "Gulf",
                                    TRUE ~ Genetic_subpop_50per),
         ECOTYPE = case_when(is.na(Ecotype) ~ "Unknown", 
                             Ecotype == "TX" ~ "Lowland",
                             Ecotype == "Texas" ~ "Lowland",
                             TRUE ~ Ecotype)) %>%
  dplyr::select(PLANT_ID, GENETIC_SUBPOP, ECOTYPE, LATITUDE, LONGITUDE, everything())

metaplot %>% group_by(GENETIC_SUBPOP) %>% tally()

metaplot$GENETIC_SUBPOP <- factor(metaplot$GENETIC_SUBPOP, levels = c("Midwest", "Atlantic", "Gulf", "Admixed", "Unknown"))
metaplot$ECOTYPE <- factor(metaplot$ECOTYPE, levels = c("Upland", "Coastal", "Lowland", "Unknown"))

theme_set(theme_poster)
ggplot() + 
  geom_polygon(data = usa, aes(x=long, y = lat, group = group), fill = "white", 
               color = "black") +
  coord_quickmap(xlim = c(-105, -66), ylim = c(25, 50)) +
  geom_jitter(data = metaplot, aes(x = LONGITUDE, y = LATITUDE, 
                                  shape = GENETIC_SUBPOP, fill = ECOTYPE, 
                                  alpha = ECOTYPE,
                                  color = GENETIC_SUBPOP:ECOTYPE), 
             size = 5, width = 0.5, height = 0.5) + 
  labs(x = "Longitude", y = "Latitude") + 
  theme(legend.position = "right") +
  scale_shape_manual(values = c(21,22,24,4,3)) + 
  scale_fill_manual(values = c("#280B54FF","#FAC127FF", "#D44842FF", "white")) +
  scale_color_manual(values = c(rep("black", 10), "#280B54FF","#FAC127FF", "#D44842FF", "white", "#280B54FF","#FAC127FF", "#D44842FF", "white"), guide = FALSE) +
  scale_alpha_manual(values = c(0.7, 0.7, 0.7, 0.7), guide = FALSE)
save_plot(filename = "Map_subpops_and_Ecotypes.svg", plot = last_plot(), base_height = 5)

library(scales)
show_col(viridis(8, option = "B"))
```

## Plot of Sites and planting numbers

```{r}

plants_2018 <- phenotypes_2018 %>%
  pivot_longer(cols = BRKG_DEAD_2018:TMPL_TC_EOS_2018, names_to = "SITE_PHE", values_to = "Measurement") %>%
  separate(col = SITE_PHE, into = c("SITE", "_", "PHE"), sep = c(4,5)) %>%
  dplyr::select(-(LATITUDE:ELEVATION), -`_`) %>%
  dplyr::filter(!is.na(Measurement)) %>%
  group_by(SITE, PLANT_ID) %>% tally() %>%
  summarise(Phenotyped_2018 = n())

died_winter <- phenotypes_2018 %>%
  pivot_longer(cols = ends_with("DEAD_2018"), names_to = "SITE_PHE", values_to = "Measurement") %>%
  filter(Measurement == 0) %>%
  separate(col = SITE_PHE, into = c("SITE", "_", "PHE"), sep = c(4,5)) %>%
  dplyr::select(-`_`) %>%
  dplyr::filter(!is.na(Measurement)) %>%
  group_by(SITE, PLANT_ID) %>% tally() %>%
  summarise(Alive_2019 = n())

plants_2019 <- phenotypes_wide_2019 %>%
  pivot_longer(cols = BRKG_BIOMASS:TMPL_TC_EOS, names_to = "SITE_PHE", values_to = "Measurement") %>%
  separate(col = SITE_PHE, into = c("SITE", "_", "PHE"), sep = c(4,5)) %>%
  dplyr::select(-`_`) %>%
  dplyr::filter(!is.na(Measurement)) %>%
  group_by(SITE, PLANT_ID) %>% tally() %>%
  summarise(Phenotyped_2019 = n())

siteplot <- sites %>%
  left_join(plants_2018, by = c("Site" = "SITE")) %>%
  left_join(died_winter, by = c("Site" = "SITE")) %>%
  left_join(plants_2019, by = c("Site" = "SITE")) %>%
  mutate(Winter_killed_2018 = Phenotyped_2018 - Alive_2019,
         Not_phenotyped_2019 = Alive_2019 - Phenotyped_2019)

theme_set(theme_poster)
map1 <- ggplot() + 
  geom_polygon(data = usa, aes(x=long, y = lat, group = group), fill = "white", 
               color = "black") +
  coord_quickmap(xlim = c(-105, -66), ylim = c(25, 50)) +
  geom_point(data = siteplot, aes(x = Longitude, y = Latitude, 
                                  fill = Phenotyped_2018), 
             size = 4, shape = 21) + 
  labs(x = "Longitude", y = "Latitude") + 
  scale_fill_viridis(option = "B", limits = c(250, 650)) + 
  theme(legend.position = c(0.95, 0.3), plot.title = element_text(size = 16)) + labs(title = "Plants Grown in 2018")
  
map2 <- ggplot() + 
  geom_polygon(data = usa, aes(x=long, y = lat, group = group), fill = "white", 
               color = "black") +
  coord_quickmap(xlim = c(-105, -66), ylim = c(25, 50)) +
  geom_point(data = siteplot, aes(x = Longitude, y = Latitude, 
                                  fill = Winter_killed_2018), 
             size = 4, shape = 21) + 
  labs(x = "Longitude", y = "Latitude") + 
  scale_fill_viridis(option = "B", limits = c(0, 180)) + 
  theme(legend.position = c(0.95, 0.3), plot.title = element_text(size = 16)) + labs(title = "Plants Killed Winter 2018")
  
plot_grid(map1, map2)
save_plot(filename = "Map_plants_winter_killed_2018.svg", plot = last_plot(), base_height = 5)

library(scales)
show_col(viridis(2, option = "D", end = 0.8))

```

# 4X vs 8X for Jason



```{r}

wb_pheno <- loadWorkbook(file.path("C:", "Users", "ahm543", "Dropbox",
                                   paste0("GWAS DOE Switchgrass, ",
                                          "Consolidated Data"),
                                   paste0("DOE_GWAS_Master Plant List.xlsx")))
lst_pheno <- readWorksheet(wb_pheno, sheet = getSheets(wb_pheno))
planting_2018 <- lst_pheno$`GWAS 2018 Master Plant List`



ploidyplot <- planting_2018 %>%
  left_join(metadata) %>%
   mutate(PLOIDY = ifelse(ACC == "AP13", "4X", PLOIDY)) %>%
   filter(!is.na(PLOIDY)) %>%
    mutate(GENETIC_SUBPOP = case_when(is.na(Genetic_subpop_50per) ~ "Unknown",
                                    Genetic_subpop_50per == "Eastcoast" ~ "Atlantic",
                                    Genetic_subpop_50per == "Texas" ~ "Gulf",
                                    TRUE ~ Genetic_subpop_50per),
         ECOTYPE = case_when(is.na(Ecotype) ~ "Unknown", 
                             Ecotype == "TX" ~ "Lowland",
                             Ecotype == "Texas" ~ "Lowland",
                             TRUE ~ Ecotype)) 

ploidyplot$PLOIDY <- factor(ploidyplot$PLOIDY, levels = c("8X", "4X"))
ploidyplot$SITE <- factor(ploidyplot$SITE, levels = c("BRKG", "KBSM", "FRMI", "LINC", "CLMB", "STIL", "OVTN", "TMPL", "PKLE", "KING"))
ploidyplot$GENETIC_SUBPOP <- factor(ploidyplot$GENETIC_SUBPOP, levels = c("Midwest", "Atlantic", "Gulf", "Admixed", "Unknown"))
ploidyplot$ECOTYPE <- factor(ploidyplot$ECOTYPE, levels = c("Upland", "Coastal", "Lowland", "Unknown"))


ploidyplot %>%
  #group_by(SITE, PLOIDY) %>% tally() %>%
   ggplot(aes(x = SITE)) + 
   geom_bar(aes(fill = PLOIDY)) +
   scale_fill_manual(values = c("black", "darkgreen")) + 
  xlab("Sites (North to South)") + ylab("Number of Plants") + 
  theme(legend.position = "right",
        axis.text.x = element_text(angle = 45, hjust = 1))

save_plot(filename = "Ploidy of Plants at Common Garden Sites.png", plot = last_plot())

ploidyplot %>%
  #group_by(SITE, PLOIDY) %>% tally() %>%
   ggplot(aes(x = SITE)) + 
   geom_bar(aes(fill = GENETIC_SUBPOP)) +
   scale_fill_manual(values = c("darkblue", "orange", "red", "grey", "lightgrey")) + 
  xlab("Sites (North to South)") + ylab("Number of Plants") + 
  theme(legend.position = "right",
        axis.text.x = element_text(angle = 45, hjust = 1))

save_plot(filename = "Ploidy of Plants at Common Garden Sites.png", plot = last_plot())

```



```{r}
winterlost <- read_csv("twoHit.Chi2PeakIntervals.csv")

winterlost %>%
  ggplot(aes(x = peak)) + geom_histogram() + facet_wrap(~chr)
```

# Frequencies lost 
```{r}
frz_freq <- readRDS("../analysis/common-garden-allele-balancing/SNPs_with_1percent_tail_frequency_change_freezing_sites.rds") 

theme_set(theme_poster)
frzlostplot <- frz_freq %>%
  filter(Loss_BRKG > 159 | Loss_LINC > 179 | Loss_FRMI > 84 | 
           Loss_CLMB > 94) %>%
  mutate(Loss_freq = (Loss_BRKG/MAC_BRKG_2018 + Loss_FRMI/MAC_FRMI_2018 +
                            Loss_LINC/MAC_LINC_2018 +
                            Loss_CLMB/MAC_CLMB_2018)/4) %>% 
  filter(Loss_freq > 0.42) %>%
  ggplot(aes(x = POS/1000000, y = Loss_freq)) +
  geom_point(aes(color = CHR), alpha = 0.35) + 
  facet_wrap(~CHR, scales = "free_x", nrow = 1,
            strip.position = "bottom") + 
  #scale_fill_viridis(trans = "log10", option = "B") + 
  xlab("Chromosome") + ylab("Fraction of Minor Allele Lost") +
    scale_x_continuous(expand = c(0.03, 0.03)) +
  theme(legend.position = "none", 
        axis.text.x = element_blank(),
          axis.ticks.x = element_blank()) +
        #axis.text.x = element_text(angle = 45, hjust = 1)) +
  scale_color_manual(values = c(rep(c("#1B0C42FF", "grey"), 9)))

save_plot(filename = paste0("Distribution_of_SNPs_1per_delta_all_four_sites_lost_50per", str_replace_all(Sys.time(), ":", "."),
                   ".png"), plot = frzlostplot, base_asp = 4, base_height = 4)

```

# Phenology histograms

```{r}
phenotypes_long_2019 <- phenotypes_wide_2019 %>%
  pivot_longer(cols = BRKG_BIOMASS:TMPL_TC_EOS, names_to = "SITE_PHE", values_to = "Measurement") %>%
  separate(col = SITE_PHE, into = c("SITE", "_", "PHE"), sep = c(4,5)) %>%
  dplyr::select(-`_`)

phenotypes_long_2019$SITE <- factor(phenotypes_long_2019$SITE, levels = c("BRKG", "KBSM", "FRMI", "LINC", "CLMB", "STIL", "OVTN", "TMPL", "PKLE", "KING"))

unique(phenotypes_long_2019$PHE) # check what phenotypes there are
phenotypes_long_2019$PHE <- factor(phenotypes_long_2019$PHE, levels = c("GR1", "GR50", "GR100", "EM1", "EM50", "FL1", "FL50", "T_GR_EM", "T_GR_FL", "T_EM_FL", "TC_EOS", "HT_PAN_EOS", "REGR", "LD_EOS", "BIOMASS", "MOISTURE", "FRZ_DMG", "CHLR_101", "CHLR_121", "CHLR_136", "CHLR_JASON_138", "CHLR_150.", "SPAD_158", "SPADCOR_158"))


phenology_long_2019 <- phenotypes_long_2019 %>%
  filter(!(grepl("CHLR", PHE)) & !(grepl("SPAD", PHE)) & !grepl("FRZ", PHE) & !is.na(PHE) & !grepl("TC_EOS", PHE) & !grepl("HT_PAN_EOS", PHE) & !grepl("BIOMASS", PHE) & !grepl("REGR", PHE) & !grepl("MOISTURE", PHE) & !grepl("LD_EOS", PHE) & !grepl("T_", PHE))

pheplot <- phenotypes_long_2019 %>%
  filter(PHE %in% c("TC_EOS", "HT_PAN_EOS", "BIOMASS")) %>%
  mutate(PHE = case_when(PHE == "TC_EOS" ~ "Tiller Count",
                         PHE == "HT_PAN_EOS" ~ "Plant Height (cm)",
                         PHE == "BIOMASS" ~ "Biomass (kg)"))

```

```{r}
library(lubridate)
phenplot <- phenology_long_2019 %>%
  filter(PHE %in% c("GR50")) %>%
  mutate(PHE = case_when(PHE == "GR50" ~ "Greenup",
                         PHE == "EM50" ~ "Panicle Emergence",
                         PHE == "FL50" ~ "Flowering"))
phenplot$PHE <- factor(phenplot$PHE, levels = c("Greenup", "Panicle Emergence", "Flowering"))

phenplot %>%
  left_join(metaplot, by = "PLANT_ID") %>%
  mutate(Days = as_date(Measurement)) %>%
  ggplot(aes(x = Days)) + 
  geom_histogram(aes(fill = ECOTYPE)) + 
  facet_grid(SITE~PHE, scales = "free") +
  scale_fill_manual(values = c("#280B54FF","#FAC127FF", "#D44842FF", "lightgrey", "grey")) +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45, hjust = 1)) + xlab("Value") + ylab("Count of Individuals") +
  theme(legend.position = "none", 
        axis.text.x = element_text(angle = 45, hjust = 1)) +#,
        #panel.spacing = unit(2, "lines")) +
  xlab("Date") + 
  ylab("Number of Individuals") +
  geom_vline(xintercept = c(120), linetype = 2) +
  geom_vline(xintercept = c(181), linetype = 3, color = "grey20") +
  geom_vline(xintercept = c(243), linetype = 4, color = "grey80") +
  scale_y_continuous(breaks = c(10, 50, 150))

save_plot("Greenup_Data_by_Site_and_Ecotype_2020-02-19.svg", plot = last_plot(), base_height = 6, base_asp = 1.3/3)
```


```{r}
pheplot$PHE <- factor(pheplot$PHE, levels = c("Biomass (kg)", "Tiller Count", "Plant Height (cm)"))

pheplot %>%
  left_join(metaplot, by = "PLANT_ID") %>%
  mutate(Measurement = ifelse(PHE == "Biomass (kg)", Measurement/1000, Measurement)) %>%
  ggplot(aes(x = Measurement)) + 
  geom_histogram(aes(fill = ECOTYPE)) + 
  facet_grid(SITE~PHE, scales = "free") +
  scale_fill_manual(values = c("#280B54FF","#FAC127FF", "#D44842FF", "lightgrey", "grey")) +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45, hjust = 1)) + xlab("Value") + ylab("Count of Individuals") +
  theme(legend.position = "right", 
        axis.text.x = element_text(angle = 45, hjust = 1)) +#,
        #panel.spacing = unit(2, "lines")) +
  xlab("") + 
  ylab("Number of Individuals") +
  scale_y_continuous(breaks = c(10, 50, 150))

save_plot("EOS_Phenotype_Data_by_Site_and_Ecotype_2020-02-19.svg", plot = last_plot(), base_height = 6, base_asp = 1.3)

pheplot2 <- phenotypes_long_2019 %>%
  filter(PHE %in% c("BIOMASS")) %>%
  mutate(PHE = case_when(PHE == "TC_EOS" ~ "Tiller Count",
                         PHE == "HT_PAN_EOS" ~ "Plant Height (cm)",
                         PHE == "BIOMASS" ~ "Biomass (kg)"))
pheplot2 %>%
  left_join(metaplot, by = "PLANT_ID") %>%
  mutate(Measurement = ifelse(PHE == "Biomass (kg)", Measurement/1000, Measurement)) %>%
  ggplot(aes(x = Measurement)) + 
  geom_histogram(aes(fill = ECOTYPE)) + 
  facet_grid(SITE~PHE, scales = "free") +
  scale_fill_manual(values = c("#280B54FF","#FAC127FF", "#D44842FF", "lightgrey", "grey")) +
  theme(legend.position = "none") +#,
        #panel.spacing = unit(2, "lines")) +
  xlab("") + 
  geom_vline(xintercept = c(2.0), linetype = 2) +
  ylab("Number of Individuals") +
  scale_y_continuous(breaks = c(10, 50, 150))

save_plot("Biomass_Phenotype_Data_by_Site_and_Ecotype_2020-02-19.svg", plot = last_plot(), base_height = 6, base_asp = 1.3/3)
```

# Manhattan plots for GR50 in KBSM and biomass in TMPL
```{r}
gwas_data <- readRDS("GWAS_datatable_TMPL_BIOMASS_5_PCs.rds")
gwas_data[,FDR_adj := p.adjust(pvalue, method = "BH")]
FDRthreshhi <- gwas_data %>%
    as_tibble() %>%
    filter(between(FDR_adj, 0.1001, 1)) %>%  # 5% FDR threshold currently.
    summarise(thresh = max(log10p))
  FDRthreshlo <- gwas_data %>%
    as_tibble() %>%
    filter(between(FDR_adj, 0, 0.0999)) %>%  # 5% FDR threshold currently.
    summarise(thresh = min(log10p))
  if(FDRthreshhi$thresh[1] > 0 & FDRthreshlo$thresh[1] > 0){
    FDRthreshold = (FDRthreshhi$thresh[1] + FDRthreshlo$thresh[1])/2
  } else if(FDRthreshhi$thresh[1] > 0){
    FDRthreshold = FDRthreshhi$thresh[1]
  } else if(FDRthreshlo$thresh[1] > 0){g
    FDRthreshold = FDRthreshlo$thresh[1]
  } else {
    FDRthreshold = NA
  }

  ggmanobject1 <- gwas_data %>%
    filter(log10p > 1) %>%
    ggplot(aes(x = pos, y = log10p)) +
    geom_hline(yintercept = c(10,15,20), color = "lightgrey") +
    geom_point(aes(color = chr, fill = chr)) +
    geom_point(aes(color = chr, fill = chr)) +
    #geom_hline(yintercept = FDRthreshold, color = "black", linetype = 2,
    #           size = 0.7) +
    geom_hline(yintercept = bonferroni, color = "#FAC127FF", linetype = 2,
               size = 1) +
    facet_wrap(~ chr, nrow = 1, scales = "free_x",
               strip.position = "bottom") +
    #scale_color_viridis(option = "B", end = 0.8, discrete = TRUE) +
    #scale_fill_viridis(option = "B", end = 0.8, discrete = TRUE) +
    theme(axis.text.x = element_blank(),
          axis.ticks.x = element_blank(),
          panel.background = element_rect(fill=NA), legend.position = "none") +
    labs(x = "Chromosome", y = "-log10(p value)") +
    scale_x_continuous(expand = c(0.18, 0.18)) +
    scale_color_manual(values = rep(c("#1B0C42FF", "grey"),9), guide = FALSE)
   #scale_shape_manual(values = rep(c(21,22),9), guide = FALSE)
  theme_set(theme_poster)
  save_plot(paste0("Manhattan_TMPL_BIOMASS_5_PCs_FDR_", str_replace_all(Sys.time(), ":", "."),
                   ".png"),
            plot = ggmanobject1, base_aspect_ratio = 4, base_height = 4)
```

```{r}
gwas_data2 <- readRDS("GWAS_datatable_KBSM_GR50_3_PCs_.rds")
gwas_data2[,FDR_adj := p.adjust(pvalue, method = "BH")]
FDRthreshhi <- gwas_data2 %>%
    as_tibble() %>%
    filter(between(FDR_adj, 0.1001, 1)) %>%  # 5% FDR threshold currently.
    summarise(thresh = max(log10p))
  FDRthreshlo <- gwas_data2 %>%
    as_tibble() %>%
    filter(between(FDR_adj, 0, 0.0999)) %>%  # 5% FDR threshold currently.
    summarise(thresh = min(log10p))
  if(FDRthreshhi$thresh[1] > 0 & FDRthreshlo$thresh[1] > 0){
    FDRthreshold = (FDRthreshhi$thresh[1] + FDRthreshlo$thresh[1])/2
  } else if(FDRthreshhi$thresh[1] > 0){
    FDRthreshold = FDRthreshhi$thresh[1]
  } else if(FDRthreshlo$thresh[1] > 0){
    FDRthreshold = FDRthreshlo$thresh[1]
  } else {
    FDRthreshold = NA
  }

  ggmanobject1 <- gwas_data2 %>%
    filter(log10p > 1) %>%
    ggplot(aes(x = pos, y = log10p)) +
    geom_hline(yintercept = c(10,15), color = "lightgrey") +
    geom_point(aes(color = chr, fill = chr)) +
    #geom_hline(yintercept = FDRthreshold, color = "black", linetype = 2,
    #           size = 0.7) +
    geom_hline(yintercept = bonferroni, color = "#FAC127FF", linetype = 2,
               size = 1) +
    facet_wrap(~ chr, nrow = 1, scales = "free_x",
               strip.position = "bottom") +
    #scale_color_viridis(option = "B", end = 0.8, discrete = TRUE) +
    #scale_fill_viridis(option = "B", end = 0.8, discrete = TRUE) +
    theme(axis.text.x = element_blank(),
          axis.ticks.x = element_blank(),
          panel.background = element_rect(fill=NA), legend.position = "none") +
    labs(x = "Chromosome", y = "-log10(p value)") +
    scale_x_continuous(expand = c(0.18, 0.18)) +
    scale_color_manual(values = rep(c("#1B0C42FF", "grey"),9), guide = FALSE)
   #scale_shape_manual(values = rep(c(21,22),9), guide = FALSE)
  theme_set(theme_poster)
  save_plot(paste0("Manhattan_KBSM_GR50_3_PCs_FDR_", str_replace_all(Sys.time(), ":", "."),
                   ".png"),
            plot = ggmanobject1, base_aspect_ratio = 4, base_height = 4)
```

# Manhattan plots for ASV's with 10% FDR & Bonferroni
```{r}
asvs <- c("ASV89_2_PCs", "ASV113_3_PCs", "ASV1197_3_PCs")
for(i in seq_along(asvs)){
gwas_data <- readRDS(paste0("../GWAS_datatable_", asvs[i], "_.rds"))
#gwas_data[,FDR_adj := p.adjust(pvalue, method = "BH")]
FDRthreshhi <- gwas_data %>%
    as_tibble() %>%
    filter(between(FDR_adj, 0.1001, 1)) %>%  # 5% FDR threshold currently.
    summarise(thresh = max(log10p))
  FDRthreshlo <- gwas_data %>%
    as_tibble() %>%
    filter(between(FDR_adj, 0, 0.0999)) %>%  # 5% FDR threshold currently.
    summarise(thresh = min(log10p))
  if(FDRthreshhi$thresh[1] > 0 & FDRthreshlo$thresh[1] > 0){
    FDRthreshold = (FDRthreshhi$thresh[1] + FDRthreshlo$thresh[1])/2
  } else if(FDRthreshhi$thresh[1] > 0){
    FDRthreshold = FDRthreshhi$thresh[1]
  } else if(FDRthreshlo$thresh[1] > 0){g
    FDRthreshold = FDRthreshlo$thresh[1]
  } else {
    FDRthreshold = NA
  }

  ggmanobject1 <- gwas_data %>%
    filter(log10p > 1) %>%
    ggplot(aes(x = pos, y = log10p)) +
    geom_hline(yintercept = c(10,15), color = "lightgrey") +
    geom_point(aes(color = chr, fill = chr)) +
    geom_point(aes(color = chr, fill = chr)) +
    geom_hline(yintercept = FDRthreshold, color = "black", linetype = 2,
               size = 0.7) +
    geom_hline(yintercept = bonferroni, color = "#FAC127FF", linetype = 2,
               size = 1) +
    facet_wrap(~ chr, nrow = 1, scales = "free_x",
               strip.position = "bottom") +
    #scale_color_viridis(option = "B", end = 0.8, discrete = TRUE) +
    #scale_fill_viridis(option = "B", end = 0.8, discrete = TRUE) +
    theme(axis.text.x = element_blank(),
          axis.ticks.x = element_blank(),
          panel.background = element_rect(fill=NA), legend.position = "none") +
    labs(x = "Chromosome", y = "-log10(p value)") +
    scale_x_continuous(expand = c(0.18, 0.18)) +
    scale_color_manual(values = rep(c("#1B0C42FF", "grey"),9), guide = FALSE)
   #scale_shape_manual(values = rep(c(21,22),9), guide = FALSE)
  theme_set(theme_poster)
  save_plot(paste0("../../presentations/Nice_Manhattans/Manhattan_", asvs[i], "_Bonferroni_10percent_FDR_", str_replace_all(Sys.time(), ":", "."),
                   ".png"),
            plot = ggmanobject1, base_aspect_ratio = 4, base_height = 4)
}
```

