---
title: "VCF file filtering by common garden site"
author: "Alice MacQueen"
date: "1/22/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(bigsnpr)
library(switchgrassGWAS)
load("~/Github/pvdiv-phenology-gxe/data/phenotypes_2019.rda")
load("~/Github/pvdiv-phenology-gxe/data/sites.rda")
```

## Goals & Outline

    1. Use the 33.9 million SNP file, located at:
    `/home/SharedUser/PV/NatPop/All_chr_combined_636g_80pct_maf0.005.vcf.bz2`
      Convert this file to vcf.gz first.
    2. Use or make a data frame of PLANT_ID's that were present at each site in 2019.
    3. Subset the 33.9 million SNP file to 5% MAF for each site.
    4. Use vcftools to find allele frequencies for each SNP at each site.

Below are the steps necessary to make a subset of a VCF file for use for GWAS.

## Subset function

```{r}
pvdiv_pop_subset_bigsnp <- function(subset, selector){
  subset_name <- paste0("Pvirgatum_4x_784g_imp_phased_", subset, "_subset")
  snpfile_subset <- subset(snp, ind.row = selector)
  snp_subset <- snp_attach(snpfile_subset)
  snp_writeBed(snp_subset, bedfile = paste0(subset_name, ".bed"))
  mac <- round(25/length(selector), digits = 2) # what should the MAF be?
  if(mac > 0.05){
    mac <- 0.05
  }
  plink <- download_plink()
  bedfile <- snp_plinkQC(plink, prefix.in = subset_name,
                         maf = mac, geno = 0.1, hwe = 1e-50, 
                         prefix.out = paste0(subset_name, "_maf", mac),
                         extra.options = "--allow-no-sex --allow-extra-chr --chr Chr01K Chr01N Chr02K Chr02N Chr03K Chr03N Chr04K Chr04N Chr05K Chr05N Chr06K Chr06N Chr07K Chr07N Chr08K Chr08N Chr09K Chr09N")
  rdsfile <- snp_readBed(bedfile)
}


# Load data - if you get errors, make sure to run R from the GWAS/ folder in pubjuenger
snp <- snp_attach("/home/pubjuenger/GWAS/Pvirgatum_4x_784g_imp_phased_maf0.005.rds")
switchgrassGWAS::phenotypes

# Use phenotypes to set up selection criteria for subsets. These need to be vectors of the row numbers in the Taxa dataframe that you have phenotypes for.
gwasct_sel <- which(!is.na(phenotypes$GWAS_CT)) # gwas_ct GWAS

## Run function to generate subset SNP files for the subset
pvdiv_pop_subset_bigsnp(subset = "gwasct", selector = gwasct_sel)

```

## 2. Use or make a data frame of PLANT_ID's that were present at each site in 2019.

Plants alive EOS 2019 (higher of HT_PAN_EOS and TC_EOS):
BRKG: 254
CLMB: 496
FRMI: 185
KBSM: 516
KING: 286
LINC: 260
OVTN: 331
PKLE: 614
STIL: 451
TMPL: 373

For each site: # SNPs above a MAF of 5%
% SNPs above a MAF of 5% (at all 10 sites), # SNPs > 5% MAF at all 10 sites, distribution of these SNPs across the chromosomes
Number of plants alive from each subpopulation (sort of did this already)

Average, minimum, maximum, standard deviation, coefficient of variation of MAF across all 10 sites
histograms of these values across all SNPs

Compare these values to their values in 2018 to see how much frequencies shifted in the last year?

```{r}
colSums(!is.na(phenotypes_wide_2019))
i=1
for(i in 1:10){
site <- sites$Site[i]

site_2019 <- phenotypes_wide_2019 %>%
  pivot_longer(cols = BRKG_EM1:TMPL_TC_EOS, names_to = "SITE_PHE", values_to = "Measurement") %>%
  separate(col = SITE_PHE, into = c("SITE", "_", "PHE"), sep = c(4,5)) %>%
  filter(SITE == site & !is.na(Measurement)) %>%
  group_by(PLANT_ID) %>%
  summarise(count = n()) 

writeLines(site_2019$PLANT_ID, con = paste0(site, "_2019"))
}

for(i in 1:10){
site <- sites$Site[i]
site_2018 <- phenotypes_2018 %>%
  pivot_longer(cols = BRKG_DEAD_2018:TMPL_TC_EOS_2018, names_to = "SITE_PHE", values_to = "Measurement") %>%
  separate(col = SITE_PHE, into = c("SITE", "_", "PHE"), sep = c(4,5)) %>%
  filter(SITE == site & !is.na(Measurement)) %>%
  group_by(PLANT_ID) %>%
  summarise(count = n()) 
writeLines(site_2018$PLANT_ID, con = paste0(site, "_2018"))
}

```

Plants alive EOS 2018, then EOS 2019:
BRKG: 412 down to 254 (377 in 2018; 234 in 2019)
CLMB: 591 down to 496 (547 in 2018; 462 in 2019)
FRMI: 273 down to 185 (245 in 2018; 163 in 2019)
KBSM: 544 down to 516 (501 in 2018; 480 in 2019)
KING: 305 down to 286 (275 in 2018; 267 in 2019)
LINC: 437 down to 260 (400 in 2018; 247 in 2019)
OVTN: 346 down to 331 (312 in 2018; 304 in 2019)
PKLE: 623 down to 614 (579 in 2018; 577 in 2019 out of 732 individuals in 2020 genome file)
STIL: 469 down to 451 (433 in 2018; 419 in 2019)
TMPL: 388 down to 373 (351 in 2018; 341 in 2019)

# VCFtools lines
```{}
vcftools --gzvcf /home/SharedUser/PV/NatPop/33dot9MSite_Unimputed/Pvirgatum_V5_732g_33M.vcf.gz --keep /home/alice/Github/pvdiv-phenology-gxe/analysis/common-garden-allele-balancing/PKLE_2019 --freq --out /home/alice/Github/pvdiv-phenology-gxe/large-files/Site-by-year-allele-frequencies/PKLE_2019 &
vcftools --gzvcf /home/SharedUser/PV/NatPop/33dot9MSite_Unimputed/Pvirgatum_V5_732g_33M.vcf.gz --keep /home/alice/Github/pvdiv-phenology-gxe/analysis/common-garden-allele-balancing/BRKG_2019 --freq --out /home/alice/Github/pvdiv-phenology-gxe/large-files/Site-by-year-allele-frequencies/BRKG_2019 &
vcftools --gzvcf /home/SharedUser/PV/NatPop/33dot9MSite_Unimputed/Pvirgatum_V5_732g_33M.vcf.gz --keep /home/alice/Github/pvdiv-phenology-gxe/analysis/common-garden-allele-balancing/CLMB_2019 --freq --out /home/alice/Github/pvdiv-phenology-gxe/large-files/Site-by-year-allele-frequencies/CLMB_2019 &
vcftools --gzvcf /home/SharedUser/PV/NatPop/33dot9MSite_Unimputed/Pvirgatum_V5_732g_33M.vcf.gz --keep /home/alice/Github/pvdiv-phenology-gxe/analysis/common-garden-allele-balancing/FRMI_2019 --freq --out /home/alice/Github/pvdiv-phenology-gxe/large-files/Site-by-year-allele-frequencies/FRMI_2019 &
vcftools --gzvcf /home/SharedUser/PV/NatPop/33dot9MSite_Unimputed/Pvirgatum_V5_732g_33M.vcf.gz --keep /home/alice/Github/pvdiv-phenology-gxe/analysis/common-garden-allele-balancing/KBSM_2019 --freq --out /home/alice/Github/pvdiv-phenology-gxe/large-files/Site-by-year-allele-frequencies/KBSM_2019 &
vcftools --gzvcf /home/SharedUser/PV/NatPop/33dot9MSite_Unimputed/Pvirgatum_V5_732g_33M.vcf.gz --keep /home/alice/Github/pvdiv-phenology-gxe/analysis/common-garden-allele-balancing/KING_2019 --freq --out /home/alice/Github/pvdiv-phenology-gxe/large-files/Site-by-year-allele-frequencies/KING_2019 &
vcftools --gzvcf /home/SharedUser/PV/NatPop/33dot9MSite_Unimputed/Pvirgatum_V5_732g_33M.vcf.gz --keep /home/alice/Github/pvdiv-phenology-gxe/analysis/common-garden-allele-balancing/LINC_2019 --freq --out /home/alice/Github/pvdiv-phenology-gxe/large-files/Site-by-year-allele-frequencies/LINC_2019 &
vcftools --gzvcf /home/SharedUser/PV/NatPop/33dot9MSite_Unimputed/Pvirgatum_V5_732g_33M.vcf.gz --keep /home/alice/Github/pvdiv-phenology-gxe/analysis/common-garden-allele-balancing/OVTN_2019 --freq --out /home/alice/Github/pvdiv-phenology-gxe/large-files/Site-by-year-allele-frequencies/OVTN_2019 &
vcftools --gzvcf /home/SharedUser/PV/NatPop/33dot9MSite_Unimputed/Pvirgatum_V5_732g_33M.vcf.gz --keep /home/alice/Github/pvdiv-phenology-gxe/analysis/common-garden-allele-balancing/STIL_2019 --freq --out /home/alice/Github/pvdiv-phenology-gxe/large-files/Site-by-year-allele-frequencies/STIL_2019 &
vcftools --gzvcf /home/SharedUser/PV/NatPop/33dot9MSite_Unimputed/Pvirgatum_V5_732g_33M.vcf.gz --keep /home/alice/Github/pvdiv-phenology-gxe/analysis/common-garden-allele-balancing/TMPL_2019 --freq --out /home/alice/Github/pvdiv-phenology-gxe/large-files/Site-by-year-allele-frequencies/TMPL_2019 &
vcftools --gzvcf /home/SharedUser/PV/NatPop/33dot9MSite_Unimputed/Pvirgatum_V5_732g_33M.vcf.gz --keep /home/alice/Github/pvdiv-phenology-gxe/analysis/common-garden-allele-balancing/PKLE_2018 --freq --out /home/alice/Github/pvdiv-phenology-gxe/large-files/Site-by-year-allele-frequencies/PKLE_2018 &
vcftools --gzvcf /home/SharedUser/PV/NatPop/33dot9MSite_Unimputed/Pvirgatum_V5_732g_33M.vcf.gz --keep /home/alice/Github/pvdiv-phenology-gxe/analysis/common-garden-allele-balancing/BRKG_2018 --freq --out /home/alice/Github/pvdiv-phenology-gxe/large-files/Site-by-year-allele-frequencies/BRKG_2018 &
vcftools --gzvcf /home/SharedUser/PV/NatPop/33dot9MSite_Unimputed/Pvirgatum_V5_732g_33M.vcf.gz --keep /home/alice/Github/pvdiv-phenology-gxe/analysis/common-garden-allele-balancing/CLMB_2018 --freq --out /home/alice/Github/pvdiv-phenology-gxe/large-files/Site-by-year-allele-frequencies/CLMB_2018 &
vcftools --gzvcf /home/SharedUser/PV/NatPop/33dot9MSite_Unimputed/Pvirgatum_V5_732g_33M.vcf.gz --keep /home/alice/Github/pvdiv-phenology-gxe/analysis/common-garden-allele-balancing/FRMI_2018 --freq --out /home/alice/Github/pvdiv-phenology-gxe/large-files/Site-by-year-allele-frequencies/FRMI_2018 &
vcftools --gzvcf /home/SharedUser/PV/NatPop/33dot9MSite_Unimputed/Pvirgatum_V5_732g_33M.vcf.gz --keep /home/alice/Github/pvdiv-phenology-gxe/analysis/common-garden-allele-balancing/KBSM_2018 --freq --out /home/alice/Github/pvdiv-phenology-gxe/large-files/Site-by-year-allele-frequencies/KBSM_2018j &
vcftools --gzvcf /home/SharedUser/PV/NatPop/33dot9MSite_Unimputed/Pvirgatum_V5_732g_33M.vcf.gz --keep /home/alice/Github/pvdiv-phenology-gxe/analysis/common-garden-allele-balancing/KING_2018 --freq --out /home/alice/Github/pvdiv-phenology-gxe/large-files/Site-by-year-allele-frequencies/KING_2018 &
vcftools --gzvcf /home/SharedUser/PV/NatPop/33dot9MSite_Unimputed/Pvirgatum_V5_732g_33M.vcf.gz --keep /home/alice/Github/pvdiv-phenology-gxe/analysis/common-garden-allele-balancing/LINC_2018 --freq --out /home/alice/Github/pvdiv-phenology-gxe/large-files/Site-by-year-allele-frequencies/LINC_2018 &
vcftools --gzvcf /home/SharedUser/PV/NatPop/33dot9MSite_Unimputed/Pvirgatum_V5_732g_33M.vcf.gz --keep /home/alice/Github/pvdiv-phenology-gxe/analysis/common-garden-allele-balancing/OVTN_2018 --freq --out /home/alice/Github/pvdiv-phenology-gxe/large-files/Site-by-year-allele-frequencies/OVTN_2018 &
vcftools --gzvcf /home/SharedUser/PV/NatPop/33dot9MSite_Unimputed/Pvirgatum_V5_732g_33M.vcf.gz --keep /home/alice/Github/pvdiv-phenology-gxe/analysis/common-garden-allele-balancing/STIL_2018 --out /home/alice/Github/pvdiv-phenology-gxe/large-files/Site-by-year-allele-frequencies/STIL_2018 &
vcftools --gzvcf /home/SharedUser/PV/NatPop/33dot9MSite_Unimputed/Pvirgatum_V5_732g_33M.vcf.gz --keep /home/alice/Github/pvdiv-phenology-gxe/analysis/common-garden-allele-balancing/TMPL_2018 --freq --out /home/alice/Github/pvdiv-phenology-gxe/large-files/Site-by-year-allele-frequencies/TMPL_2018 &
```

```{r}
snpfile <- read_csv(file = "33M_SNP_file_PLANT_ID")
pkle2019 <- read_csv(file = "PKLE_2019", col_names = "PLANT_ID")


snpfile %>%
  left_join(phenotypes_wide_2019)

metadata %>%
  filter(LIB_CLIMATE == "Y") %>%
  anti_join(snpfile)

snpfile %>%
  anti_join(metadata) # NL94SL93 is the only one not in the metadata. So what happened?

metadata2 %>%
  anti_join(snpfile) # NL94SL93 is the only one not in the 732 metadata. Ehhhh? 732 metadata has NL94 & SL93. 
# So the error isn't because I gave Taslima the wrong metadata for the 732 file.

snpfile %>%
  anti_join(pkle2019)
pkle2019 %>%
  anti_join(snpfile)
site_2019 %>%
  anti_join(snpfile) %>%
  left_join(metadata)

metadata %>%
  filter(LIB_CLIMATE == "Y")
metadata %>%
  filter(LIB_PHENO == "Y")
```
NL94SL93 is the only one not in the 732 metadata. Ehhhh? 732 metadata has NL94 & SL93. 

So the error isn't because I gave Taslima the wrong metadata for the 732 file. But there are still 44 missing individuals that were planted in the field but aren't in the 732 genotype file. 

I think maybe this file has the 

```{r}

phenotypes_wide_2019 %>%
  anti_join(snpfile) %>%
  dplyr::select(PLANT_ID) %>%
  left_join(metadata) %>%
  arrange(LIB_PHENO) %>%
  filter(!(PLANT_ID %in% c("J661.C", "J491.A", "J385.C", "J471.A", "J243.A")))

phenotypes_wide_2019 %>%
  filter(!(PLANT_ID %in% c("J661.C", "J491.A", "J385.C", "J471.A", "J243.A", "J428.A", "J399.B", "J344.B", "J003.C", "J624.B", "J682.C", "NL94×SL93_370", "NL94×SL93_305"))) # None of these have any 2019 data.

phe_2019 <- phenotypes_wide_2019[which(rowSums(!is.na(phenotypes_wide_2019))>1),] 
# Only 630 individuals have phenotyping data in 2019. But 38 aren't in 732 file. That's 10% of data.
phe_2018 <- phenotypes_2018[which(rowSums(!is.na(phenotypes_2018))>1),] # 636 have 2018 data.
phe_csv <- phe_2018 %>%
  dplyr::select(PLANT_ID) %>%
  #anti_join(snpfile) %>%
  left_join(metadata) %>% 
  filter(LIB_PHENO == "Y") %>%
  dplyr::select(PLANT_ID:LIB_PHENO) #%>% anti_join(snpfile)

write_csv(phe_csv, path = "Individuals_needed_for_33M_SNPs.csv")
```

Individuals that are in the phenotyping panel and should be resequenced:

J428.A	IGRB	N	N	N	 Highest heterozygoisty and low-ish coverage. Fishy, but also, probaably OK. Exclude
J399.B	IHZG	N	N	N	low coverage, high het. Exclude
J344.B	IHZN	N	N	N	low coverage, high het. Exclude
J003.C	IIAH	N	N	N	J003 is a mess. Exclude
J243.A	IRUH	N	N	N	This has really bad coverage and should be excluded. Exclude
J661.C	IRXF	N	N	N		Probable Error, Plant does not match others. Exclude
J682.C	ITCI	N	N N Very high heterozygosity, coverage on the low side. Exclude
