---
title: "Allele Frequencies Common Gardens"
author: "Alice MacQueen"
date: "1/24/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(switchgrassGWAS)
library(viridis)
library(cowplot)
source("~/Github/Functions_ggplot-theme-adjustments_2018-01-03.R")
```

# Join data on server

```{r}
load("~/Github/pvdiv-phenology-gxe/data/sites.rda")
year <- "2019"
i=1
k= i+1
nmax = 33890000

site_year <- read_delim(paste0(sites$Site[i],"_", year, ".frq"), delim = "\t", 
                        n_max = nmax, col_types = "cdddcc", skip = 1, 
                        col_names=c("CHR", "POS", "N_ALLELES", 
                                    paste0("N_CHR_", sites$Site[i],"_", year),
                                    paste0("ALLELEFREQ1_", sites$Site[i],"_", year),
                                    paste0("ALLELEFREQ2_", sites$Site[i],"_", year))) %>%
  separate(col = paste0("ALLELEFREQ1_", sites$Site[i],"_", year), 
           into = c("MAJOR",
                    paste0("MAJF_", sites$Site[i],"_", year)), sep = ":", 
           convert = TRUE) %>%
  separate(col = paste0("ALLELEFREQ2_", sites$Site[i],"_", year), 
           into = c("MINOR",
                    paste0("MINF_", sites$Site[i],"_", year)), sep = ":", 
           convert = TRUE)
site_year2 <- read_delim(paste0(sites$Site[k],"_", year, ".frq"), delim = "\t", 
                        n_max = nmax, col_types = "cdddcc", skip = 1, 
                        col_names=c("CHR", "POS", "N_ALLELES", 
                                    paste0("N_CHR_", sites$Site[k],"_", year),
                                    paste0("ALLELEFREQ1_", sites$Site[k],"_", year),
                                    paste0("ALLELEFREQ2_", sites$Site[k],"_", year))) %>%
  separate(col = paste0("ALLELEFREQ1_", sites$Site[k],"_", year), 
           into = c("MAJOR",
                    paste0("MAJF_", sites$Site[k],"_", year)), sep = ":", 
           convert = TRUE) %>%
  separate(col = paste0("ALLELEFREQ2_", sites$Site[k],"_", year), 
           into = c("MINOR",
                    paste0("MINF_", sites$Site[k],"_", year)), sep = ":", 
           convert = TRUE)
freq_sitebyyear <- site_year %>%
  full_join(site_year2, by = c("CHR", "POS", "N_ALLELES", "MAJOR", "MINOR"))

for(k in 3:10){
  site_year2 <- read_delim(paste0(sites$Site[k],"_", year, ".frq"), delim = "\t", 
                        n_max = nmax, col_types = "cdddcc", skip = 1, 
                        col_names=c("CHR", "POS", "N_ALLELES", 
                                    paste0("N_CHR_", sites$Site[k],"_", year),
                                    paste0("ALLELEFREQ1_", sites$Site[k],"_", year),
                                    paste0("ALLELEFREQ2_", sites$Site[k],"_", year))) %>%
  separate(col = paste0("ALLELEFREQ1_", sites$Site[k],"_", year), 
           into = c("MAJOR",
                    paste0("MAJF_", sites$Site[k],"_", year)), sep = ":", 
           convert = TRUE) %>%
  separate(col = paste0("ALLELEFREQ2_", sites$Site[k],"_", year), 
           into = c("MINOR",
                    paste0("MINF_", sites$Site[k],"_", year)), sep = ":", 
           convert = TRUE)
  freq_sitebyyear <- freq_sitebyyear %>%
    full_join(site_year2, by = c("CHR", "POS", "N_ALLELES", "MAJOR", "MINOR"))
}
year <- "2018"
for(k in 1:10){
  site_year2 <- read_delim(paste0(sites$Site[k],"_", year, ".frq"), delim = "\t", 
                        n_max = nmax, col_types = "cdddcc", skip = 1, 
                        col_names=c("CHR", "POS", "N_ALLELES", 
                                    paste0("N_CHR_", sites$Site[k],"_", year),
                                    paste0("ALLELEFREQ1_", sites$Site[k],"_", year),
                                    paste0("ALLELEFREQ2_", sites$Site[k],"_", year))) %>%
  separate(col = paste0("ALLELEFREQ1_", sites$Site[k],"_", year), 
           into = c("MAJOR",
                    paste0("MAJF_", sites$Site[k],"_", year)), sep = ":", 
           convert = TRUE) %>%
  separate(col = paste0("ALLELEFREQ2_", sites$Site[k],"_", year), 
           into = c("MINOR",
                    paste0("MINF_", sites$Site[k],"_", year)), sep = ":", 
           convert = TRUE)
  freq_sitebyyear <- freq_sitebyyear %>%
    full_join(site_year2, by = c("CHR", "POS", "N_ALLELES", "MAJOR", "MINOR"))
}
saveRDS(freq_sitebyyear, file = paste0("Allele_frequencies_site_by_year_", nmax, "SNPs.rds"))
```

## Load data

```{r}
freq_sitebyyear <- readRDS("Allele_frequencies_site_by_year_100000SNPs.rds")
freq_sitebyyear
```

## Tally SNPs by site

```{r}
sites
freq_sitebyyear %>%
  group_by(between(MINF_PKLE_2018, 0.05, 0.95)) %>% tally()
freq_sitebyyear %>%
  filter(between(MINF_BRKG_2018, 0.05, 0.95) & 
           between(MINF_PKLE_2018, 0.05, 0.95) &
           between(MINF_FRMI_2018, 0.05, 0.95) & 
           between(MINF_KING_2018, 0.05, 0.95) & 
           between(MINF_OVTN_2018, 0.05, 0.95) & 
           between(MINF_KBSM_2018, 0.05, 0.95) & 
           between(MINF_LINC_2018, 0.05, 0.95) & 
           between(MINF_STIL_2018, 0.05, 0.95) & 
           between(MINF_TMPL_2018, 0.05, 0.95) & 
           between(MINF_CLMB_2018, 0.05, 0.95)) 
freq_sitebyyear %>%
  group_by(between(MINF_BRKG_2018, 0.05, 0.95), 
           between(MINF_PKLE_2018, 0.05, 0.95),
           between(MINF_FRMI_2018, 0.05, 0.95), 
           between(MINF_KING_2018, 0.05, 0.95), 
           between(MINF_OVTN_2018, 0.05, 0.95),  
           between(MINF_KBSM_2018, 0.05, 0.95),  
           between(MINF_LINC_2018, 0.05, 0.95),  
           between(MINF_STIL_2018, 0.05, 0.95),  
           between(MINF_TMPL_2018, 0.05, 0.95),  
           between(MINF_CLMB_2018, 0.05, 0.95),
           between(MINF_BRKG_2019, 0.05, 0.95), 
           between(MINF_PKLE_2019, 0.05, 0.95),
           between(MINF_FRMI_2019, 0.05, 0.95), 
           between(MINF_KING_2019, 0.05, 0.95), 
           between(MINF_OVTN_2019, 0.05, 0.95),  
           between(MINF_KBSM_2019, 0.05, 0.95),  
           between(MINF_LINC_2019, 0.05, 0.95),  
           between(MINF_STIL_2019, 0.05, 0.95),  
           between(MINF_TMPL_2019, 0.05, 0.95),  
           between(MINF_CLMB_2019, 0.05, 0.95)) %>% 
  tally() %>% arrange(desc(n)) %>%
  mutate(frac = n/100000)# %>%
  #write_csv(., path = "MAF_allows_gwas_10_sites_2_years.csv")
freq_sitebyyear %>%
  group_by(between(MINF_BRKG_2018, 0.05, 0.95), 
           between(MINF_PKLE_2018, 0.05, 0.95),
           between(MINF_FRMI_2018, 0.05, 0.95), 
           between(MINF_KING_2018, 0.05, 0.95), 
           between(MINF_OVTN_2018, 0.05, 0.95),  
           between(MINF_KBSM_2018, 0.05, 0.95),  
           between(MINF_LINC_2018, 0.05, 0.95),  
           between(MINF_STIL_2018, 0.05, 0.95),  
           between(MINF_TMPL_2018, 0.05, 0.95),  
           between(MINF_CLMB_2018, 0.05, 0.95)) %>% 
  tally() %>% arrange(desc(n)) %>%
  mutate(approx = n/100000*33880000) %>%
  write_csv(., path = "MAF_allows_gwas_10_sites_2018.csv")

freq_sitebyyear %>%
  filter(between(MINF_BRKG_2018, 0.05, 0.95)) %>%
  ggplot(aes(x = MINF_BRKG_2018, y = MINF_BRKG_2019)) + 
  geom_point(alpha = 0.05) + geom_abline(slope = 1, intercept = 0, color = "red")

36007-28544
28544/36007
36007/100000*33880000
28544/100000*33880000
```

    *36007 out of 100k sites had MAF > 0.05 at Pickle in 2018, our best site.
    *12.2M SNPs total.
    *28544 out of 100k sites have MAF > 0.05 at all 10 locations.
    
That's 7463 SNPs less, or 79.27% of the SNPs we would have had.
9.67M SNPs total. Lose 20% of the SNPs by constraining to 5% MAF at all sites.

BRKG 35452

# ------------------
# Start of Plots

```{r}
freq_sitebyyear %>%
  filter(between(MINF_TMPL_2018, 0.05, 0.95)) %>%
  ggplot(aes(x = MINF_TMPL_2018, y = MINF_TMPL_2019)) + 
  geom_point(alpha = 0.05) + geom_abline(slope = 1, intercept = 0, color = "red")
freq_sitebyyear %>%
  filter(between(MINF_LINC_2018, 0.05, 0.95)) %>%
  ggplot(aes(x = MINF_LINC_2018, y = MINF_LINC_2019)) + 
  geom_point(alpha = 0.05) + geom_abline(slope = 1, intercept = 0, color = "red")
freq_sitebyyear %>%
  filter(between(MINF_CLMB_2018, 0.05, 0.95)) %>%
  ggplot(aes(x = MINF_CLMB_2018, y = MINF_CLMB_2019)) + 
  geom_point(alpha = 0.05) + geom_abline(slope = 1, intercept = 0, color = "red")
freq_sitebyyear %>%
  filter(between(MINF_BRKG_2018, 0.05, 0.95)) %>%
  ggplot(aes(x = MINF_BRKG_2018, y = MINF_BRKG_2019)) + 
  geom_point(alpha = 0.05) + geom_abline(slope = 1, intercept = 0, color = "red")
freq_sitebyyear %>%
  ggplot(aes(x = MINF_BRKG_2019, y = MINF_KING_2019)) + 
  geom_point(alpha = 0.05) + geom_abline(slope = 1, intercept = 0, color = "red")
```

```{r}
freq_sitebyyear %>%
  filter(between(MINF_BRKG_2018, 0.05, 0.95) & between(MINF_KING_2018, 0.05, 0.95)) %>%
  ggplot(aes(x = MINF_BRKG_2018, y = MINF_KING_2018)) + 
  geom_hex() + geom_abline(slope = 1, intercept = 0, color = "red") + 
  scale_fill_viridis(option = "B", trans = "log2") 
freq_sitebyyear %>%
  filter(between(MINF_BRKG_2019, 0.05, 0.95) & between(MINF_KING_2019, 0.05, 0.95)) %>%
  ggplot(aes(x = MINF_BRKG_2019, y = MINF_KING_2019)) + 
  geom_hex() + geom_abline(slope = 1, intercept = 0, color = "red") + 
  scale_fill_viridis(option = "B", trans = "log2") 
freq_sitebyyear %>%
  filter(between(MINF_LINC_2019, 0.05, 0.95) & between(MINF_PKLE_2019, 0.05, 0.95)) %>%
  ggplot(aes(x = MINF_LINC_2019, y = MINF_PKLE_2019)) + 
  geom_hex() + geom_abline(slope = 1, intercept = 0, color = "red") + 
  scale_fill_viridis(option = "B", trans = "log2") 
```

## Labeled hex plots

```{r}
freq_sitebyyear %>%
  filter(between(MINF_BRKG_2018, 0.05, 0.95) & between(MINF_BRKG_2019, 0.05, 0.95)) %>%
  ggplot(aes(x = MINF_BRKG_2018, y = MINF_BRKG_2019)) + 
  geom_hline(yintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_vline(xintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.95, 0.43)) + 
  xlab("MAF in Brookings, 2018") +
  ylab("MAF in Brookings, 2019")

freq_sitebyyear %>%
  filter(between(MINF_KING_2019, 0.05, 0.95) & between(MINF_BRKG_2019, 0.05, 0.95)) %>%
  ggplot(aes(x = MINF_KING_2019, y = MINF_BRKG_2019)) + 
  geom_hline(yintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_vline(xintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.95, 0.43)) + 
  xlab("MAF in Kingsville, 2019") +
  ylab("MAF in Brookings, 2019")

freq_sitebyyear %>%
  filter(between(MINF_OVTN_2019, 0.05, 0.95) & between(MINF_BRKG_2019, 0.05, 0.95)) %>%
  ggplot(aes(x = MINF_OVTN_2019, y = MINF_BRKG_2019)) + 
  geom_hline(yintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_vline(xintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.95, 0.43)) + 
  xlab("MAF in Overton, 2019") +
  ylab("MAF in Brookings, 2019")

freq_sitebyyear %>%
  filter(between(MINF_OVTN_2018, 0.05, 0.95) & between(MINF_BRKG_2018, 0.05, 0.95)) %>%
  ggplot(aes(x = MINF_OVTN_2018, y = MINF_BRKG_2018)) + 
  geom_hline(yintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_vline(xintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.95, 0.43)) + 
  xlab("MAF in Overton, 2018") +
  ylab("MAF in Brookings, 2018")

freq_sitebyyear %>%
  filter(between(MINF_OVTN_2018, 0.05, 0.95) & between(MINF_LINC_2018, 0.05, 0.95)) %>%
  ggplot(aes(x = MINF_OVTN_2018, y = MINF_LINC_2018)) + 
  geom_hline(yintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_vline(xintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.95, 0.43)) + 
  xlab("MAF in Overton, 2018") +
  ylab("MAF in Lincoln, 2018")
```

## Grid of plots?


```{r}
maf_long <- freq_sitebyyear %>%
  dplyr::select(-starts_with("MAJF"), -starts_with("N_CHR_")) %>%
  pivot_longer(cols = starts_with("MINF"), names_to = "SITE_YEAR", values_to = "MINF") %>%
  separate(col = "SITE_YEAR", into = c("MINF_", "SITE", "YEAR"), sep = "_")

```

# ---------------------------
## Full SNP file analyses

```{r}

freq_sitebyyear %>%
  group_by(between(MINF_BRKG_2018, 0.05, 0.95), 
           between(MINF_PKLE_2018, 0.05, 0.95),
           between(MINF_FRMI_2018, 0.05, 0.95), 
           between(MINF_KING_2018, 0.05, 0.95), 
           between(MINF_OVTN_2018, 0.05, 0.95),  
           between(MINF_KBSM_2018, 0.05, 0.95),  
           between(MINF_LINC_2018, 0.05, 0.95),  
           between(MINF_STIL_2018, 0.05, 0.95),  
           between(MINF_TMPL_2018, 0.05, 0.95),  
           between(MINF_CLMB_2018, 0.05, 0.95),
           between(MINF_BRKG_2019, 0.05, 0.95), 
           between(MINF_PKLE_2019, 0.05, 0.95),
           between(MINF_FRMI_2019, 0.05, 0.95), 
           between(MINF_KING_2019, 0.05, 0.95), 
           between(MINF_OVTN_2019, 0.05, 0.95),  
           between(MINF_KBSM_2019, 0.05, 0.95),  
           between(MINF_LINC_2019, 0.05, 0.95),  
           between(MINF_STIL_2019, 0.05, 0.95),  
           between(MINF_TMPL_2019, 0.05, 0.95),  
           between(MINF_CLMB_2019, 0.05, 0.95)) %>% 
  tally() %>% arrange(desc(n)) %>%
  write_csv(., path = "SNPs_where_MAF_allows_gwas_10_sites_2_years.csv")
freq_sitebyyear %>%
  group_by(between(MINF_BRKG_2018, 0.05, 0.95), 
           between(MINF_PKLE_2018, 0.05, 0.95),
           between(MINF_FRMI_2018, 0.05, 0.95), 
           between(MINF_KING_2018, 0.05, 0.95), 
           between(MINF_OVTN_2018, 0.05, 0.95),  
           between(MINF_KBSM_2018, 0.05, 0.95),  
           between(MINF_LINC_2018, 0.05, 0.95),  
           between(MINF_STIL_2018, 0.05, 0.95),  
           between(MINF_TMPL_2018, 0.05, 0.95),  
           between(MINF_CLMB_2018, 0.05, 0.95)) %>% 
  tally() %>% arrange(desc(n)) %>%
  write_csv(., path = "SNPs_where_MAF_allows_gwas_10_sites_2018.csv")
```

```{r}
freq_maf_2018 <- freq_sitebyyear %>%
  filter(between(MINF_BRKG_2018, 0.05, 0.95) & 
           between(MINF_PKLE_2018, 0.05, 0.95) &
           between(MINF_FRMI_2018, 0.05, 0.95) & 
           between(MINF_KING_2018, 0.05, 0.95) & 
           between(MINF_OVTN_2018, 0.05, 0.95) & 
           between(MINF_KBSM_2018, 0.05, 0.95) & 
           between(MINF_LINC_2018, 0.05, 0.95) & 
           between(MINF_STIL_2018, 0.05, 0.95) & 
           between(MINF_TMPL_2018, 0.05, 0.95) & 
           between(MINF_CLMB_2018, 0.05, 0.95)) 
saveRDS(freq_maf_2018, "~/Github/pvdiv-phenology-gxe/large-files/Site-by-year-allele-frequencies/Allele_frequencies_with_maf_above_5percent.rds")
```

## Labeled hex plots

```{r}
brkg2yr <- freq_maf_2018 %>%
  ggplot(aes(x = MINF_BRKG_2018, y = MINF_BRKG_2019)) + 
  geom_hline(yintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_vline(xintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.95, 0.475)) + 
  xlab("MAF in Brookings, 2018") +
  ylab("MAF in Brookings, 2019")
save_plot(filename = "BRKG_2_years_MAF_change.png", plot = brkg2yr)

brkg_king9 <- freq_maf_2018 %>%
  ggplot(aes(x = MINF_KING_2019, y = MINF_BRKG_2019)) + 
  geom_hline(yintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_vline(xintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.95, 0.475)) + 
  xlab("MAF in Kingsville, 2019") +
  ylab("MAF in Brookings, 2019")
save_plot(filename = "BRKG_2019_vs_KING_2019_MAF.png", plot = brkg_king9)

brkg_ovtn9 <- freq_maf_2018 %>%
  ggplot(aes(x = MINF_OVTN_2019, y = MINF_BRKG_2019)) + 
  geom_hline(yintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_vline(xintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.95, 0.475)) + 
  xlab("MAF in Overton, 2019") +
  ylab("MAF in Brookings, 2019")
save_plot(filename = "BRKG_2019_vs_OVTN_2019_MAF.png", plot = brkg_ovtn9)

brkg_ovtn8 <- freq_maf_2018 %>%
  ggplot(aes(x = MINF_OVTN_2018, y = MINF_BRKG_2018)) + 
  geom_hline(yintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_vline(xintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.95, 0.475)) + 
  xlab("MAF in Overton, 2018") +
  ylab("MAF in Brookings, 2018")
save_plot(filename = "BRKG_2018_vs_OVTN_2018_MAF.png", plot = brkg_ovtn8)


linc_ovtn8 <- freq_maf_2018 %>%
  ggplot(aes(x = MINF_OVTN_2018, y = MINF_LINC_2018)) + 
  geom_hline(yintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_vline(xintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.95, 0.475)) + 
  xlab("MAF in Overton, 2018") +
  ylab("MAF in Lincoln, 2018")
save_plot(filename = "LINC_2018_vs_OVTN_2018_MAF.png", plot = linc_ovtn8)
###
linc_ovtn9 <- freq_maf_2018 %>%
  ggplot(aes(x = MINF_OVTN_2019, y = MINF_LINC_2019)) + 
  geom_hline(yintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_vline(xintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.95, 0.475)) + 
  xlab("MAF in Overton, 2019") +
  ylab("MAF in Lincoln, 2019")
save_plot(filename = "LINC_2019_vs_OVTN_2019_MAF.png", plot = linc_ovtn9)

frmi_king9 <- freq_maf_2018 %>%
  ggplot(aes(x = MINF_FRMI_2019, y = MINF_KING_2019)) + 
  geom_hline(yintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_vline(xintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.95, 0.475)) + 
  xlab("MAF in Fermilab, IL, 2019") +
  ylab("MAF in Kingsville, TX, 2019")
save_plot(filename = "FRMI_2019_vs_KING_2019_MAF.png", plot = frmi_king9)
##

frmi_2yr <- freq_maf_2018 %>%
  ggplot(aes(x = MINF_FRMI_2018, y = MINF_FRMI_2019)) + 
  geom_hline(yintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_vline(xintercept = c(0.25, 0.5, 0.75), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.95, 0.475)) + 
  xlab("MAF in Fermilab, IL, 2018") +
  ylab("MAF in Fermilab, IL, 2019")
save_plot(filename = "FRMI_2018_vs_2019_MAF.png", plot = frmi_2yr)
##

```


# Minor allele count instead of MAF

Minor allele needs to be present at least 15 times (for a MAF of 5% in 2019 at FRMI). 18 for mean

But for BRKG/LINC/KING 21 - 24 present is better. Let's do 25, every location has to have 25 or more of an allele to keep that SNP.

Or (if FRMI leads to dropping a ton of SNPs) possibly keep some SNPs that are above 15 for FRMI, and above 25 everywhere else.

```{r}
freq
mean(freq$N_CHR_FRMI_2019)*.05 # 286 # 416 BRKG 428 LINC 478 KING
(286)*.05
mean(freq$N_CHR_BRKG_2019)*.05
mean(freq$N_CHR_LINC_2019)*.05
mean(freq$N_CHR_KING_2019)*.05
```
STIL 762

```{r}
mac_sitebyyear <- freq_sitebyyear %>%
  mutate(MAC_PKLE_2019 = ifelse(round(N_CHR_PKLE_2019*MINF_PKLE_2019)>=N_CHR_PKLE_2019/2, round(N_CHR_PKLE_2019*MAJF_PKLE_2019), round(N_CHR_PKLE_2019*MINF_PKLE_2019)),
         MAC_FRMI_2019 = ifelse(round(N_CHR_FRMI_2019*MINF_FRMI_2019)>=N_CHR_FRMI_2019/2, round(N_CHR_FRMI_2019*MAJF_FRMI_2019), round(N_CHR_FRMI_2019*MINF_FRMI_2019)),
         MAC_LINC_2019 = ifelse(round(N_CHR_LINC_2019*MINF_LINC_2019)>=N_CHR_LINC_2019/2, round(N_CHR_LINC_2019*MAJF_LINC_2019), round(N_CHR_LINC_2019*MINF_LINC_2019)),
         MAC_BRKG_2019 = ifelse(round(N_CHR_BRKG_2019*MINF_BRKG_2019)>=N_CHR_BRKG_2019/2, round(N_CHR_BRKG_2019*MAJF_BRKG_2019), round(N_CHR_BRKG_2019*MINF_BRKG_2019)),
         MAC_CLMB_2019 = ifelse(round(N_CHR_CLMB_2019*MINF_CLMB_2019)>=N_CHR_CLMB_2019/2, round(N_CHR_CLMB_2019*MAJF_CLMB_2019), round(N_CHR_CLMB_2019*MINF_CLMB_2019)),
         MAC_OVTN_2019 = ifelse(round(N_CHR_OVTN_2019*MINF_OVTN_2019)>=N_CHR_OVTN_2019/2, round(N_CHR_OVTN_2019*MAJF_OVTN_2019), round(N_CHR_OVTN_2019*MINF_OVTN_2019)),
         MAC_STIL_2019 = ifelse(round(N_CHR_STIL_2019*MINF_STIL_2019)>=N_CHR_STIL_2019/2, round(N_CHR_STIL_2019*MAJF_STIL_2019), round(N_CHR_STIL_2019*MINF_STIL_2019)),
         MAC_TMPL_2019 = ifelse(round(N_CHR_TMPL_2019*MINF_TMPL_2019)>=N_CHR_TMPL_2019/2, round(N_CHR_TMPL_2019*MAJF_TMPL_2019), round(N_CHR_TMPL_2019*MINF_TMPL_2019)),
         MAC_KBSM_2019 = ifelse(round(N_CHR_KBSM_2019*MINF_KBSM_2019)>=N_CHR_KBSM_2019/2, round(N_CHR_KBSM_2019*MAJF_KBSM_2019), round(N_CHR_KBSM_2019*MINF_KBSM_2019)),
         MAC_KING_2019 = ifelse(round(N_CHR_KING_2019*MINF_KING_2019)>=N_CHR_KING_2019/2, round(N_CHR_KING_2019*MAJF_KING_2019), round(N_CHR_KING_2019*MINF_KING_2019)),
         MAC_KING_2018 = ifelse(round(N_CHR_KING_2018*MINF_KING_2018)>=N_CHR_KING_2018/2, round(N_CHR_KING_2018*MAJF_KING_2018), round(N_CHR_KING_2018*MINF_KING_2018)),
         MAC_KBSM_2018 = ifelse(round(N_CHR_KBSM_2018*MINF_KBSM_2018)>=N_CHR_KBSM_2018/2, round(N_CHR_KBSM_2018*MAJF_KBSM_2018), round(N_CHR_KBSM_2018*MINF_KBSM_2018)),
         MAC_PKLE_2018 = ifelse(round(N_CHR_PKLE_2018*MINF_PKLE_2018)>=N_CHR_PKLE_2018/2, round(N_CHR_PKLE_2018*MAJF_PKLE_2018), round(N_CHR_PKLE_2018*MINF_PKLE_2018)),
         MAC_OVTN_2018 = ifelse(round(N_CHR_OVTN_2018*MINF_OVTN_2018)>=N_CHR_OVTN_2018/2, round(N_CHR_OVTN_2018*MAJF_OVTN_2018), round(N_CHR_OVTN_2018*MINF_OVTN_2018)),
         MAC_BRKG_2018 = ifelse(round(N_CHR_BRKG_2018*MINF_BRKG_2018)>=N_CHR_BRKG_2018/2, round(N_CHR_BRKG_2018*MAJF_BRKG_2018), round(N_CHR_BRKG_2018*MINF_BRKG_2018)),
         MAC_CLMB_2018 = ifelse(round(N_CHR_CLMB_2018*MINF_CLMB_2018)>=N_CHR_CLMB_2018/2, round(N_CHR_CLMB_2018*MAJF_CLMB_2018), round(N_CHR_CLMB_2018*MINF_CLMB_2018)),
         MAC_TMPL_2018 = ifelse(round(N_CHR_TMPL_2018*MINF_TMPL_2018)>=N_CHR_TMPL_2018/2, round(N_CHR_TMPL_2018*MAJF_TMPL_2018), round(N_CHR_TMPL_2018*MINF_TMPL_2018)),
         MAC_LINC_2018 = ifelse(round(N_CHR_LINC_2018*MINF_LINC_2018)>=N_CHR_LINC_2018/2, round(N_CHR_LINC_2018*MAJF_LINC_2018), round(N_CHR_LINC_2018*MINF_LINC_2018)),
         MAC_STIL_2018 = ifelse(round(N_CHR_STIL_2018*MINF_STIL_2018)>=N_CHR_STIL_2018/2, round(N_CHR_STIL_2018*MAJF_STIL_2018), round(N_CHR_STIL_2018*MINF_STIL_2018)),
         MAC_FRMI_2018 = ifelse(round(N_CHR_FRMI_2018*MINF_FRMI_2018)>=N_CHR_FRMI_2018/2, round(N_CHR_FRMI_2018*MAJF_FRMI_2018), round(N_CHR_FRMI_2018*MINF_FRMI_2018))) %>%
  dplyr::select(CHR, POS, MAC_PKLE_2019:MAC_FRMI_2018)

saveRDS(mac_sitebyyear, file =  "SNPs_where_minor_allele_count_allows_gwas_10_sites_2_years.rds")

mac_sitebyyear %>%
    group_by(MAC_PKLE_2019 > 25, MAC_PKLE_2018 > 25,
           MAC_FRMI_2019 > 15, MAC_FRMI_2018 > 15,
           MAC_OVTN_2019 > 25, MAC_OVTN_2018 > 25,
           MAC_LINC_2019 > 25, MAC_LINC_2018 > 25,
           MAC_STIL_2019 > 25, MAC_STIL_2018 > 25,
           MAC_TMPL_2019 > 25, MAC_TMPL_2018 > 25,
           MAC_CLMB_2019 > 25, MAC_CLMB_2018 > 25,
           MAC_BRKG_2019 > 25, MAC_BRKG_2018 > 25,
           MAC_KBSM_2019 > 25, MAC_KBSM_2018 > 25,
           MAC_KING_2019 > 25, MAC_KING_2018 > 25) %>%
  tally() %>%
  write_csv(., path = "Minor_allele_count_that_allows_gwas_10_sites_2_years_relaxed_FRMI.csv")

mac_sitebyyear %>%
    group_by(MAC_PKLE_2019 > 25, MAC_PKLE_2018 > 25,
           MAC_FRMI_2019 > 25, MAC_FRMI_2018 > 25,
           MAC_OVTN_2019 > 25, MAC_OVTN_2018 > 25,
           MAC_LINC_2019 > 25, MAC_LINC_2018 > 25,
           MAC_STIL_2019 > 25, MAC_STIL_2018 > 25,
           MAC_TMPL_2019 > 25, MAC_TMPL_2018 > 25,
           MAC_CLMB_2019 > 25, MAC_CLMB_2018 > 25,
           MAC_BRKG_2019 > 25, MAC_BRKG_2018 > 25,
           MAC_KBSM_2019 > 25, MAC_KBSM_2018 > 25,
           MAC_KING_2019 > 25, MAC_KING_2018 > 25) %>%
  tally() %>%
  write_csv(., path = "Minor_allele_count_that_allows_gwas_10_sites_2_years.csv")
```

```{r}
mac %>%
  group_by(MAC_PKLE_2019 > 25, MAC_PKLE_2018 > 25,
           MAC_FRMI_2019 > 15, MAC_FRMI_2018 > 15,
           MAC_OVTN_2019 > 25, MAC_OVTN_2018 > 25,
           MAC_LINC_2019 > 25, MAC_LINC_2018 > 25,
           MAC_STIL_2019 > 25, MAC_STIL_2018 > 25,
           MAC_TMPL_2019 > 25, MAC_TMPL_2018 > 25,
           MAC_CLMB_2019 > 25, MAC_CLMB_2018 > 25,
           MAC_BRKG_2019 > 25, MAC_BRKG_2018 > 25,
           MAC_KBSM_2019 > 25, MAC_KBSM_2018 > 25,
           MAC_KING_2019 > 25, MAC_KING_2018 > 25) %>%
  tally()
```

```{r}
mac_sitebyyear <- readRDS(file =  "SNPs_where_minor_allele_count_allows_gwas_10_sites_2_years.rds")

mac_2018_gt25 <- mac_sitebyyear %>%
  filter(MAC_PKLE_2018 >= 25,
         MAC_FRMI_2018 >= 15,
         MAC_OVTN_2018 >= 25,
         MAC_LINC_2018 >= 25,
         MAC_STIL_2018 >= 25,
         MAC_TMPL_2018 >= 25,
         MAC_CLMB_2018 >= 25,
         MAC_BRKG_2018 >= 25,
         MAC_KBSM_2018 >= 25,
         MAC_KING_2018 >= 25)

brkg_2yr <- mac_2018_gt25 %>%
  ggplot(aes(x = MAC_BRKG_2018, y = MAC_BRKG_2019)) + 
  geom_hline(yintercept = c(0, 100, 200), color = "lightgrey") +
  geom_vline(xintercept = c(0, 100, 200), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.2, 0.7)) + 
  xlab("Minor Allele Count in Brookings, 2018") +
  ylab("Minor Allele Count in Brookings, 2019")
save_plot(filename = "BRKG_2018_v_2019_minor_allele_count_change.png", plot = brkg_2yr)

brkg_king9 <- mac_2018_gt25 %>%
  ggplot(aes(x = MAC_KING_2019, y = MAC_BRKG_2019)) + 
  geom_hline(yintercept = c(0, 100, 200), color = "lightgrey") +
  geom_vline(xintercept = c(0, 100, 200), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.2, 0.7)) + 
  xlab("Minor Allele Count in Kingsville, 2019") +
  ylab("Minor Allele Count in Brookings, 2019")
save_plot(filename = "BRKG_2019_vs_KING_2019_minor_allele_count.png", plot = brkg_king9)

brkg_ovtn9 <- mac_2018_gt25 %>%
  ggplot(aes(x = MAC_OVTN_2019, y = MAC_BRKG_2019)) + 
  geom_hline(yintercept = c(0, 100, 200), color = "lightgrey") +
  geom_vline(xintercept = c(0, 100, 200), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.2, 0.7)) + + 
  xlab("Minor Allele Count in Overton, 2019") +
  ylab("Minor Allele Count in Brookings, 2019")
save_plot(filename = "BRKG_2019_vs_OVTN_2019_minor_allele_count.png", plot = brkg_ovtn9)

brkg_ovtn8 <- mac_2018_gt25 %>%
  ggplot(aes(x = MAC_OVTN_2018, y = MAC_BRKG_2018)) + 
  geom_hline(yintercept = c(0, 100, 200), color = "lightgrey") +
  geom_vline(xintercept = c(0, 100, 200), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.2, 0.7)) + 
  xlab("Minor Allele Count in Overton, TX, 2018") +
  ylab("Minor Allele Count in Brookings, SD, 2018")
save_plot(filename = "BRKG_2018_vs_OVTN_2018_minor_allele_count.png", plot = brkg_ovtn8)


linc_ovtn8 <- mac_2018_gt25 %>%
  ggplot(aes(x = MAC_OVTN_2018, y = MAC_LINC_2018)) + 
  geom_hline(yintercept = c(0, 100, 200), color = "lightgrey") +
  geom_vline(xintercept = c(0, 100, 200), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.2, 0.7)) +  
  xlab("Minor Allele Count in Overton, 2018") +
  ylab("Minor Allele Count in Lincoln, 2018")
save_plot(filename = "LINC_2018_vs_OVTN_2018_minor_allele_count.png", plot = linc_ovtn8)
###








linc_ovtn9 <- mac_2018_gt25 %>%
  ggplot(aes(x = MAC_OVTN_2019, y = MAC_LINC_2019)) + 
  geom_hline(yintercept = c(0, 100, 200), color = "lightgrey") +
  geom_vline(xintercept = c(0, 100, 200), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.2, 0.7)) + 
  xlab("Minor Allele Count in Overton, TX, 2019") +
  ylab("Minor Allele Count in Lincoln, NE, 2019")
save_plot(filename = "LINC_2019_vs_OVTN_2019_minor_allele_count.png", plot = linc_ovtn9)

frmi_king9 <- mac_2018_gt25 %>%
  ggplot(aes(x = MAC_FRMI_2019, y = MAC_KING_2019)) + 
  geom_hline(yintercept = c(0, 100, 200), color = "lightgrey") +
  geom_vline(xintercept = c(0, 100, 200), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.2, 0.7)) + 
  xlab("Minor Allele Count in Fermilab, IL, 2019") +
  ylab("Minor Allele Count in Kingsville, TX, 2019")
save_plot(filename = "FRMI_2019_vs_KING_2019_minor_allele_count.png", plot = frmi_king9)
##

frmi_2yr <- mac_2018_gt25 %>%
  ggplot(aes(x = MAC_FRMI_2018, y = MAC_FRMI_2019)) + 
  geom_hline(yintercept = c(0, 100, 200), color = "lightgrey") +
  geom_vline(xintercept = c(0, 100, 200), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.2, 0.7)) + 
  xlab("Minor Allele Count in Fermilab, IL, 2018") +
  ylab("Minor Allele Count in Fermilab, IL, 2019")
save_plot(filename = "FRMI_2018_vs_2019_minor_allele_count.png", plot = frmi_2yr)
##
linc_2yr <- mac_2018_gt25 %>%
  ggplot(aes(x = MAC_LINC_2018, y = MAC_LINC_2019)) + 
  geom_hline(yintercept = c(0, 100, 200), color = "lightgrey") +
  geom_vline(xintercept = c(0, 100, 200), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.2, 0.7)) + 
  xlab("Minor Allele Count in Lincoln, NE, 2018") +
  ylab("Minor Allele Count in Lincoln, NE, 2019")
save_plot(filename = "LINC_2018_vs_2019_minor_allele_count.png", plot = linc_2yr)

clmb_2yr <- mac_2018_gt25 %>%
  ggplot(aes(x = MAC_CLMB_2018, y = MAC_CLMB_2019)) + 
  geom_hline(yintercept = c(0, 100, 200), color = "lightgrey") +
  geom_vline(xintercept = c(0, 100, 200), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.2, 0.7)) + 
  xlab("Minor Allele Count in Columbia, MO, 2018") +
  ylab("Minor Allele Count in Columbia, MO, 2019")
save_plot(filename = "CLMB_2018_vs_2019_minor_allele_count.png", plot = clmb_2yr)

kbsm_2yr <- mac_2018_gt25 %>%
  ggplot(aes(x = MAC_KBSM_2018, y = MAC_KBSM_2019)) + 
  geom_hline(yintercept = c(0, 100, 200), color = "lightgrey") +
  geom_vline(xintercept = c(0, 100, 200), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.2, 0.7)) + 
  xlab("Minor Allele Count in KBS, MI, 2018") +
  ylab("Minor Allele Count in KBS, MI, 2019")
save_plot(filename = "KBSM_2018_vs_2019_minor_allele_count.png", plot = kbsm_2yr)
###
frmi_king8 <- mac_2018_gt25 %>%
  ggplot(aes(x = MAC_FRMI_2018, y = MAC_KING_2018)) + 
  geom_hline(yintercept = c(0, 100, 200), color = "lightgrey") +
  geom_vline(xintercept = c(0, 100, 200), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.2, 0.7)) + 
  xlab("Minor Allele Count in Fermilab, IL, 2018") +
  ylab("Minor Allele Count in Kingsville, TX, 2018")
save_plot(filename = "FRMI_2018_vs_KING_2018_minor_allele_count.png", plot = frmi_king9)
brkg_king8 <- mac_2018_gt25 %>%
  ggplot(aes(x = MAC_KING_2018, y = MAC_BRKG_2018)) + 
  geom_hline(yintercept = c(0, 100, 200), color = "lightgrey") +
  geom_vline(xintercept = c(0, 100, 200), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.2, 0.7)) + 
  xlab("Minor Allele Count in Kingsville, 2018") +
  ylab("Minor Allele Count in Brookings, 2018")
save_plot(filename = "BRKG_2018_vs_KING_2018_minor_allele_count.png", plot = brkg_king9)
#
clmb_pkle9 <- mac_2018_gt25 %>%
  ggplot(aes(x = MAC_CLMB_2019, y = MAC_PKLE_2019)) + 
  geom_hline(yintercept = c(0, 100, 200), color = "lightgrey") +
  geom_vline(xintercept = c(0, 100, 200), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.2, 0.7)) + 
  xlab("Minor Allele Count in Columbia, MO, 2019") +
  ylab("Minor Allele Count in Pickle, TX, 2019")
save_plot(filename = "CLMB_2019_vs_PKLE_2019_minor_allele_count.png", plot = clmb_pkle9)

kbsm_pkle9 <- mac_2018_gt25 %>%
  ggplot(aes(x = MAC_KBSM_2019, y = MAC_PKLE_2019)) + 
  geom_hline(yintercept = c(0, 100, 200), color = "lightgrey") +
  geom_vline(xintercept = c(0, 100, 200), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.2, 0.7)) + 
  xlab("Minor Allele Count in KBS, MI, 2019") +
  ylab("Minor Allele Count in Pickle, TX, 2019")
save_plot(filename = "KBSM_2019_vs_PKLE_2019_minor_allele_count.png", plot = kbsm_pkle9)

brkg_pkle9 <- mac_2018_gt25 %>%
  ggplot(aes(x = MAC_BRKG_2019, y = MAC_PKLE_2019)) + 
  geom_hline(yintercept = c(0, 100, 200), color = "lightgrey") +
  geom_vline(xintercept = c(0, 100, 200), color = "lightgrey") +
  geom_hex() + 
  geom_abline(slope = 1, intercept = 0, linetype = 2) + 
  scale_fill_viridis(option = "B", trans = "log2") + 
  theme(legend.position = c(0.2, 0.7)) + 
  xlab("Minor Allele Count in Brookings, SD, 2019") +
  ylab("Minor Allele Count in Pickle, TX, 2019")
save_plot(filename = "BRKG_2019_vs_PKLE_2019_minor_allele_count.png", plot = brkg_pkle9)
```

5 plots of 2018 vs 2019: any place where more than 20 plants died. That's basically just the north.

Some 2019 north vs south plots.


# -----------------

# Impute on 11M SNPs

```{r}
snps <- readRDS("SNPs_where_minor_allele_count_allows_gwas_10_sites_2_years_.rds")
```

Here is a breakdown of the number of individuals we have for GWAS at each site, for phenotypes from 2018 and from 2019. The first two data columns have individuals with high quality SNP data and phenotypes. The next two have tetraploid plants with phenotypes – there is a lot of overlap, and only 5-6 plants that could have their libraries resequenced to help the GWAS data.

	Plants with SNP data & phenotypes	Plants with phenotypes	Plants Dead
	2018	2019	2018	2019	2018 to 2019
BRKG	408	254	412	254	154
KBSM	539	517	544	517	22
FRMI	272	185	273	185	87
LINC	433	262	437	262	171
CLMB	585	496	591	496	89
STIL	465	451	469	451	14
OVTN	343	332	346	332	11
TMPL	384	373	388	373	11
PKLE	617	615	623	615	2
KING	303	294	305	294	9


Given that information, I then subdivided the SNPs from John’s new 33.9M SNP dataset. These are SNPs missing in 20% or less of individuals from each subpopulation. Instead of a minor allele frequency, I used a minor allele count of 25 or more (except for FRMI where I used 15). This was the same as a MAF of 5% at BRKG, FRMI, and LINC, and a slightly lower MAF for the other sites with more individuals. Here’s the breakdown of the site*year combinations where SNPs can and can’t be used for GWAS:

Number of SNPs	Category
12536071
	Can't be used for GWAS – these SNPs have less than 25 copies of the minor allele in the 631 individuals planted at the common gardens.
10452584	Can be used for GWAS at all 10 sites and both 2018 and 2019 data.
1160326	Can’t be used in LINC & BRKG in 2019
1155906	Only at high enough frequency to be used at the big 3 sites
546381	Can only be used at PKLE & CLMB
439918	Can only be used at PKLE
346189	Can only be used at big 4 sites – KBSM CLMB PKLE STIL
341773	Can’t be used at KING OVTN FRMI
331082	Can’t be used at LINC in 2019 
327364	Can’t be used at KING
315522	Can only be used at big 4 and LINC 2018
305573	Missing 2019 at 4 freezing sites – LINC FRMI CLMB BRKG
285838	Can only be used at PKLE x2yr & CLMB 2018
246340	Missing 2019 at 3 freezing sites, not big site (BRKG)
231650	Missing KING x2yr and FRMI 2019
192819	Missing KING x2yr and FRMI x2yr
……	4.67M SNPs have some other combination of Site x Year that is useable.
 
10_site_2_yr 10452584 all MAC > 25 except FRMI 2019 > 15.
big_three   10452584 + 1155906 PKLE KBSM CLMB > 25
fourway_seven   PKLE TMPL BRKG KBSM LINC CLMB STIL > 25
 10452584 + 341773 + 327364 + 231650 + 192819 + ...
miss_freeze 305573 + 246340 + 1160326 + 331082 4 sites 2018 > 25 and 1-4 sites 2019 < 25

## 10site_2yr
all MAC >= 25 except FRMI 2019 > 15.

10,183,856 SNPs
```{r}
tensite_twoyear <- snps %>%
  filter(MAC_PKLE_2019 >= 25, MAC_PKLE_2018 >= 25,
           MAC_FRMI_2019 >= 15, MAC_FRMI_2018 >= 25,
           MAC_OVTN_2019 >= 25, MAC_OVTN_2018 >= 25,
           MAC_LINC_2019 >= 25, MAC_LINC_2018 >= 25,
           MAC_STIL_2019 >= 25, MAC_STIL_2018 >= 25,
           MAC_TMPL_2019 >= 25, MAC_TMPL_2018 >= 25,
           MAC_CLMB_2019 >= 25, MAC_CLMB_2018 >= 25,
           MAC_BRKG_2019 >= 25, MAC_BRKG_2018 >= 25,
           MAC_KBSM_2019 >= 25, MAC_KBSM_2018 >= 25,
           MAC_KING_2019 >= 25, MAC_KING_2018 >= 25)
saveRDS(tensite_twoyear, file = "SNPs_above_MAC_25_in_10sites_2years.rds")
```

## big_three

18,333,684

PKLE KBSM CLMB >= 25
```{r}
big_three <- snps %>%
  filter(MAC_PKLE_2019 >= 25, MAC_PKLE_2018 >= 25,
           MAC_CLMB_2019 >= 25, MAC_CLMB_2018 >= 25,
           MAC_KBSM_2019 >= 25, MAC_KBSM_2018 >= 25)
saveRDS(big_three, "SNPs_above_MAC_25_in_big_three_sites.rds")
```

## fourway_seven

12,358,701 SNPs

PKLE TMPL BRKG KBSM LINC CLMB STIL >= 25
```{r}
fourway_seven <- snps %>%
  filter(MAC_PKLE_2019 >= 25, MAC_PKLE_2018 >= 25,
           MAC_LINC_2019 >= 25, MAC_LINC_2018 >= 25,
           MAC_STIL_2019 >= 25, MAC_STIL_2018 >= 25,
           MAC_TMPL_2019 >= 25, MAC_TMPL_2018 >= 25,
           MAC_CLMB_2019 >= 25, MAC_CLMB_2018 >= 25,
           MAC_BRKG_2019 >= 25, MAC_BRKG_2018 >= 25,
           MAC_KBSM_2019 >= 25, MAC_KBSM_2018 >= 25)
saveRDS(fourway_seven, file = "SNPs_above_MAC_25_in_fourway_7_sites.rds")
```

## missing_frz

4 sites 2018 >= 25 and 1-4 sites 2019 < 25

1,963,771 SNPs
```{r}
missing_frz <- snps %>%
  filter(MAC_FRMI_2018 >= 25 & MAC_LINC_2018 >= 25 & MAC_CLMB_2018 >= 25 &
           MAC_BRKG_2018 >= 25, MAC_FRMI_2019 < 15 | MAC_LINC_2018 < 25 | 
           MAC_CLMB_2018 < 25 | MAC_BRKG_2019 < 25) %>%
  mutate(avediff = ((MAC_FRMI_2018 - MAC_FRMI_2019) + (MAC_LINC_2018 - MAC_LINC_2019) + (MAC_CLMB_2018 - MAC_CLMB_2019) + (MAC_BRKG_2018 - MAC_BRKG_2019))/4)

saveRDS(missing_frz, file = "Missing_SNPs_2019_4_freezing_sites.rds")
```

```{r}
missing_frz <- readRDS(file = "Missing_SNPs_2019_4_freezing_sites.rds")
```

```{r}
missing_frz %>%
  ggplot(aes(x = POS, y = avediff)) + 
  geom_point() + 
  facet_wrap(~CHR)
save_plot(filename = "Distribution_of_SNPs_lost_in_freezing_four_sites.png", plot = last_plot())


missing_frz %>%
  ggplot(aes(x = POS, y = avediff)) + 
  geom_hex() + 
  facet_wrap(~CHR) + 
  scale_color_viridis() + 
  scale_fill_viridis()
save_plot(filename = "Distribution_of_SNPs_lost_in_freezing_four_sites_hex.png", plot = last_plot())
```

