---
title: "3_subset_SNP_files"
author: "Alice MacQueen"
date: "2/7/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(switchgrassGWAS)
library(viridis)
library(cowplot)
library(bigsnpr)
source("~/Github/Functions_ggplot-theme-adjustments_2018-01-03.R")
```

## 

```{r}
snp_subsets <- readRDS("~/Github/pvdiv-phenology-gxe/analysis/common-garden-allele-balancing/SNP_subsets_33M_markers.rds")

rds <- snp_readBed("Pvirgatum_V5_GWAS_630g_33M.bed")
snp <- snp_attach(rds)
```

```{r}
snp_subsets
```

# Impute 33M SNP file

```{r}


G2 <- snp_fastImputeSimple(snp$genotypes, method = "random", ncores = NCORES)
snp$genotypes$code256 <- CODE_IMPUTE_PRED
big_counts(snp$genotypes, ind.col = 1:10) # Make sure imputation succeeded
snp <- snp_save(snp)

```



# Use previously determined SNP sets as selection criteria for subsets. 

These need to be vectors of the row numbers in the Taxa dataframe that you have phenotypes for.

```{r}  
tensite_twoyear_sel <- which(snp_subsets$SNP_tensite_twoyear == "Y")
fourway_seven_sel <- which(snp_subsets$SNP_fourway_seven == "Y")
microbiome_sel <- which(snp_subsets$SNP_microbiome == "Y")
missing_frz_sel <- which(snp_subsets$SNP_missing_frz == "Y")


tensite_twoyear_subset <- subset(snp, ind.col = tensite_twoyear_sel)
fourway_seven_subset <- subset(snp, ind.col = fourway_seven_sel)
microbiome_subset <- subset(snp, ind.col = microbiome_sel)
missing_frz_subset <- subset(snp, ind.colstr = missing_frz_sel)

```

```{r}
freq <- readRDS("~/Github/pvdiv-climate-gwas/results-upland-vs-lowland/2DSFS/Alleles_five_populations_subpops0.95.rds")
missing_frz_freq <- snp_subsets %>% filter(SNP_missing_frz == "Y") %>% left_join(freq, by = c("CHR", "POS"))

 frz_freq_subpop <- frz_freq %>% left_join(freq)
saveRDS(frz_freq_subpop, "Five_subpop_allele_frequencies_for_SNPs_with_1percent_tail_frequency_change_freezing_sites.rds")

```



```{r}
ind_gwas <- read_csv("~/Github/pvdiv-climate-gwas/results-33M-SNPs/Individuals_needed_for_climate_and_phenotype_GWAS.csv")
```


# Adding AP13 to vcf files

For each chromosome, sequentially (to cut down on the size of the intermediate files):

gunzip Pvirgatum_V5_GWAS_758g_Chr01K_33dot9M.vcf.gz
head -n 9 Pvirgatum_V5_GWAS_758g_Chr01K_33dot9M.vcf > Pvirgatum_header.txt
# Edit this file so it has AP13 as the last column; make sure there is a \n after AP13
# Save Pvirgatum_header.txt as Pvirgatum_V5_GWAS_758g_33M.vcf

tail -n +9 Pvirgatum_V5_GWAS_758g_Chr01K_33dot9M.vcf | sed 's/$/\t0,0:50,0/' >> Pvirgatum_V5_GWAS_758g_33M.vcf
rm Pvirgatum_V5_GWAS_758g_Chr01K_33dot9M.vcf

gunzip Pvirgatum_V5_GWAS_758g_Chr01N_33dot9M.vcf.gz
tail -n +9 Pvirgatum_V5_GWAS_758g_Chr01N_33dot9M.vcf | sed 's/$/\t0,0:50,0/' >> Pvirgatum_V5_GWAS_758g_33M.vcf
rm Pvirgatum_V5_GWAS_758g_Chr01N_33dot9M.vcf

gunzip Pvirgatum_V5_GWAS_758g_Chr02K_33dot9M.vcf.gz
tail -n +9 Pvirgatum_V5_GWAS_758g_Chr02K_33dot9M.vcf | sed 's/$/\t0,0:50,0/' >> Pvirgatum_V5_GWAS_758g_33M.vcf
rm Pvirgatum_V5_GWAS_758g_Chr02K_33dot9M.vcf

gunzip Pvirgatum_V5_GWAS_758g_Chr02N_33dot9M.vcf.gz
tail -n +9 Pvirgatum_V5_GWAS_758g_Chr02N_33dot9M.vcf | sed 's/$/\t0,0:50,0/' >> Pvirgatum_V5_GWAS_758g_33M.vcf
rm Pvirgatum_V5_GWAS_758g_Chr02N_33dot9M.vcf

gunzip Pvirgatum_V5_GWAS_758g_Chr03K_33dot9M.vcf.gz
tail -n +9 Pvirgatum_V5_GWAS_758g_Chr03K_33dot9M.vcf | sed 's/$/\t0,0:50,0/' >> Pvirgatum_V5_GWAS_758g_33M.vcf
rm Pvirgatum_V5_GWAS_758g_Chr03K_33dot9M.vcf

gunzip Pvirgatum_V5_GWAS_758g_Chr03N_33dot9M.vcf.gz
tail -n +9 Pvirgatum_V5_GWAS_758g_Chr03N_33dot9M.vcf | sed 's/$/\t0,0:50,0/' >> Pvirgatum_V5_GWAS_758g_33M.vcf
rm Pvirgatum_V5_GWAS_758g_Chr03N_33dot9M.vcf

gunzip Pvirgatum_V5_GWAS_758g_Chr04K_33dot9M.vcf.gz
tail -n +9 Pvirgatum_V5_GWAS_758g_Chr04K_33dot9M.vcf | sed 's/$/\t0,0:50,0/' >> Pvirgatum_V5_GWAS_758g_33M.vcf
rm Pvirgatum_V5_GWAS_758g_Chr04K_33dot9M.vcf

gunzip Pvirgatum_V5_GWAS_758g_Chr04N_33dot9M.vcf.gz
tail -n +9 Pvirgatum_V5_GWAS_758g_Chr04N_33dot9M.vcf | sed 's/$/\t0,0:50,0/' >> Pvirgatum_V5_GWAS_758g_33M.vcf
rm Pvirgatum_V5_GWAS_758g_Chr04N_33dot9M.vcf

gunzip Pvirgatum_V5_GWAS_758g_Chr05K_33dot9M.vcf.gz
tail -n +9 Pvirgatum_V5_GWAS_758g_Chr05K_33dot9M.vcf | sed 's/$/\t0,0:50,0/' >> Pvirgatum_V5_GWAS_758g_33M.vcf
rm Pvirgatum_V5_GWAS_758g_Chr05K_33dot9M.vcf

gunzip Pvirgatum_V5_GWAS_758g_Chr05N_33dot9M.vcf.gz
tail -n +9 Pvirgatum_V5_GWAS_758g_Chr05N_33dot9M.vcf | sed 's/$/\t0,0:50,0/' >> Pvirgatum_V5_GWAS_758g_33M.vcf
rm Pvirgatum_V5_GWAS_758g_Chr05N_33dot9M.vcf

gunzip Pvirgatum_V5_GWAS_758g_Chr06K_33dot9M.vcf.gz
tail -n +9 Pvirgatum_V5_GWAS_758g_Chr06K_33dot9M.vcf | sed 's/$/\t0,0:50,0/' >> Pvirgatum_V5_GWAS_758g_33M.vcf
rm Pvirgatum_V5_GWAS_758g_Chr06K_33dot9M.vcf

gunzip Pvirgatum_V5_GWAS_758g_Chr06N_33dot9M.vcf.gz
tail -n +9 Pvirgatum_V5_GWAS_758g_Chr06N_33dot9M.vcf | sed 's/$/\t0,0:50,0/' >> Pvirgatum_V5_GWAS_758g_33M.vcf
rm Pvirgatum_V5_GWAS_758g_Chr06N_33dot9M.vcf

gunzip Pvirgatum_V5_GWAS_758g_Chr07K_33dot9M.vcf.gz
tail -n +9 Pvirgatum_V5_GWAS_758g_Chr07K_33dot9M.vcf | sed 's/$/\t0,0:50,0/' >> Pvirgatum_V5_GWAS_758g_33M.vcf
rm Pvirgatum_V5_GWAS_758g_Chr07K_33dot9M.vcf

gunzip Pvirgatum_V5_GWAS_758g_Chr07N_33dot9M.vcf.gz
tail -n +9 Pvirgatum_V5_GWAS_758g_Chr07N_33dot9M.vcf | sed 's/$/\t0,0:50,0/' >> Pvirgatum_V5_GWAS_758g_33M.vcf
rm Pvirgatum_V5_GWAS_758g_Chr07N_33dot9M.vcf

gunzip Pvirgatum_V5_GWAS_758g_Chr08K_33dot9M.vcf.gz
tail -n +9 Pvirgatum_V5_GWAS_758g_Chr08K_33dot9M.vcf | sed 's/$/\t0,0:50,0/' >> Pvirgatum_V5_GWAS_758g_33M.vcf
rm Pvirgatum_V5_GWAS_758g_Chr08K_33dot9M.vcf

gunzip Pvirgatum_V5_GWAS_758g_Chr08N_33dot9M.vcf.gz
tail -n +9 Pvirgatum_V5_GWAS_758g_Chr08N_33dot9M.vcf | sed 's/$/\t0,0:50,0/' >> Pvirgatum_V5_GWAS_758g_33M.vcf
rm Pvirgatum_V5_GWAS_758g_Chr08N_33dot9M.vcf

gunzip Pvirgatum_V5_GWAS_758g_Chr09K_33dot9M.vcf.gz
tail -n +9 Pvirgatum_V5_GWAS_758g_Chr09K_33dot9M.vcf | sed 's/$/\t0,0:50,0/' >> Pvirgatum_V5_GWAS_758g_33M.vcf
rm Pvirgatum_V5_GWAS_758g_Chr09K_33dot9M.vcf

gunzip Pvirgatum_V5_GWAS_758g_Chr09N_33dot9M.vcf.gz
tail -n +9 Pvirgatum_V5_GWAS_758g_Chr09N_33dot9M.vcf | sed 's/$/\t0,0:50,0/' >> Pvirgatum_V5_GWAS_758g_33M.vcf
rm Pvirgatum_V5_GWAS_758g_Chr09N_33dot9M.vcf

gzip Pvirgatum_V5_GWAS_758g_33M.vcf


###

sed 's/$/\t0,0:50,0/' Pvirgatum_header.txt > Pvirgatum_header2.txt

Then join all chromosomes

Combine individual chromosome files:  
			
  	ls * > filelist.txt
    bcftools concat -n -f filelist.txt -o Pvirgatum_V5_GWAS_757g_33M.vcf.gz -O z
